<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data collation</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.11.3/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.11.3/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">About</a>
</li>
<li>
  <a href="How_to.html">Upload your data</a>
</li>
<li>
  <a href="Data_collation.html">Data collation</a>
</li>
<li>
  <a href="Trait_correlations.html">Calculating genetic trait correlations</a>
</li>
<li>
  <a href="SNP_heritabilities.html">Calculating 'SNP-heritability'</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Data collation</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220705code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220705code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220705)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtree494a5b3b17ed89bb0a912a29ae533d9d3e0f4784targetblank494a5b3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tomkeaney/DGRP/tree/494a5b3b17ed89bb0a912a29ae533d9d3e0f4784" target="_blank">494a5b3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtree494a5b3b17ed89bb0a912a29ae533d9d3e0f4784targetblank494a5b3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tomkeaney/DGRP/tree/494a5b3b17ed89bb0a912a29ae533d9d3e0f4784" target="_blank">494a5b3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/derived/.DS_Store
    Ignored:    data/input/.DS_Store
    Ignored:    data/input/Raw_data_files/.DS_Store
    Ignored:    data/input/Raw_data_files/Zwoinska_2020/.DS_Store

Untracked files:
    Untracked:  Guild_variance_estimates.pdf
    Untracked:  Trait_variance_analysis.Rmd
    Untracked:  analysis/Google_sheets_private.Rmd
    Untracked:  analysis/Trait_correlations.Rmd
    Untracked:  broad_sense_heritability.csv
    Untracked:  code/LICENSE
    Untracked:  code/gcta64
    Untracked:  code/plink
    Untracked:  code/prettify
    Untracked:  code/run_mashr.R
    Untracked:  code/toy.map
    Untracked:  code/toy.ped
    Untracked:  data/DGRP_gwas_bfiles/
    Untracked:  data/Literature_table.csv
    Untracked:  data/Reported_heritability.xlsx
    Untracked:  data/derived/all.dgrp.phenos_combined.csv
    Untracked:  data/derived/all.dgrp.phenos_unscaled.csv
    Untracked:  data/derived/rg_female.csv
    Untracked:  data/derived/rg_male.csv
    Untracked:  data/derived/selected.dgrp.phenos_unscaled.csv
    Untracked:  data/gwas_data/
    Untracked:  fits/

Unstaged changes:
    Deleted:    analysis/Analysis.Rmd
    Modified:   data/derived/SNP_heritability_cleaned.csv
    Modified:   data/derived/all.dgrp.phenos_scaled.csv
    Modified:   data/derived/rg.csv
    Modified:   data/derived/selected.dgrp.phenos_scaled.csv
    Deleted:    data/input/Appel_2016_memory_learning.csv
    Deleted:    data/input/Chapman_2020.csv
    Deleted:    data/input/Chowdhury_2021_aggression.csv
    Deleted:    data/input/Dean_2018.csv
    Deleted:    data/input/Dobson_2015/gwasB.filtered.TAG_glycerol.ab.csv
    Deleted:    data/input/Dobson_2015/gwasB.filtered.dw.ab.csv
    Deleted:    data/input/Dobson_2015/gwasB.filtered.feeding.ab.csv
    Deleted:    data/input/Dobson_2015/gwasB.filtered.glucose.ab.csv
    Deleted:    data/input/Dobson_2015/gwasB.filtered.glycogen.ab.csv
    Deleted:    data/input/Dobson_2015/gwasB.filtered.protein.ab.csv
    Deleted:    data/input/Duneau_2018/Duneau_2018_deltamethrin.csv
    Deleted:    data/input/Duneau_2018/Duneau_2018_parathion.csv
    Deleted:    data/input/Everett_microbiome_2020.csv
    Deleted:    data/input/Everman_2019_starvation_res.csv
    Deleted:    data/input/Freda_2017_ThermalHardiness.csv
    Deleted:    data/input/Freda_2019/Cold18.csv
    Deleted:    data/input/Freda_2019/Cold25.csv
    Deleted:    data/input/Freda_2019/Heat18.csv
    Deleted:    data/input/Freda_2019/Heat25.csv
    Modified:   data/input/Future_line_means/template.csv
    Deleted:    data/input/Gaertner_2015_male_mating.csv
    Deleted:    data/input/Grubbs_2013_leg_props.csv
    Deleted:    data/input/He_et_al_2014.csv
    Deleted:    data/input/Jumbo_Lucioni_2012.csv
    Deleted:    data/input/Najarro_et_al_2015_caffeine_resistance.csv
    Deleted:    data/input/Palmer_2018_kallithea.csv
    Deleted:    data/input/Pitchers_2019_wing_multivariate.csv
    Deleted:    data/input/Rohde_2019.csv
    Deleted:    data/input/Saltz_2017_larvae_learning_data.csv
    Deleted:    data/input/Saltz_2017_pupation_substrate_data.csv
    Deleted:    data/input/Srivastav_Kelleher_2017_data.csv
    Deleted:    data/input/Teets_Hahn_2018_data.csv
    Deleted:    data/input/Watanabe_Riddle_2021.csv
    Deleted:    data/input/Yanagawa_total_groom_time.csv
    Deleted:    data/input/Zhou_metabolome_2020.xlsx
    Deleted:    data/input/Zwoinska_2020/Zwoinska_females_2020.csv
    Deleted:    data/input/Zwoinska_2020/Zwoinska_males_2020.csv
    Deleted:    data/input/munoz_munoz_2016.csv
    Deleted:    data/input/najarro_2017.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Data_collation.Rmd</code>) and
HTML (<code>docs/Data_collation.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/494a5b3b17ed89bb0a912a29ae533d9d3e0f4784/analysis/Data_collation.Rmd" target="_blank">494a5b3</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-29
</td>
<td>
adding reproducibility features
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/a5e979c7c2daa7ebace32c121d6cdd958456c448/docs/Data_collation.html" target="_blank">a5e979c</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/d9a3e127afa5495cee1013ed6677b96a4e6bf50a/analysis/Data_collation.Rmd" target="_blank">d9a3e12</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-19
</td>
<td>
Adding Zhou 2020 data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/93e183b84c40eef9d1249c5d6225a52904137bb5/docs/Data_collation.html" target="_blank">93e183b</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/668940d1c3f60a36a6a79fc4cce939abb57ffa98/analysis/Data_collation.Rmd" target="_blank">668940d</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-15
</td>
<td>
Making things more reproducible
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/0a9b7985e07e9447459818ef426bd9c2e6cc6478/analysis/Data_collation.Rmd" target="_blank">0a9b798</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-15
</td>
<td>
Merge branch ‘master’ of <a href="https://github.com/tomkeaney/DGRP"
class="uri">https://github.com/tomkeaney/DGRP</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/bef5beef2b12be706aa51496362823ac14593981/analysis/Data_collation.Rmd" target="_blank">bef5bee</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-15
</td>
<td>
committing for merge with Luke’s edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/be494c457e030aad09fe01ec606a7cc16439aefc/analysis/Data_collation.Rmd" target="_blank">be494c4</a>
</td>
<td>
lukeholman
</td>
<td>
2022-09-08
</td>
<td>
Luke work on Data_collation Sept 2022
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/b1f1ed75e6d4ff07a6113ab8a93a4bc4a62b10b1/docs/Data_collation.html" target="_blank">b1f1ed7</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/665a84d1d1fe8c49134ddded27a9ced0fea03f6b/docs/Data_collation.html" target="_blank">665a84d</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/9651fe93fc237dce875b4ed5110b5c4771776ef7/analysis/Data_collation.Rmd" target="_blank">9651fe9</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
ready for shiny app roll out
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/200cafa543f978668b729c096bae8a289ac631b9/docs/Data_collation.html" target="_blank">200cafa</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/e3c284a7e2df921ce2a709c619e51517cfaf582b/analysis/Data_collation.Rmd" target="_blank">e3c284a</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
final version for analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/dcb2918aef2511da6e4d1b4e9ff50f2ee0b0dd29/docs/Data_collation.html" target="_blank">dcb2918</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/ed527563b89e7462962afe6b2cfce58d27e2d809/analysis/Data_collation.Rmd" target="_blank">ed52756</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
final (lol) version for analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/2ac2cbcefab1fa1941cc6c9c0e95b159e06ce9fe/docs/Data_collation.html" target="_blank">2ac2cbc</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/006deee4c5fd8db5e0fc02301d9005cc0addb93d/analysis/Data_collation.Rmd" target="_blank">006deee</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
final version for analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/f924e99a74781b05a8a36e8b7d8b16a83930da65/docs/Data_collation.html" target="_blank">f924e99</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/86612ae2696c27e674526d1d1f89d18d7abcda74/analysis/Data_collation.Rmd" target="_blank">86612ae</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-06
</td>
<td>
Publish edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/8b03a2f9a539a3555ff033a2ad4269c72c8e389c/docs/Data_collation.html" target="_blank">8b03a2f</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/5133de98630e74431cae4005a4f55365b1d59c76/analysis/Data_collation.Rmd" target="_blank">5133de9</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-05
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-in-necessary-packages" class="section level1">
<h1>Load in necessary packages</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(googlesheets4) <span class="co"># for reading google sheets</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms) <span class="co"># for bayesian models</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidybayes) <span class="co"># for more bayesian things</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT) <span class="co"># for interactive tables</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom) <span class="co"># convert results of functions into tables</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(osfr) <span class="co"># For accessing data on the repository OSF</span></span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
<div id="loading-the-line-mean-phenotype-data" class="section level2">
<h2>Loading the line mean phenotype data</h2>
<p><span class="math inline">\(~\)</span></p>
<p>A high percentage of the papers that report DGRP data provide
estimates of the mean phenotype of multiple DGRP lines (“line mean
data”). We worked collaboratively to conduct a non-exhaustive search of
the literature for phenotypic trait measurements for DGRP lines, and
collated all the line mean data in two csv files;
<code>all.line.mean.phenos.csv</code> which contains line mean values
for each trait and <code>all.line.mean.phenos_metadata.csv</code> which
contains study meta-data. The R code below combines these files.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>all.line.mean.phenos <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/all.line.mean.phenos.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">read_csv</span>(<span class="st">&quot;data/input/all.line.mean.phenos_metadata.csv&quot;</span>))</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="estimating-line-means-from-raw-data" class="section level2">
<h2>Estimating line means from raw data</h2>
<p>Some studies only provide the raw data they collected when measuring
DGRP phenotypes. We need line mean data to calculate genetic
correlations between traits and run typical GWA analyses. To get line
means, we fit Bayesian mixed effect models and use these to estimate
mean trait values for each line. Each analysis is presented in a
separate tab below.</p>
<p>Where the response follows a Gaussian distribution, we standardise
the response to have a mean of 0 and a standard deviation of 1. We do
this to make prior specification more straightforward.</p>
<p>The Bayesian models won’t run unless you have <code>stan</code> and
the <code>brms</code> package installed. To install <code>stan</code>
follow these <a
href="https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started">instructions</a>.
Otherwise, you can skip to the data cleaning section and load in the csv
file <code>"data/input/all.dgrp.phenos_combined.csv"</code> as a
tibble/dataframe.</p>
</div>
<div id="analysis-of-raw-data" class="section level2 tabset">
<h2 class="tabset">Analysis of raw data</h2>
<div id="jumbo-lucioni-et-al-2012" class="section level3">
<h3>Jumbo-Lucioni et al 2012</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>Jumbo_mito_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Jumbo_Lucioni_2012.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Gender),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Rep =</span> <span class="fu">as.factor</span>(Rep),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">po_ratio =</span> <span class="fu">na_if</span>(<span class="st">`</span><span class="at">po ratio</span><span class="st">`</span>, <span class="st">&quot;.&quot;</span>),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">po_ratio =</span> <span class="fu">as.numeric</span>(po_ratio)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">state_3 =</span> <span class="st">`</span><span class="at">st 3 (pmol/s*mg protein)</span><span class="st">`</span>,</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">state_4 =</span> <span class="st">`</span><span class="at">st 4 (pmol/s*mg protein)</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Sex, Rep, state_3, state_4, po_ratio) <span class="sc">%&gt;%</span> </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(line))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co"># standardise </span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>Jumbo_mito_data <span class="ot">&lt;-</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  Jumbo_mito_data <span class="sc">%&gt;%</span> </span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">st_state_3 =</span> (state_3 <span class="sc">-</span> <span class="fu">mean</span>(state_3))<span class="sc">/</span><span class="fu">sd</span>(state_3),</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_state_4 =</span> (state_4 <span class="sc">-</span> <span class="fu">mean</span>(state_4))<span class="sc">/</span><span class="fu">sd</span>(state_4),</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_po_ratio =</span> (po_ratio <span class="sc">-</span> <span class="fu">mean</span>(po_ratio, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))<span class="sc">/</span><span class="fu">sd</span>(po_ratio, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co"># respiration state 3 model</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>state_3_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_state_3 <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Jumbo_mito_data, <span class="at">family =</span> gaussian,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/state_3.model&quot;</span>)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(state_3_model)</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Jumbo_mito_data<span class="sc">$</span>line,</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>, <span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>fitted_state_3 <span class="ot">&lt;-</span> <span class="fu">fitted</span>(state_3_model,</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_3) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_3),</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_3) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_3),</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_3) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_3))</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>state_3_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_state_3) <span class="sc">%&gt;%</span> </span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding sex specific line means</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>Jumbo_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, y, Trait, description){</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> y) <span class="sc">%&gt;%</span> </span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Jumbo-Lucioni et al (2012) BMC Genomics&quot;</span>,</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Physiological&quot;</span>,</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a><span class="co"># create the state 3 tibbles</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>state_3_f_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_3_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;mitochondrial.state.3.respiration.f&quot;</span>, <span class="st">&quot;Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>state_3_m_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_3_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;mitochondrial.state.3.respiration.m&quot;</span>, <span class="st">&quot;Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a><span class="co"># respiration state 4 model</span></span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>state_4_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_state_4 <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data =</span> Jumbo_mito_data, <span class="at">family =</span> gaussian,</span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>                     <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>                     <span class="at">iter =</span> <span class="dv">10000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>                     <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/state_4.model&quot;</span>)</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(state_4_model)</span></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>fitted_state_4 <span class="ot">&lt;-</span> <span class="fu">fitted</span>(state_4_model,</span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_4) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_4),</span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_4) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_4),</span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_4) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_4))</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>state_4_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_state_4) <span class="sc">%&gt;%</span> </span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a><span class="co"># create the state 4 tibbles</span></span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a>state_4_f_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_4_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;mitochondrial.state.4.respiration.f&quot;</span>, <span class="st">&quot;Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-109"><a href="#cb3-109" aria-hidden="true" tabindex="-1"></a>state_4_m_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_4_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;mitochondrial.state.4.respiration.m&quot;</span>, <span class="st">&quot;Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a><span class="co"># P:O ratio model</span></span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-115"><a href="#cb3-115" aria-hidden="true" tabindex="-1"></a>PO_ratio_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_po_ratio <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb3-116"><a href="#cb3-116" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Jumbo_mito_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(po_ratio <span class="sc">!=</span> <span class="st">&quot;NA&quot;</span>), <span class="at">family =</span> gaussian,</span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-119"><a href="#cb3-119" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-120"><a href="#cb3-120" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/PO_ratio.model&quot;</span>)</span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(PO_ratio_model)</span></span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a>Jumbo_PO_data <span class="ot">&lt;-</span> Jumbo_mito_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(po_ratio <span class="sc">!=</span> <span class="st">&quot;NA&quot;</span>)</span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a>fitted_PO_ratio <span class="ot">&lt;-</span> <span class="fu">fitted</span>(PO_ratio_model, <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio),</span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio),</span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio))</span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a>PO_ratio_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_PO_ratio) <span class="sc">%&gt;%</span> </span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-136"><a href="#cb3-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-137"><a href="#cb3-137" aria-hidden="true" tabindex="-1"></a><span class="co"># create the P:O ratio tibbles</span></span>
<span id="cb3-138"><a href="#cb3-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-139"><a href="#cb3-139" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a>PO_ratio_f_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(PO_ratio_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;mitochondrial.PO.ratio.f&quot;</span>, <span class="st">&quot;Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.&quot;</span>)</span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a>PO_ratio_m_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(PO_ratio_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;mitochondrial.PO.ratio.m&quot;</span>, <span class="st">&quot;Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.&quot;</span>)</span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a>Jumbo_traits <span class="ot">&lt;-</span> </span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(state_3_f_line_means, </span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a>        state_3_m_line_means, </span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a>        state_4_f_line_means, </span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a>        state_4_m_line_means, </span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a>        PO_ratio_f_line_means, </span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a>        PO_ratio_m_line_means)</span></code></pre></div>
</div>
<div id="grubbs-et-al-2013" class="section level3">
<h3>Grubbs et al 2013</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Grubbs et al 2013</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>Grubbs_2013_leg_props <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Grubbs_2013_leg_props.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Femur =</span> (Femur <span class="sc">-</span> <span class="fu">mean</span>(Femur))<span class="sc">/</span><span class="fu">sd</span>(Femur),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Tibia =</span> (Tibia <span class="sc">-</span> <span class="fu">mean</span>(Tibia))<span class="sc">/</span><span class="fu">sd</span>(Tibia),</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Tarsus =</span> (Tarsus <span class="sc">-</span> <span class="fu">mean</span>(Tarsus))<span class="sc">/</span><span class="fu">sd</span>(Tarsus),</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Total =</span> (Total <span class="sc">-</span> <span class="fu">mean</span>(Total))<span class="sc">/</span><span class="fu">sd</span>(Total))</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>Grubbs_2013_T1_leg_props <span class="ot">&lt;-</span> Grubbs_2013_leg_props <span class="sc">%&gt;%</span> </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(T1_T2 <span class="sc">==</span> <span class="st">&quot;T1&quot;</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>Grubbs_2013_T2_leg_props <span class="ot">&lt;-</span> Grubbs_2013_leg_props <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(T1_T2 <span class="sc">==</span> <span class="st">&quot;T2&quot;</span>)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co"># mean total T1 leg length</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>T1_leg_length_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Total <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1_leg.length.model&quot;</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">Line =</span> Grubbs_2013_leg_props<span class="sc">$</span>Line,</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>, <span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Line, Sex)</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>fitted_T1_leg_total <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_leg_length_model, <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total),</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total),</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total))</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding sex specific line means</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>Grubbs_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, y, Trait, guild, description){</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> y) <span class="sc">%&gt;%</span> </span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(Line, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> Trait,</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Grubbs et al (2013) PLOS One&quot;</span>,</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> guild,</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>T1_leg_length_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_leg_total) <span class="sc">%&gt;%</span> </span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>T1_leg_length_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_leg_length_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.leg.length.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>T1_leg_length_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_leg_length_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.leg.length.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a><span class="co"># mean total T2 leg length</span></span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>T2_leg_length_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Total <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> <span class="st">&quot;Gaussian&quot;</span>,</span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2_leg.length.model&quot;</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>fitted_T2_leg_total <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_leg_length_model,</span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total),</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total),</span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total))</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>T2_leg_length_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_leg_total) <span class="sc">%&gt;%</span> </span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a>T2_leg_length_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_leg_length_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.leg.length.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>T2_leg_length_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_leg_length_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.leg.length.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Femur length</span></span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>T1_femur_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Femur <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1.femur.model&quot;</span>, </span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a>fitted_T1_femur <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_femur_model,</span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur))</span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>T1_femur_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_femur) <span class="sc">%&gt;%</span> </span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>femur_T1_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_femur_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.femur.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a>femur_T1_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_femur_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.femur.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a>T2_femur_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Femur <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2.femur.model&quot;</span>, </span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>fitted_T2_femur <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_femur_model,</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur))</span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a>T2_femur_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_femur) <span class="sc">%&gt;%</span> </span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>femur_T2_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_femur_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.femur.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a>femur_T2_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_femur_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.femur.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a><span class="co"># Tibia length</span></span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a>T1_tibia_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tibia <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1.tibia.model&quot;</span>, </span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>fitted_T1_tibia <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_tibia_model,</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia))</span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>T1_tibia_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_tibia) <span class="sc">%&gt;%</span> </span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a>tibia_T1_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tibia_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.tibia.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>tibia_T1_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tibia_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.tibia.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a>T2_tibia_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tibia <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">10000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2.tibia.model&quot;</span>, </span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a>fitted_T2_tibia <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_tibia_model,</span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia))</span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a>T2_tibia_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_tibia) <span class="sc">%&gt;%</span> </span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-207"><a href="#cb4-207" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb4-208"><a href="#cb4-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a>tibia_T2_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tibia_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.tibia.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a>tibia_T2_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tibia_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.tibia.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a><span class="co"># Tarsus length</span></span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-217"><a href="#cb4-217" aria-hidden="true" tabindex="-1"></a>T1_tarsus_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tarsus <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-218"><a href="#cb4-218" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1.tarsus.model&quot;</span>, </span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a>fitted_T1_tarsus <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_tarsus_model,</span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus))</span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a>T1_tarsus_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_tarsus) <span class="sc">%&gt;%</span> </span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a>tarsus_T1_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tarsus_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.tarsus.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a>tarsus_T1_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tarsus_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.tarsus.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a>T2_tarsus_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tarsus <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-251"><a href="#cb4-251" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2.tarsus.model&quot;</span>, </span>
<span id="cb4-252"><a href="#cb4-252" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a>fitted_T2_tarsus <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_tarsus_model,</span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus))</span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a>T2_tarsus_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_tarsus) <span class="sc">%&gt;%</span> </span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a>tarsus_T2_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tarsus_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.tarsus.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a>tarsus_T2_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tarsus_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.tarsus.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a>Grubbs_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(T1_leg_length_f_line_means, </span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a>                       T1_leg_length_m_line_means, </span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a>                       T2_leg_length_f_line_means, </span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a>                       T2_leg_length_m_line_means,</span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a>                       femur_T1_f_line_means, </span>
<span id="cb4-278"><a href="#cb4-278" aria-hidden="true" tabindex="-1"></a>                       femur_T1_m_line_means, </span>
<span id="cb4-279"><a href="#cb4-279" aria-hidden="true" tabindex="-1"></a>                       femur_T2_f_line_means, </span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a>                       femur_T2_m_line_means,</span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a>                       tibia_T1_f_line_means, </span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a>                       tibia_T1_m_line_means, </span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a>                       tibia_T2_f_line_means, </span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a>                       tibia_T2_m_line_means,</span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a>                       tarsus_T1_f_line_means, </span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a>                       tarsus_T1_m_line_means, </span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>                       tarsus_T2_f_line_means, </span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a>                       tarsus_T2_m_line_means)</span></code></pre></div>
</div>
<div id="dobson-et-al-2015" class="section level3">
<h3>Dobson et al 2015</h3>
<p>From <strong>Supplementary Table 1</strong>’s caption: “In the second
approach (c), simpler models were fitted, in which microbiota, weight
and Wolbachia status were additive, with genotype nested in experimental
block as a random effect. The statistical analysis presented in Fig.1b
uses these models simplified by removal of non-significant factors.”</p>
<p>We follow this approach to best recapitulate the study’s analysis,
except that we allow the intercept for the random effect line to vary
with microbiota treatment. This is needed in our modelling approach to
estimate line specific effects of microbiota. We also include day as a
blocking factor, which represents a block of data collection involving a
subset of lines. This controls for micro-environmental variance between
blocks of phenotyping.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># weight</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>Dobson_NI_microbiome.dryweight <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.dw.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_weight.5.flies =</span> (weight.<span class="fl">5.</span>flies <span class="sc">-</span> <span class="fu">mean</span>(weight.<span class="fl">5.</span>flies)) <span class="sc">/</span> <span class="fu">sd</span>(weight.<span class="fl">5.</span>flies)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(PC1, PC2, PC3, Code, RAL, ...<span class="dv">1</span>))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_NI_microbiome.dryweight <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_NI_microbiome.dryweight <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measurements in both treatments </span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>weight_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding specific line means</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>Dobson_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, Trait, guild, description){</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Dobson et al (2015) Nature Communications&quot;</span>,</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> guild,</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>weight_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_weight.<span class="fl">5.</span>flies <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_NI_microbiome.dryweight, <span class="at">family =</span> gaussian,</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/weight.microbiome.model&quot;</span>)</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(weight_microbiome_model)</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_NI_microbiome.dryweight<span class="sc">$</span>line,</span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_NI_microbiome.dryweight<span class="sc">$</span>trt,</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day)</span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>fitted_dryweight <span class="ot">&lt;-</span> <span class="fu">fitted</span>(weight_microbiome_model, <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies),</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies),</span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies))</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>dry_weight_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_dryweight) <span class="sc">%&gt;%</span> </span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>weight.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(dry_weight_estimates_axenic, <span class="st">&quot;weight.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.&quot;</span>)</span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>dry_weight_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_dryweight) <span class="sc">%&gt;%</span> </span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>weight.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(dry_weight_estimates_gnotobiotic, <span class="st">&quot;weight.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.&quot;</span>)</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>axenic <span class="ot">&lt;-</span> </span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(weight_microbiome_effect_lines, dry_weight_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>gnoto <span class="ot">&lt;-</span></span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(weight_microbiome_effect_lines, dry_weight_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>dry_weight_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic, gnoto) <span class="sc">%&gt;%</span> </span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_weight_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(dry_weight_estimates_diff, <span class="st">&quot;weight.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.&quot;</span>)</span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Glucose</span></span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>Dobson_glucose_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.glucose.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_glucose =</span> (glucose <span class="sc">-</span> <span class="fu">mean</span>(glucose)) <span class="sc">/</span> <span class="fu">sd</span>(glucose)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, wolb, glucose, trt, day, weight.<span class="fl">5.</span>flies, st_glucose))</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_glucose_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_glucose_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a>glucose_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that we include weight as a moderator variable. This means we only estimate the direct effect of line on glucose levels, instead of the total effect. This is because line likely also affects weight which is likely to have a flow on effect on glucose levels. </span></span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a>glucose_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_glucose <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_glucose_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/glucose.microbiome.model&quot;</span>)</span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(glucose_microbiome_model)</span></span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_glucose_microbiome<span class="sc">$</span>line,</span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_glucose_microbiome<span class="sc">$</span>trt,</span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>fitted_glucose <span class="ot">&lt;-</span> <span class="fu">fitted</span>(glucose_microbiome_model,</span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data, <span class="at">allow_new_levels =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="co"># Why do we need to allow new levels? Might be an issue</span></span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose),</span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose),</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose))</span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a>glucose_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glucose) <span class="sc">%&gt;%</span> </span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>glucose.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glucose_estimates_axenic, <span class="st">&quot;glucose.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.&quot;</span>)</span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a>glucose_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glucose) <span class="sc">%&gt;%</span> </span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a>glucose.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glucose_estimates_gnotobiotic, <span class="st">&quot;glucose.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.&quot;</span>)</span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a>axenic_glucose <span class="ot">&lt;-</span> </span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glucose_microbiome_effect_lines, glucose_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a>gnoto_glucose <span class="ot">&lt;-</span></span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glucose_microbiome_effect_lines, glucose_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a>glucose_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_glucose, gnoto_glucose) <span class="sc">%&gt;%</span> </span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_glucose_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glucose_estimates_diff, <span class="st">&quot;glucose.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.&quot;</span>)</span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-166"><a href="#cb5-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-167"><a href="#cb5-167" aria-hidden="true" tabindex="-1"></a><span class="co"># Glycogen</span></span>
<span id="cb5-168"><a href="#cb5-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-169"><a href="#cb5-169" aria-hidden="true" tabindex="-1"></a>Dobson_glycogen_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.glycogen.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-170"><a href="#cb5-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb5-171"><a href="#cb5-171" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb5-172"><a href="#cb5-172" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb5-173"><a href="#cb5-173" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb5-174"><a href="#cb5-174" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb5-175"><a href="#cb5-175" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_glycogen =</span> (glycogen <span class="sc">-</span> <span class="fu">mean</span>(glycogen)) <span class="sc">/</span> <span class="fu">sd</span>(glycogen)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-176"><a href="#cb5-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, wolb, glycogen, st_glycogen, trt, day, weight.<span class="fl">5.</span>flies))</span>
<span id="cb5-177"><a href="#cb5-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-178"><a href="#cb5-178" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb5-179"><a href="#cb5-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-180"><a href="#cb5-180" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_glycogen_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-181"><a href="#cb5-181" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_glycogen_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-182"><a href="#cb5-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-183"><a href="#cb5-183" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb5-184"><a href="#cb5-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-185"><a href="#cb5-185" aria-hidden="true" tabindex="-1"></a>glycogen_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb5-186"><a href="#cb5-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-187"><a href="#cb5-187" aria-hidden="true" tabindex="-1"></a>glycogen_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_glycogen <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb5-188"><a href="#cb5-188" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_glycogen_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb5-189"><a href="#cb5-189" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-190"><a href="#cb5-190" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb5-191"><a href="#cb5-191" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb5-192"><a href="#cb5-192" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-193"><a href="#cb5-193" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/glycogen.microbiome.model&quot;</span>)</span>
<span id="cb5-194"><a href="#cb5-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-195"><a href="#cb5-195" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(glycogen_microbiome_model)</span></span>
<span id="cb5-196"><a href="#cb5-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-197"><a href="#cb5-197" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb5-198"><a href="#cb5-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_glycogen_microbiome<span class="sc">$</span>line,</span>
<span id="cb5-199"><a href="#cb5-199" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_glycogen_microbiome<span class="sc">$</span>trt,</span>
<span id="cb5-200"><a href="#cb5-200" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb5-201"><a href="#cb5-201" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb5-202"><a href="#cb5-202" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-203"><a href="#cb5-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb5-204"><a href="#cb5-204" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb5-205"><a href="#cb5-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-206"><a href="#cb5-206" aria-hidden="true" tabindex="-1"></a>fitted_glycogen <span class="ot">&lt;-</span> <span class="fu">fitted</span>(glycogen_microbiome_model,</span>
<span id="cb5-207"><a href="#cb5-207" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span></span>
<span id="cb5-208"><a href="#cb5-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-209"><a href="#cb5-209" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb5-210"><a href="#cb5-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen),</span>
<span id="cb5-211"><a href="#cb5-211" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen),</span>
<span id="cb5-212"><a href="#cb5-212" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen))</span>
<span id="cb5-213"><a href="#cb5-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-214"><a href="#cb5-214" aria-hidden="true" tabindex="-1"></a>glycogen_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glycogen) <span class="sc">%&gt;%</span> </span>
<span id="cb5-215"><a href="#cb5-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb5-216"><a href="#cb5-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-217"><a href="#cb5-217" aria-hidden="true" tabindex="-1"></a>glycogen.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glycogen_estimates_axenic, <span class="st">&quot;glycogen.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined glycogen content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glycogen content.&quot;</span>)</span>
<span id="cb5-218"><a href="#cb5-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-219"><a href="#cb5-219" aria-hidden="true" tabindex="-1"></a>glycogen_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glycogen) <span class="sc">%&gt;%</span> </span>
<span id="cb5-220"><a href="#cb5-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb5-221"><a href="#cb5-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-222"><a href="#cb5-222" aria-hidden="true" tabindex="-1"></a>glycogen.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glycogen_estimates_gnotobiotic, <span class="st">&quot;glycogen.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>,  <span class="st">&quot;Mean combined glycogen content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glycogen content.&quot;</span>)</span>
<span id="cb5-223"><a href="#cb5-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-224"><a href="#cb5-224" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb5-225"><a href="#cb5-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-226"><a href="#cb5-226" aria-hidden="true" tabindex="-1"></a>axenic_glycogen <span class="ot">&lt;-</span> </span>
<span id="cb5-227"><a href="#cb5-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glycogen_microbiome_effect_lines, glycogen_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-228"><a href="#cb5-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-229"><a href="#cb5-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb5-230"><a href="#cb5-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-231"><a href="#cb5-231" aria-hidden="true" tabindex="-1"></a>gnoto_glycogen <span class="ot">&lt;-</span></span>
<span id="cb5-232"><a href="#cb5-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glycogen_microbiome_effect_lines, glycogen_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-233"><a href="#cb5-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-234"><a href="#cb5-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb5-235"><a href="#cb5-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-236"><a href="#cb5-236" aria-hidden="true" tabindex="-1"></a>glycogen_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb5-237"><a href="#cb5-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_glycogen, gnoto_glycogen) <span class="sc">%&gt;%</span> </span>
<span id="cb5-238"><a href="#cb5-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb5-239"><a href="#cb5-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-240"><a href="#cb5-240" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_glycogen_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glycogen_estimates_diff, <span class="st">&quot;glycogen.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined glycogen content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glycogen content. Note that the presence of micriobiota on average increased glycogen content.&quot;</span>)</span>
<span id="cb5-241"><a href="#cb5-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-242"><a href="#cb5-242" aria-hidden="true" tabindex="-1"></a><span class="co"># Protein</span></span>
<span id="cb5-243"><a href="#cb5-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-244"><a href="#cb5-244" aria-hidden="true" tabindex="-1"></a>Dobson_protein_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.protein.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-245"><a href="#cb5-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb5-246"><a href="#cb5-246" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb5-247"><a href="#cb5-247" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb5-248"><a href="#cb5-248" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb5-249"><a href="#cb5-249" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb5-250"><a href="#cb5-250" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_protein =</span> (protein <span class="sc">-</span> <span class="fu">mean</span>(protein)) <span class="sc">/</span> <span class="fu">sd</span>(protein)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-251"><a href="#cb5-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, wolb, protein, st_protein, trt, day, weight.<span class="fl">5.</span>flies))</span>
<span id="cb5-252"><a href="#cb5-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-253"><a href="#cb5-253" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb5-254"><a href="#cb5-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-255"><a href="#cb5-255" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_protein_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-256"><a href="#cb5-256" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_protein_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-257"><a href="#cb5-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-258"><a href="#cb5-258" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb5-259"><a href="#cb5-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-260"><a href="#cb5-260" aria-hidden="true" tabindex="-1"></a>protein_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb5-261"><a href="#cb5-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-262"><a href="#cb5-262" aria-hidden="true" tabindex="-1"></a>protein_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_protein <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb5-263"><a href="#cb5-263" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_protein_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb5-264"><a href="#cb5-264" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-265"><a href="#cb5-265" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb5-266"><a href="#cb5-266" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb5-267"><a href="#cb5-267" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-268"><a href="#cb5-268" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/protein.microbiome.model&quot;</span>)</span>
<span id="cb5-269"><a href="#cb5-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-270"><a href="#cb5-270" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(protein_microbiome_model)</span></span>
<span id="cb5-271"><a href="#cb5-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-272"><a href="#cb5-272" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb5-273"><a href="#cb5-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_protein_microbiome<span class="sc">$</span>line,</span>
<span id="cb5-274"><a href="#cb5-274" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_protein_microbiome<span class="sc">$</span>trt,</span>
<span id="cb5-275"><a href="#cb5-275" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb5-276"><a href="#cb5-276" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb5-277"><a href="#cb5-277" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-278"><a href="#cb5-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb5-279"><a href="#cb5-279" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb5-280"><a href="#cb5-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-281"><a href="#cb5-281" aria-hidden="true" tabindex="-1"></a>fitted_protein <span class="ot">&lt;-</span> <span class="fu">fitted</span>(protein_microbiome_model,</span>
<span id="cb5-282"><a href="#cb5-282" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span></span>
<span id="cb5-283"><a href="#cb5-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-284"><a href="#cb5-284" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb5-285"><a href="#cb5-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_protein_microbiome<span class="sc">$</span>protein) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_protein_microbiome<span class="sc">$</span>protein),</span>
<span id="cb5-286"><a href="#cb5-286" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_protein_microbiome<span class="sc">$</span>protein) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_protein_microbiome<span class="sc">$</span>protein),</span>
<span id="cb5-287"><a href="#cb5-287" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_protein_microbiome<span class="sc">$</span>protein) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_protein_microbiome<span class="sc">$</span>protein))</span>
<span id="cb5-288"><a href="#cb5-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-289"><a href="#cb5-289" aria-hidden="true" tabindex="-1"></a>protein_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_protein) <span class="sc">%&gt;%</span> </span>
<span id="cb5-290"><a href="#cb5-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb5-291"><a href="#cb5-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-292"><a href="#cb5-292" aria-hidden="true" tabindex="-1"></a>protein.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(protein_estimates_axenic, <span class="st">&quot;protein.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.&quot;</span>)</span>
<span id="cb5-293"><a href="#cb5-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-294"><a href="#cb5-294" aria-hidden="true" tabindex="-1"></a>protein_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_protein) <span class="sc">%&gt;%</span> </span>
<span id="cb5-295"><a href="#cb5-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb5-296"><a href="#cb5-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-297"><a href="#cb5-297" aria-hidden="true" tabindex="-1"></a>protein.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(protein_estimates_gnotobiotic, <span class="st">&quot;protein.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.&quot;</span>)</span>
<span id="cb5-298"><a href="#cb5-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-299"><a href="#cb5-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb5-300"><a href="#cb5-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-301"><a href="#cb5-301" aria-hidden="true" tabindex="-1"></a>axenic_protein <span class="ot">&lt;-</span> </span>
<span id="cb5-302"><a href="#cb5-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_microbiome_effect_lines, protein_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-303"><a href="#cb5-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-304"><a href="#cb5-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb5-305"><a href="#cb5-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-306"><a href="#cb5-306" aria-hidden="true" tabindex="-1"></a>gnoto_protein <span class="ot">&lt;-</span></span>
<span id="cb5-307"><a href="#cb5-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_microbiome_effect_lines, protein_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-308"><a href="#cb5-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-309"><a href="#cb5-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb5-310"><a href="#cb5-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-311"><a href="#cb5-311" aria-hidden="true" tabindex="-1"></a>protein_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb5-312"><a href="#cb5-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_protein, gnoto_protein) <span class="sc">%&gt;%</span> </span>
<span id="cb5-313"><a href="#cb5-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb5-314"><a href="#cb5-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-315"><a href="#cb5-315" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_protein_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(protein_estimates_diff, <span class="st">&quot;protein.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined protein content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher protein content. Note that the presence of micriobiota on average increased protein content.&quot;</span>)</span>
<span id="cb5-316"><a href="#cb5-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-317"><a href="#cb5-317" aria-hidden="true" tabindex="-1"></a><span class="co"># TAG glycerol</span></span>
<span id="cb5-318"><a href="#cb5-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-319"><a href="#cb5-319" aria-hidden="true" tabindex="-1"></a>Dobson_TAG_glycerol_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.TAG_glycerol.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-320"><a href="#cb5-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb5-321"><a href="#cb5-321" aria-hidden="true" tabindex="-1"></a>         <span class="co">#wolb = as.factor(wolb),</span></span>
<span id="cb5-322"><a href="#cb5-322" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb5-323"><a href="#cb5-323" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb5-324"><a href="#cb5-324" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb5-325"><a href="#cb5-325" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_TAG_glycerol =</span> (TAG_glycerol <span class="sc">-</span> <span class="fu">mean</span>(TAG_glycerol)) <span class="sc">/</span> <span class="fu">sd</span>(TAG_glycerol)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-326"><a href="#cb5-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, TAG_glycerol, st_TAG_glycerol, trt, day, weight.<span class="fl">5.</span>flies))</span>
<span id="cb5-327"><a href="#cb5-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-328"><a href="#cb5-328" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb5-329"><a href="#cb5-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-330"><a href="#cb5-330" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_TAG_glycerol_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-331"><a href="#cb5-331" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_TAG_glycerol_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb5-332"><a href="#cb5-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-333"><a href="#cb5-333" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb5-334"><a href="#cb5-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-335"><a href="#cb5-335" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb5-336"><a href="#cb5-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-337"><a href="#cb5-337" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_TAG_glycerol <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb5-338"><a href="#cb5-338" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_TAG_glycerol_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb5-339"><a href="#cb5-339" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-340"><a href="#cb5-340" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb5-341"><a href="#cb5-341" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb5-342"><a href="#cb5-342" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-343"><a href="#cb5-343" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/TAG_glycerol.microbiome.model&quot;</span>)</span>
<span id="cb5-344"><a href="#cb5-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-345"><a href="#cb5-345" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(TAG_glycerol_microbiome_model)</span></span>
<span id="cb5-346"><a href="#cb5-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-347"><a href="#cb5-347" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb5-348"><a href="#cb5-348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_TAG_glycerol_microbiome<span class="sc">$</span>line,</span>
<span id="cb5-349"><a href="#cb5-349" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_TAG_glycerol_microbiome<span class="sc">$</span>trt,</span>
<span id="cb5-350"><a href="#cb5-350" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb5-351"><a href="#cb5-351" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-352"><a href="#cb5-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb5-353"><a href="#cb5-353" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb5-354"><a href="#cb5-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-355"><a href="#cb5-355" aria-hidden="true" tabindex="-1"></a>fitted_TAG_glycerol <span class="ot">&lt;-</span> <span class="fu">fitted</span>(TAG_glycerol_microbiome_model,</span>
<span id="cb5-356"><a href="#cb5-356" aria-hidden="true" tabindex="-1"></a>                              <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span></span>
<span id="cb5-357"><a href="#cb5-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-358"><a href="#cb5-358" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb5-359"><a href="#cb5-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol),</span>
<span id="cb5-360"><a href="#cb5-360" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol),</span>
<span id="cb5-361"><a href="#cb5-361" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol))</span>
<span id="cb5-362"><a href="#cb5-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-363"><a href="#cb5-363" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_TAG_glycerol) <span class="sc">%&gt;%</span> </span>
<span id="cb5-364"><a href="#cb5-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb5-365"><a href="#cb5-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-366"><a href="#cb5-366" aria-hidden="true" tabindex="-1"></a>TAG_glycerol.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(TAG_glycerol_estimates_axenic, <span class="st">&quot;triglyceride.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher TAG content.&quot;</span>)</span>
<span id="cb5-367"><a href="#cb5-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-368"><a href="#cb5-368" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_TAG_glycerol) <span class="sc">%&gt;%</span> </span>
<span id="cb5-369"><a href="#cb5-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb5-370"><a href="#cb5-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-371"><a href="#cb5-371" aria-hidden="true" tabindex="-1"></a>TAG_glycerol.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(TAG_glycerol_estimates_gnotobiotic, <span class="st">&quot;triglyceride.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.&quot;</span>)</span>
<span id="cb5-372"><a href="#cb5-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-373"><a href="#cb5-373" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb5-374"><a href="#cb5-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-375"><a href="#cb5-375" aria-hidden="true" tabindex="-1"></a>axenic_TAG_glycerol <span class="ot">&lt;-</span> </span>
<span id="cb5-376"><a href="#cb5-376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(TAG_glycerol_microbiome_effect_lines, TAG_glycerol_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-377"><a href="#cb5-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-378"><a href="#cb5-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb5-379"><a href="#cb5-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-380"><a href="#cb5-380" aria-hidden="true" tabindex="-1"></a>gnoto_TAG_glycerol <span class="ot">&lt;-</span></span>
<span id="cb5-381"><a href="#cb5-381" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(TAG_glycerol_microbiome_effect_lines, TAG_glycerol_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb5-382"><a href="#cb5-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-383"><a href="#cb5-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb5-384"><a href="#cb5-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-385"><a href="#cb5-385" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb5-386"><a href="#cb5-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_TAG_glycerol, gnoto_TAG_glycerol) <span class="sc">%&gt;%</span> </span>
<span id="cb5-387"><a href="#cb5-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb5-388"><a href="#cb5-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-389"><a href="#cb5-389" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_TAG_glycerol_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(TAG_glycerol_estimates_diff, <span class="st">&quot;TAG.glycerol.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in combined triglyceride content after glycerol substraction of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher TAG content. Note though that the presence of micriobiota on average decreased TAG content.&quot;</span>)</span>
<span id="cb5-390"><a href="#cb5-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-391"><a href="#cb5-391" aria-hidden="true" tabindex="-1"></a>Dobson_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(weight.anexic_line_means, </span>
<span id="cb5-392"><a href="#cb5-392" aria-hidden="true" tabindex="-1"></a>                       weight.gnotobiotic_line_means, </span>
<span id="cb5-393"><a href="#cb5-393" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_weight_line_means,</span>
<span id="cb5-394"><a href="#cb5-394" aria-hidden="true" tabindex="-1"></a>                       glucose.anexic_line_means, </span>
<span id="cb5-395"><a href="#cb5-395" aria-hidden="true" tabindex="-1"></a>                       glucose.gnotobiotic_line_means, </span>
<span id="cb5-396"><a href="#cb5-396" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_glucose_line_means,</span>
<span id="cb5-397"><a href="#cb5-397" aria-hidden="true" tabindex="-1"></a>                       glycogen.anexic_line_means, </span>
<span id="cb5-398"><a href="#cb5-398" aria-hidden="true" tabindex="-1"></a>                       glycogen.gnotobiotic_line_means, </span>
<span id="cb5-399"><a href="#cb5-399" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_glycogen_line_means,</span>
<span id="cb5-400"><a href="#cb5-400" aria-hidden="true" tabindex="-1"></a>                       protein.anexic_line_means, </span>
<span id="cb5-401"><a href="#cb5-401" aria-hidden="true" tabindex="-1"></a>                       protein.gnotobiotic_line_means, </span>
<span id="cb5-402"><a href="#cb5-402" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_protein_line_means,</span>
<span id="cb5-403"><a href="#cb5-403" aria-hidden="true" tabindex="-1"></a>                       TAG_glycerol.anexic_line_means, </span>
<span id="cb5-404"><a href="#cb5-404" aria-hidden="true" tabindex="-1"></a>                       TAG_glycerol.gnotobiotic_line_means, </span>
<span id="cb5-405"><a href="#cb5-405" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_TAG_glycerol_line_means)</span></code></pre></div>
</div>
<div id="gaertner-et-al-2015" class="section level3">
<h3>Gaertner et al 2015</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>Gaertner_2015_male_mating <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Gaertner_2015_male_mating.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Block, Day, Assay, Chamber, Line, Subject, SUM, orient_female, approach, wing, genital_lick, attempt_cop, copulate, sum_court) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">MMP =</span> SUM) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Block =</span> <span class="fu">as.factor</span>(Block),</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">courtship_events =</span> orient_female <span class="sc">+</span> approach <span class="sc">+</span> wing <span class="sc">+</span> genital_lick <span class="sc">+</span> attempt_cop,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">mating_latency =</span> <span class="dv">31</span> <span class="sc">-</span> copulate, <span class="co"># we subtract from 31 to tell the model that flies did not start the assay mating </span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">censor =</span> <span class="fu">if_else</span>(mating_latency <span class="sc">==</span> <span class="dv">31</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding specific line means</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>Gaertner_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, Trait, guild, description){</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Gaertner et al (2015) G3&quot;</span>,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> guild,</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co"># We will model the 5 courtship processes individually (note that the courtship data is very zero inflated)</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co"># orientation towards females</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>orient_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(orient_female <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> zero_inflated_poisson,</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>                    <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>                    <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/orient.female.model&quot;</span>,</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Gaertner_2015_male_mating<span class="sc">$</span>Line,</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">Block =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Block)</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>fitted_orient <span class="ot">&lt;-</span> <span class="fu">fitted</span>(orient_model,</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> orient_female <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>orient_to_female_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_orient) </span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>orient_to_female_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(orient_to_female_m_estimates, <span class="st">&quot;orient.towards.female.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a><span class="co"># approaching females</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>approach_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(approach <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> zero_inflated_poisson,</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/approach.female.model&quot;</span>,</span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>fitted_approach <span class="ot">&lt;-</span> <span class="fu">fitted</span>(approach_model,</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> approach <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a>approach_female_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_approach) </span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>approach_female_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(approach_female_m_estimates, <span class="st">&quot;approach.female.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a><span class="co"># wing display (singing)</span></span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>wing_display_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(wing <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a>                          <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a>                          <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a>                          <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/wing.display.model&quot;</span>,</span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>                          <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a>fitted_wing_display <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_display_model,</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> wing <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a>wing_display_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_display) </span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a>wing_display_frequency_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(wing_display_m_estimates, <span class="st">&quot;wing.display.frequency.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a><span class="co"># genital licking</span></span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a>genital_licking_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(genital_lick <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a>                             <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a>                             <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a>                             <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/genital.lick.model&quot;</span>,</span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a>                             <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a>fitted_genital_licking <span class="ot">&lt;-</span> <span class="fu">fitted</span>(genital_licking_model,</span>
<span id="cb6-108"><a href="#cb6-108" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb6-109"><a href="#cb6-109" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> genital_lick <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-110"><a href="#cb6-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb6-111"><a href="#cb6-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-112"><a href="#cb6-112" aria-hidden="true" tabindex="-1"></a>genital_licking_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_genital_licking) </span>
<span id="cb6-113"><a href="#cb6-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-114"><a href="#cb6-114" aria-hidden="true" tabindex="-1"></a>genital_licking_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(genital_licking_m_estimates, <span class="st">&quot;genital.licking.frequency.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb6-115"><a href="#cb6-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-116"><a href="#cb6-116" aria-hidden="true" tabindex="-1"></a><span class="co"># attempted copulations</span></span>
<span id="cb6-117"><a href="#cb6-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-118"><a href="#cb6-118" aria-hidden="true" tabindex="-1"></a>attempted_copulations_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(attempt_cop <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-119"><a href="#cb6-119" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb6-120"><a href="#cb6-120" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb6-121"><a href="#cb6-121" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb6-122"><a href="#cb6-122" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb6-123"><a href="#cb6-123" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb6-124"><a href="#cb6-124" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-125"><a href="#cb6-125" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/attempt.cop.model&quot;</span>,</span>
<span id="cb6-126"><a href="#cb6-126" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb6-127"><a href="#cb6-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-128"><a href="#cb6-128" aria-hidden="true" tabindex="-1"></a>fitted_attempted_copulations <span class="ot">&lt;-</span> </span>
<span id="cb6-129"><a href="#cb6-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fitted</span>(attempted_copulations_model,</span>
<span id="cb6-130"><a href="#cb6-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb6-131"><a href="#cb6-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb6-132"><a href="#cb6-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-133"><a href="#cb6-133" aria-hidden="true" tabindex="-1"></a>attempted_copulations_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_attempted_copulations) </span>
<span id="cb6-134"><a href="#cb6-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-135"><a href="#cb6-135" aria-hidden="true" tabindex="-1"></a>attempted_copulations_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(attempted_copulations_m_estimates, <span class="st">&quot;attempted.copulation.frequency.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb6-136"><a href="#cb6-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-137"><a href="#cb6-137" aria-hidden="true" tabindex="-1"></a><span class="co"># Mating latency (like a no choice trial)</span></span>
<span id="cb6-138"><a href="#cb6-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-139"><a href="#cb6-139" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is right censored for males that did not mate</span></span>
<span id="cb6-140"><a href="#cb6-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-141"><a href="#cb6-141" aria-hidden="true" tabindex="-1"></a>mating_latency_model <span class="ot">&lt;-</span></span>
<span id="cb6-142"><a href="#cb6-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(mating_latency <span class="sc">|</span> <span class="fu">cens</span>(censor) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-143"><a href="#cb6-143" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Gaertner_2015_male_mating, </span>
<span id="cb6-144"><a href="#cb6-144" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> <span class="fu">weibull</span>(), <span class="at">inits =</span> <span class="dv">0</span>,</span>
<span id="cb6-145"><a href="#cb6-145" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">class =</span> Intercept), <span class="co"># the intercept in this model is the rate of decline in un-mated males</span></span>
<span id="cb6-146"><a href="#cb6-146" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb6-147"><a href="#cb6-147" aria-hidden="true" tabindex="-1"></a>      <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-148"><a href="#cb6-148" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/mating.latency.model&quot;</span>,</span>
<span id="cb6-149"><a href="#cb6-149" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb6-150"><a href="#cb6-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-151"><a href="#cb6-151" aria-hidden="true" tabindex="-1"></a>fitted_mating_latency <span class="ot">&lt;-</span> <span class="fu">fitted</span>(mating_latency_model,</span>
<span id="cb6-152"><a href="#cb6-152" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb6-153"><a href="#cb6-153" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> mating_latency <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-154"><a href="#cb6-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb6-155"><a href="#cb6-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-156"><a href="#cb6-156" aria-hidden="true" tabindex="-1"></a>mating_latency_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_mating_latency) </span>
<span id="cb6-157"><a href="#cb6-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-158"><a href="#cb6-158" aria-hidden="true" tabindex="-1"></a>mating_latency_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(mating_latency_m_estimates, <span class="st">&quot;mating.latency.2015.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.&quot;</span>)</span>
<span id="cb6-159"><a href="#cb6-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-160"><a href="#cb6-160" aria-hidden="true" tabindex="-1"></a>Gaertner_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(orient_to_female_m_line_means, </span>
<span id="cb6-161"><a href="#cb6-161" aria-hidden="true" tabindex="-1"></a>                         approach_female_m_line_means, </span>
<span id="cb6-162"><a href="#cb6-162" aria-hidden="true" tabindex="-1"></a>                         wing_display_frequency_m_line_means, </span>
<span id="cb6-163"><a href="#cb6-163" aria-hidden="true" tabindex="-1"></a>                         genital_licking_m_line_means, </span>
<span id="cb6-164"><a href="#cb6-164" aria-hidden="true" tabindex="-1"></a>                         attempted_copulations_m_line_means, </span>
<span id="cb6-165"><a href="#cb6-165" aria-hidden="true" tabindex="-1"></a>                         mating_latency_m_line_means)</span></code></pre></div>
</div>
<div id="najarro-et-al-2015" class="section level3">
<h3>Najarro et al 2015</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Line 34061 doesn&#39;t exist on Bloomington or flybase. There is a 34062 though (line 49). Currently removed</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>Najarro_2015_caff_resistance <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Najarro_et_al_2015_caffeine_resistance.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove unidentified line</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(line <span class="sc">!=</span> <span class="dv">34061</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_CaffeineResistance =</span> (CaffeineResistance <span class="sc">-</span> <span class="fu">mean</span>(CaffeineResistance)) <span class="sc">/</span> <span class="fu">sd</span>(CaffeineResistance))</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a> <span class="co"># mean caffeine resistance model</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>caffeine_resistance_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_CaffeineResistance <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Najarro_2015_caff_resistance, <span class="at">family =</span> <span class="st">&quot;Gaussian&quot;</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/caffeine.resistance.model&quot;</span>)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Najarro_2015_caff_resistance<span class="sc">$</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line)</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>fitted_caffeine_resistance <span class="ot">&lt;-</span> <span class="fu">fitted</span>(caffeine_resistance_model,</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance),</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance),</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance))</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>caffeine_resistance_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_caffeine_resistance) <span class="sc">%&gt;%</span> </span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;caffeine.resistance.f&quot;</span>,</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Najarro et al (2015) PLOS Genetics&quot;</span>,</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Drug response&quot;</span>,</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine&quot;</span>,</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="appel-et-al-2016" class="section level3">
<h3>Appel et al 2016</h3>
<p>We need to do some wrangling to fit an appropriate model for these
data. The response is a learning score that is bounded between -100 to
100. It is the percentage of flies that approached a trained target
compared with a control target. Negative values indicate that the flies
preferred the control target; positive values indicate a preference for
the trained target. This data generating process can be approximated
using the beta distribution. However, we need to re-express the data so
that it falls between 0 and 1.</p>
<p>The authors calculate the learning score using the formula <span
class="math inline">\(learning.score =\frac{(Trained.odour -
Control.odour)*100}{Trained.odour + control.odour}\)</span>. A value of
0 represents an even split of flies between both odours. We can convert
this to the proportion scale, where this even split now = 0.5.</p>
<p>We can transform our data using the following formula:</p>
<p><span class="math inline">\(normalised.learning.score =
\frac{learning.score - min(learning.score)}{max(learing.score) -
min(learning.score)}\)</span></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Punishment learning</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>Appel_2016_memory_punishment <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Appel_2016_memory_learning.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Test <span class="sc">==</span> <span class="st">&quot;Punishment&quot;</span>,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>         memory_score <span class="sc">!=</span> <span class="st">&quot;-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group_ID =</span> <span class="fu">as.factor</span>(Group_ID),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">memory_score =</span> <span class="fu">as.numeric</span>(memory_score),</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>         <span class="co"># ok here&#39;s our normalising step. Note that a value of -100 now = 0</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>         <span class="co"># this means that lower values = greater learning i.e. a value of 0.1 means 90% of flies avoided the punishment associated odour</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">norm_mem_score =</span> (memory_score <span class="sc">-</span> <span class="fu">min</span>(memory_score)) <span class="sc">/</span> (<span class="fu">max</span>(memory_score) <span class="sc">-</span> <span class="fu">min</span>(memory_score))</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>                           )</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>punishment_memory_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(norm_mem_score <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Appel_2016_memory_punishment, <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>(),</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/punishment.memory.model&quot;</span>,</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Appel_2016_memory_punishment<span class="sc">$</span>line,</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>fitted_punishment_memory <span class="ot">&lt;-</span> <span class="fu">fitted</span>(punishment_memory_model,</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>punishment_memory_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_punishment_memory) <span class="sc">%&gt;%</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;punishment.memory.learning.f&quot;</span>,</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with &#39;punishment&#39;. Higher values also indicate a higher learning score&quot;</span>,</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>punishment_memory_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_punishment_memory) <span class="sc">%&gt;%</span></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;punishment.memory.learning.m&quot;</span>,</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with &#39;punishment&#39;. Higher values also indicate a greater learning score.&quot;</span>,</span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a><span class="do">## Relief learning</span></span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>Appel_2016_memory_relief <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Appel_2016_memory_learning.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Test <span class="sc">==</span> <span class="st">&quot;Relief&quot;</span>,</span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>         memory_score <span class="sc">!=</span> <span class="st">&quot;-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group_ID =</span> <span class="fu">as.factor</span>(Group_ID),</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>         <span class="at">memory_score =</span> <span class="fu">as.numeric</span>(memory_score),</span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>          <span class="co"># ok here&#39;s our normalising step. Note that a value of -100 now = 0</span></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>         <span class="co"># this means that lower values = greater learning i.e. a value of 0.1 means 90% of flies avoided the punishment associated odour</span></span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>         <span class="at">norm_mem_score =</span> (memory_score <span class="sc">-</span> <span class="fu">min</span>(memory_score)) <span class="sc">/</span> (<span class="fu">max</span>(memory_score) <span class="sc">-</span> <span class="fu">min</span>(memory_score)))</span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a><span class="co"># I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults</span></span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>relief_memory_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(norm_mem_score <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>                           <span class="at">data =</span> Appel_2016_memory_relief, <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>(),</span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/relief.memory.model&quot;</span>,</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>                           <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(relief_memory_model)</span></span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Appel_2016_memory_relief<span class="sc">$</span>line,</span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>fitted_relief_memory <span class="ot">&lt;-</span> <span class="fu">fitted</span>(relief_memory_model,</span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>relief_memory_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_relief_memory) <span class="sc">%&gt;%</span></span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;relief.memory.learning.f&quot;</span>,</span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with &#39;relief&#39;. Higher values also indicate a greater learning score.&quot;</span>,</span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>relief_memory_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_relief_memory) <span class="sc">%&gt;%</span></span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;relief.memory.learning.m&quot;</span>,</span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with &#39;relief&#39;. Higher values also indicate a greater learning score.&quot;</span>,</span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>Appel_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(punishment_memory_f_line_means,</span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>                      punishment_memory_m_line_means,</span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>                      relief_memory_f_line_means,</span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>                      relief_memory_m_line_means)</span></code></pre></div>
</div>
<div id="munoz-munoz-et-al-2016" class="section level3">
<h3>Munoz-Munoz et al 2016</h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>Munoz_2016_wing_size <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/munoz_munoz_2016.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Strain),</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Repl =</span> (<span class="fu">as.factor</span>(Repl)),</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">CS =</span> <span class="fu">as.double</span>(CS),</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_CS =</span> (CS <span class="sc">-</span> <span class="fu">mean</span>(CS)) <span class="sc">/</span> <span class="fu">sd</span>(CS))</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>wing_size_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_CS <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Munoz_2016_wing_size, <span class="at">family =</span> gaussian,</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/wing.size.model&quot;</span>,</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Munoz_2016_wing_size<span class="sc">$</span>line,</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>, <span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>fitted_wing_size <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_size_model,</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Munoz_2016_wing_size<span class="sc">$</span>CS) <span class="sc">+</span> <span class="fu">mean</span>(Munoz_2016_wing_size<span class="sc">$</span>CS),</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Munoz_2016_wing_size<span class="sc">$</span>CS) <span class="sc">+</span> <span class="fu">mean</span>(Munoz_2016_wing_size<span class="sc">$</span>CS),</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Munoz_2016_wing_size<span class="sc">$</span>CS) <span class="sc">+</span> <span class="fu">mean</span>(Munoz_2016_wing_size<span class="sc">$</span>CS))</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>wing_size_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_size) <span class="sc">%&gt;%</span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.size.f&quot;</span>,</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Muñoz-Muñoz et al (2016) Evolution&quot;</span>,</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>wing_size_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_size) <span class="sc">%&gt;%</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.size.m&quot;</span>,</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Muñoz-Muñoz et al (2016) Evolution&quot;</span>,</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>Munoz_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(wing_size_f_line_means,</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>                      wing_size_m_line_means)</span></code></pre></div>
</div>
<div id="freda-2017" class="section level3">
<h3>Freda, 2017</h3>
<p>We have line mean data for larvae, but raw data for adults.
Therefore, we only need to model the adult data here.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>Freda_2017_cold_hardiness <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2017_ThermalHardiness.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex))</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>cold_hardiness_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(Proportion_alive <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">data =</span> Freda_2017_cold_hardiness, <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>(),</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/cold_hardiness.model&quot;</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(cold_hardiness_model)</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Freda_2017_cold_hardiness<span class="sc">$</span>line,</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>fitted_cold_hardiness <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_model,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>cold_hardiness_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cold_hardiness) <span class="sc">%&gt;%</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.f&quot;</span>,</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2017) Integrative and Comparative Biology&quot;</span>,</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.&quot;</span>,</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>cold_hardiness_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cold_hardiness) <span class="sc">%&gt;%</span></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.m&quot;</span>,</span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2017) Integrative and Comparative Biology&quot;</span>,</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.&quot;</span>,</span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>Freda_2017_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cold_hardiness_f_line_means,</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>                      cold_hardiness_m_line_means)</span></code></pre></div>
</div>
<div id="najarro-hackett-and-macdonald-2017" class="section level3">
<h3>Najarro, Hackett and Macdonald, 2017</h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>Najarro_2017_boric_acid <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/najarro_2017.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">lifespan =</span> <span class="st">`</span><span class="at">life-span (hr) on 1.5% boric acid</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_lifespan =</span> (lifespan <span class="sc">-</span> <span class="fu">mean</span>(lifespan)) <span class="sc">/</span> <span class="fu">sd</span>(lifespan))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>boric_acid_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_lifespan <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">data =</span> Najarro_2017_boric_acid, <span class="at">family =</span> <span class="st">&quot;Gaussian&quot;</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/boric.acid.model&quot;</span>)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Najarro_2017_boric_acid<span class="sc">$</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line)</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>fitted_boric_acid <span class="ot">&lt;-</span> <span class="fu">fitted</span>(boric_acid_model,</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan),</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan),</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan))</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>boric_acid_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_boric_acid) <span class="sc">%&gt;%</span> </span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL&quot;</span>),</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="st">&quot;boric.acid.resistance.f&quot;</span>,</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Najarro, Hackett and Macdonald (2017) G3&quot;</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Insecticide response&quot;</span>,</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.&quot;</span>,</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="dean-et-al-2018" class="section level3">
<h3>Dean et al 2018</h3>
<p>To model copulation latency Dean <em>et al</em> log transform the
data and fit a linear model with a gaussian distribution. This model
struggles to converge using <code>brms</code>, so we treat the data as
time to event with right censoring for the flies that failed to mate
within a 2 hour window (coded as 120 in the dataset). To model this data
we fit a survival model, specifying a weibull distribution.</p>
<p>To model insemination capacity Dean <em>et al</em> specify a
quasipoisson distribution family because they treat the data as
underdispersed counts. There is no family like this in
<code>brms</code>. However, we can treat the data as binomial, with each
female treated as a YES/NO for mating success. The posterior predictive
check suggests this is a good fit to the data. Our predictions are also
a good match with those of Dean et al.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>Dean_2018_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dean_2018.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">Rep =</span> <span class="fu">as.factor</span>(Rep),</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">Block =</span> <span class="fu">as.factor</span>(Block),</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">censor =</span> <span class="fu">if_else</span>(LatCop <span class="sc">==</span> <span class="dv">120</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>copulation_latency_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(LatCop <span class="sc">|</span> <span class="fu">cens</span>(censor) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Dean_2018_data, <span class="at">family =</span> <span class="fu">weibull</span>(),</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">class =</span> b),</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/copulation_latency.model&quot;</span>,</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(copulation_latency_model)</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dean_2018_data<span class="sc">$</span>line,</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">Block =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Block)</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>fitted_cop_lat <span class="ot">&lt;-</span> <span class="fu">fitted</span>(copulation_latency_model,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>cop_lat_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cop_lat) <span class="sc">%&gt;%</span></span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;copulation.latency.m&quot;</span>,</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Dean et al (2020) Evolution&quot;</span>,</span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean time (mins) from a male flies introduction to a female until the onset of mating. Lower values indicate shorter copulation latencies and potentially more attractive males&quot;</span>,</span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Dean et al fit a quasipoisson model because they treat the data as underdispersed counts. There is no family like this in brms. However, I think we can treat the data as binomial with each female treated as a YES/NO for mating success. The posterior predictive check suggests this is a good fit to the data.</span></span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>Dean_2018_data <span class="ot">&lt;-</span></span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a>  Dean_2018_data <span class="sc">%&gt;%</span> </span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Cohabiting_females =</span> <span class="dv">10</span>)</span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a>insemination_capacity_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(promiscuity <span class="sc">|</span> <span class="fu">trials</span>(Cohabiting_females) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">data =</span> Dean_2018_data, <span class="at">family =</span> binomial,</span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb12-47"><a href="#cb12-47" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb12-48"><a href="#cb12-48" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb12-49"><a href="#cb12-49" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb12-50"><a href="#cb12-50" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/insemination_capacity.model&quot;</span>,</span>
<span id="cb12-51"><a href="#cb12-51" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb12-52"><a href="#cb12-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-53"><a href="#cb12-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-54"><a href="#cb12-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-55"><a href="#cb12-55" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(insemination_capacity_model)</span></span>
<span id="cb12-56"><a href="#cb12-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-57"><a href="#cb12-57" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb12-58"><a href="#cb12-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dean_2018_data<span class="sc">$</span>line,</span>
<span id="cb12-59"><a href="#cb12-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">Cohabiting_females =</span> <span class="dv">1</span>,</span>
<span id="cb12-60"><a href="#cb12-60" aria-hidden="true" tabindex="-1"></a>         <span class="at">Block =</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-61"><a href="#cb12-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Cohabiting_females, Block)</span>
<span id="cb12-62"><a href="#cb12-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-63"><a href="#cb12-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-64"><a href="#cb12-64" aria-hidden="true" tabindex="-1"></a>fitted_insem_capac <span class="ot">&lt;-</span> <span class="fu">fitted</span>(insemination_capacity_model,</span>
<span id="cb12-65"><a href="#cb12-65" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb12-66"><a href="#cb12-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb12-67"><a href="#cb12-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-68"><a href="#cb12-68" aria-hidden="true" tabindex="-1"></a>insem_capac_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_insem_capac) <span class="sc">%&gt;%</span></span>
<span id="cb12-69"><a href="#cb12-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;insemination.capacity.m&quot;</span>,</span>
<span id="cb12-70"><a href="#cb12-70" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb12-71"><a href="#cb12-71" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Dean et al (2020) Evolution&quot;</span>,</span>
<span id="cb12-72"><a href="#cb12-72" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Life history&quot;</span>,</span>
<span id="cb12-73"><a href="#cb12-73" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of 10 females that a single male successfully inseminated in 48 hours. Insemination was verified by identification of viable pupal progeny. Higher values indicate males with a greater insemination capacity.&quot;</span>,</span>
<span id="cb12-74"><a href="#cb12-74" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb12-75"><a href="#cb12-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb12-76"><a href="#cb12-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb12-77"><a href="#cb12-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-78"><a href="#cb12-78" aria-hidden="true" tabindex="-1"></a>Dean_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cop_lat_line_means,</span>
<span id="cb12-79"><a href="#cb12-79" aria-hidden="true" tabindex="-1"></a>                     insem_capac_line_means)</span></code></pre></div>
</div>
<div id="duneau-et-al-2018" class="section level3">
<h3>Duneau et al 2018</h3>
<p>We fit a slightly different parathion model to that presented in the
manuscript. This is because some lines are resistant to parathion,
leading to heavy sensoring. This results in two groups of lines: those
that are susceptible and those that are resistant. The resistant lines
have estimated mean survival times of 400+ days, which is not realistic
for D. melanogaster. Therefore, I instead model resistance to parathion
using a binomial response, which describes whether the line had
surviving flies 48 hours after exposure to parathion.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Deltamethrin</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>Duneau_2018_deltamethrin_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Duneau_2018/Duneau_2018_deltamethrin.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(DGRP_lines, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">Experiment =</span> <span class="fu">as.factor</span>(Experiment),</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">line =</span> <span class="fu">as.factor</span>(line)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample.size <span class="sc">!=</span> <span class="st">&quot;NA&quot;</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>deltamethrin_model <span class="ot">&lt;-</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Delta_Alive.at<span class="fl">.48</span>h <span class="sc">|</span> <span class="fu">trials</span>(Sample.size) <span class="sc">~</span> <span class="dv">1</span>  <span class="sc">+</span> Experiment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Duneau_2018_deltamethrin_data,</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">iter =</span> <span class="dv">10000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/deltamethrin.model&quot;</span>,</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(deltamethrin_model)</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Duneau_2018_deltamethrin_data<span class="sc">$</span>line,</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Experiment =</span> <span class="st">&quot;Exp 1&quot;</span>,</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sample.size =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Experiment, Sample.size)</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>fitted_deltamethrin <span class="ot">&lt;-</span> <span class="fu">fitted</span>(deltamethrin_model,</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>deltamethrin_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_deltamethrin) <span class="sc">%&gt;%</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;deltamethrin.resistance.m&quot;</span>,</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Duneau et al (2018) G3&quot;</span>,</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Insecticide response&quot;</span>,</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.&quot;</span>,</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Parathion</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a><span class="co"># set up data for right censoring in brms - it is the opposite to what the survival frequentist package uses. We don&#39;t use this but it is useful to know if we do choose to fit a time to event model</span></span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>Duneau_2018_parathion_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Duneau_2018/Duneau_2018_parathion.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(DGRP_lines, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>          <span class="at">Experiment =</span> <span class="fu">as.factor</span>(Experiment)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">Alive.at.48hr =</span> <span class="fu">if_else</span>(Censor <span class="sc">==</span> <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">Censor =</span> <span class="fu">if_else</span>(Censor <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;none&quot;</span>, <span class="st">&quot;right&quot;</span>)) </span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the new binomial model</span></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>parathion_binomial_model <span class="ot">&lt;-</span></span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Alive.at<span class="fl">.48</span>hr <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Experiment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Duneau_2018_parathion_data,</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> bernoulli,</span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/parathion_binomial.model&quot;</span>)</span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(parathion_binomial_model) </span></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Duneau_2018_parathion_data<span class="sc">$</span>line,</span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a>         <span class="at">Experiment =</span> <span class="st">&quot;Experiment_1&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Experiment)</span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a>fitted_parathion_binomial <span class="ot">&lt;-</span> <span class="fu">fitted</span>(parathion_binomial_model,</span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a>parathion_binomial_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_parathion_binomial) <span class="sc">%&gt;%</span></span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;parathion.resistance.m&quot;</span>,</span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Duneau et al (2018) G3&quot;</span>,</span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Insecticide response&quot;</span>,</span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.&quot;</span>,</span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a>Duneau_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(deltamethrin_line_means,</span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a>                       parathion_binomial_line_means)</span></code></pre></div>
</div>
<div id="palmer-et-al-2018" class="section level3">
<h3>Palmer et al 2018</h3>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>Palmer_2018_KV_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Palmer_2018_kallithea.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Injection.date) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">injection_date =</span> <span class="fu">as_factor</span>(<span class="fu">cur_group_id</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">value =</span> <span class="fu">as.numeric</span>(value))</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>Palmer_2018_LT50 <span class="ot">&lt;-</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  Palmer_2018_KV_data <span class="sc">%&gt;%</span> </span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait.column <span class="sc">==</span> <span class="st">&quot;Mortality&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_value =</span> (value <span class="sc">-</span> <span class="fu">mean</span>(value)) <span class="sc">/</span> <span class="fu">sd</span>(value))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co"># LT50 model</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>kv_LT50_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>injection_date),</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Palmer_2018_LT50, <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/kv.LT50.model&quot;</span>,</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.97</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(kv_LT50_model)</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Palmer_2018_LT50<span class="sc">$</span>line,</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>fitted_kv_LT50 <span class="ot">&lt;-</span> <span class="fu">fitted</span>(kv_LT50_model,</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> value <span class="sc">~</span> (Sex<span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_LT50<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_LT50<span class="sc">$</span>value),</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_LT50<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_LT50<span class="sc">$</span>value),</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_LT50<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_LT50<span class="sc">$</span>value))</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>kv_LT50_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_kv_LT50) <span class="sc">%&gt;%</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;kallithea.resistance.LT50.f&quot;</span>,</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Palmer et al (2018) PLOS Pathogens&quot;</span>,</span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.&quot;</span>,</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>kv_LT50_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_kv_LT50) <span class="sc">%&gt;%</span></span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;kallithea.resistance.LT50.m&quot;</span>,</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Palmer et al (2018) PLOS Pathogens&quot;</span>,</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.&quot;</span>,</span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a><span class="co"># titre model</span></span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>Palmer_2018_KV_titre_data <span class="ot">&lt;-</span></span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a>  Palmer_2018_KV_data <span class="sc">%&gt;%</span> </span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait.column <span class="sc">==</span> <span class="st">&quot;titre&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_value =</span> (value <span class="sc">-</span> <span class="fu">mean</span>(value)) <span class="sc">/</span> <span class="fu">sd</span>(value))</span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a>kv_titre_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>injection_date),</span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Palmer_2018_KV_titre_data, <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb14-80"><a href="#cb14-80" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb14-81"><a href="#cb14-81" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb14-82"><a href="#cb14-82" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/kv.titre.model&quot;</span>,</span>
<span id="cb14-83"><a href="#cb14-83" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb14-84"><a href="#cb14-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-85"><a href="#cb14-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-86"><a href="#cb14-86" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb14-87"><a href="#cb14-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Palmer_2018_KV_titre_data<span class="sc">$</span>line,</span>
<span id="cb14-88"><a href="#cb14-88" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-89"><a href="#cb14-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb14-90"><a href="#cb14-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-91"><a href="#cb14-91" aria-hidden="true" tabindex="-1"></a>fitted_kv_titre <span class="ot">&lt;-</span> <span class="fu">fitted</span>(kv_titre_model,</span>
<span id="cb14-92"><a href="#cb14-92" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb14-93"><a href="#cb14-93" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> value <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-94"><a href="#cb14-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-95"><a href="#cb14-95" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb14-96"><a href="#cb14-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value),</span>
<span id="cb14-97"><a href="#cb14-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value),</span>
<span id="cb14-98"><a href="#cb14-98" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value))</span>
<span id="cb14-99"><a href="#cb14-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-100"><a href="#cb14-100" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb14-101"><a href="#cb14-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-102"><a href="#cb14-102" aria-hidden="true" tabindex="-1"></a>kv_titre_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_kv_titre) <span class="sc">%&gt;%</span></span>
<span id="cb14-103"><a href="#cb14-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-104"><a href="#cb14-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;kallithea.viral.load.f&quot;</span>,</span>
<span id="cb14-105"><a href="#cb14-105" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb14-106"><a href="#cb14-106" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Palmer et al (2018) PLOS Pathogens&quot;</span>,</span>
<span id="cb14-107"><a href="#cb14-107" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb14-108"><a href="#cb14-108" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.&quot;</span>,</span>
<span id="cb14-109"><a href="#cb14-109" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb14-110"><a href="#cb14-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb14-111"><a href="#cb14-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb14-112"><a href="#cb14-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-113"><a href="#cb14-113" aria-hidden="true" tabindex="-1"></a>Palmer_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(kv_LT50_f_line_means,</span>
<span id="cb14-114"><a href="#cb14-114" aria-hidden="true" tabindex="-1"></a>                       kv_LT50_m_line_means,</span>
<span id="cb14-115"><a href="#cb14-115" aria-hidden="true" tabindex="-1"></a>                       kv_titre_f_line_means)</span></code></pre></div>
</div>
<div id="teets-and-hahn-2018" class="section level3">
<h3>Teets and Hahn 2018</h3>
<p>Cumulative cold tolerance data was supplied by the authors in line
mean form, so we do not model it here.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>Teets_2018_cold_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Teets_Hahn_2018_data.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="st">`</span><span class="at">Line (DGRP)</span><span class="st">`</span>),</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temp =</span> <span class="fu">as.factor</span>(Temp))</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>cold_survival_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(Live <span class="sc">|</span> <span class="fu">trials</span>(Total) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Temp <span class="sc">+</span> (Temp<span class="sc">|</span>line),</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Teets_2018_cold_data, <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/cold.survival.model&quot;</span>,</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(cold_survival_model)</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Teets_2018_cold_data<span class="sc">$</span>line,</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">Temp =</span> Teets_2018_cold_data<span class="sc">$</span>Temp,</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">Total =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Temp, Total)</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>fitted_cold_survival <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_survival_model,</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>cold_survival_line_means_data <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cold_survival) <span class="sc">%&gt;%</span> </span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding line and temp specific line means</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>Teets_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, value, Trait, description){</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temp <span class="sc">==</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Teets and Hahn (2018) Journal of Evolutionary Biology&quot;</span>,</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span><span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>neg1_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;cold.tolerance.-1.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -1 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>neg2_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-2&quot;</span>, <span class="st">&quot;cold.tolerance.-2.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -2 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>neg3_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-3&quot;</span>, <span class="st">&quot;cold.tolerance.-3.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -3 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>neg4_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-4&quot;</span>, <span class="st">&quot;cold.tolerance.-4.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -4 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>neg5_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-5&quot;</span>, <span class="st">&quot;cold.tolerance.-5.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -5 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>neg6_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-6&quot;</span>, <span class="st">&quot;cold.tolerance.-6.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -6 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>neg7_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-7&quot;</span>, <span class="st">&quot;cold.tolerance.-7.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -7 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>Teets_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(neg1_cold_tolerance_line_means,</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>                      neg2_cold_tolerance_line_means,</span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>                      neg3_cold_tolerance_line_means,</span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>                      neg4_cold_tolerance_line_means,</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>                      neg5_cold_tolerance_line_means,</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>                      neg6_cold_tolerance_line_means,</span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>                      neg7_cold_tolerance_line_means)</span></code></pre></div>
</div>
<div id="everman-et-al-2019" class="section level3">
<h3>Everman et al 2019</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>Everman_2019_starvation_resistance <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Everman_2019_starvation_res.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Lifespan_hrs =</span> <span class="fu">as.numeric</span>(Lifespan_hrs),</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Lifespan_hrs =</span> (Lifespan_hrs <span class="sc">-</span> <span class="fu">mean</span>(Lifespan_hrs)) <span class="sc">/</span> <span class="fu">sd</span>(Lifespan_hrs))</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>starvation_resistance_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Lifespan_hrs <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Everman_2019_starvation_resistance, <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/starvation.res.model&quot;</span>,</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(starvation_resistance_model)</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Everman_2019_starvation_resistance<span class="sc">$</span>line,</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>fitted_starvation_resistance <span class="ot">&lt;-</span> <span class="fu">fitted</span>(starvation_resistance_model,</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs) <span class="sc">+</span> <span class="fu">mean</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs),</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs) <span class="sc">+</span> <span class="fu">mean</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs),</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs) <span class="sc">+</span> <span class="fu">mean</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs))</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>starvation_resistance_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_starvation_resistance) <span class="sc">%&gt;%</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;starvation.resistance.2019.f&quot;</span>,</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Everman et al (2019) Genetics&quot;</span>,</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Stress response&quot;</span>,</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.&quot;</span>,</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>starvation_resistance_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_starvation_resistance) <span class="sc">%&gt;%</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;starvation.resistance.2019.m&quot;</span>,</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Everman et al (2019) Genetics&quot;</span>,</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Stress response&quot;</span>,</span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.&quot;</span>,</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>Everman_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(starvation_resistance_f_line_means,</span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>                        starvation_resistance_m_line_means)</span></code></pre></div>
</div>
<div id="freda-2019" class="section level3">
<h3>Freda 2019</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>Freda_2019a <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Cold18.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline))</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>Freda_2019b <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Cold25.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>Freda_2019c <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Heat18.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline))</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>Freda_2019d <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Heat25.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline))</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 18C for larvae model</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_18_l.model&quot;</span>,</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_18_larvae_model)</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>Freda_2019a_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019a_larvae_data<span class="sc">$</span>line,</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>fitted_ch_18_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_18_larvae_model,</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a><span class="co"># larvae</span></span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_18_l) <span class="sc">%&gt;%</span></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.18C.larvae&quot;</span>,</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 18C for females model</span></span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_18_f.model&quot;</span>,</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_18_f_model)</span></span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a>Freda_2019a_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>  Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019a_larvae_data<span class="sc">$</span>line,</span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>fitted_ch_18_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_18_f_model,</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a><span class="co"># female</span></span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_18_f) <span class="sc">%&gt;%</span></span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.18C.f&quot;</span>,</span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 18C for males model</span></span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_18_m.model&quot;</span>,</span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_18_m_model)</span></span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a>Freda_2019a_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a>  Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-103"><a href="#cb17-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019a_larvae_data<span class="sc">$</span>line,</span>
<span id="cb17-104"><a href="#cb17-104" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a>fitted_ch_18_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_18_m_model,</span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-110"><a href="#cb17-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-111"><a href="#cb17-111" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_18_m) <span class="sc">%&gt;%</span></span>
<span id="cb17-112"><a href="#cb17-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.18C.m&quot;</span>,</span>
<span id="cb17-113"><a href="#cb17-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a><span class="do">#################</span></span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 25C for larvae model</span></span>
<span id="cb17-124"><a href="#cb17-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-125"><a href="#cb17-125" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_25_l.model&quot;</span>,</span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_25_larvae_model)</span></span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a>Freda_2019b_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a>  Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019b_larvae_data<span class="sc">$</span>line,</span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a>fitted_ch_25_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_25_larvae_model,</span>
<span id="cb17-144"><a href="#cb17-144" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-145"><a href="#cb17-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-146"><a href="#cb17-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-147"><a href="#cb17-147" aria-hidden="true" tabindex="-1"></a><span class="co"># larvae</span></span>
<span id="cb17-148"><a href="#cb17-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-149"><a href="#cb17-149" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_l) <span class="sc">%&gt;%</span></span>
<span id="cb17-150"><a href="#cb17-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.25C.larvae&quot;</span>,</span>
<span id="cb17-151"><a href="#cb17-151" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb17-152"><a href="#cb17-152" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-153"><a href="#cb17-153" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-154"><a href="#cb17-154" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb17-155"><a href="#cb17-155" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-156"><a href="#cb17-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-157"><a href="#cb17-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-158"><a href="#cb17-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-159"><a href="#cb17-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-160"><a href="#cb17-160" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 25C for females model</span></span>
<span id="cb17-161"><a href="#cb17-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-162"><a href="#cb17-162" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-163"><a href="#cb17-163" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-164"><a href="#cb17-164" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-165"><a href="#cb17-165" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-166"><a href="#cb17-166" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-167"><a href="#cb17-167" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_25_f.model&quot;</span>,</span>
<span id="cb17-168"><a href="#cb17-168" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-169"><a href="#cb17-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-170"><a href="#cb17-170" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_25_f_model)</span></span>
<span id="cb17-171"><a href="#cb17-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-172"><a href="#cb17-172" aria-hidden="true" tabindex="-1"></a>Freda_2019b_f_data <span class="ot">&lt;-</span></span>
<span id="cb17-173"><a href="#cb17-173" aria-hidden="true" tabindex="-1"></a>  Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-174"><a href="#cb17-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-175"><a href="#cb17-175" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-176"><a href="#cb17-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019b_f_data<span class="sc">$</span>line,</span>
<span id="cb17-177"><a href="#cb17-177" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-178"><a href="#cb17-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb17-179"><a href="#cb17-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-180"><a href="#cb17-180" aria-hidden="true" tabindex="-1"></a>fitted_ch_25_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_25_f_model,</span>
<span id="cb17-181"><a href="#cb17-181" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-182"><a href="#cb17-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-183"><a href="#cb17-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-184"><a href="#cb17-184" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_f) <span class="sc">%&gt;%</span></span>
<span id="cb17-185"><a href="#cb17-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.25C.f&quot;</span>,</span>
<span id="cb17-186"><a href="#cb17-186" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb17-187"><a href="#cb17-187" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-188"><a href="#cb17-188" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-189"><a href="#cb17-189" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb17-190"><a href="#cb17-190" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-191"><a href="#cb17-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-192"><a href="#cb17-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-193"><a href="#cb17-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-194"><a href="#cb17-194" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 25C for males model</span></span>
<span id="cb17-195"><a href="#cb17-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-196"><a href="#cb17-196" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-197"><a href="#cb17-197" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-198"><a href="#cb17-198" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-199"><a href="#cb17-199" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-200"><a href="#cb17-200" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-201"><a href="#cb17-201" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_25_m.model&quot;</span>,</span>
<span id="cb17-202"><a href="#cb17-202" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-203"><a href="#cb17-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-204"><a href="#cb17-204" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_25_m_model)</span></span>
<span id="cb17-205"><a href="#cb17-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-206"><a href="#cb17-206" aria-hidden="true" tabindex="-1"></a>Freda_2019b_m_data <span class="ot">&lt;-</span></span>
<span id="cb17-207"><a href="#cb17-207" aria-hidden="true" tabindex="-1"></a>  Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-208"><a href="#cb17-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-209"><a href="#cb17-209" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-210"><a href="#cb17-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019b_m_data<span class="sc">$</span>line,</span>
<span id="cb17-211"><a href="#cb17-211" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-212"><a href="#cb17-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb17-213"><a href="#cb17-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-214"><a href="#cb17-214" aria-hidden="true" tabindex="-1"></a>fitted_ch_25_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_25_m_model,</span>
<span id="cb17-215"><a href="#cb17-215" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-216"><a href="#cb17-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-217"><a href="#cb17-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-218"><a href="#cb17-218" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_m) <span class="sc">%&gt;%</span></span>
<span id="cb17-219"><a href="#cb17-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.25C.m&quot;</span>,</span>
<span id="cb17-220"><a href="#cb17-220" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb17-221"><a href="#cb17-221" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-222"><a href="#cb17-222" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-223"><a href="#cb17-223" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb17-224"><a href="#cb17-224" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-225"><a href="#cb17-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-226"><a href="#cb17-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-227"><a href="#cb17-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-228"><a href="#cb17-228" aria-hidden="true" tabindex="-1"></a><span class="do">##############</span></span>
<span id="cb17-229"><a href="#cb17-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-230"><a href="#cb17-230" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 18C for larvae model</span></span>
<span id="cb17-231"><a href="#cb17-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-232"><a href="#cb17-232" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-233"><a href="#cb17-233" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-234"><a href="#cb17-234" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-235"><a href="#cb17-235" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-236"><a href="#cb17-236" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-237"><a href="#cb17-237" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_18_l.model&quot;</span>,</span>
<span id="cb17-238"><a href="#cb17-238" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-239"><a href="#cb17-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-240"><a href="#cb17-240" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_18_larvae_model)</span></span>
<span id="cb17-241"><a href="#cb17-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-242"><a href="#cb17-242" aria-hidden="true" tabindex="-1"></a>Freda_2019c_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb17-243"><a href="#cb17-243" aria-hidden="true" tabindex="-1"></a>  Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb17-244"><a href="#cb17-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-245"><a href="#cb17-245" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-246"><a href="#cb17-246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019c_larvae_data<span class="sc">$</span>line,</span>
<span id="cb17-247"><a href="#cb17-247" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-248"><a href="#cb17-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb17-249"><a href="#cb17-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-250"><a href="#cb17-250" aria-hidden="true" tabindex="-1"></a>fitted_hh_18_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_18_larvae_model,</span>
<span id="cb17-251"><a href="#cb17-251" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-252"><a href="#cb17-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-253"><a href="#cb17-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-254"><a href="#cb17-254" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_18_l) <span class="sc">%&gt;%</span></span>
<span id="cb17-255"><a href="#cb17-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.18C.larvae&quot;</span>,</span>
<span id="cb17-256"><a href="#cb17-256" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb17-257"><a href="#cb17-257" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-258"><a href="#cb17-258" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-259"><a href="#cb17-259" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb17-260"><a href="#cb17-260" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-261"><a href="#cb17-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-262"><a href="#cb17-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-263"><a href="#cb17-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-264"><a href="#cb17-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-265"><a href="#cb17-265" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 18C for females model</span></span>
<span id="cb17-266"><a href="#cb17-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-267"><a href="#cb17-267" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-268"><a href="#cb17-268" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-269"><a href="#cb17-269" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-270"><a href="#cb17-270" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-271"><a href="#cb17-271" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-272"><a href="#cb17-272" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_18_f.model&quot;</span>,</span>
<span id="cb17-273"><a href="#cb17-273" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-274"><a href="#cb17-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-275"><a href="#cb17-275" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_18_f_model)</span></span>
<span id="cb17-276"><a href="#cb17-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-277"><a href="#cb17-277" aria-hidden="true" tabindex="-1"></a>Freda_2019c_f_data <span class="ot">&lt;-</span></span>
<span id="cb17-278"><a href="#cb17-278" aria-hidden="true" tabindex="-1"></a>  Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-279"><a href="#cb17-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-280"><a href="#cb17-280" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-281"><a href="#cb17-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019c_f_data<span class="sc">$</span>line,</span>
<span id="cb17-282"><a href="#cb17-282" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-283"><a href="#cb17-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb17-284"><a href="#cb17-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-285"><a href="#cb17-285" aria-hidden="true" tabindex="-1"></a>fitted_hh_18_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_18_f_model,</span>
<span id="cb17-286"><a href="#cb17-286" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-287"><a href="#cb17-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-288"><a href="#cb17-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-289"><a href="#cb17-289" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_18_f) <span class="sc">%&gt;%</span></span>
<span id="cb17-290"><a href="#cb17-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.18C.f&quot;</span>,</span>
<span id="cb17-291"><a href="#cb17-291" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb17-292"><a href="#cb17-292" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-293"><a href="#cb17-293" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-294"><a href="#cb17-294" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb17-295"><a href="#cb17-295" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-296"><a href="#cb17-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-297"><a href="#cb17-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-298"><a href="#cb17-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-299"><a href="#cb17-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 18C for males model</span></span>
<span id="cb17-300"><a href="#cb17-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-301"><a href="#cb17-301" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-302"><a href="#cb17-302" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-303"><a href="#cb17-303" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-304"><a href="#cb17-304" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-305"><a href="#cb17-305" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-306"><a href="#cb17-306" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_18_m.model&quot;</span>,</span>
<span id="cb17-307"><a href="#cb17-307" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-308"><a href="#cb17-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-309"><a href="#cb17-309" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_18_m_model)</span></span>
<span id="cb17-310"><a href="#cb17-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-311"><a href="#cb17-311" aria-hidden="true" tabindex="-1"></a>Freda_2019c_m_data <span class="ot">&lt;-</span></span>
<span id="cb17-312"><a href="#cb17-312" aria-hidden="true" tabindex="-1"></a>  Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-313"><a href="#cb17-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-314"><a href="#cb17-314" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-315"><a href="#cb17-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019c_m_data<span class="sc">$</span>line,</span>
<span id="cb17-316"><a href="#cb17-316" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-317"><a href="#cb17-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb17-318"><a href="#cb17-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-319"><a href="#cb17-319" aria-hidden="true" tabindex="-1"></a>fitted_hh_18_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_18_m_model,</span>
<span id="cb17-320"><a href="#cb17-320" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-321"><a href="#cb17-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-322"><a href="#cb17-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-323"><a href="#cb17-323" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_18_m) <span class="sc">%&gt;%</span></span>
<span id="cb17-324"><a href="#cb17-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.18C.m&quot;</span>,</span>
<span id="cb17-325"><a href="#cb17-325" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb17-326"><a href="#cb17-326" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-327"><a href="#cb17-327" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-328"><a href="#cb17-328" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb17-329"><a href="#cb17-329" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-330"><a href="#cb17-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-331"><a href="#cb17-331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-332"><a href="#cb17-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-333"><a href="#cb17-333" aria-hidden="true" tabindex="-1"></a><span class="do">#################</span></span>
<span id="cb17-334"><a href="#cb17-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-335"><a href="#cb17-335" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 25C for larvae model</span></span>
<span id="cb17-336"><a href="#cb17-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-337"><a href="#cb17-337" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-338"><a href="#cb17-338" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-339"><a href="#cb17-339" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-340"><a href="#cb17-340" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-341"><a href="#cb17-341" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-342"><a href="#cb17-342" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_25_l.model&quot;</span>,</span>
<span id="cb17-343"><a href="#cb17-343" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-344"><a href="#cb17-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-345"><a href="#cb17-345" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_25_larvae_model)</span></span>
<span id="cb17-346"><a href="#cb17-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-347"><a href="#cb17-347" aria-hidden="true" tabindex="-1"></a>Freda_2019d_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb17-348"><a href="#cb17-348" aria-hidden="true" tabindex="-1"></a>  Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb17-349"><a href="#cb17-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-350"><a href="#cb17-350" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-351"><a href="#cb17-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019d_larvae_data<span class="sc">$</span>line,</span>
<span id="cb17-352"><a href="#cb17-352" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-353"><a href="#cb17-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb17-354"><a href="#cb17-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-355"><a href="#cb17-355" aria-hidden="true" tabindex="-1"></a>fitted_hh_25_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_25_larvae_model,</span>
<span id="cb17-356"><a href="#cb17-356" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-357"><a href="#cb17-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-358"><a href="#cb17-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-359"><a href="#cb17-359" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb17-360"><a href="#cb17-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-361"><a href="#cb17-361" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_l) <span class="sc">%&gt;%</span></span>
<span id="cb17-362"><a href="#cb17-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.25C.larvae&quot;</span>,</span>
<span id="cb17-363"><a href="#cb17-363" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb17-364"><a href="#cb17-364" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-365"><a href="#cb17-365" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-366"><a href="#cb17-366" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb17-367"><a href="#cb17-367" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-368"><a href="#cb17-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-369"><a href="#cb17-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-370"><a href="#cb17-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-371"><a href="#cb17-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-372"><a href="#cb17-372" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 25C for females model</span></span>
<span id="cb17-373"><a href="#cb17-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-374"><a href="#cb17-374" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-375"><a href="#cb17-375" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-376"><a href="#cb17-376" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-377"><a href="#cb17-377" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-378"><a href="#cb17-378" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-379"><a href="#cb17-379" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_25_f.model&quot;</span>,</span>
<span id="cb17-380"><a href="#cb17-380" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-381"><a href="#cb17-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-382"><a href="#cb17-382" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_25_f_model)</span></span>
<span id="cb17-383"><a href="#cb17-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-384"><a href="#cb17-384" aria-hidden="true" tabindex="-1"></a>Freda_2019d_f_data <span class="ot">&lt;-</span></span>
<span id="cb17-385"><a href="#cb17-385" aria-hidden="true" tabindex="-1"></a>  Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-386"><a href="#cb17-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-387"><a href="#cb17-387" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-388"><a href="#cb17-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019d_f_data<span class="sc">$</span>line,</span>
<span id="cb17-389"><a href="#cb17-389" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-390"><a href="#cb17-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb17-391"><a href="#cb17-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-392"><a href="#cb17-392" aria-hidden="true" tabindex="-1"></a>fitted_hh_25_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_25_f_model,</span>
<span id="cb17-393"><a href="#cb17-393" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-394"><a href="#cb17-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-395"><a href="#cb17-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-396"><a href="#cb17-396" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_25_f) <span class="sc">%&gt;%</span></span>
<span id="cb17-397"><a href="#cb17-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.25C.f&quot;</span>,</span>
<span id="cb17-398"><a href="#cb17-398" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb17-399"><a href="#cb17-399" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-400"><a href="#cb17-400" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-401"><a href="#cb17-401" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb17-402"><a href="#cb17-402" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-403"><a href="#cb17-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-404"><a href="#cb17-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-405"><a href="#cb17-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-406"><a href="#cb17-406" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 25C for males model</span></span>
<span id="cb17-407"><a href="#cb17-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-408"><a href="#cb17-408" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-409"><a href="#cb17-409" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb17-410"><a href="#cb17-410" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-411"><a href="#cb17-411" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-412"><a href="#cb17-412" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-413"><a href="#cb17-413" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_25_m.model&quot;</span>,</span>
<span id="cb17-414"><a href="#cb17-414" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-415"><a href="#cb17-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-416"><a href="#cb17-416" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_25_m_model)</span></span>
<span id="cb17-417"><a href="#cb17-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-418"><a href="#cb17-418" aria-hidden="true" tabindex="-1"></a>Freda_2019d_m_data <span class="ot">&lt;-</span></span>
<span id="cb17-419"><a href="#cb17-419" aria-hidden="true" tabindex="-1"></a>  Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb17-420"><a href="#cb17-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-421"><a href="#cb17-421" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-422"><a href="#cb17-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019d_m_data<span class="sc">$</span>line,</span>
<span id="cb17-423"><a href="#cb17-423" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-424"><a href="#cb17-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb17-425"><a href="#cb17-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-426"><a href="#cb17-426" aria-hidden="true" tabindex="-1"></a>fitted_hh_25_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_25_m_model,</span>
<span id="cb17-427"><a href="#cb17-427" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-428"><a href="#cb17-428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb17-429"><a href="#cb17-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-430"><a href="#cb17-430" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_25_m) <span class="sc">%&gt;%</span></span>
<span id="cb17-431"><a href="#cb17-431" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.25C.m&quot;</span>,</span>
<span id="cb17-432"><a href="#cb17-432" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb17-433"><a href="#cb17-433" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb17-434"><a href="#cb17-434" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb17-435"><a href="#cb17-435" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb17-436"><a href="#cb17-436" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-437"><a href="#cb17-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-438"><a href="#cb17-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-439"><a href="#cb17-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-440"><a href="#cb17-440" aria-hidden="true" tabindex="-1"></a>Freda_2019_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cold_hardiness_18_l_line_means,</span>
<span id="cb17-441"><a href="#cb17-441" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_18_f_line_means,</span>
<span id="cb17-442"><a href="#cb17-442" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_18_m_line_means,</span>
<span id="cb17-443"><a href="#cb17-443" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_25_l_line_means,</span>
<span id="cb17-444"><a href="#cb17-444" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_25_f_line_means,</span>
<span id="cb17-445"><a href="#cb17-445" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_25_m_line_means,</span>
<span id="cb17-446"><a href="#cb17-446" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_18_l_line_means,</span>
<span id="cb17-447"><a href="#cb17-447" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_18_f_line_means,</span>
<span id="cb17-448"><a href="#cb17-448" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_18_m_line_means,</span>
<span id="cb17-449"><a href="#cb17-449" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_25_l_line_means,</span>
<span id="cb17-450"><a href="#cb17-450" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_25_f_line_means,</span>
<span id="cb17-451"><a href="#cb17-451" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_25_m_line_means)</span></code></pre></div>
</div>
<div id="pitchers-et-al-2019" class="section level3">
<h3>Pitchers et al 2019</h3>
<p>This study measures wing shape and area. They then run a multivariate
GWA. Here, we just model the mean centroid size for each line.</p>
<p>For whatever reason, this is a hard trait to model.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>Pitchers_2019_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Pitchers_2019_wing_multivariate.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Linenum),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Lab =</span> <span class="fu">as.factor</span>(Lab),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Perp =</span> <span class="fu">as.factor</span>(Perp),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">block =</span> <span class="fu">as.factor</span>(block),</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex))</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>Houle_data <span class="ot">&lt;-</span> Pitchers_2019_data <span class="sc">%&gt;%</span> </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Lab <span class="sc">==</span> <span class="st">&quot;Hou&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardise</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_csmm =</span> (csmm <span class="sc">-</span> <span class="fu">mean</span>(csmm)) <span class="sc">/</span> <span class="fu">sd</span>(csmm))</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>Dworkin_data <span class="ot">&lt;-</span> Pitchers_2019_data <span class="sc">%&gt;%</span> </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Lab <span class="sc">==</span> <span class="st">&quot;Dwo&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardise</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_csmm =</span> (csmm <span class="sc">-</span> <span class="fu">mean</span>(csmm)) <span class="sc">/</span> <span class="fu">sd</span>(csmm))</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Houle lab model</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>wing_centroid_size_model_Hou <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_csmm <span class="sc">~</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> gaussian, <span class="at">data =</span> Houle_data,</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/Houle_cs_wing.model&quot;</span>,</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(wing_centroid_size_model_Hou)</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Houle_data<span class="sc">$</span>line,</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> Houle_data<span class="sc">$</span>Sex) <span class="sc">%&gt;%</span> </span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>fitted_wing_houle <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_centroid_size_model_Hou,</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Houle_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Houle_data<span class="sc">$</span>csmm),</span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Houle_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Houle_data<span class="sc">$</span>csmm),</span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Houle_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Houle_data<span class="sc">$</span>csmm))</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>centroid_size_Houle_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_houle) <span class="sc">%&gt;%</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Houle.f&quot;</span>,</span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the &#39;Wingmachine&#39; system. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a>centroid_size_Houle_m_line_means <span class="ot">&lt;-</span> </span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(new_data, fitted_wing_houle) <span class="sc">%&gt;%</span></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Houle.m&quot;</span>,</span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the &#39;Wingmachine&#39; system. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-68"><a href="#cb18-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-69"><a href="#cb18-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Dworkin lab model</span></span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a>wing_centroid_size_model_Dwo_f <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_csmm <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> student, <span class="at">data =</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>),</span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#prior(normal(0, 0.5), class = b),</span></span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">20000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/Dwo_cs_wing.f.model&quot;</span>,</span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-80"><a href="#cb18-80" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(wing_centroid_size_model_Dwo_f)</span></span>
<span id="cb18-81"><a href="#cb18-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-82"><a href="#cb18-82" aria-hidden="true" tabindex="-1"></a>dworkin_f <span class="ot">&lt;-</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>)</span>
<span id="cb18-83"><a href="#cb18-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-84"><a href="#cb18-84" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb18-85"><a href="#cb18-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> dworkin_f<span class="sc">$</span>line,</span>
<span id="cb18-86"><a href="#cb18-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;F&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-87"><a href="#cb18-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb18-88"><a href="#cb18-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-89"><a href="#cb18-89" aria-hidden="true" tabindex="-1"></a>fitted_wing_dworkin_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_centroid_size_model_Dwo_f,</span>
<span id="cb18-90"><a href="#cb18-90" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb18-91"><a href="#cb18-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-92"><a href="#cb18-92" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb18-93"><a href="#cb18-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb18-94"><a href="#cb18-94" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb18-95"><a href="#cb18-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm))</span>
<span id="cb18-96"><a href="#cb18-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-97"><a href="#cb18-97" aria-hidden="true" tabindex="-1"></a>centroid_size_Dworkin_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_dworkin_f) <span class="sc">%&gt;%</span></span>
<span id="cb18-98"><a href="#cb18-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Dworkin.f&quot;</span>,</span>
<span id="cb18-99"><a href="#cb18-99" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb18-100"><a href="#cb18-100" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb18-101"><a href="#cb18-101" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb18-102"><a href="#cb18-102" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb18-103"><a href="#cb18-103" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-104"><a href="#cb18-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-105"><a href="#cb18-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-106"><a href="#cb18-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-107"><a href="#cb18-107" aria-hidden="true" tabindex="-1"></a>wing_centroid_size_model_Dwo_m <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_csmm <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb18-108"><a href="#cb18-108" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> student, <span class="at">data =</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>),</span>
<span id="cb18-109"><a href="#cb18-109" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb18-110"><a href="#cb18-110" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#prior(normal(0, 0.5), class = b),</span></span>
<span id="cb18-111"><a href="#cb18-111" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb18-112"><a href="#cb18-112" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">20000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb18-113"><a href="#cb18-113" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/Dwo_cs_wing.m.model&quot;</span>,</span>
<span id="cb18-114"><a href="#cb18-114" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb18-115"><a href="#cb18-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-116"><a href="#cb18-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-117"><a href="#cb18-117" aria-hidden="true" tabindex="-1"></a>dworkin_m <span class="ot">&lt;-</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>)</span>
<span id="cb18-118"><a href="#cb18-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-119"><a href="#cb18-119" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb18-120"><a href="#cb18-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> dworkin_m<span class="sc">$</span>line,</span>
<span id="cb18-121"><a href="#cb18-121" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;M&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-122"><a href="#cb18-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb18-123"><a href="#cb18-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-124"><a href="#cb18-124" aria-hidden="true" tabindex="-1"></a>fitted_wing_dworkin_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_centroid_size_model_Dwo_m,</span>
<span id="cb18-125"><a href="#cb18-125" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb18-126"><a href="#cb18-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-127"><a href="#cb18-127" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb18-128"><a href="#cb18-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb18-129"><a href="#cb18-129" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb18-130"><a href="#cb18-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm))</span>
<span id="cb18-131"><a href="#cb18-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-132"><a href="#cb18-132" aria-hidden="true" tabindex="-1"></a>centroid_size_Dworkin_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_dworkin_m) <span class="sc">%&gt;%</span></span>
<span id="cb18-133"><a href="#cb18-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Dworkin.m&quot;</span>,</span>
<span id="cb18-134"><a href="#cb18-134" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb18-135"><a href="#cb18-135" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb18-136"><a href="#cb18-136" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb18-137"><a href="#cb18-137" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb18-138"><a href="#cb18-138" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-139"><a href="#cb18-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-140"><a href="#cb18-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-141"><a href="#cb18-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-142"><a href="#cb18-142" aria-hidden="true" tabindex="-1"></a>Pitchers_traits <span class="ot">&lt;-</span> </span>
<span id="cb18-143"><a href="#cb18-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(centroid_size_Houle_f_line_means,</span>
<span id="cb18-144"><a href="#cb18-144" aria-hidden="true" tabindex="-1"></a>        centroid_size_Houle_m_line_means,</span>
<span id="cb18-145"><a href="#cb18-145" aria-hidden="true" tabindex="-1"></a>        centroid_size_Dworkin_f_line_means,</span>
<span id="cb18-146"><a href="#cb18-146" aria-hidden="true" tabindex="-1"></a>        centroid_size_Dworkin_m_line_means)</span></code></pre></div>
</div>
<div id="rohde-et-al-2019" class="section level3">
<h3>Rohde et al 2019</h3>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>Rohde_2019_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Rohde_2019.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(DGRP_id),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">arena_id =</span> <span class="fu">as.factor</span>(arena_id),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> <span class="fu">as.factor</span>(treatment),</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">day_id =</span> <span class="fu">as.factor</span>(day_id),</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">neighbour =</span> <span class="fu">as.factor</span>(neighbour),</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">plate_id =</span> <span class="fu">as.factor</span>(plate_id),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_distance_cm =</span> (distance_cm <span class="sc">-</span> <span class="fu">mean</span>(distance_cm)) <span class="sc">/</span> <span class="fu">sd</span>(distance_cm))</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co"># distance travelled</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>activity_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_distance_cm <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>day_id) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>plate_id) <span class="sc">+</span> (treatment<span class="sc">|</span>line),</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> gaussian, <span class="at">data =</span> Rohde_2019_data,</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/activity.model&quot;</span>,</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(activity_model)</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Rohde_2019_data<span class="sc">$</span>line,</span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> Rohde_2019_data<span class="sc">$</span>treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, treatment)</span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>fitted_activity <span class="ot">&lt;-</span> <span class="fu">fitted</span>(activity_model,</span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> distance_cm <span class="sc">~</span> (treatment<span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Rohde_2019_data<span class="sc">$</span>distance_cm) <span class="sc">+</span> <span class="fu">mean</span>(Rohde_2019_data<span class="sc">$</span>distance_cm),</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Rohde_2019_data<span class="sc">$</span>distance_cm) <span class="sc">+</span> <span class="fu">mean</span>(Rohde_2019_data<span class="sc">$</span>distance_cm),</span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Rohde_2019_data<span class="sc">$</span>distance_cm) <span class="sc">+</span> <span class="fu">mean</span>(Rohde_2019_data<span class="sc">$</span>distance_cm))</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>activity_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_activity) <span class="sc">%&gt;%</span> </span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, treatment, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">treatment =</span> <span class="fu">if_else</span>(treatment <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;MPH&quot;</span>, <span class="st">&quot;SUC&quot;</span>))</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a><span class="co"># first lets get line means for activity in the control sucrose treatment</span></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>activity_line_means <span class="ot">&lt;-</span> activity_line_estimates <span class="sc">%&gt;%</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">&quot;SUC&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;activity.m&quot;</span>,</span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Rohde et al (2019) Genetics&quot;</span>,</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.&quot;</span>,</span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a><span class="co"># now get line means for activity in the mph treatment</span></span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a>activity_mph_line_means <span class="ot">&lt;-</span> activity_line_estimates <span class="sc">%&gt;%</span></span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">&quot;MPH&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;activity.MPH.m&quot;</span>,</span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Rohde et al (2019) Genetics&quot;</span>,</span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.&quot;</span>,</span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a><span class="co"># now lets get means for the effect of mph on movement by getting the difference between treatments</span></span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a>SUC_activity <span class="ot">&lt;-</span> </span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a>  activity_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">SUC_estimate =</span> trait_value)</span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a>mph_activity <span class="ot">&lt;-</span> </span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>  activity_mph_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">mph_estimate =</span> trait_value)</span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a>activity_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(SUC_activity, mph_activity) <span class="sc">%&gt;%</span> </span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> mph_estimate <span class="sc">-</span> SUC_estimate)</span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a>mph_effect_line_means <span class="ot">&lt;-</span> activity_estimates_diff <span class="sc">%&gt;%</span></span>
<span id="cb19-86"><a href="#cb19-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;MPH.effect.on.activity.m&quot;</span>,</span>
<span id="cb19-87"><a href="#cb19-87" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Rohde et al (2019) Genetics&quot;</span>,</span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Drug response&quot;</span>,</span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.&quot;</span>,</span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb19-94"><a href="#cb19-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-95"><a href="#cb19-95" aria-hidden="true" tabindex="-1"></a>Rohde_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(activity_line_means,</span>
<span id="cb19-96"><a href="#cb19-96" aria-hidden="true" tabindex="-1"></a>                      activity_mph_line_means,</span>
<span id="cb19-97"><a href="#cb19-97" aria-hidden="true" tabindex="-1"></a>                      mph_effect_line_means)</span></code></pre></div>
</div>
<div id="everett-et-al-2020" class="section level3">
<h3>Everett et al 2020</h3>
<p>This is mainly data wrangling. We find the mean from two data points
for each line simply by using the <code>mean()</code> function.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>Everett_2020_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Everett_microbiome_2020.csv&quot;</span>) </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>microbiome_line_means <span class="ot">&lt;-</span> </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  Everett_2020_data <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="st">`</span><span class="at">Microbial Species</span><span class="st">`</span>, <span class="at">names_to =</span> <span class="st">&quot;line&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;trait_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(line, <span class="st">&quot;F&quot;</span>), <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>),</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_F1&quot;</span>),</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_F2&quot;</span>),</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_M1&quot;</span>),</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_M2&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Microbial Species</span><span class="st">`</span>, line, Sex) <span class="sc">%&gt;%</span> </span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">trait_value =</span> <span class="fu">mean</span>(trait_value)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>( <span class="at">Reference =</span> <span class="st">&quot;Everett et al (2020) Genome Research&quot;</span>,</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Microbiome&quot;</span>,</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.&quot;</span>,</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">Trait =</span> <span class="fu">str_replace</span>(<span class="st">`</span><span class="at">Microbial Species</span><span class="st">`</span>, <span class="st">&quot; &quot;</span>, <span class="st">&quot;.&quot;</span>),</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">Trait_1 =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(Sex, <span class="st">&quot;Female&quot;</span>), <span class="st">&quot;.f&quot;</span>, <span class="st">&quot;.m&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Trait, Trait, Trait_1, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait, line, Sex)</span></code></pre></div>
</div>
<div id="chapman-et-al-2020" class="section level3">
<h3>Chapman et al 2020</h3>
<p><strong>Note that I haven’t included Wolbachia status in the model
yet, which they do in the paper…</strong></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>Chapman_2020_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Chapman_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">Treatment =</span> <span class="fu">as.factor</span>(Treatment),</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">Date =</span> <span class="fu">as.factor</span>(Date))</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>E.faecalis_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(Alive_at_day5 <span class="sc">|</span> <span class="fu">trials</span>(number_infected) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Infector <span class="sc">+</span> Date <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Vial) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Chapman_2020_data, <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/E.faecalis.model&quot;</span>,</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(E.faecalis_model)</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co"># set the fixed effects in the new data for which we will derive predictions to the reference levels in the model i.e. Infector = jo and Date = `5/3/17`</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Chapman_2020_data<span class="sc">$</span>line,</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">number_infected =</span> <span class="dv">1</span>,</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">Infector =</span> <span class="st">&quot;Jo&quot;</span>,</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Date =</span> <span class="st">&#39;5/3/17&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, number_infected, Infector, Date) </span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>fitted_E.faecalis <span class="ot">&lt;-</span> <span class="fu">fitted</span>(E.faecalis_model,</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> Alive_at_day5 <span class="sc">|</span> <span class="fu">trials</span>(number_infected) <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>E.faecalis_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_E.faecalis) <span class="sc">%&gt;%</span></span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;E.faecalis.resistance.m&quot;</span>,</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Chapman et al (2020) Genes&quot;</span>,</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis&quot;</span>,</span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="zhou-et-al-2020" class="section level3">
<h3>Zhou et al 2020</h3>
<p>Line means are provided for the non-metabolome traits in the
supplementary material. Raw data are provided for the metabolome traits,
which we model below.</p>
<p>There are 453 metabolites. We need a quick and easy way to model
these and find line means. To do this, we fit relatively simple gaussian
models with Sex as a fixed effect (where this is applicable) and line as
a random effect, with a sex-specific intercept.</p>
<p>Five traits show no genetic variation in one of the sexes; that is,
each line has the same value. We do not model these, but still need to
model the trait for the sex that shows genetic variation. We fit a very
similar model, except without the sex fixed effect.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>Zhou_data <span class="ot">&lt;-</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Zhou_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">SUB PATHWAY</span><span class="st">`</span>, <span class="st">`</span><span class="at">METABOLITE  ID</span><span class="st">`</span>, PLATFORM, <span class="st">`</span><span class="at">CHEMICAL ID</span><span class="st">`</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">RETENTION INDEX</span><span class="st">`</span>, MASS, CAS, PUBCHEM, KEGG, <span class="st">`</span><span class="at">CLIENT IDENTIFIER</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">242</span>, <span class="at">names_to =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Trait_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(ID, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;Line&quot;</span>, <span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Rep&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;-&quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line), <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Line) <span class="sc">%&gt;%</span> </span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># standardise</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(METABOLITE, Sex) <span class="sc">%&gt;%</span> </span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">st_trait_value =</span> (Trait_value <span class="sc">-</span> <span class="fu">mean</span>(Trait_value)) <span class="sc">/</span> <span class="fu">sd</span>(Trait_value)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># some traits have the same value for every line; these can&#39;t be standardised because the SD = 0. We remove these traits from the dataset</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(st_trait_value <span class="sc">!=</span> <span class="st">&quot;NaN&quot;</span>)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co"># this is our model outline - pick your favourite metabolite! We can then update this iteratively with new filtered data to model every trait. We do this because we don&#39;t want brms to recompile every time. With the model already compiled we can simply update, which saves a load of time.</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> Zhou_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(METABOLITE <span class="sc">==</span> <span class="st">&quot;xanthurenate&quot;</span>)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>model_outline <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="at">data =</span> data,</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>             st_trait_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>             <span class="at">family =</span> gaussian,</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>             <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, </span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>             <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, </span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span></code></pre></div>
<pre><code>Running /Library/Frameworks/R.framework/Resources/bin/R CMD SHLIB foo.c
clang -mmacosx-version-min=10.13 -I&quot;/Library/Frameworks/R.framework/Resources/include&quot; -DNDEBUG   -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/Rcpp/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/unsupported&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/BH/include&quot; -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/src/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppParallel/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/rstan/include&quot; -DEIGEN_NO_DEBUG  -DBOOST_DISABLE_ASSERTS  -DBOOST_PENDING_INTEGER_LOG2_HPP  -DSTAN_THREADS  -DBOOST_NO_AUTO_PTR  -include &#39;/Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp&#39;  -D_REENTRANT -DRCPP_PARALLEL_USE_TBB=1   -I/usr/local/include   -fPIC  -Wall -g -O2  -c foo.c -o foo.o
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Dense:1:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Core:88:
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:1: error: unknown type name &#39;namespace&#39;
namespace Eigen {
^
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:16: error: expected &#39;;&#39; after top level declarator
namespace Eigen {
               ^
               ;
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Dense:1:
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RcppEigen/include/Eigen/Core:96:10: fatal error: &#39;complex&#39; file not found
#include &lt;complex&gt;
         ^~~~~~~~~
3 errors generated.
make: *** [foo.o] Error 1</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Now make a function to run a model for each trait and get the posterior line means for each sex </span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>metabolite_modeller <span class="ot">&lt;-</span> <span class="cf">function</span>(selected_trait){</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> Zhou_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(METABOLITE <span class="sc">==</span> selected_trait)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">update</span>(model_outline, <span class="at">newdata =</span> data) <span class="co"># same model, but with new data each time</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>fitted_new_data <span class="ot">&lt;-</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> data<span class="sc">$</span>line,</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> data<span class="sc">$</span>Sex,</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> data<span class="sc">$</span>METABOLITE,</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolome&quot;</span>,</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.&quot;</span>,</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zhou et al (2020) Genome Research&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Trait, Sex, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>fitted_metabolite <span class="ot">&lt;-</span> <span class="fu">fitted</span>(model,</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> fitted_new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">trait_value =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value))</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>metabolite_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(fitted_new_data, fitted_metabolite) <span class="sc">%&gt;%</span> </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>),</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, <span class="fu">paste</span>(Trait, <span class="st">&quot;f&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>), <span class="fu">paste</span>(Trait, <span class="st">&quot;m&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>)))</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a><span class="co"># create a list of all trait to iterate our function over</span></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>Both_sexes <span class="ot">&lt;-</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>  Zhou_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(METABOLITE, Sex) <span class="sc">%&gt;%</span></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(METABOLITE) <span class="sc">%&gt;%</span> </span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>trait_list_both_sexes <span class="ot">&lt;-</span> Both_sexes<span class="sc">$</span>METABOLITE</span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Now make a function to run a model for each trait and get the posterior line means for each sex </span></span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>metabolite_modeller_single_sex <span class="ot">&lt;-</span> <span class="cf">function</span>(selected_trait){</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> Zhou_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(METABOLITE <span class="sc">==</span> selected_trait)</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">update</span>(model_outline, <span class="at">newdata =</span> data,</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>                  st_trait_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>                  <span class="at">family =</span> gaussian,</span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>                  <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd))) <span class="co"># same model, but with new data each time</span></span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>  fitted_new_data <span class="ot">&lt;-</span></span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tibble</span>(<span class="at">line =</span> data<span class="sc">$</span>line,</span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">Sex =</span> data<span class="sc">$</span>Sex,</span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> data<span class="sc">$</span>METABOLITE,</span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolome&quot;</span>,</span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.&quot;</span>,</span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a>           <span class="at">Reference =</span> <span class="st">&quot;Zhou et al (2020) Genome Research&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(line, Trait, Sex, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference)</span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a>  fitted_metabolite <span class="ot">&lt;-</span> <span class="fu">fitted</span>(model,</span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a>                              <span class="at">newdata =</span> fitted_new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a>    <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">trait_value =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a>           <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>           <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value))</span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>  metabolite_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(fitted_new_data, fitted_metabolite) <span class="sc">%&gt;%</span> </span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>),</span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, <span class="fu">paste</span>(Trait, <span class="st">&quot;f&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>), <span class="fu">paste</span>(Trait, <span class="st">&quot;m&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>)))</span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a><span class="co"># create trait list to run function over</span></span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a>Single_sex <span class="ot">&lt;-</span></span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a>  Zhou_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(METABOLITE, Sex) <span class="sc">%&gt;%</span></span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(METABOLITE) <span class="sc">%&gt;%</span> </span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a>trait_list_single_sex <span class="ot">&lt;-</span> Single_sex<span class="sc">$</span>METABOLITE</span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the models</span></span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>Run_function <span class="ot">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># Change this to TRUE to run the models</span></span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(Run_function){</span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>  models_both_sexes <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(trait_list_both_sexes, metabolite_modeller) <span class="co"># map_dfr returns a data frame created by row-binding each output</span></span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a>  models_single_sex <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(trait_list_single_sex, metabolite_modeller_single_sex)</span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a>  Zhou_line_means <span class="ot">&lt;-</span></span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbind</span>(models_both_sexes,</span>
<span id="cb24-105"><a href="#cb24-105" aria-hidden="true" tabindex="-1"></a>          models_single_sex)</span>
<span id="cb24-106"><a href="#cb24-106" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-107"><a href="#cb24-107" aria-hidden="true" tabindex="-1"></a>  Zhou_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb24-108"><a href="#cb24-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_csv</span>(<span class="st">&quot;data/derived/Zhou_line_means.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-109"><a href="#cb24-109" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">str_replace_all</span>(Trait, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>),</span>
<span id="cb24-110"><a href="#cb24-110" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> <span class="fu">str_remove</span>(Trait, <span class="st">&quot;[*]&quot;</span>))  <span class="co"># replace invalid names</span></span>
<span id="cb24-111"><a href="#cb24-111" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb24-112"><a href="#cb24-112" aria-hidden="true" tabindex="-1"></a>  Zhou_line_means <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/Zhou_line_means.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-113"><a href="#cb24-113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">str_replace_all</span>(Trait, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>),</span>
<span id="cb24-114"><a href="#cb24-114" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> <span class="fu">str_remove</span>(Trait, <span class="st">&quot;[*]&quot;</span>))  <span class="co"># replace invalid names</span></span>
<span id="cb24-115"><a href="#cb24-115" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="zwoinska-et-al-2020" class="section level3">
<h3>Zwoinska et al 2020</h3>
<p>The authors treat temperature as a continuous variable. But they only
have measures at 3 temperatures, so we choose to consider temperature as
an ordered predictor, following Solomon Kurz’s <a
href="https://bookdown.org/content/4857/monsters-and-mixtures.html#ordered-categorical-predictors">translation</a>
of McElreath (2020) to the <code>brms</code> framework. This approach
results in an improved model fit.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>Zwoinska_male_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Zwoinska_2020/Zwoinska_males_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">line =</span> DGRP) <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="fu">as.factor</span>(Batch),</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> <span class="fu">recode</span>(Temperature,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;25&quot;</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;27&quot;</span> <span class="ot">=</span> <span class="dv">2</span>,</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;29&quot;</span> <span class="ot">=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.integer</span>())</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>fertility_male_model_ordered <span class="ot">&lt;-</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Larvae <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> <span class="fu">mo</span>(Temperature_new) <span class="sc">+</span> (<span class="fu">mo</span>(Temperature_new)<span class="sc">|</span>line),</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Zwoinska_male_data,</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">dirichlet</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">class =</span> simo, <span class="at">coef =</span> moTemperature_new1),</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> bernoulli,</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">4000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/fertility_male.model_ordered&quot;</span>)</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Zwoinska_male_data<span class="sc">$</span>line,</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> Zwoinska_male_data<span class="sc">$</span>Temperature_new,</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Temperature_new, Batch)</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>fitted_male_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_male_model_ordered,</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>fertility_m_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_male_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">.keep =</span> <span class="st">&quot;unused&quot;</span>, </span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">Temperature =</span> <span class="fu">recode</span>(Temperature_new,</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="dv">25</span>,</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;2&quot;</span> <span class="ot">=</span> <span class="dv">27</span>,</span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;3&quot;</span> <span class="ot">=</span> <span class="dv">29</span>)) </span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create the line means tibbles</span></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a>Zwoinska_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, value, Trait, sex, description){</span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span></span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature <span class="sc">==</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> sex,</span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Life history&quot;</span>,</span>
<span id="cb25-52"><a href="#cb25-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb25-53"><a href="#cb25-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb25-54"><a href="#cb25-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb25-55"><a href="#cb25-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb25-56"><a href="#cb25-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-57"><a href="#cb25-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-58"><a href="#cb25-58" aria-hidden="true" tabindex="-1"></a>fertility_25C_m_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_m_line_estimates, <span class="st">&quot;25&quot;</span>, <span class="st">&quot;fertility.25C.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.&quot;</span>)</span>
<span id="cb25-59"><a href="#cb25-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-60"><a href="#cb25-60" aria-hidden="true" tabindex="-1"></a>fertility_27C_m_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_m_line_estimates, <span class="st">&quot;27&quot;</span>, <span class="st">&quot;fertility.27C.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb25-61"><a href="#cb25-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-62"><a href="#cb25-62" aria-hidden="true" tabindex="-1"></a>fertility_29C_m_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_m_line_estimates, <span class="st">&quot;29&quot;</span>, <span class="st">&quot;fertility.29C.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb25-63"><a href="#cb25-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-64"><a href="#cb25-64" aria-hidden="true" tabindex="-1"></a><span class="co"># we need a plasticity measure</span></span>
<span id="cb25-65"><a href="#cb25-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-66"><a href="#cb25-66" aria-hidden="true" tabindex="-1"></a><span class="do">## The code below uses the slope estimated by the model for each line as the plasticity measure.</span></span>
<span id="cb25-67"><a href="#cb25-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-68"><a href="#cb25-68" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). The line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).</span></span>
<span id="cb25-69"><a href="#cb25-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-70"><a href="#cb25-70" aria-hidden="true" tabindex="-1"></a>fertility_m_temp_effect_line_means <span class="ot">&lt;-</span> <span class="fu">as_draws_df</span>(fertility_male_model_ordered) <span class="sc">%&gt;%</span> </span>
<span id="cb25-71"><a href="#cb25-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;r_line&quot;</span>) <span class="sc">&amp;</span> <span class="fu">contains</span>(<span class="st">&quot;moTemperature&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-72"><a href="#cb25-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-73"><a href="#cb25-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(key) <span class="sc">%&gt;%</span></span>
<span id="cb25-74"><a href="#cb25-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the trait in question is plasticity, which is measured by the line specific random slope between treatments - we can get this straight from our posterior samples, without adding anything else </span></span>
<span id="cb25-75"><a href="#cb25-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(value) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-76"><a href="#cb25-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(key, <span class="st">&quot;r_line&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-77"><a href="#cb25-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;,moTemperature_new&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-78"><a href="#cb25-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">[|</span><span class="sc">\\</span><span class="st">]&quot;</span>, <span class="st">&quot;&quot;</span>, line),</span>
<span id="cb25-79"><a href="#cb25-79" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.m&quot;</span>,</span>
<span id="cb25-80"><a href="#cb25-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb25-81"><a href="#cb25-81" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb25-82"><a href="#cb25-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb25-83"><a href="#cb25-83" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.&quot;</span>,</span>
<span id="cb25-84"><a href="#cb25-84" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> mean</span>
<span id="cb25-85"><a href="#cb25-85" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb25-86"><a href="#cb25-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb25-87"><a href="#cb25-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb25-88"><a href="#cb25-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-89"><a href="#cb25-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Slopes are hard to interpret, so we also provide this data in a different format. Below we calculate the change in fertility between 25C and 29C for each line. Greater values indicate a greater loss in fertility. Note this approach produces a very similar line ordering to the bove approach</span></span>
<span id="cb25-90"><a href="#cb25-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-91"><a href="#cb25-91" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb25-92"><a href="#cb25-92" aria-hidden="true" tabindex="-1"></a>  new_data <span class="sc">%&gt;%</span> </span>
<span id="cb25-93"><a href="#cb25-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">paste0</span>(<span class="st">&quot;V&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">372</span>))</span>
<span id="cb25-94"><a href="#cb25-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-95"><a href="#cb25-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-96"><a href="#cb25-96" aria-hidden="true" tabindex="-1"></a>fitted_male_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_male_model_ordered,</span>
<span id="cb25-97"><a href="#cb25-97" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data, <span class="at">summary =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb25-98"><a href="#cb25-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-99"><a href="#cb25-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Estimate&quot;</span>)</span>
<span id="cb25-100"><a href="#cb25-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-101"><a href="#cb25-101" aria-hidden="true" tabindex="-1"></a>fitted_male_fertility <span class="ot">&lt;-</span> <span class="fu">right_join</span>(new_data, fitted_male_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb25-102"><a href="#cb25-102" aria-hidden="true" tabindex="-1"></a>  as_tibble <span class="sc">%&gt;%</span> </span>
<span id="cb25-103"><a href="#cb25-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ID)</span>
<span id="cb25-104"><a href="#cb25-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-105"><a href="#cb25-105" aria-hidden="true" tabindex="-1"></a><span class="co"># Different numbers of lines have been phenotyped for fertility in the 25C and 29C conditions. The code below finds the lines that are measured in one temp but not the other and removes them. We then find the difference in fertility between 25 and 29C conditions</span></span>
<span id="cb25-106"><a href="#cb25-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-107"><a href="#cb25-107" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> </span>
<span id="cb25-108"><a href="#cb25-108" aria-hidden="true" tabindex="-1"></a>  fitted_male_fertility <span class="sc">%&gt;%</span> </span>
<span id="cb25-109"><a href="#cb25-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">1</span>, line <span class="sc">!=</span> <span class="st">&quot;38&quot;</span> <span class="sc">&amp;</span> line <span class="sc">!=</span> <span class="st">&quot;83&quot;</span> <span class="sc">&amp;</span> line <span class="sc">!=</span> <span class="st">&quot;395&quot;</span>, line <span class="sc">!=</span> <span class="st">&quot;707&quot;</span> <span class="sc">&amp;</span> line <span class="sc">!=</span> <span class="st">&quot;727&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-110"><a href="#cb25-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp25 =</span> Estimate)</span>
<span id="cb25-111"><a href="#cb25-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-112"><a href="#cb25-112" aria-hidden="true" tabindex="-1"></a>c <span class="ot">&lt;-</span> </span>
<span id="cb25-113"><a href="#cb25-113" aria-hidden="true" tabindex="-1"></a>  fitted_male_fertility <span class="sc">%&gt;%</span></span>
<span id="cb25-114"><a href="#cb25-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">3</span>, line <span class="sc">!=</span> <span class="st">&quot;911&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-115"><a href="#cb25-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp29 =</span> Estimate) <span class="sc">%&gt;%</span></span>
<span id="cb25-116"><a href="#cb25-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Temperature_new, Batch, line))</span>
<span id="cb25-117"><a href="#cb25-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-118"><a href="#cb25-118" aria-hidden="true" tabindex="-1"></a><span class="co">#a_line &lt;- a %&gt;% distinct(line) </span></span>
<span id="cb25-119"><a href="#cb25-119" aria-hidden="true" tabindex="-1"></a><span class="co">#c_line &lt;- c %&gt;% distinct(line) # we remove the line column in the above code to make binding possible. This line of code will therefore not work, but it has already done its job.</span></span>
<span id="cb25-120"><a href="#cb25-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-121"><a href="#cb25-121" aria-hidden="true" tabindex="-1"></a><span class="co">#anti_join(a_line, c_line) # this line finds the lines present in the a_line vector but not in the b_line vector</span></span>
<span id="cb25-122"><a href="#cb25-122" aria-hidden="true" tabindex="-1"></a><span class="co">#anti_join(c_line, a_line) # this line finds the lines present in the b_line vector but not in the a_line vector</span></span>
<span id="cb25-123"><a href="#cb25-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-124"><a href="#cb25-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-125"><a href="#cb25-125" aria-hidden="true" tabindex="-1"></a>backup_approach <span class="ot">&lt;-</span> <span class="fu">cbind</span>(a, c) <span class="sc">%&gt;%</span> </span>
<span id="cb25-126"><a href="#cb25-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-127"><a href="#cb25-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fertility_diff =</span> Temp25 <span class="sc">-</span> Temp29) <span class="sc">%&gt;%</span> </span>
<span id="cb25-128"><a href="#cb25-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb25-129"><a href="#cb25-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">trait_value =</span> <span class="fu">mean</span>(fertility_diff)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-130"><a href="#cb25-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.m&quot;</span>,</span>
<span id="cb25-131"><a href="#cb25-131" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb25-132"><a href="#cb25-132" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb25-133"><a href="#cb25-133" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb25-134"><a href="#cb25-134" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in male fertility (the proportion of males that produced viable larvae), between 25C and 29C rearing conditions. Higher values indicate a greater decrease in fertility. This is a measure of thermal susceptibility or thermal plasticity in fertility.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-135"><a href="#cb25-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb25-136"><a href="#cb25-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>trait_value)</span>
<span id="cb25-137"><a href="#cb25-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-138"><a href="#cb25-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-139"><a href="#cb25-139" aria-hidden="true" tabindex="-1"></a><span class="co"># female fertility</span></span>
<span id="cb25-140"><a href="#cb25-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-141"><a href="#cb25-141" aria-hidden="true" tabindex="-1"></a>Zwoinska_female_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Zwoinska_2020/Zwoinska_females_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-142"><a href="#cb25-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">line =</span> DGRP) <span class="sc">%&gt;%</span> </span>
<span id="cb25-143"><a href="#cb25-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb25-144"><a href="#cb25-144" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="fu">as.factor</span>(Batch),</span>
<span id="cb25-145"><a href="#cb25-145" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> <span class="fu">recode</span>(Temperature,</span>
<span id="cb25-146"><a href="#cb25-146" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;25&quot;</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb25-147"><a href="#cb25-147" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;27&quot;</span> <span class="ot">=</span> <span class="dv">2</span>,</span>
<span id="cb25-148"><a href="#cb25-148" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;29&quot;</span> <span class="ot">=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-149"><a href="#cb25-149" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.integer</span>())</span>
<span id="cb25-150"><a href="#cb25-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-151"><a href="#cb25-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-152"><a href="#cb25-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-153"><a href="#cb25-153" aria-hidden="true" tabindex="-1"></a>fertility_female_model_ordered <span class="ot">&lt;-</span></span>
<span id="cb25-154"><a href="#cb25-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Larvae <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> <span class="fu">mo</span>(Temperature_new) <span class="sc">+</span> (<span class="fu">mo</span>(Temperature_new)<span class="sc">|</span>line),</span>
<span id="cb25-155"><a href="#cb25-155" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Zwoinska_female_data,</span>
<span id="cb25-156"><a href="#cb25-156" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb25-157"><a href="#cb25-157" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb25-158"><a href="#cb25-158" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">dirichlet</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">class =</span> simo, <span class="at">coef =</span> moTemperature_new1),</span>
<span id="cb25-159"><a href="#cb25-159" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb25-160"><a href="#cb25-160" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> bernoulli,</span>
<span id="cb25-161"><a href="#cb25-161" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">4000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb25-162"><a href="#cb25-162" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb25-163"><a href="#cb25-163" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/fertility_female.model_ordered&quot;</span>)</span>
<span id="cb25-164"><a href="#cb25-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-165"><a href="#cb25-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-166"><a href="#cb25-166" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb25-167"><a href="#cb25-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Zwoinska_female_data<span class="sc">$</span>line,</span>
<span id="cb25-168"><a href="#cb25-168" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> Zwoinska_female_data<span class="sc">$</span>Temperature_new,</span>
<span id="cb25-169"><a href="#cb25-169" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-170"><a href="#cb25-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Temperature_new, Batch)</span>
<span id="cb25-171"><a href="#cb25-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-172"><a href="#cb25-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-173"><a href="#cb25-173" aria-hidden="true" tabindex="-1"></a>fitted_female_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_female_model_ordered,</span>
<span id="cb25-174"><a href="#cb25-174" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb25-175"><a href="#cb25-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb25-176"><a href="#cb25-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-177"><a href="#cb25-177" aria-hidden="true" tabindex="-1"></a>fertility_f_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_female_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb25-178"><a href="#cb25-178" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">.keep =</span> <span class="st">&quot;unused&quot;</span>, </span>
<span id="cb25-179"><a href="#cb25-179" aria-hidden="true" tabindex="-1"></a>            <span class="at">Temperature =</span> <span class="fu">recode</span>(Temperature_new,</span>
<span id="cb25-180"><a href="#cb25-180" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="dv">25</span>,</span>
<span id="cb25-181"><a href="#cb25-181" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;2&quot;</span> <span class="ot">=</span> <span class="dv">27</span>,</span>
<span id="cb25-182"><a href="#cb25-182" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;3&quot;</span> <span class="ot">=</span> <span class="dv">29</span>)) </span>
<span id="cb25-183"><a href="#cb25-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-184"><a href="#cb25-184" aria-hidden="true" tabindex="-1"></a>fertility_25C_f_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_f_line_estimates, <span class="st">&quot;25&quot;</span>, <span class="st">&quot;fertility.25C.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean proportion of females that produced viable larvae. Females were reared at 25C. Higher values indicates greater fertility.&quot;</span>)</span>
<span id="cb25-185"><a href="#cb25-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-186"><a href="#cb25-186" aria-hidden="true" tabindex="-1"></a>fertility_27C_f_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_f_line_estimates, <span class="st">&quot;27&quot;</span>, <span class="st">&quot;fertility.27C.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean proportion of females that produced viable larvae. Females were reared at 27C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb25-187"><a href="#cb25-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-188"><a href="#cb25-188" aria-hidden="true" tabindex="-1"></a>fertility_29C_f_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_f_line_estimates, <span class="st">&quot;29&quot;</span>, <span class="st">&quot;fertility.29C.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean proportion of females that produced viable larvae. Females were reared at 29C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb25-189"><a href="#cb25-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-190"><a href="#cb25-190" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). Once again, the line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).</span></span>
<span id="cb25-191"><a href="#cb25-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-192"><a href="#cb25-192" aria-hidden="true" tabindex="-1"></a><span class="co"># we need a plasticity measure</span></span>
<span id="cb25-193"><a href="#cb25-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-194"><a href="#cb25-194" aria-hidden="true" tabindex="-1"></a><span class="do">## The code below uses the slope estimated by the model for each line as the plasticity measure.</span></span>
<span id="cb25-195"><a href="#cb25-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-196"><a href="#cb25-196" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). The line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).</span></span>
<span id="cb25-197"><a href="#cb25-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-198"><a href="#cb25-198" aria-hidden="true" tabindex="-1"></a>fertility_f_temp_effect_line_means <span class="ot">&lt;-</span> <span class="fu">as_draws_df</span>(fertility_female_model_ordered) <span class="sc">%&gt;%</span> </span>
<span id="cb25-199"><a href="#cb25-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;r_line&quot;</span>) <span class="sc">&amp;</span> <span class="fu">contains</span>(<span class="st">&quot;moTemperature&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-200"><a href="#cb25-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-201"><a href="#cb25-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(key) <span class="sc">%&gt;%</span></span>
<span id="cb25-202"><a href="#cb25-202" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the trait in question is plasticity, which is measured by the line specific random slope between treatments - we can get this straight from our posterior samples, without adding anything else </span></span>
<span id="cb25-203"><a href="#cb25-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(value) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-204"><a href="#cb25-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(key, <span class="st">&quot;r_line&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-205"><a href="#cb25-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;,moTemperature_new&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-206"><a href="#cb25-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">[|</span><span class="sc">\\</span><span class="st">]&quot;</span>, <span class="st">&quot;&quot;</span>, line),</span>
<span id="cb25-207"><a href="#cb25-207" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.f&quot;</span>,</span>
<span id="cb25-208"><a href="#cb25-208" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb25-209"><a href="#cb25-209" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb25-210"><a href="#cb25-210" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb25-211"><a href="#cb25-211" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean slope of the change in female fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.&quot;</span>,</span>
<span id="cb25-212"><a href="#cb25-212" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> mean</span>
<span id="cb25-213"><a href="#cb25-213" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb25-214"><a href="#cb25-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb25-215"><a href="#cb25-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb25-216"><a href="#cb25-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-217"><a href="#cb25-217" aria-hidden="true" tabindex="-1"></a><span class="co"># Slopes are hard to interpret, so we also provide this data in a different format. Below we calculate the change in fertility between 25C and 29C for each line. Greater values indicate a greater loss in fertility. Note this approach produces a very similar line ordering to the bove approach</span></span>
<span id="cb25-218"><a href="#cb25-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-219"><a href="#cb25-219" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb25-220"><a href="#cb25-220" aria-hidden="true" tabindex="-1"></a>  new_data <span class="sc">%&gt;%</span> </span>
<span id="cb25-221"><a href="#cb25-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">paste0</span>(<span class="st">&quot;V&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">120</span>))</span>
<span id="cb25-222"><a href="#cb25-222" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-223"><a href="#cb25-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-224"><a href="#cb25-224" aria-hidden="true" tabindex="-1"></a>fitted_female_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_female_model_ordered,</span>
<span id="cb25-225"><a href="#cb25-225" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data, <span class="at">summary =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb25-226"><a href="#cb25-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-227"><a href="#cb25-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Estimate&quot;</span>)</span>
<span id="cb25-228"><a href="#cb25-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-229"><a href="#cb25-229" aria-hidden="true" tabindex="-1"></a>fitted_female_fertility <span class="ot">&lt;-</span> <span class="fu">right_join</span>(new_data, fitted_female_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb25-230"><a href="#cb25-230" aria-hidden="true" tabindex="-1"></a>  as_tibble <span class="sc">%&gt;%</span> </span>
<span id="cb25-231"><a href="#cb25-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ID)</span>
<span id="cb25-232"><a href="#cb25-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-233"><a href="#cb25-233" aria-hidden="true" tabindex="-1"></a><span class="co"># phenotypes for all 40 lines are measured so no fancy steps for binding are needed</span></span>
<span id="cb25-234"><a href="#cb25-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-235"><a href="#cb25-235" aria-hidden="true" tabindex="-1"></a>a_female <span class="ot">&lt;-</span> </span>
<span id="cb25-236"><a href="#cb25-236" aria-hidden="true" tabindex="-1"></a>  fitted_female_fertility <span class="sc">%&gt;%</span> </span>
<span id="cb25-237"><a href="#cb25-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-238"><a href="#cb25-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp25 =</span> Estimate)</span>
<span id="cb25-239"><a href="#cb25-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-240"><a href="#cb25-240" aria-hidden="true" tabindex="-1"></a>c_female <span class="ot">&lt;-</span> </span>
<span id="cb25-241"><a href="#cb25-241" aria-hidden="true" tabindex="-1"></a>  fitted_female_fertility <span class="sc">%&gt;%</span></span>
<span id="cb25-242"><a href="#cb25-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-243"><a href="#cb25-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp29 =</span> Estimate) <span class="sc">%&gt;%</span></span>
<span id="cb25-244"><a href="#cb25-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Temperature_new, Batch, line))</span>
<span id="cb25-245"><a href="#cb25-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-246"><a href="#cb25-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-247"><a href="#cb25-247" aria-hidden="true" tabindex="-1"></a>backup_approach <span class="ot">&lt;-</span> <span class="fu">cbind</span>(a_female, c_female) <span class="sc">%&gt;%</span> </span>
<span id="cb25-248"><a href="#cb25-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-249"><a href="#cb25-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fertility_diff =</span> Temp25 <span class="sc">-</span> Temp29) <span class="sc">%&gt;%</span> </span>
<span id="cb25-250"><a href="#cb25-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb25-251"><a href="#cb25-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">trait_value =</span> <span class="fu">mean</span>(fertility_diff)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-252"><a href="#cb25-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.f&quot;</span>,</span>
<span id="cb25-253"><a href="#cb25-253" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb25-254"><a href="#cb25-254" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb25-255"><a href="#cb25-255" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb25-256"><a href="#cb25-256" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in female fertility (the proportion of males that produced viable larvae), between 25C and 29C rearing conditions. Higher values indicate a greater decrease in fertility. This is a measure of thermal susceptibility or thermal plasticity in fertility.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-257"><a href="#cb25-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb25-258"><a href="#cb25-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>trait_value)</span>
<span id="cb25-259"><a href="#cb25-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-260"><a href="#cb25-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-261"><a href="#cb25-261" aria-hidden="true" tabindex="-1"></a><span class="co"># combine all the data from this study into a simple dataframe</span></span>
<span id="cb25-262"><a href="#cb25-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-263"><a href="#cb25-263" aria-hidden="true" tabindex="-1"></a>Zwoinska_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(fertility_25C_m_line_means,</span>
<span id="cb25-264"><a href="#cb25-264" aria-hidden="true" tabindex="-1"></a>                         fertility_27C_m_line_means,</span>
<span id="cb25-265"><a href="#cb25-265" aria-hidden="true" tabindex="-1"></a>                         fertility_29C_m_line_means,</span>
<span id="cb25-266"><a href="#cb25-266" aria-hidden="true" tabindex="-1"></a>                         fertility_m_temp_effect_line_means,</span>
<span id="cb25-267"><a href="#cb25-267" aria-hidden="true" tabindex="-1"></a>                         fertility_25C_f_line_means,</span>
<span id="cb25-268"><a href="#cb25-268" aria-hidden="true" tabindex="-1"></a>                         fertility_27C_f_line_means,</span>
<span id="cb25-269"><a href="#cb25-269" aria-hidden="true" tabindex="-1"></a>                         fertility_29C_f_line_means,</span>
<span id="cb25-270"><a href="#cb25-270" aria-hidden="true" tabindex="-1"></a>                         fertility_f_temp_effect_line_means)</span></code></pre></div>
</div>
<div id="chowdhury-et-al-2021" class="section level3">
<h3>Chowdhury et al 2021</h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>Chowdhury_2021_aggressive_lunges <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Chowdhury_2021_aggression.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>aggression_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(agressive.lunges.per<span class="fl">.20</span>mins.m <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Chowdhury_2021_aggressive_lunges, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/aggression.model&quot;</span>,</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(aggression_model)</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Chowdhury_2021_aggressive_lunges<span class="sc">$</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line)</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>fitted_aggression <span class="ot">&lt;-</span> <span class="fu">fitted</span>(aggression_model,</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>aggression_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_aggression) <span class="sc">%&gt;%</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;aggression.2021.m&quot;</span>,</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Chowdhury et al Nature Communications Biology&quot;</span>,</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean number of aggressive lunges made in 20 min window following removal of a divider, by pairs of DGRP males of same genotype. Trials were run after males were kept isolated for five days. Higher values indicate more aggressive males.&quot;</span>,</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="watanabe-and-riddle-2021" class="section level3">
<h3>Watanabe and Riddle 2021</h3>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>Watanabe_climbing_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Watanabe_Riddle_2021.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">line =</span> Line) <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Vial =</span> <span class="fu">as.factor</span>(Vial))</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Females</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>Watanabe_female_data <span class="ot">&lt;-</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  Watanabe_climbing_data <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_Climbing =</span> (Climbing <span class="sc">-</span> <span class="fu">mean</span>(Climbing)) <span class="sc">/</span> <span class="fu">sd</span>(Climbing))</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>climbing_female_model <span class="ot">&lt;-</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(st_Climbing <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span>  Treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Vial) <span class="sc">+</span> (Treatment<span class="sc">|</span>line),</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Watanabe_female_data,</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> gaussian,</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/climbing_female.model&quot;</span>)</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Watanabe_female_data<span class="sc">$</span>line,</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Treatment =</span> Watanabe_female_data<span class="sc">$</span>Treatment,</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Vial =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Treatment, Vial)</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>fitted_female_climbing <span class="ot">&lt;-</span> <span class="fu">fitted</span>(climbing_female_model,</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Watanabe_female_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_female_data<span class="sc">$</span>Climbing),</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_female_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_female_data<span class="sc">$</span>Climbing),</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_female_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_female_data<span class="sc">$</span>Climbing))</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>climbing_f_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_female_climbing)</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create the line means tibbles</span></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>Watanabe_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, value, Trait, sex, description){</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span></span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Treatment <span class="sc">==</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> sex,</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;</span>,</span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span><span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>climbing_f_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_f_line_estimates, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;climbing.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>climbing_post_exercise_f_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_f_line_estimates, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;climbing.post.exercise.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). Once again, the line rankings produced here are very similar to the slopes estimates presented in Watanabe and Riddle (2021).</span></span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a>control_climbing_f <span class="ot">&lt;-</span> </span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a>  climbing_f_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">control_estimate =</span> trait_value)</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a>exercise_climbing_f <span class="ot">&lt;-</span> </span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a>  climbing_post_exercise_f_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">exercise_estimate =</span> trait_value)</span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a>climbing_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(control_climbing_f, exercise_climbing_f) <span class="sc">%&gt;%</span> </span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> exercise_estimate <span class="sc">-</span> control_estimate)</span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a>climbing_f_exercise_effect_line_means <span class="ot">&lt;-</span> climbing_estimates_diff <span class="sc">%&gt;%</span></span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;climbing.exercise.effect.f&quot;</span>,</span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;</span>,</span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.&quot;</span>,</span>
<span id="cb27-86"><a href="#cb27-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a>Watanabe_male_data <span class="ot">&lt;-</span></span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a>  Watanabe_climbing_data <span class="sc">%&gt;%</span> </span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb27-96"><a href="#cb27-96" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_Climbing =</span> (Climbing <span class="sc">-</span> <span class="fu">mean</span>(Climbing)) <span class="sc">/</span> <span class="fu">sd</span>(Climbing))</span>
<span id="cb27-97"><a href="#cb27-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-98"><a href="#cb27-98" aria-hidden="true" tabindex="-1"></a>climbing_male_model <span class="ot">&lt;-</span></span>
<span id="cb27-99"><a href="#cb27-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(st_Climbing <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span>  Treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Vial) <span class="sc">+</span> (Treatment<span class="sc">|</span>line),</span>
<span id="cb27-100"><a href="#cb27-100" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Watanabe_male_data,</span>
<span id="cb27-101"><a href="#cb27-101" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb27-102"><a href="#cb27-102" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb27-103"><a href="#cb27-103" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb27-104"><a href="#cb27-104" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> gaussian,</span>
<span id="cb27-105"><a href="#cb27-105" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb27-106"><a href="#cb27-106" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.99</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb27-107"><a href="#cb27-107" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/climbing_male.model&quot;</span>)</span>
<span id="cb27-108"><a href="#cb27-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-109"><a href="#cb27-109" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(climbing_male_model)</span></span>
<span id="cb27-110"><a href="#cb27-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-111"><a href="#cb27-111" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb27-112"><a href="#cb27-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Watanabe_male_data<span class="sc">$</span>line,</span>
<span id="cb27-113"><a href="#cb27-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">Treatment =</span> Watanabe_male_data<span class="sc">$</span>Treatment,</span>
<span id="cb27-114"><a href="#cb27-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Vial =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-115"><a href="#cb27-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Treatment, Vial)</span>
<span id="cb27-116"><a href="#cb27-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-117"><a href="#cb27-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-118"><a href="#cb27-118" aria-hidden="true" tabindex="-1"></a>fitted_male_climbing <span class="ot">&lt;-</span> <span class="fu">fitted</span>(climbing_male_model,</span>
<span id="cb27-119"><a href="#cb27-119" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb27-120"><a href="#cb27-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb27-121"><a href="#cb27-121" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb27-122"><a href="#cb27-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Watanabe_male_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_male_data<span class="sc">$</span>Climbing),</span>
<span id="cb27-123"><a href="#cb27-123" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_male_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_male_data<span class="sc">$</span>Climbing),</span>
<span id="cb27-124"><a href="#cb27-124" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_male_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_male_data<span class="sc">$</span>Climbing))</span>
<span id="cb27-125"><a href="#cb27-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-126"><a href="#cb27-126" aria-hidden="true" tabindex="-1"></a>climbing_m_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_male_climbing)</span>
<span id="cb27-127"><a href="#cb27-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-128"><a href="#cb27-128" aria-hidden="true" tabindex="-1"></a>climbing_m_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_m_line_estimates, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;climbing.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb27-129"><a href="#cb27-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-130"><a href="#cb27-130" aria-hidden="true" tabindex="-1"></a>climbing_post_exercise_m_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_m_line_estimates, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;climbing.post.exercise.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb27-131"><a href="#cb27-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-132"><a href="#cb27-132" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb27-133"><a href="#cb27-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-134"><a href="#cb27-134" aria-hidden="true" tabindex="-1"></a>control_climbing_m <span class="ot">&lt;-</span> </span>
<span id="cb27-135"><a href="#cb27-135" aria-hidden="true" tabindex="-1"></a>  climbing_m_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb27-136"><a href="#cb27-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb27-137"><a href="#cb27-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">control_estimate =</span> trait_value)</span>
<span id="cb27-138"><a href="#cb27-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-139"><a href="#cb27-139" aria-hidden="true" tabindex="-1"></a>exercise_climbing_m <span class="ot">&lt;-</span> </span>
<span id="cb27-140"><a href="#cb27-140" aria-hidden="true" tabindex="-1"></a>  climbing_post_exercise_m_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb27-141"><a href="#cb27-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb27-142"><a href="#cb27-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">exercise_estimate =</span> trait_value)</span>
<span id="cb27-143"><a href="#cb27-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-144"><a href="#cb27-144" aria-hidden="true" tabindex="-1"></a>climbing_estimates_diff_m <span class="ot">&lt;-</span></span>
<span id="cb27-145"><a href="#cb27-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(control_climbing_m, exercise_climbing_m) <span class="sc">%&gt;%</span> </span>
<span id="cb27-146"><a href="#cb27-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> exercise_estimate <span class="sc">-</span> control_estimate)</span>
<span id="cb27-147"><a href="#cb27-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-148"><a href="#cb27-148" aria-hidden="true" tabindex="-1"></a>climbing_m_exercise_effect_line_means <span class="ot">&lt;-</span> climbing_estimates_diff_m <span class="sc">%&gt;%</span></span>
<span id="cb27-149"><a href="#cb27-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;climbing.exercise.effect.m&quot;</span>,</span>
<span id="cb27-150"><a href="#cb27-150" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb27-151"><a href="#cb27-151" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;</span>,</span>
<span id="cb27-152"><a href="#cb27-152" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb27-153"><a href="#cb27-153" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.&quot;</span>,</span>
<span id="cb27-154"><a href="#cb27-154" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb27-155"><a href="#cb27-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb27-156"><a href="#cb27-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb27-157"><a href="#cb27-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-158"><a href="#cb27-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-159"><a href="#cb27-159" aria-hidden="true" tabindex="-1"></a>Watanabe_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(climbing_f_line_means,</span>
<span id="cb27-160"><a href="#cb27-160" aria-hidden="true" tabindex="-1"></a>                         climbing_f_exercise_effect_line_means,</span>
<span id="cb27-161"><a href="#cb27-161" aria-hidden="true" tabindex="-1"></a>                         climbing_post_exercise_f_line_means,</span>
<span id="cb27-162"><a href="#cb27-162" aria-hidden="true" tabindex="-1"></a>                         climbing_m_line_means,</span>
<span id="cb27-163"><a href="#cb27-163" aria-hidden="true" tabindex="-1"></a>                         climbing_m_exercise_effect_line_means,</span>
<span id="cb27-164"><a href="#cb27-164" aria-hidden="true" tabindex="-1"></a>                         climbing_post_exercise_m_line_means)</span></code></pre></div>
<p><strong>Combine all modelled trait means into a single
tibble</strong></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>modelled_trait_means <span class="ot">&lt;-</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(Jumbo_traits,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>        Grubbs_traits,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>        Dobson_traits,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>        Gaertner_traits,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>        caffeine_resistance_line_means,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>        Appel_traits,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>        Munoz_traits,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>        Freda_2017_traits,</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>        boric_acid_line_means,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>        Dean_traits,</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>        Duneau_traits,</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>        Palmer_traits,</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>        Teets_traits,</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>        Everman_traits,</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>        Freda_2019_traits,</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>        Pitchers_traits,</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>        Rohde_traits,</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>        microbiome_line_means,</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>        E.faecalis_line_means,</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>        Zhou_line_means,</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>        Zwoinska_traits,</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>        aggression_m_line_means,</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>        Watanabe_traits)</span></code></pre></div>
<p><strong>Combine all data and save</strong></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the all.dgrp.phenos_combined object as a csv file</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;data/derived/all.dgrp.phenos_combined.csv&quot;</span>)){</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  all.dgrp.phenos_combined <span class="ot">&lt;-</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbind</span>(all.line.mean.phenos, </span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>          modelled_trait_means) <span class="sc">%&gt;%</span> </span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">recode</span>(Sex, <span class="at">Both =</span> <span class="st">&quot;Pooled&quot;</span>))</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv</span>(all.dgrp.phenos_combined, <span class="at">file =</span> <span class="st">&quot;data/derived/all.dgrp.phenos_combined.csv&quot;</span>)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> all.dgrp.phenos_combined <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/all.dgrp.phenos_combined.csv&quot;</span>)</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="cleaning-the-data" class="section level2">
<h2>Cleaning the data</h2>
<p><span class="math inline">\(~\)</span></p>
<div id="removing-unusual-lines" class="section level3">
<h3>Removing unusual lines</h3>
<p><span class="math inline">\(~\)</span></p>
<p>We have phenotype line means for 238 lines. As of the start of 2018,
there are only 205 DGRP lines. There appear to be some lines that have
been used in very few studies, suggesting they have been mistakenly
labelled at some point.</p>
<p>After checking Bloomington (<a
href="https://bdsc.indiana.edu/Home/Search"
class="uri">https://bdsc.indiana.edu/Home/Search</a>) and flybase (<a
href="http://flybase.org/reports/FBlc0000504.html"
class="uri">http://flybase.org/reports/FBlc0000504.html</a>) we found
the following anomalies.</p>
<ul>
<li><p>Chapman et al (2020): lines 273, 331, 501, 568, 709, 831, 846 -
these lines are not phenotyped in any other study and are not documented
on flybase.</p></li>
<li><p>Palmer et al (2018): line 575 - not on flybase</p></li>
<li><p>Chowdhury et al (2019): line 471 - not on flybase (only aberrant
line from study)</p></li>
<li><p>Houston (2019): line 285 - not on flybase</p></li>
<li><p>Watanabe ad Riddle (2021): line 854 - not on flybase</p></li>
<li><p>Battlay et al (2016): lines 226, 847 and 917 - there is no
obvious reason for mysterious lines. No obvious typos and the paper
states they GWA for 178 lines, which matches the number reported in File
S2, which is where we sourced the data.</p></li>
<li><p>Katzenberger et al (2015): line 521 - no obvious typo (data is
ordered by increasing trait value). One-off issue. Not on
flybase.</p></li>
<li><p>Turner, Miller and Cochrane (2013): line 760 - no obvious typo.
One off. Not on flybase.</p></li>
<li><p>Montgomery et al (2014): lines 424 and 766 - Errors appear to be
mislabelled lines. Table S3 includes RAL lines 424 and 766. Line 766 is
out of numeric order in the Table. Most likely to actually be line 776
or 786. No way to tell.</p></li>
<li><p>He et al (2014): Bloomington stock number 28120 - Error appears
to be a typo. Line 28120 is included in their raw data, but isn’t listed
in Table S2 which shows the lines used in the study. One-off
issue.</p></li>
<li><p>Lafluente, Duneau and Beldade (2018): line 809 - Likely to be a
typo or a labelling error. One-off issue.</p></li>
</ul>
<p>Remove these lines from the dataset:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check how many traits each line has been phenotyped for</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>line_test <span class="ot">&lt;-</span> all.dgrp.phenos_combined <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_traits =</span> <span class="fu">length</span>(Trait)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(n_traits)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co"># check specific lines</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co">#print(all.dgrp.phenos_combined %&gt;% </span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a> <span class="co"># filter(line == &quot;556&quot;), n = 31)</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the aberrant lines</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>all.dgrp.phenos_combined <span class="ot">&lt;-</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  all.dgrp.phenos_combined <span class="sc">%&gt;%</span> </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(line <span class="sc">!=</span> <span class="st">&quot;226&quot;</span>,</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;273&quot;</span>,</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;285&quot;</span>,</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;331&quot;</span>,</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;424&quot;</span>,</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;471&quot;</span>,</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;501&quot;</span>,</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;521&quot;</span>,</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;568&quot;</span>,</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;575&quot;</span>,</span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;709&quot;</span>,</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;760&quot;</span>,</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;766&quot;</span>,</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;809&quot;</span>,</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;831&quot;</span>,</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;846&quot;</span>,</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;847&quot;</span>,</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;854&quot;</span>,</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>         line <span class="sc">!=</span> <span class="st">&quot;917&quot;</span></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a> <span class="co">#all.dgrp.phenos_combined %&gt;% distinct(line)</span></span></code></pre></div>
<p>This still leaves 219 lines, which considering most studies report
the DGRP to consist of 205 lines, is unexpected.</p>
<p>However, the DGRP record on flybase (<a
href="http://flybase.org/reports/FBlc0000504.html"
class="uri">http://flybase.org/reports/FBlc0000504.html</a>) indicates
that a total of 219 lines have been included in the DGRP at some point
since its creation. We have phenotype measurements for all 219 lines,
and this can be explained by some lines being withdrawn from the
Bloomington stock centre. Lines were withdrawn because they were very
sick, sequence data was not available, or they were essentially
duplicates (that is, they exhibit high genomic relatedness with another
one of the lines) of other lines.</p>
<p>The lines that have been withdrawn but are included in our dataset
are: 80, 272, 343, 378, 387, 393, 398, 476, 514, 554, 556, 591, 750,
771. We choose <strong>not to remove</strong> these lines from our
dataset, as many of the studies we include in our dataset make use of
these lines.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="removing-data-duplicates" class="section level3">
<h3>Removing data duplicates</h3>
<p><span class="math inline">\(~\)</span></p>
<p>Some studies report data that has been previously used, leading to
perfect or near perfect correlations.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>all.dgrp.phenos_combined_subset <span class="ot">&lt;-</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  all.dgrp.phenos_combined <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Trait <span class="sc">!=</span> <span class="st">&quot;rapid.cold.hardening.25C&quot;</span>)</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="removing-traits-measured-in-heterozyogtes"
class="section level3">
<h3>Removing traits measured in heterozyogtes</h3>
<p><span class="math inline">\(~\)</span></p>
<p>Some traits have been measured in heterozygotes. This generally
occurs when the goal of a study is to observe whether genetic variation
affects the response to a genetic disease encoded by a particular
variant or the expression of a transgenic construct. These studies are
not ideal for our database, as the phenotypes they observe may be
affected be dominance and epistatic interactions that are absent in the
nearly completely homozygous DGRP lines. We remove these studies from
our dataset.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove binary traits and those measured in heterozyogtes</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>all.dgrp.phenos_combined_subset <span class="ot">&lt;-</span> all.dgrp.phenos_combined_subset <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>( Trait <span class="sc">!=</span> <span class="st">&quot;recombination.rate.3R.f&quot;</span>, </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;recombination.rate.X.f&quot;</span>, </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.diabetes.susceptibility.2014a.m&quot;</span>, </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.diabetes.susceptibility.2014a.f&quot;</span>,</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.diabetes.susceptibility.2014b.m&quot;</span>,</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.Drop.susceptibility.2014a.f&quot;</span>, </span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.Drop.susceptibility.2014a.m&quot;</span>, </span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.Lobe.susceptibility.2014a.f&quot;</span>, </span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;eye.area.Lobe.susceptibility.2014a.m&quot;</span>, </span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;notum.bristles.diabetes.susceptibility.2014a.f&quot;</span>, </span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;notum.bristles.diabetes.susceptibility.2014a.m&quot;</span>, </span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;CRISPR.full.resistance.embryo.formation.f&quot;</span>, </span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;CRISPR.germline.conversion.rate.f&quot;</span>, </span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;CRISPR.mosaic.resistance.embryo.formation.f&quot;</span>, </span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;CRISPR.wildtype.rate.m&quot;</span>, </span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;paternal.induction.hybrid.dysgenesis.m&quot;</span>, </span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;Canton S(B).olfactory.response.benzaldehyde.f&quot;</span>, </span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;neur.olfactory.response.benzaldehyde.f&quot;</span>, </span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>         Trait <span class="sc">!=</span> <span class="st">&quot;Sema5c.olfactory.response.benzaldehyde.f&quot;</span>)</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="removing-lines-that-have-multiple-trait-values"
class="section level3">
<h3>Removing lines that have multiple trait values</h3>
<p>For some traits, a small number of lines have &gt;1 trait value. We
fix the dataset up before it enters R, but document it here for
transparency.</p>
<ul>
<li><p>Unckless, Rottschaefer and Lazzaro (2015) G3, include line 287
twice in their provided data. Values are identical, so we simply remove
the duplication.</p></li>
<li><p>Chow, Wolfner and Clark (2013) PNAS, include line 287 twice in
their provided data (coincidence or not that it’s line 287 again?). They
measure two traits: a hazard ratio for tunicamycin-induced survival and
a LT50 value. Line 287 has two different hazard ratios, but the same
LT50 value. We therefore remove line 287 completely from the hazard
ratio dataset, but only remove the duplicate in the LT50
dataset.</p></li>
<li><p>Hotson and Schneider (2015) G3, include line 712 twice in their
provided data, for the trait <code>L.monocytogenes.load.m</code>. The
duplicates provide different values. We are unable to tell which is the
true value, so we remove line 712 from the dataset.</p></li>
<li><p>Adebambo et al (2020) Frontiers in Genetics, include the line 229
twice. Values are identical so we simply remove the duplication. It is
unclear whether this is a scribing error on our side or in the provided
data.</p></li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the code below reports lines that have &gt; 1 measurement for any of the traits in our dataset </span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># If there are 0 rows in the tibble, it means we&#39;ve fixed them all**</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>problems <span class="ot">&lt;-</span> all.dgrp.phenos_combined_subset <span class="sc">%&gt;%</span> </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(line, Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>n) <span class="sc">%&gt;%</span> </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="save-and-show-the-cleaned-dataset" class="section level3">
<h3>Save and show the cleaned dataset</h3>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># now save our cleaned dataset</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(all.dgrp.phenos_combined_subset, <span class="at">file =</span> <span class="st">&quot;data/derived/all.dgrp.phenos_unscaled.csv&quot;</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="co"># how many traits did we start with: all.dgrp.phenos_combined %&gt;% distinct(Trait)</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="co">#after cleaning: all.dgrp.phenos_combined_subset %&gt;% distinct(Trait)</span></span></code></pre></div>
<p>All up we have removed 18 traits, taking our total from 2133 to
2115</p>
<p>The full list of traits we document can be viewed here:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>lines_measured <span class="ot">&lt;-</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  all.dgrp.phenos_combined_subset <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="st">`</span><span class="at"># lines measured</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>table <span class="ot">&lt;-</span> all.dgrp.phenos_combined_subset <span class="sc">%&gt;%</span> </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(trait_value))</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>full_table <span class="ot">&lt;-</span> <span class="fu">left_join</span>(lines_measured, table) <span class="sc">%&gt;%</span> </span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait)</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to build HTML searchable tables</span></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>my_data_table <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">datatable</span>(</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>    df, <span class="at">rownames=</span><span class="cn">FALSE</span>,</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">autoHideNavigation =</span> <span class="cn">TRUE</span>,</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">extensions =</span> <span class="fu">c</span>(<span class="st">&quot;Scroller&quot;</span>,  <span class="st">&quot;Buttons&quot;</span>),</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">autoWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">dom =</span> <span class="st">&#39;Bfrtip&#39;</span>,</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">deferRender=</span><span class="cn">TRUE</span>,</span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">scrollX=</span><span class="cn">TRUE</span>, <span class="at">scrollY=</span><span class="dv">1000</span>,</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">scrollCollapse=</span><span class="cn">TRUE</span>,</span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">buttons =</span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="st">&#39;pageLength&#39;</span>, <span class="st">&#39;colvis&#39;</span>, <span class="st">&#39;csv&#39;</span>, <span class="fu">list</span>(</span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">extend =</span> <span class="st">&#39;pdf&#39;</span>,</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>          <span class="at">pageSize =</span> <span class="st">&#39;A4&#39;</span>,</span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>          <span class="at">orientation =</span> <span class="st">&#39;landscape&#39;</span>,</span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>          <span class="at">filename =</span> <span class="st">&#39;Trait_table&#39;</span>)),</span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">pageLength =</span> <span class="dv">1214</span></span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a><span class="fu">my_data_table</span>(full_table)</span></code></pre></div>
<div id="htmlwidget-85fbb532f8c593ef22f2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-85fbb532f8c593ef22f2">{"x":{"filter":"none","vertical":false,"extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["1_3-Methylhistidine.high.yeast.f","1_3-Methylhistidine.low.yeast.f","1-(1-enyl-palmitoyl)-2-linoleoyl-GPC (P-16:0_18:2).m","1-(1-enyl-palmitoyl)-2-linoleoyl-GPE (P-16:0_18:2).f","1-(1-enyl-palmitoyl)-2-linoleoyl-GPE (P-16:0_18:2).m","1-(1-enyl-palmitoyl)-2-oleoyl-GPC (P-16:0_18:1).f","1-(1-enyl-palmitoyl)-2-oleoyl-GPC (P-16:0_18:1).m","1-(1-enyl-palmitoyl)-2-oleoyl-GPE (P-16:0_18:1).f","1-(1-enyl-palmitoyl)-2-oleoyl-GPE (P-16:0_18:1).m","1-(1-enyl-palmitoyl)-2-palmitoleoyl-GPC (P-16:0_16:1).m","1-(1-enyl-palmitoyl)-GPE (P-16:0).f","1-(1-enyl-palmitoyl)-GPE (P-16:0).m","1-(1-enyl-stearoyl)-2-linoleoyl-GPE (P-18:0_18:2).f","1-(1-enyl-stearoyl)-2-linoleoyl-GPE (P-18:0_18:2).m","1-(1-enyl-stearoyl)-2-oleoyl-GPE (P-18:0_18:1).f","1-(1-enyl-stearoyl)-2-oleoyl-GPE (P-18:0_18:1).m","1-(1-enyl-stearoyl)-GPE (P-18:0).f","1-(1-enyl-stearoyl)-GPE (P-18:0).m","1-kestose.f","1-kestose.m","1-linolenoyl-GPC (18:3).f","1-linolenoyl-GPC (18:3).m","1-linoleoyl-2-linolenoyl-galactosylglycerol (18:2_18:3).f","1-linoleoyl-2-linolenoyl-galactosylglycerol (18:2_18:3).m","1-linoleoyl-2-linolenoyl-GPC (18:2_18:3).f","1-linoleoyl-2-linolenoyl-GPC (18:2_18:3).m","1-linoleoyl-GPA (18:2).f","1-linoleoyl-GPA (18:2).m","1-linoleoyl-GPC (18:2).f","1-linoleoyl-GPC (18:2).m","1-linoleoyl-GPE (18:2).f","1-linoleoyl-GPE (18:2).m","1-linoleoyl-GPG (18:2).f","1-linoleoyl-GPG (18:2).m","1-linoleoyl-GPI (18:2).f","1-linoleoyl-GPI (18:2).m","1-linoleoyl-GPS (18:2).f","1-linoleoyl-GPS (18:2).m","1-linoleoylglycerol (18:2).f","1-linoleoylglycerol (18:2).m","1-methyladenine.f","1-methyladenine.m","1-Methyladenosine.high.yeast.f","1-Methyladenosine.low.yeast.f","1-Methylguanosine.high.yeast.f","1-Methylguanosine.low.yeast.f","1-methylhistamine.f","1-methylhistamine.m","1-methylhistidine.f","1-methylhistidine.m","1-methylimidazoleacetate.f","1-methylimidazoleacetate.m","1-methylurate.f","1-methylurate.m","1-oleoyl-2-linoleoyl-glycerol (18:1_18:2).f","1-oleoyl-2-linoleoyl-glycerol (18:1_18:2).m","1-oleoyl-2-linoleoyl-GPC (18:1_18:2).f","1-oleoyl-2-linoleoyl-GPC (18:1_18:2).m","1-oleoyl-2-linoleoyl-GPE (18:1_18:2).f","1-oleoyl-2-linoleoyl-GPE (18:1_18:2).m","1-oleoyl-2-linoleoyl-GPI (18:1_18:2).f","1-oleoyl-2-linoleoyl-GPI (18:1_18:2).m","1-oleoyl-3-linoleoyl-glycerol (18:1_18:2).f","1-oleoyl-3-linoleoyl-glycerol (18:1_18:2).m","1-oleoyl-GPA (18:1).f","1-oleoyl-GPA (18:1).m","1-oleoyl-GPC (18:1).f","1-oleoyl-GPC (18:1).m","1-oleoyl-GPE (18:1).f","1-oleoyl-GPE (18:1).m","1-oleoyl-GPG (18:1).f","1-oleoyl-GPG (18:1).m","1-oleoyl-GPI (18:1).f","1-oleoyl-GPI (18:1).m","1-oleoyl-GPS (18:1).f","1-oleoyl-GPS (18:1).m","1-palmitoleoyl-2-linolenoyl-GPC (16:1_18:3).f","1-palmitoleoyl-2-linolenoyl-GPC (16:1_18:3).m","1-palmitoleoyl-2-linoleoyl-GPC (16:1_18:2).f","1-palmitoleoyl-2-linoleoyl-GPC (16:1_18:2).m","1-palmitoleoyl-2-oleoyl-glycerol (16:1_18:1).f","1-palmitoleoyl-2-oleoyl-glycerol (16:1_18:1).m","1-palmitoleoyl-3-oleoyl-glycerol (16:1_18:1).f","1-palmitoleoyl-3-oleoyl-glycerol (16:1_18:1).m","1-palmitoleoyl-GPC (16:1).f","1-palmitoleoyl-GPC (16:1).m","1-palmitoyl-2-linolenoyl-galactosylglycerol (16:0_18:3).f","1-palmitoyl-2-linolenoyl-galactosylglycerol (16:0_18:3).m","1-palmitoyl-2-linoleoyl-galactosylglycerol (16:0_18:2).f","1-palmitoyl-2-linoleoyl-galactosylglycerol (16:0_18:2).m","1-palmitoyl-2-linoleoyl-glycerol (16:0_18:2).f","1-palmitoyl-2-linoleoyl-glycerol (16:0_18:2).m","1-palmitoyl-2-linoleoyl-GPC (16:0_18:2).f","1-palmitoyl-2-linoleoyl-GPC (16:0_18:2).m","1-palmitoyl-2-linoleoyl-GPE (16:0_18:2).f","1-palmitoyl-2-linoleoyl-GPE (16:0_18:2).m","1-palmitoyl-2-linoleoyl-GPI (16:0_18:2).f","1-palmitoyl-2-linoleoyl-GPI (16:0_18:2).m","1-palmitoyl-2-oleoyl-GPC (16:0_18:1).f","1-palmitoyl-2-oleoyl-GPC (16:0_18:1).m","1-palmitoyl-2-oleoyl-GPE (16:0_18:1).f","1-palmitoyl-2-oleoyl-GPE (16:0_18:1).m","1-palmitoyl-2-palmitoleoyl-GPC (16:0_16:1).f","1-palmitoyl-2-palmitoleoyl-GPC (16:0_16:1).m","1-palmitoyl-2-stearoyl-GPC (16:0_18:0).f","1-palmitoyl-2-stearoyl-GPC (16:0_18:0).m","1-palmitoyl-3-linoleoyl-glycerol (16:0_18:2).f","1-palmitoyl-3-linoleoyl-glycerol (16:0_18:2).m","1-palmitoyl-GPA (16:0).f","1-palmitoyl-GPA (16:0).m","1-palmitoyl-GPC (16:0).f","1-palmitoyl-GPC (16:0).m","1-palmitoyl-GPE (16:0).f","1-palmitoyl-GPE (16:0).m","1-palmitoyl-GPG (16:0).f","1-palmitoyl-GPG (16:0).m","1-palmitoyl-GPI (16:0).f","1-palmitoyl-GPI (16:0).m","1-stearoyl-2-linoleoyl-GPC (18:0_18:2).f","1-stearoyl-2-linoleoyl-GPC (18:0_18:2).m","1-stearoyl-2-linoleoyl-GPE (18:0_18:2).f","1-stearoyl-2-linoleoyl-GPE (18:0_18:2).m","1-stearoyl-2-oleoyl-GPC (18:0_18:1).f","1-stearoyl-2-oleoyl-GPC (18:0_18:1).m","1-stearoyl-2-oleoyl-GPE (18:0_18:1).f","1-stearoyl-2-oleoyl-GPE (18:0_18:1).m","1-stearoyl-GPC (18:0).f","1-stearoyl-GPC (18:0).m","1-stearoyl-GPE (18:0).f","1-stearoyl-GPE (18:0).m","1-stearoyl-GPI (18:0).f","1,2-dilinoleoyl-galactosylglycerol (18:2_18:2).f","1,2-dilinoleoyl-galactosylglycerol (18:2_18:2).m","1,2-dilinoleoyl-GPC (18:2_18:2).f","1,2-dilinoleoyl-GPC (18:2_18:2).m","1,2-dilinoleoyl-GPE (18:2_18:2).f","1,2-dilinoleoyl-GPE (18:2_18:2).m","1,2-dioleoyl-GPC (18:1_18:1).f","1,2-dioleoyl-GPC (18:1_18:1).m","1,2-dioleoyl-GPE (18:1_18:1).f","1,2-dioleoyl-GPE (18:1_18:1).m","1,2-dipalmitoleoyl-GPC (16:1_16:1).f","1,2-dipalmitoleoyl-GPC (16:1_16:1).m","1,2-dipalmitoyl-GPC (16:0_16:0).f","1,2-dipalmitoyl-GPC (16:0_16:0).m","10-heptadecenoate (17:1n7).f","10-heptadecenoate (17:1n7).m","10-nonadecenoate (19:1n9).f","10-nonadecenoate (19:1n9).m","12,13-DiHOME.f","12,13-DiHOME.m","13-HODE + 9-HODE.f","13-HODE + 9-HODE.m","13-HODE.high.yeast.f","13-HODE.low.yeast.f","13-methylmyristate.f","13-methylmyristate.m","15-methylpalmitate.f","15-methylpalmitate.m","16-hydroxypalmitate.f","16-hydroxypalmitate.m","17-methylstearate.f","17-methylstearate.m","2-aminooctanoate.f","2-aminooctanoate.m","2-hydroxy-3-methylvalerate.f","2-hydroxy-3-methylvalerate.m","2-hydroxybutyrate_2-hydroxyisobutyrate.f","2-hydroxybutyrate_2-hydroxyisobutyrate.m","2-hydroxyglutarate.f","2-Hydroxyglutarate.high.yeast.f","2-Hydroxyglutarate.low.yeast.f","2-hydroxyglutarate.m","2-isopropylmalate.f","2-isopropylmalate.m","2-keto-3-deoxy-gluconate.f","2-keto-3-deoxy-gluconate.m","2-ketogluconate.f","2-ketogluconate.m","2-linoleoylglycerol (18:2).f","2-linoleoylglycerol (18:2).m","2-methylcitrate.f","2-methylcitrate.m","2-methylserine.f","2-methylserine.m","2-palmitoleoyl-GPC (16:1).f","2-palmitoleoyl-GPC (16:1).m","2-palmitoleoylglycerol (16:1).f","2-palmitoleoylglycerol (16:1).m","2-palmitoyl-GPC (16:0).f","2-palmitoyl-GPC (16:0).m","2-phosphoglycerate.f","2-phosphoglycerate.m","2-Phosphoglyceric.acid.high.yeast.f","2-Phosphoglyceric.acid.low.yeast.f","2-stearoyl-GPE (18:0).f","2-stearoyl-GPE (18:0).m","2,3-dihydroxyisovalerate.f","2,3-dihydroxyisovalerate.m","2.3-butanedione.olfactory.attraction.f","2.3-butanedione.olfactory.attraction.m","2.3-butanedione.olfactory.latency.30.percentile.f","2.3-butanedione.olfactory.latency.30.percentile.m","2.3-butanedione.olfactory.latency.90.percentile.f","2.3-butanedione.olfactory.latency.90.percentile.m","2.3-butanedione.olfactory.response.intensity.f","2.3-butanedione.olfactory.response.intensity.m","2.3-butanedione.olfactory.strength.attraction.f","2.3-butanedione.olfactory.strength.attraction.m","2'-deoxycytidine.f","2'-deoxycytidine.m","2'-deoxyguanosine.f","2'-deoxyguanosine.m","2'-deoxyinosine.f","2'-deoxyinosine.m","2'-deoxyuridine.f","2'-deoxyuridine.m","2'-O-methylguanosine.f","2'-O-methylguanosine.m","3-(4-hydroxyphenyl)lactate.f","3-(4-hydroxyphenyl)lactate.m","3-aminoisobutyrate.f","3-aminoisobutyrate.m","3-hydroxy-3-methylglutarate.f","3-hydroxy-3-methylglutarate.m","3-hydroxybutyrate (BHBA).f","3-hydroxybutyrate (BHBA).m","3-Hydroxybutyric.acid.high.yeast.f","3-Hydroxybutyric.acid.low.yeast.f","3-hydroxybutyrylcarnitine (2).f","3-hydroxybutyrylcarnitine (2).m","3-hydroxydecanoate.f","3-hydroxydecanoate.m","3-hydroxyisobutyrate.f","3-hydroxyisobutyrate.m","3-hydroxykynurenine.f","3-Hydroxykynurenine.high.yeast.f","3-Hydroxykynurenine.low.yeast.f","3-hydroxykynurenine.m","3-hydroxylaurate.f","3-hydroxylaurate.m","3-hydroxymyristate.f","3-hydroxymyristate.m","3-hydroxyoctanoate.f","3-hydroxyoctanoate.m","3-hydroxypalmitate.f","3-hydroxypalmitate.m","3-hydroxypropanoate.f","3-hydroxypropanoate.m","3-hydroxysebacate.f","3-hydroxysebacate.m","3-hydroxysuberate.f","3-hydroxysuberate.m","3-ketosphinganine.f","3-ketosphinganine.m","3-methyl-2-oxobutyrate.f","3-methyl-2-oxobutyrate.m","3-methyl-2-oxovalerate.f","3-methyl-2-oxovalerate.m","3-methylcytidine.f","3-methylcytidine.m","3-methylhistidine.f","3-methylhistidine.m","3-phosphoglycerate.f","3-phosphoglycerate.m","3-ureidopropionate.f","3-ureidopropionate.m","3,4-dihydroxyphenylacetate.f","3,4-dihydroxyphenylacetate.m","4-guanidinobutanoate.f","4-guanidinobutanoate.m","4-hydroxybenzoate.f","4-hydroxybenzoate.m","4-hydroxybutyrate (GHB).f","4-hydroxybutyrate (GHB).m","4-hydroxyhippurate.f","4-hydroxyhippurate.m","4-hydroxyphenylpyruvate.f","4-hydroxyphenylpyruvate.m","4-imidazoleacetate.f","4-imidazoleacetate.m","4-methyl-2-oxopentanoate.f","4-methyl-2-oxopentanoate.m","4-Pyridoxic.acid.high.yeast.f","4-Pyridoxic.acid.low.yeast.f","5-aminolevulinate.f","5-aminolevulinate.m","5-Aminovalerate.high.yeast.f","5-Aminovalerate.low.yeast.f","5-dodecenoate (12:1n7).f","5-dodecenoate (12:1n7).m","5-methylthioadenosine (MTA).f","5-methylthioadenosine (MTA).m","5-methyluridine (ribothymidine).f","5-methyluridine (ribothymidine).m","5-oxoproline.f","5-oxoproline.m","6-phosphogluconate.f","6-phosphogluconate.m","7-methylguanine.f","7-methylguanine.m","7-methylguanosine.f","7-methylguanosine.m","9,10-DiHOME.f","9,10-DiHOME.m","A.tropicalis.load.m","abdomen.curling.same.sex.m","abdomen.length.18C.f","abdomen.length.28C.f","abdomen.length.thermal.plasticity.f","acclimated.acute.cold.exposure.resistance.18C.f","acclimated.acute.cold.exposure.resistance.18C.m","acclimated.acute.cold.exposure.resistance.25C.f","acclimated.acute.cold.exposure.resistance.25C.m","acetoacetate.high.yeast.f","acetoacetate.low.yeast.f","Acetobacter.aceti.f","Acetobacter.aceti.m","Acetobacter.cibinongensis.f","Acetobacter.cibinongensis.m","Acetobacter.ghanensis.f","Acetobacter.ghanensis.m","Acetobacter.indonesiensis.f","Acetobacter.indonesiensis.m","Acetobacter.malorum.f","Acetobacter.malorum.m","Acetobacter.nitrogenifigens.f","Acetobacter.nitrogenifigens.m","Acetobacter.okinawensis.f","Acetobacter.okinawensis.m","Acetobacter.orientalis.f","Acetobacter.orientalis.m","Acetobacter.papayae.f","Acetobacter.papayae.m","Acetobacter.pasteurianus.f","Acetobacter.pasteurianus.m","Acetobacter.persici.f","Acetobacter.persici.m","Acetobacter.pomorum.f","Acetobacter.pomorum.m","Acetobacter.senegalensis.f","Acetobacter.senegalensis.m","Acetobacter.syzygii.f","Acetobacter.syzygii.m","Acetobacter.tropicalis.f","Acetobacter.tropicalis.m","acetylcarnitine.f","acetylcarnitine.high.yeast.f","acetylcarnitine.low.yeast.f","acetylcarnitine.m","acetylcholine.high.yeast.f","acetylcholine.low.yeast.f","acetylphosphate.f","acetylphosphate.m","Acidiphilium.angustum.f","Acidiphilium.angustum.m","Acidomonas.methanolica.f","Acidomonas.methanolica.m","Acidovorax.sp..f","Acidovorax.sp..m","Acidovorax.temperans.f","Acidovorax.temperans.m","Acinetobacter.gyllenbergii.f","Acinetobacter.gyllenbergii.m","Acinetobacter.johnsonii.f","Acinetobacter.johnsonii.m","Acinetobacter.sp..f","Acinetobacter.sp..m","aconitate [cis or trans].f","aconitate [cis or trans].m","activity.m","activity.MPH.m","activity.same.sex.m","acute.cold.exposure.resistance.18C.-6.f","acute.cold.exposure.resistance.18C.-6.m","acute.cold.exposure.resistance.18C.-8.f","acute.cold.exposure.resistance.18C.-8.m","acute.cold.exposure.resistance.25C.f","acute.cold.exposure.resistance.25C.m","adenine.f","adenine.high.yeast.f","adenine.low.yeast.f","adenine.m","adenosine 3'-monophosphate (3'-AMP).f","adenosine 3'-monophosphate (3'-AMP).m","adenosine 5'-monophosphate (AMP).f","adenosine 5'-monophosphate (AMP).m","adenosine-2',3'-cyclic monophosphate.f","adenosine-2',3'-cyclic monophosphate.m","adenosine.f","adenosine.high.yeast.f","adenosine.low.yeast.f","adenosine.m","adipate.f","adipate.m","ADP.high.yeast.f","ADP.low.yeast.f","aggression.2015.m","aggression.2021.m","agmatine.high.yeast.f","agmatine.low.yeast.f","Agrobacterium.tumefaciens.f","Agrobacterium.tumefaciens.m","alanine.f","alanine.high.yeast.f","alanine.low.yeast.f","alanine.m","alanylleucine.f","alanylleucine.m","allantoin.f","allantoin.high.yeast.f","allantoin.low.yeast.f","allantoin.m","alpha-amanitin.resistance.0.2","alpha-amanitin.resistance.2","alpha-amanitin.resistance.LC50","alpha-hydroxyisocaproate.f","alpha-hydroxyisocaproate.m","alpha-hydroxyisovalerate.f","alpha-hydroxyisovalerate.m","alpha-ketoglutarate.f","alpha-ketoglutarate.m","alpha-Ketoglutaric.acid.high.yeast.f","alpha-Ketoglutaric.acid.low.yeast.f","alpha-tocopherol.f","alpha-tocopherol.m","alpha-tocotrienol.f","alpha-tocotrienol.m","amino.acid.pref.post.limitation.f","amino.acid.pref.post.limitation.m","aminoisobutyrate.high.yeast.f","aminoisobutyrate.low.yeast.f","AMP.high.yeast.f","AMP.low.yeast.f","anoxia.tolerance.adult.f","anoxia.tolerance.adult.m","anoxia.tolerance.larval","anthranilate.high.yeast.f","anthranilate.low.yeast.f","approach.female.m","arabitol_xylitol.f","arabitol_xylitol.m","arabonate_xylonate.f","arabonate_xylonate.m","arachidate (20:0).f","arachidate (20:0).m","arachidonate.high.yeast.f","arachidonate.low.yeast.f","arginine.f","arginine.high.yeast.f","arginine.low.yeast.f","arginine.m","argininosuccinate.f","argininosuccinate.m","Asaia.prunellae.f","Asaia.prunellae.m","Asaia.sp..f","Asaia.sp..m","asparagine.f","asparagine.high.yeast.f","asparagine.low.yeast.f","asparagine.m","aspartate.f","aspartate.m","aspartic.acid.high.yeast.f","aspartic.acid.low.yeast.f","Aspergillus.nidulans.f","Aspergillus.nidulans.m","Aspergillus.terreus,Malassezia globosa.f","Aspergillus.terreus,Malassezia globosa.m","attempted.copulation.frequency.m","azelate (nonanedioate).f","azelate (nonanedioate).m","azinphos.methyl.resistance.0.25","azinphos.methyl.resistance.0.5","azinphos.methyl.resistance.1","azinphos.methyl.resistance.2","azinphos.methyl.resistance.LD50","Azospirillum.brasilense.f","Azospirillum.brasilense.m","Azospirillum.thiophilum.f","Azospirillum.thiophilum.m","basal.activity.f","basal.activity.m","benzoate.f","benzoate.m","beta-alanine.f","beta-alanine.m","beta-hydroxyisovalerate.f","beta-hydroxyisovalerate.m","beta-sitosterol.f","beta-sitosterol.m","betaine.f","betaine.high.yeast.f","betaine.low.yeast.f","betaine.m","biliverdin.f","biliverdin.m","biopterin.f","biopterin.m","biotin.f","biotin.high.yeast.f","biotin.low.yeast.f","biotin.m","boric.acid.resistance.f","Borrelia.coriaceae.f","Borrelia.coriaceae.m","cadaverine.high.yeast.f","cadaverine.low.yeast.f","caffeine.resistance.f","campesterol.f","campesterol.m","Campylobacter.mucosalis.f","Campylobacter.mucosalis.m","caprate (10:0).f","caprate (10:0).m","carboxyethyl-GABA.f","carboxyethyl-GABA.m","Cardinium.endosymbiont.f","Cardinium.endosymbiont.m","carnitine.f","carnitine.high.yeast.f","carnitine.low.yeast.f","carnitine.m","Caulobacter.henricii.f","Caulobacter.henricii.m","cGMP.high.yeast.f","cGMP.low.yeast.f","Chaetomium.globosum.f","Chaetomium.globosum.m","change.cocaine.preferenceA.f","change.cocaine.preferenceA.m","change.cocaine.preferenceB.f","change.cocaine.preferenceB.m","change.methamphetamine.preferenceA.f","change.methamphetamine.preferenceA.m","change.methamphetamine.preferenceB.f","change.methamphetamine.preferenceB.m","CHC.11 &amp; 9-Me-C23.f","CHC.13- &amp; 11-Me-C25.f","CHC.2-Me-C22.f","CHC.2-Me-C22.m","CHC.2-Me-C24.f","CHC.2-Me-C24.m","CHC.2-Me-C25.f","CHC.2-Me-C25.m","CHC.2-Me-C26.m","CHC.2-Me-C27.m","CHC.2-Me-C28 &amp; 7,11-C29:2.f","CHC.2-Me-C28.m","CHC.2-Me-C29 &amp; 8,12-C30:2.f","CHC.2-Me-C29.m","CHC.2-Me-C30.f","CHC.3-Me-C23.m","CHC.3-Me-C25 &amp; 9-C26:1.f","CHC.3-Me-C25.m","CHC.3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1.f","CHC.5-C23:1.f","CHC.5-C23:1.m","CHC.5-C24:1.f","CHC.5-C24:1.m","CHC.5-C25:1.f","CHC.5-C25:1.m","CHC.5-C27:1.f","CHC.5-Me-C25.f","CHC.5,9-C25:2 &amp; 9-C25:1.f","CHC.5,9-C27:2 &amp; 9-C27:1.f","CHC.6-C23:1.f","CHC.6-C23:1.m","CHC.6-C24:1.f","CHC.6-C24:1.m","CHC.6-C26:1 &amp; i.m","CHC.6,10-C25:2.f","CHC.6,10-C25:2.m","CHC.6,10-C26:2.f","CHC.7-C23:1.f","CHC.7-C23:1.m","CHC.7-C24:1.m","CHC.7-C25:1.f","CHC.7-C25:1.m","CHC.7-C26:1.f","CHC.7-C26:1.m","CHC.7-C27:1.f","CHC.7-C27:1.m","CHC.7-C29:1.f","CHC.7-C29:1.m","CHC.7,11-C23:2.f","CHC.7,11-C24:2.f","CHC.7,11-C25:2.f","CHC.7,11-C26:2.f","CHC.7,11-C27:2 &amp; 2-Me-C26.f","CHC.7,11-C28:2 &amp; 2-Me-C27.f","CHC.7,11-C31:2.f","CHC.8-C25:1.f","CHC.8,12-C26:2.f","CHC.8,12-C28:2.f","CHC.9-C23:1.f","CHC.9-C23:1.m","CHC.9-C25:1.m","CHC.9-C27:1.m","CHC.9-C29:1 &amp; 5,9-C29:2.f","CHC.9,13-C25:2.f","CHC.9,13-C27:2.f","CHC.n-C21.f","CHC.n-C21.m","CHC.n-C22.f","CHC.n-C22.m","CHC.n-C23.f","CHC.n-C23.m","CHC.n-C24.f","CHC.n-C24.m","CHC.n-C25.m","CHC.n-C26.f","CHC.n-C26.m","CHC.n-C27.f","CHC.n-C27.m","CHC.n-C28.f","CHC.n-C28.m","CHC.n-C29.f","CHC.n-C29.m","CHC.n-C31.f","CHC.n-C31.m","CHC.NI.20.f","CHC.NI.33.f","CHC.NI.34.f","CHC.NI.54.f","CHC.NI.54.m","CHC.NI.7.m","CHC.x-C22:1a.m","CHC.x-C22:1b.m","CHC.x-C23:1.f","CHC.x,y-C24:2.f","CHC.x,y-C24:2.m","chill.coma.recovery.f","chill.coma.recovery.m","chill.coma.recovery.variance.f","chill.coma.recovery.variance.m","chlorantraniliprole.resistance.0.5","chlorantraniliprole.resistance.1","chlorantraniliprole.resistance.2","chlorantraniliprole.resistance.3","chlorantraniliprole.resistance.4","chlorantraniliprole.resistance.5","chlorantraniliprole.resistance.LC50","choline phosphate.f","choline phosphate.m","choline.f","choline.high.yeast.f","choline.low.yeast.f","choline.m","chronic.cold.exposure.resistance.f","chronic.cold.exposure.resistance.m","circadian.period.chi.f","circadian.period.chi.m","circadian.period.mesa.f","circadian.period.mesa.m","circadian.rythmicity.index.f","circadian.rythmicity.index.m","citraconate_glutaconate.f","citraconate_glutaconate.m","citrate.f","citrate.m","citrate.synthase.activity.f","citrate.synthase.activity.m","citrulline.f","citrulline.high.yeast.f","citrulline.low.yeast.f","citrulline.m","climbing.ability.control.exercise.m","climbing.ability.f","climbing.ability.m","climbing.ability.paraquat.f","climbing.ability.paraquat.m","climbing.ability.post.exercise.m","climbing.ability.pre.exercise.m","climbing.age.decline.high.yeast.f","climbing.age.decline.low.yeast.f","climbing.exercise.effect.f","climbing.exercise.effect.m","climbing.f","climbing.m","climbing.post.exercise.f","climbing.post.exercise.m","Clostridium.lentocellum.f","Clostridium.lentocellum.m","CMP.high.yeast.f","CMP.low.yeast.f","CO2.production.f","cocaine.preferenceA.E1.f","cocaine.preferenceA.E1.m","cocaine.preferenceA.E3.f","cocaine.preferenceA.E3.m","cocaine.preferenceB.E1.f","cocaine.preferenceB.E1.m","cocaine.preferenceB.E3.f","cocaine.preferenceB.E3.m","cold.hardiness.-5C.larvae","cold.hardiness.18C.f","cold.hardiness.18C.larvae","cold.hardiness.18C.m","cold.hardiness.25C.f","cold.hardiness.25C.larvae","cold.hardiness.25C.m","cold.hardiness.f","cold.hardiness.m","cold.shock.acclimated.old.f","cold.shock.acclimated.old.m","cold.shock.acclimated.young.f","cold.shock.acclimated.young.m","cold.shock.old.f","cold.shock.old.m","cold.shock.young.f","cold.shock.young.m","cold.tolerance.-1.f","cold.tolerance.-2.f","cold.tolerance.-3.f","cold.tolerance.-4.f","cold.tolerance.-5.f","cold.tolerance.-6.f","cold.tolerance.-7.f","copulation.latency.m","Corynebacterium.minutissimum.f","Corynebacterium.minutissimum.m","Corynebacterium.pseudogenitalium.f","Corynebacterium.pseudogenitalium.m","Corynebacterium.sp..f","Corynebacterium.sp..m","Corynebacterium.terpenotabidum.f","Corynebacterium.terpenotabidum.m","critical.thermal.max.2018.f","critical.thermal.max.2018.m","critical.thermal.max.2020.f","critical.thermal.max.2020.m","critical.thermal.min.2020.f","critical.thermal.min.2020.m","critical.thermal.min.23C.m","critical.thermal.min.developmental.stability.m","critical.thermal.min.plasticity.m","critical.thermal.min.plasticity.variation.m","critical.thermal.min.within.environment.variation.plasticity.m","cumulative.cold.tolerance.f","cystathionine.f","cystathionine.high.yeast.f","cystathionine.low.yeast.f","cystathionine.m","cysteine.f","cysteine.m","cysteinylglycine.f","cysteinylglycine.m","cystine.f","cystine.high.yeast.f","cystine.low.yeast.f","cystine.m","cytidine 2',3'-cyclic monophosphate.f","cytidine 2',3'-cyclic monophosphate.m","cytidine 3'-monophosphate (3'-CMP).f","cytidine 3'-monophosphate (3'-CMP).m","cytidine 5'-diphosphocholine.f","cytidine 5'-diphosphocholine.m","cytidine 5'-monophosphate (5'-CMP).f","cytidine 5'-monophosphate (5'-CMP).m","cytidine-5'-diphosphoethanolamine.f","cytidine-5'-diphosphoethanolamine.m","cytidine.f","cytidine.high.yeast.f","cytidine.low.yeast.f","cytidine.m","cytosine.f","cytosine.high.yeast.f","cytosine.low.yeast.f","cytosine.m","D-GA3P.high.yeast.f","D-GA3P.low.yeast.f","D-Leucic.acid.high.yeast.f","D-Leucic.acid.low.yeast.f","day.sleep.bout.length.f","day.sleep.bout.length.m","day.sleep.bout.length.SD.f","day.sleep.bout.length.SD.m","day.sleep.bouts.f","day.sleep.bouts.m","day.sleep.bouts.SD.f","day.sleep.bouts.SD.m","day.sleep.f","day.sleep.m","day.sleep.SD.f","day.sleep.SD.m","DDT.resistance.knockdown.f","DDT.resistance.mortality.f","dehydroascorbate.f","dehydroascorbate.m","deltamethrin.resistance.m","deoxycarnitine.f","deoxycarnitine.m","dessication.tolerance.f","dessication.tolerance.m","developmental.acclimation.cold.exposure.f","developmental.acclimation.cold.exposure.m","dihomo-linoleate (20:2n6).f","dihomo-linoleate (20:2n6).m","dihydoxyphenylalanine (L-DOPA).f","dihydoxyphenylalanine (L-DOPA).m","dihydrobiopterin.f","dihydrobiopterin.m","dihydroxyacetone phosphate (DHAP).f","dihydroxyacetone phosphate (DHAP).m","dimethylarginine (SDMA + ADMA).f","dimethylarginine (SDMA + ADMA).m","docosadienoate (22:2n6).f","docosadienoate (22:2n6).m","dodecanedioate.f","dodecanedioate.m","dopamine.content.2021.m","dopamine.content.paraquat.2021.m","dopamine.response.to.paraquat.2021.m","Drosophila.A Virus.f","Drosophila.A Virus.m","drosophila.C.virus.resistance.m","dry.weight.2021.f","dry.weight.2021.m","duration.embryogenesis","E.faecalis.load.m","E.faecalis.resistance.m","eclosion.to.oviposition.time.16C.f","eclosion.to.oviposition.time.23C.f","egg.adult.dev.time.20C.f","egg.adult.dev.time.25C.f","egg.adult.dev.time.30C.f","egg.adult.development.time.2016.f","egg.adult.development.time.2016.m","egg.adult.development.time.2018","egg.adult.development.time.EtOH.exposure","egg.adult.viability.2014.25C","egg.adult.viability.2016","egg.adult.viability.2016.25C","egg.adult.viability.2018","egg.adult.viability.20C","egg.adult.viability.28C","egg.adult.viability.30C","egg.adult.viability.ethanol.exposure","egg.larva.viability","egg.pupa.viability.20C","egg.pupa.viability.25C","egg.pupa.viability.30C","eggs.per.day.16C.f","eggs.per.day.23C.f","eicosenoate (20:1).f","eicosenoate (20:1).m","Enterococcus.hirae.f","Enterococcus.hirae.m","epinephrine.high.yeast.f","epinephrine.low.yeast.f","ergosterol.f","ergosterol.m","ergothioneine.f","ergothioneine.m","erucate (22:1n9).f","erucate (22:1n9).m","erythritol.f","erythritol.m","erythronate.f","erythronate.m","Escherichia.coli.f","Escherichia.coli.m","ethanol.sensitivity.development.time","ethanol.sensitivity.first.exposure.f","ethanol.sensitivity.first.exposure.m","ethanol.sensitivity.second.exposure.f","ethanol.sensitivity.second.exposure.m","ethanol.sensitivity.startle.response.f","ethanol.sensitivity.startle.response.m","ethanol.sensitivity.viability","ethanol.tolerance.f","ethanol.tolerance.m","ethylmalonate.f","ethylmalonate.m","exercise.induced.activity.f","exercise.induced.activity.m","Exophiala.xenobiotica.f","Exophiala.xenobiotica.m","fecundity.f","fertilised.egg.retention.score.f","fertility.25C.f","fertility.25C.m","fertility.27C.f","fertility.27C.m","fertility.29C.f","fertility.29C.m","fertility.thermal.susceptibility.f","fertility.thermal.susceptibility.m","Fimbriimonas.ginsengisoli.f","Fimbriimonas.ginsengisoli.m","fipronil.resistance","fitness.early.life.f","fitness.early.life.m","fitness.late.life.f","fitness.late.life.m","flavin mononucleotide (FMN).f","flavin mononucleotide (FMN).m","flight.performance.f","flight.performance.m","flock.house.virus.resistance.m","following.same.sex.m","food.intake.control.exercise.m","food.intake.f","food.intake.m","food.intake.post.exercise.m","foraging.survival.m","fructose.f","fructose.m","fumarate.f","fumarate.m","G1P.G6P.F6P.F1P.high.yeast.f","G1P.G6P.F6P.F1P.low.yeast.f","galactonate.f","galactonate.m","galactose 1-phosphate.f","galactose 1-phosphate.m","gamma-aminobutyrate (GABA).f","gamma-aminobutyrate (GABA).m","gamma-glutamylalanine.f","gamma-glutamylalanine.m","gamma-glutamylglutamate.f","gamma-glutamylglutamate.m","gamma-glutamylglutamine.f","gamma-glutamylglutamine.m","gamma-glutamylhistidine.f","gamma-glutamylhistidine.m","gamma-glutamylisoleucine.f","gamma-glutamylisoleucine.m","gamma-glutamylleucine.f","gamma-glutamylleucine.m","gamma-glutamylmethionine.f","gamma-glutamylmethionine.m","gamma-tocopherol_beta-tocopherol.f","gamma-tocopherol_beta-tocopherol.m","Gardnerella.vaginalis.f","Gardnerella.vaginalis.m","Gemmata.obscuriglobus.f","Gemmata.obscuriglobus.m","Gemmata.sp..f","Gemmata.sp..m","genital.licking.frequency.m","genome.size","geranyl.pyrophosphate.high.yeast.f","geranyl.pyrophosphate.low.yeast.f","Gluconacetobacter.diazotrophicus.f","Gluconacetobacter.diazotrophicus.m","gluconate.f","gluconate.m","Gluconobacter.frateurii.f","Gluconobacter.frateurii.m","Gluconobacter.morbifer.f","Gluconobacter.morbifer.m","Gluconobacter.oxydans.f","Gluconobacter.oxydans.m","Gluconobacter.thailandicus.f","Gluconobacter.thailandicus.m","glucoronate.high.yeast.f","glucoronate.low.yeast.f","glucosamine-6-phosphate.f","glucosamine-6-phosphate.m","glucose 1-phosphate.f","glucose 1-phosphate.m","glucose 6-phosphate.f","glucose 6-phosphate.m","glucose.2015.m","glucose.2018.f","glucose.2018.m","glucose.2019.f","glucose.2019.m","glucose.abdomen.content.control.exercise.m","glucose.abdomen.content.post.exercise.m","glucose.anexic.m","glucose.f","glucose.fructose.preference.f","glucose.fructose.preference.m","glucose.gnotobiotic.m","glucose.high.glucose.diet.m","glucose.high.yeast.f","glucose.larvae.2018","glucose.low.glucose.diet.m","glucose.low.yeast.f","glucose.m","glucose.medium.glucose.diet.m","glucose.microbiome.effect.m","glucose.sucrose.preference.f","glucose.sucrose.preference.m","glucose.thorax.content.control.exercise.m","glucose.thorax.content.post.exercise.m","glucose.trehalose.preference.f","glucose.trehalose.preference.m","glucose.yeast.poor.diet.f","glucose.yeast.rich.diet.f","glucuronate.f","glucuronate.m","glutamate, gamma-methyl ester.f","glutamate, gamma-methyl ester.m","glutamate.f","glutamate.m","glutamic.acid.high.yeast.f","glutamic.acid.low.yeast.f","glutamine.f","glutamine.high.yeast.f","glutamine.low.yeast.f","glutamine.m","glutaminylleucine.f","glutaminylleucine.m","glutarate (pentanedioate).f","glutarate (pentanedioate).m","glyceraldehyde.high.yeast.f","glyceraldehyde.low.yeast.f","glycerate.f","glycerate.high.yeast.f","glycerate.low.yeast.f","glycerate.m","glycerol 2-phosphate.m","glycerol 3-phosphate.f","glycerol 3-phosphate.m","glycerol-3-P.high.yeast.f","glycerol-3-P.low.yeast.f","glycerol.2015.m","glycerol.2018.f","glycerol.2018.m","glycerol.2019.f","glycerol.2019.m","glycerol.abdomen.content.control.exercise.m","glycerol.abdomen.content.post.exercise.m","glycerol.f","glycerol.high.glucose.diet.m","glycerol.larvae.2018","glycerol.low.glucose.diet.m","glycerol.m","glycerol.medium.glucose.diet.m","glycerol.thorax.content.control.exercise.m","glycerol.thorax.content.post.exercise.m","glycerophosphoethanolamine.f","glycerophosphoethanolamine.m","glycerophosphoglycerol.f","glycerophosphoglycerol.m","glycerophosphoinositol.f","glycerophosphoinositol.m","glycerophosphorylcholine (GPC).f","glycerophosphorylcholine (GPC).m","glycine.f","glycine.high.yeast.f","glycine.low.yeast.f","glycine.m","glycine.pref.post.limitation.f","glycine.pref.post.limitation.m","glycogen.2015.m","glycogen.2018.f","glycogen.2018.m","glycogen.2019.f","glycogen.2019.m","glycogen.anexic.m","glycogen.gnotobiotic.m","glycogen.high.glucose.diet.m","glycogen.larvae.2018","glycogen.low.glucose.diet.m","glycogen.medium.glucose.diet.m","glycogen.microbiome.effect.m","glycosyl-N-palmitoyl-sphingosine.f","glycosyl-N-palmitoyl-sphingosine.m","glycosyl-N-stearoyl-sphingosine.f","glycosyl-N-stearoyl-sphingosine.m","glycylisoleucine.f","glycylisoleucine.m","glycylleucine.f","glycylleucine.m","glycylvaline.f","glycylvaline.m","GMP.high.yeast.f","GMP.low.yeast.f","grooming.f","grooming.m","guanine.f","guanine.m","guanosine 3'-monophosphate (3'-GMP).f","guanosine 3'-monophosphate (3'-GMP).m","guanosine-2',3'-cyclic monophosphate.f","guanosine-2',3'-cyclic monophosphate.m","guanosine.f","guanosine.high.yeast.f","guanosine.low.yeast.f","guanosine.m","gulonic acid.f","gulonic acid.m","gustatory.response.ethanol.f","gustatory.response.ethanol.m","gut.permeability.f","gut.permeability.radiation.f","heat.hardiness.18C.f","heat.hardiness.18C.larvae","heat.hardiness.18C.m","heat.hardiness.25C.f","heat.hardiness.25C.larvae","heat.hardiness.25C.m","heat.resistance.f","heme.f","heme.m","hexanoylglycine.f","hexanoylglycine.m","hippurate.f","hippurate.m","histamine.f","histamine.high.yeast.f","histamine.low.yeast.f","histamine.m","histidine.f","histidine.high.yeast.f","histidine.low.yeast.f","histidine.m","homogentisate.high.yeast.f","homogentisate.low.yeast.f","homoserine lactone.f","homoserine lactone.m","Hsp70.expression.f","hydrogen.peroxide.resistance.f","Hymenobacter.sp..f","Hymenobacter.sp..m","Hymenobacter.swuensis.f","Hymenobacter.swuensis.m","hypoxanthine.f","hypoxanthine.high.yeast.f","hypoxanthine.low.yeast.f","hypoxanthine.m","Ideonella.sp..f","Ideonella.sp..m","imidacloprid.resistance.100ppm","imidacloprid.resistance.25ppm","IMP.high.yeast.f","IMP.low.yeast.f","inosine.f","inosine.high.yeast.f","inosine.low.yeast.f","inosine.m","inositol 1-phosphate (I1P).f","inositol 1-phosphate (I1P).m","inositol.high.yeast.f","inositol.low.yeast.f","insemination.capacity.m","interocular.distance.f","interocular.distance.m","iso-leucine.high.yeast.f","iso-leucine.low.yeast.f","Isobar: fructose 1,6-diphosphate, glucose 1,6-diphosphate, myo-inositol 1,4 or 1,3-diphosphate.f","Isobar: fructose 1,6-diphosphate, glucose 1,6-diphosphate, myo-inositol 1,4 or 1,3-diphosphate.m","isobutyrylcarnitine.f","isobutyrylcarnitine.m","isolated.aggression.m","isoleucine.f","isoleucine.m","isoleucylglycine.f","isoleucylglycine.m","isomaltose.f","isomaltose.m","isoValeric.acid.high.yeast.f","isoValeric.acid.low.yeast.f","isoxanthopterin.f","isoxanthopterin.m","kallithea.resistance.LT50.f","kallithea.resistance.LT50.m","kallithea.viral.load.f","Klebsiella.pneumoniae.f","Klebsiella.pneumoniae.m","Kocuria.polaris.f","Kocuria.polaris.m","Kocuria.sp..f","Kocuria.sp..m","kojibiose.f","kojibiose.m","Komagataeibacter.europaeus.f","Komagataeibacter.europaeus.m","Komagataeibacter.hansenii.f","Komagataeibacter.hansenii.m","Komagataeibacter.intermedius.f","Komagataeibacter.intermedius.m","Komagataeibacter.kakiaceti.f","Komagataeibacter.kakiaceti.m","Komagataeibacter.medellinensis.f","Komagataeibacter.medellinensis.m","Komagataeibacter.oboediens.f","Komagataeibacter.oboediens.m","Komagataeibacter.rhaeticus.f","Komagataeibacter.rhaeticus.m","Komagataeibacter.xylinus.f","Komagataeibacter.xylinus.m","Kozakia.baliensis.f","Kozakia.baliensis.m","kujin.virus.resistance.f","kynurenate.f","kynurenate.m","kynurenine.f","kynurenine.m","L-Kynurenine.high.yeast.f","L-Kynurenine.low.yeast.f","L.bevis.load.m","L.monocytogenes.load.m","L.monocytogenes.MTD.m","L.plantarum.load.m","lactate.2018.f","lactate.2018.m","lactate.f","lactate.high.yeast.f","lactate.larvae.2018","lactate.low.yeast.f","lactate.m","Lactobacillus.brevis.f","Lactobacillus.brevis.m","Lactobacillus.fabifermentans.f","Lactobacillus.fabifermentans.m","Lactobacillus.fermentum.f","Lactobacillus.fermentum.m","Lactobacillus.kisonensis.f","Lactobacillus.kisonensis.m","Lactobacillus.plantarum.f","Lactobacillus.plantarum.m","Lactobacillus.rhamnosus.f","Lactobacillus.rhamnosus.m","Lactococcus.lactis.f","Lactococcus.lactis.m","lactose.high.yeast.f","lactose.low.yeast.f","lactosyl-N-palmitoyl-sphingosine.f","lactosyl-N-palmitoyl-sphingosine.m","lanthionine.f","lanthionine.m","larvae.length.low.yeast.anexic","larvae.length.low.yeast.L.plantarum","laurate (12:0).f","laurate (12:0).m","lead.effect.development.time.f","lead.effect.development.time.m","lead.effect.locomotor.activity.f","lead.effect.locomotor.activity.m","lead.effect.viability","lead.egg.adult.development.time.2016.f","lead.egg.adult.development.time.2016.m","lead.egg.adult.viability.2016","lead.locomotor.activity.2016.f","lead.locomotor.activity.2016.m","leucine.f","leucine.high.yeast.f","leucine.low.yeast.f","leucine.m","Leuconostoc.carnosum.f","Leuconostoc.carnosum.m","Leuconostoc.pseudomesenteroides.f","Leuconostoc.pseudomesenteroides.m","leucylalanine.f","leucylalanine.m","leucylglutamine.f","leucylglutamine.m","leucylglycine.f","leucylglycine.m","licking.same.sex.m","lifespan.16C.f","lifespan.18C.f","lifespan.18C.m","lifespan.2019.f","lifespan.23C.f","lifespan.25C.2014.f","lifespan.25C.2015.f","lifespan.25C.2020.f","lifespan.25C.2020.m","lifespan.25C.high.yeast.f","lifespan.25C.low.yeast.f","lifespan.28C.f","lifespan.28C.m","lifespan.following.diapause.2019.f","lifespan.high.sugar.diet.f","lifespan.td50.m","lifetime.fecundity.25C.f","linoleate (18:2n6).f","linoleate (18:2n6).m","linoleic.acid.high.yeast.f","linoleic.acid.low.yeast.f","linolenate [alpha or gamma; (18:3n3 or 6)].f","linolenate [alpha or gamma; (18:3n3 or 6)].m","linolenic.acid.high.yeast.f","linolenic.acid.low.yeast.f","linoleoylcarnitine.f","linoleoylcarnitine.m","locomotor.activity.2016.f","locomotor.activity.2016.m","locomotor.activity.m","locomotor.frequency.f","Luteipulveratus.mongoliensis.f","Luteipulveratus.mongoliensis.m","lysine.f","lysine.high.yeast.f","lysine.low.yeast.f","lysine.m","lysylleucine.f","lysylleucine.m","M.anisopliae.resistance.f","M.anisopliae.resistance.m","malate.f","malate.m","malathion.mortality.12hr.f","malathion.mortality.12hr.m","malathion.mortality.24hr.f","malathion.mortality.24hr.m","malathion.mortality.3hr.f","malathion.mortality.3hr.m","malathion.mortality.6hr.f","malathion.mortality.6hr.m","malonate.f","malonate.m","malondialdehyde.high.yeast.f","malondialdehyde.low.yeast.f","maltopentaose.f","maltopentaose.m","maltose.f","maltose.m","maltotetraose.f","maltotetraose.m","maltotriose.f","maltotriose.m","mannitol_sorbitol.f","mannitol_sorbitol.m","mannose-6-phosphate.f","mannose-6-phosphate.m","mannose.f","mannose.m","margarate (17:0).f","margarate (17:0).m","margaric.acid.high.yeast.f","margaric.acid.low.yeast.f","mass.2016.f","mass.2019.f","mass.2019.m","mass.2021.f","mass.2021.m","mass.post.exercise.f","mass.post.exercise.m","mass.yeast.poor.diet.f","mass.yeast.rich.diet.f","mating.latency.2015.m","mating.latency.2019.f","mating.latency.2019.m","MeHg.tolerance","mesaconate (methylfumarate).f","mesaconate (methylfumarate).m","Mesorhizobium.sp..f","Mesorhizobium.sp..m","Metarhizium.brunneum.f","Metarhizium.brunneum.m","methamphetamine.preferenceA.E1.f","methamphetamine.preferenceA.E1.m","methamphetamine.preferenceA.E3.f","methamphetamine.preferenceA.E3.m","methamphetamine.preferenceB.E1.f","methamphetamine.preferenceB.E1.m","methamphetamine.preferenceB.E3.f","methamphetamine.preferenceB.E3.m","methionine sulfone.f","methionine sulfone.m","methionine sulfoxide.f","methionine sulfoxide.m","methionine.f","methionine.high.yeast.f","methionine.low.yeast.f","methionine.m","methyl glucopyranoside (alpha + beta).f","methyl glucopyranoside (alpha + beta).m","methyl-4-hydroxybenzoate.f","methyl-4-hydroxybenzoate.m","methylmalonate (MMA).f","methylmalonate (MMA).m","methylmalonate.high.yeast.f","methylmalonate.low.yeast.f","Methylobacterium.extorquens.f","Methylobacterium.extorquens.m","Methylobacterium.sp..f","Methylobacterium.sp..m","methylphosphate.f","methylphosphate.m","methylsuccinate.f","methylsuccinate.m","mevalonate 5-phosphate.m","mevalonate.f","mevalonate.m","mevalonolactone.f","mevalonolactone.m","micro.environmental.plasticity.lifespan.18C.f","micro.environmental.plasticity.lifespan.18C.m","micro.environmental.plasticity.lifespan.25C.f","micro.environmental.plasticity.lifespan.25C.m","micro.environmental.plasticity.lifespan.28C.f","micro.environmental.plasticity.lifespan.28C.m","Microbacterium.sp..f","Microbacterium.sp..m","min.pupation.time.20C","min.pupation.time.25C","min.pupation.time.30C","mitochondrial.PO.ratio.f","mitochondrial.PO.ratio.m","mitochondrial.state.3.respiration.f","mitochondrial.state.3.respiration.m","mitochondrial.state.4.respiration.f","mitochondrial.state.4.respiration.m","MPH.effect.on.activity.m","MSB.resistance.f","MSB.resistance.m","mtDNA.copy.number","mushroom.abnormal.alpha.m","mushroom.abnormal.beta.m","mushroom.fusion.beta.m","mushroom.missing.alpha.m","mushroom.missing.beta.m","mushroom.short.alpha.m","mushroom.short.beta.m","mushroom.thin.alpha.m","mushroom.thin.beta.m","myo-inositol.f","myo-inositol.m","myristate (14:0).f","myristate (14:0).m","myristic.acid.high.yeast.f","myristic.acid.low.yeast.f","myristoleate (14:1n5).f","myristoleate (14:1n5).m","myristoylcarnitine.f","myristoylcarnitine.m","N-acetyl-1-methylhistidine.f","N-acetyl-1-methylhistidine.m","N-acetyl-beta-alanine.f","N-acetyl-beta-alanine.m","N-acetyl-glucosamine 1-phosphate.f","N-acetyl-glucosamine 1-phosphate.m","N-acetylalanine.f","N-acetylalanine.m","N-acetylarginine.f","N-acetylarginine.m","N-acetylasparagine.f","N-acetylasparagine.m","N-acetylaspartate (NAA).f","N-acetylaspartate (NAA).m","N-acetylcysteine.f","N-acetylcysteine.m","N-acetylglucosamine 6-phosphate.f","N-acetylglucosamine 6-phosphate.m","N-acetylglucosamine_N-acetylgalactosamine.f","N-acetylglucosamine_N-acetylgalactosamine.m","N-acetylglutamate.f","N-acetylglutamate.m","N-acetylglutamine.f","N-acetylglutamine.m","N-acetylglycine.f","N-AcetylGlycine.high.yeast.f","N-AcetylGlycine.low.yeast.f","N-acetylglycine.m","N-acetylhistamine.f","N-acetylhistamine.m","N-acetylhistidine.f","N-acetylhistidine.m","N-acetylleucine.f","N-acetylleucine.m","N-acetylmethionine sulfoxide.f","N-acetylmethionine sulfoxide.m","N-acetylmethionine.f","N-acetylmethionine.m","N-acetylphenylalanine.f","N-acetylphenylalanine.m","N-acetylputrescine.f","N-acetylputrescine.m","N-acetylserine.f","N-acetylserine.m","N-acetyltaurine.f","N-acetyltaurine.m","N-acetylthreonine.f","N-acetylthreonine.m","N-acetylvaline.f","N-acetylvaline.m","N-alpha-acetylornithine.f","N-alpha-acetylornithine.m","N-delta-acetylornithine.f","N-delta-acetylornithine.m","N-methylalanine.f","N-methylalanine.m","N-monomethylarginine.f","N-monomethylarginine.m","N-palmitoyl-sphinganine (d18:0_16:0).f","N-palmitoyl-sphinganine (d18:0_16:0).m","N-palmitoyl-sphingosine (d18:1_16:0).f","N-palmitoyl-sphingosine (d18:1_16:0).m","N1-methyladenosine.f","N1-methyladenosine.m","N1-methylinosine.f","N1-methylinosine.m","N2,N2-dimethylguanine.f","N2,N2-dimethylguanine.m","N2,N2-dimethylguanosine.f","N2,N2-dimethylguanosine.m","N6-acetyllysine.f","N6-acetyllysine.m","N6-carboxymethyllysine.f","N6-carboxymethyllysine.m","N6-succinyladenosine.f","N6-succinyladenosine.m","N6,N6-dimethyladenosine.f","N6,N6-dimethyladenosine.m","N6,N6,N6-trimethyllysine.f","N6,N6,N6-trimethyllysine.m","negative.geotaxis.f","negative.geotaxis.m","negative.geotaxis.msb.f","negative.geotaxis.msb.m","negative.geotaxis.sensitivity.msb.f","negative.geotaxis.sensitivity.msb.m","net.reproductive.rate.16C","net.reproductive.rate.23C","nicotinamide ribonucleotide (NMN).f","nicotinamide ribonucleotide (NMN).m","nicotinamide riboside.f","nicotinamide riboside.m","nicotinamide.f","nicotinamide.m","nicotinate ribonucleoside.f","nicotinate ribonucleoside.m","nicotinate.f","nicotinate.m","night.sleep.bout.length.f","night.sleep.bout.length.m","night.sleep.bout.length.SD.f","night.sleep.bout.length.SD.m","night.sleep.bouts.f","night.sleep.bouts.m","night.sleep.bouts.SD.f","night.sleep.bouts.SD.m","night.sleep.f","night.sleep.m","night.sleep.SD.f","night.sleep.SD.m","nonadecanoate (19:0).f","nonadecanoate (19:0).m","Nora.virus.f","Nora.virus.m","O-sulfo-L-tyrosine.f","O-sulfo-L-tyrosine.m","oleate_vaccenate (18:1).f","oleate_vaccenate (18:1).m","oleoylcarnitine.f","oleoylcarnitine.m","oleoylcholine.f","oleoylcholine.m","olfactory.response.1-hexanol.f","olfactory.response.1-hexanol.m","olfactory.response.2-heptanone.f","olfactory.response.2-heptanone.m","olfactory.response.2-phenyl.ethyl.alcohol.f","olfactory.response.2-phenyl.ethyl.alcohol.m","olfactory.response.acetophenone.f","olfactory.response.acetophenone.m","olfactory.response.benzaldehyde.2013.f","olfactory.response.benzaldehyde.2013.m","olfactory.response.benzaldehyde.2015.f","olfactory.response.benzaldehyde.2015.m","olfactory.response.citral.f","olfactory.response.citral.m","olfactory.response.D-carvone.f","olfactory.response.D-carvone.m","olfactory.response.ethanol.f","olfactory.response.ethanol.m","olfactory.response.ethylacetate.f","olfactory.response.ethylacetate.m","olfactory.response.ethylbutyrate.f","olfactory.response.ethylbutyrate.m","olfactory.response.eugenol.f","olfactory.response.eugenol.m","olfactory.response.heliol.f","olfactory.response.heliol.m","olfactory.response.hexal.f","olfactory.response.hexal.m","olfactory.response.L-carvone.f","olfactory.response.L-carvone.m","olfactory.response.methylSalicylate.f","olfactory.response.methylSalicylate.m","orient.towards.female.m","orientating.same.sex.m","ornithine.f","ornithine.high.yeast.f","ornithine.low.yeast.f","ornithine.m","orotate.f","orotate.high.yeast.f","orotate.low.yeast.f","orotate.m","ovariole.asymmetry.f","ovariole.number.f","oxalate (ethanedioate).f","oxalate (ethanedioate).m","oxaloacetate.f","oxaloacetate.high.yeast.f","oxaloacetate.low.yeast.f","oxaloacetate.m","oxypurinol.high.yeast.f","oxypurinol.low.yeast.f","p-xylene.susceptibility","P.aeruginosa.defecation.f","P.aeruginosa.fecal.pH.f","P.aeruginosa.intestinal.mitosis.f","P.aeruginosa.resistance.2017.f","P.aeruginosa.resistance.2017.m","P.aeruginosa.resistance.2019.f","P.entomophila.resistance.f","P.rettgeri.load.f","P.rettgeri.load.high.glucose.diet.m","P.rettgeri.load.low.glucose.diet.m","P.rettgeri.load.medium.glucose.diet.m","P.rettgeri.resistance.f","P.rettgeri.tolerance.f","Paenibacillus.borealis.f","Paenibacillus.borealis.m","Paenibacillus.sp..f","Paenibacillus.sp..m","palmitate (16:0).f","palmitate (16:0).m","palmitoleate (16:1n7).f","palmitoleate (16:1n7).m","palmitoleoylcarnitine.f","palmitoleoylcarnitine.m","palmitoloelycholine.f","palmitoloelycholine.m","palmitoylcarnitine.f","palmitoylcarnitine.m","palmitoylcholine.f","palmitoylcholine.m","panose.f","panose.m","pantethine.f","pantethine.m","pantothenate.f","pantothenate.high.yeast.f","pantothenate.low.yeast.f","pantothenate.m","paraquat.resistance.2021.f","paraquat.resistance.2021.m","paraquat.resistance.f","paraquat.resistance.m","parathion.resistance.m","Pectobacterium.carotovorum.f","Pectobacterium.carotovorum.m","pentadecanoate (15:0).f","pentadecanoate (15:0).m","PEP.high.yeast.f","PEP.low.yeast.f","permethrin.knockdown.f","permethrin.knockdown.m","permethrin.mortality.24hr.f","permethrin.mortality.24hr.m","phenylalanine.f","phenylalanine.high.yeast.f","phenylalanine.low.yeast.f","phenylalanine.m","phenylalanine.pref.post.limitation.f","phenylalanine.pref.post.limitation.m","phenylalanylalanine.f","phenylalanylalanine.m","phenylalanylglycine.f","phenylalanylglycine.m","phenyllactate (PLA).f","phenyllactate (PLA).m","phosphate.f","phosphate.m","phosphoenolpyruvate (PEP).f","phosphoenolpyruvate (PEP).m","phosphoethanolamine.f","phosphoethanolamine.m","phosphopantetheine.f","phosphopantetheine.m","phosphothreonine.f","phosphothreonine.m","phototaxis.dark.f","phototaxis.dark.m","phototaxis.mid.age.f","phototaxis.mid.age.m","phototaxis.old.f","phototaxis.old.m","phototaxis.young.f","phototaxis.young.m","phytosphingosine.f","phytosphingosine.m","pipecolate.f","pipecolate.high.yeast.f","pipecolate.low.yeast.f","pipecolate.m","Plasmodium.vivax.f","Plasmodium.vivax.m","Podospora.anserina.f","Podospora.anserina.m","pop.growth.rate.16C","pop.growth.rate.23C","posterior.lobe.shape.PC1.m","posterior.lobe.shape.PC2.m","posterior.lobe.size.m","progeny.production.f","progeny.sired.by.first.male.f","proline.f","proline.high.yeast.f","proline.low.yeast.f","proline.m","prolylglycine.f","prolylglycine.m","Propionibacterium.acidipropionici.f","Propionibacterium.acidipropionici.m","Propionibacterium.acnes.f","Propionibacterium.acnes.m","propionylcarnitine.f","propionylcarnitine.m","propionylglycine.f","propionylglycine.m","protein.2015.m","protein.2018.f","protein.2018.m","protein.2019.f","protein.2019.m","protein.abdomen.content.control.exercise.m","protein.abdomen.content.post.exercise.m","protein.anexic.m","protein.gnotobiotic.m","protein.high.glucose.diet.m","protein.larvae.2018","protein.low.glucose.diet.m","protein.medium.glucose.diet.m","protein.microbiome.effect.m","protein.thorax.content.control.exercise.m","protein.thorax.content.post.exercise.m","Pseudodonghicola.xiamenensis.f","Pseudodonghicola.xiamenensis.m","Pseudomonas.fuscovaginae.f","Pseudomonas.fuscovaginae.m","Pseudomonas.moraviensis.f","Pseudomonas.moraviensis.m","Pseudomonas.putida.f","Pseudomonas.putida.m","Pseudomonas.sp..f","Pseudomonas.sp..m","Pseudomonas.stutzeri.f","Pseudomonas.stutzeri.m","Pseudomonas.syringae.f","Pseudomonas.syringae.m","pseudouridine.f","pseudouridine.m","Psychrobacter.glacincola.f","Psychrobacter.glacincola.m","Psychrobacter.sp..f","Psychrobacter.sp..m","pterin.f","pterin.m","punishment.memory.learning.f","punishment.memory.learning.m","pupal.case.length","pupal.mass.20C.f","pupal.mass.25C.f","pupal.mass.30C.f","pupation.height","putrescine.f","putrescine.high.yeast.f","putrescine.low.yeast.f","putrescine.m","pyridoxal-5-P.high.yeast.f","pyridoxal-5-P.low.yeast.f","pyridoxal.f","pyridoxal.m","pyridoxamine.f","pyridoxamine.m","pyridoxate.f","pyridoxate.m","pyriprole.resistance","pyroglutamine.f","pyroglutamine.m","pyruvate.f","pyruvate.high.yeast.f","pyruvate.low.yeast.f","pyruvate.m","quinate.f","quinate.m","radiation.resistance.m","rapid.cold.hardening.16C","rapid.cold.hardening.18C.f","rapid.cold.hardening.18C.m","rapid.cold.hardening.23C","rapid.cold.hardening.25C.f","rapid.cold.hardening.25C.m","rapid.cold.hardening.old.f","rapid.cold.hardening.old.m","rapid.cold.hardening.young.f","rapid.cold.hardening.young.m","reactive.oxygen.species.in.gut.m","relief.memory.learning.f","relief.memory.learning.m","remating.proportion.f","Rhizobacter.sp..f","Rhizobacter.sp..m","ribitol.f","ribitol.m","riboflavin (Vitamin B2).f","riboflavin (Vitamin B2).m","ribose 1-phosphate.f","ribose 1-phosphate.m","ribose 5-phosphate.f","ribose 5-phosphate.m","ribose-5-P.high.yeast.f","ribose-5-P.low.yeast.f","ribose.f","ribose.m","ribulose_xylulose.f","ribulose_xylulose.m","S-1-pyrroline-5-carboxylate.f","S-1-pyrroline-5-carboxylate.m","S-adenosylhomocysteine (SAH).f","S-adenosylhomocysteine (SAH).m","S-adenosylmethionine (SAM).f","S-adenosylmethionine (SAM).m","Saccharomyces.cerevisiae.f","Saccharomyces.cerevisiae.m","saccharopine.f","saccharopine.m","sarcosine.f","sarcosine.high.yeast.f","sarcosine.low.yeast.f","sarcosine.m","sedoheptulose-7-phosphate.f","sedoheptulose-7-phosphate.m","sedoheptulose.f","sedoheptulose.m","sepiapterin.f","sepiapterin.m","serine.f","serine.high.yeast.f","serine.low.yeast.f","serine.m","sigma.virus.sensitivity.f","singing.same.sex.m","social.aggression.m","social.effect.aggression.m","social.spacing.f","social.spacing.m","song.interpulse.interval.m","sorbitol.high.yeast.f","sorbitol.low.yeast.f","spermidine.f","spermidine.m","sphinganine.f","sphinganine.m","Sphingomonas.hengshuiensis.f","Sphingomonas.hengshuiensis.m","Sphingomonas.parapaucimobilis.f","Sphingomonas.parapaucimobilis.m","Sphingomonas.phyllosphaerae.f","Sphingomonas.phyllosphaerae.m","Sphingomonas.sp..f","Sphingomonas.sp..m","Sphingomonas.wittichii.f","Sphingomonas.wittichii.m","Sporolactobacillus.terrae.f","Sporolactobacillus.terrae.m","sprinter.marathoner.f","sprinter.marathoner.m","stachydrine.f","stachydrine.m","Staphylococcus.aureus.f","Staphylococcus.aureus.m","Staphylococcus.epidermidis.f","Staphylococcus.epidermidis.m","Staphylococcus.gallinarum.f","Staphylococcus.gallinarum.m","startle.response.2012.f","startle.response.2012.m","startle.response.2012.msb.f","startle.response.2012.msb.m","startle.response.2018.f","startle.response.2018.m","startle.response.EtOH.f","startle.response.EtOH.m","startle.response.f","startle.response.m","startle.response.variance.f","startle.response.variance.m","startle.sensitivity.msb.f","startle.sensitivity.msb.m","starvation.resistance.2012.f","starvation.resistance.2012.m","starvation.resistance.2017.m","starvation.resistance.2019.f","starvation.resistance.2019.m","starvation.resistance.high.carb.diet.m","starvation.resistance.high.fat.diet.m","starvation.resistance.high.protein.diet.m","starvation.resistance.larvae","starvation.resistance.old.f","starvation.resistance.old.m","starvation.resistance.standard.diet.m","starvation.resistance.variance.f","starvation.resistance.variance.m","starvation.resistance.yeast.poor.diet.f","starvation.resistance.yeast.rich.diet.f","starvation.resistance.young.f","starvation.resistance.young.m","stearate (18:0).f","stearate (18:0).m","stearidonate (18:4n3).f","stearidonate (18:4n3).m","stearoylcarnitine.f","stearoylcarnitine.m","Streptococcus.pneumoniae.f","Streptococcus.pneumoniae.m","Streptococcus.salivarius.f","Streptococcus.salivarius.m","Streptomyces.griseoflavus.f","Streptomyces.griseoflavus.m","Streptomyces.pristinaespiralis.f","Streptomyces.pristinaespiralis.m","Streptomyces.sp..f","Streptomyces.sp..m","succinate.f","succinate.high.yeast.f","succinate.low.yeast.f","succinate.m","succinimide.f","succinimide.m","sucrose.f","sucrose.high.yeast.f","sucrose.low.yeast.f","sucrose.m","sulfate.f","sulfate.m","Sum.of Non-Wolbachia Species.f","Sum.of Non-Wolbachia Species.m","T1.femur.f","T1.femur.m","T1.leg.length.f","T1.leg.length.m","T1.tarsus.f","T1.tarsus.m","T1.tibia.f","T1.tibia.m","T2.femur.f","T2.femur.m","T2.leg.length.f","T2.leg.length.m","T2.tarsus.f","T2.tarsus.m","T2.tibia.f","T2.tibia.m","TAG.glycerol.microbiome.effect.m","tapped.same.sex.m","tapping.same.sex.m","taurine.f","taurine.high.yeast.f","taurine.low.yeast.f","taurine.m","tergite5.f","tergite6.f","Terriglobus.roseus.f","Terriglobus.roseus.m","tetradecanedioate.f","tetradecanedioate.m","thiamin (Vitamin B1).f","thiamin (Vitamin B1).m","thiamin monophosphate.f","thiamin monophosphate.m","thioproline.f","thioproline.m","thorax.length.18C.f","thorax.length.28C.f","thorax.length.f","thorax.length.m","thorax.length.thermal.plasticity.f","thorax.width.f","thorax.width.m","threonate.f","threonate.m","threonine.f","threonine.high.yeast.f","threonine.low.yeast.f","threonine.m","threonylphenylalanine.f","threonylphenylalanine.m","thymidine.f","thymidine.m","thymine.f","thymine.m","trans-4-hydroxyproline.f","trans-4-hydroxyproline.m","traumatic.injury.mortality","Tremella.mesenterica.f","Tremella.mesenterica.m","triacylglycerol.2018.f","triacylglycerol.2018.m","triacylglycerol.larvae.2018","tricarballylate.f","tricarballylate.m","triglyceride.2015.m","triglyceride.2019.f","triglyceride.2019.m","triglyceride.abdomen.content.control.exercise.m","triglyceride.abdomen.content.post.exercise.m","triglyceride.anexic.m","triglyceride.gnotobiotic.m","triglyceride.high.glucose.diet.m","triglyceride.low.glucose.diet.m","triglyceride.medium.glucose.diet.m","triglyceride.poor.diet.f","triglyceride.thorax.content.control.exercise.m","triglyceride.thorax.content.post.exercise.m","triglyceride.yeast.rich.diet.f","trigonelline (N'-methylnicotinate).f","trigonelline (N'-methylnicotinate).m","trimethylamine.high.yeast.f","trimethylamine.low.yeast.f","tryptophan.f","tryptophan.high.yeast.f","tryptophan.low.yeast.f","tryptophan.m","tunicamycin.ER.stress.m","tunicamycin.LT50.m","tyrosine.f","tyrosine.high.yeast.f","tyrosine.low.yeast.f","tyrosine.m","tyrosylglycine.f","tyrosylglycine.m","UDP-N-acetylglucosamine.f","UDP-N-acetylglucosamine.m","undecanedioate.f","undecanedioate.m","unfertilised.eggs.post.mating","uracil.f","uracil.high.yeast.f","uracil.low.yeast.f","uracil.m","urate.f","urate.high.yeast.f","urate.low.yeast.f","urate.m","uridine 3'-monophosphate (3'-UMP).f","uridine 3'-monophosphate (3'-UMP).m","uridine-2',3'-cyclic monophosphate.f","uridine-2',3'-cyclic monophosphate.m","uridine.f","uridine.high.yeast.f","uridine.low.yeast.f","uridine.m","valine.f","valine.high.yeast.f","valine.low.yeast.f","valine.m","valylglutamine.f","valylglutamine.m","valylglycine.f","valylglycine.m","valylleucinamide.f","valylleucinamide.m","valylleucine.f","valylleucine.m","virgin.egg.retention.f","waking.activity.f","waking.activity.m","waking.activity.SD.f","waking.activity.SD.m","week1.fecundity.25C.f","week3.fecundity.25C.f","week5.fecundity.25C.f","week7.fecundity.25C.f","weight.anexic.m","weight.control.exercise.m","weight.gnotobiotic.m","weight.high.glucose.diet.m","weight.low.glucose.diet.m","weight.medium.glucose.diet.m","weight.microbiome.effect.m","weight.post.exercise.m","wet.weight.m","wing.centroid.size.2019_Dworkin.f","wing.centroid.size.2019_Dworkin.m","wing.centroid.size.2019_Houle.f","wing.centroid.size.2019_Houle.m","wing.centroid.size.f","wing.centroid.size.m","wing.display.frequency.m","wing.imaginal.disk.size.absolute.f","wing.imaginal.disk.size.absolute.m","wing.imaginal.disk.size.relative.f","wing.imaginal.disk.size.relative.m","wing.size.f","wing.size.m","wolbachia.copy.number","Wolbachia.pipentis.f","Wolbachia.pipentis.m","xanthine.f","xanthine.m","Xanthomonas.sp..f","Xanthomonas.sp..m","Xanthomonas.vesicatoria.f","Xanthomonas.vesicatoria.m","xanthopterin.f","xanthopterin.m","xanthosine.f","xanthosine.high.yeast.f","xanthosine.low.yeast.f","xanthosine.m","xanthurenate.f","xanthurenate.high.yeast.f","xanthurenate.low.yeast.f","xanthurenate.m","Yersinia.mollaretii.f","Yersinia.mollaretii.m","Yersinia.pekkanenii.f","Yersinia.pekkanenii.m","Zymomonas.mobilis.f","Zymomonas.mobilis.m"],[178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,157,157,157,157,157,157,157,157,157,157,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,36,50,194,196,191,181,181,187,187,178,178,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,40,178,178,40,178,178,40,40,200,200,200,200,200,200,200,200,200,200,200,200,200,200,40,40,172,172,50,185,186,181,181,184,184,40,178,178,40,40,40,40,40,40,40,40,178,178,40,40,40,178,178,200,48,178,178,200,200,40,178,178,40,40,40,40,178,178,40,186,186,36,40,40,40,40,40,40,178,178,40,40,40,40,36,36,178,178,178,178,171,171,169,178,178,166,40,40,40,40,40,40,178,178,40,178,178,40,40,40,200,200,200,200,40,178,178,40,40,40,178,178,200,200,200,200,166,40,40,175,175,175,175,175,200,200,200,200,153,151,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,178,178,40,163,200,200,178,178,164,40,40,200,200,40,40,40,40,200,200,40,178,178,40,200,200,178,178,200,200,46,46,46,46,46,46,46,46,169,169,169,157,169,157,169,157,157,157,169,157,169,157,169,157,169,157,169,169,157,169,157,169,157,169,169,169,169,169,157,169,157,157,169,157,169,169,157,157,169,157,169,157,169,157,169,157,169,169,169,169,169,169,169,169,169,169,169,157,157,157,169,169,169,169,157,169,157,169,157,169,157,157,169,157,169,157,169,157,169,157,169,157,169,169,169,169,157,157,157,157,169,169,157,176,176,174,174,152,152,151,152,152,152,152,40,40,40,178,178,40,175,175,161,166,161,166,167,167,40,40,40,40,35,36,40,178,178,40,4,99,99,99,99,4,4,155,155,35,33,35,33,34,34,200,200,178,178,21,46,46,46,46,46,46,46,46,139,39,40,39,39,39,39,139,139,96,96,96,96,96,96,96,96,36,36,36,36,36,36,36,33,200,200,200,200,200,200,200,200,34,34,100,100,100,100,162,166,166,166,166,36,40,178,178,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,178,178,40,178,178,178,178,168,168,167,167,168,168,167,167,168,168,167,167,184,184,40,40,195,40,40,175,175,182,183,40,40,40,40,40,40,40,40,40,40,40,40,40,40,20,20,99,200,200,155,75,75,43,26,199,39,39,50,50,50,198,198,201,201,50,200,27,201,50,26,50,201,43,45,49,45,39,39,40,40,200,200,178,178,40,40,40,40,40,40,40,40,40,40,200,200,201,205,205,205,205,38,38,201,205,205,40,40,149,148,200,200,142,91,40,126,40,124,40,122,40,127,200,200,11,125,124,125,124,40,40,197,197,182,50,4,183,182,4,201,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,200,200,200,200,200,200,166,205,178,178,200,200,40,40,200,200,200,200,200,200,200,200,178,178,40,40,40,40,40,40,145,36,36,40,40,4,4,103,40,76,76,117,147,178,35,150,178,40,152,99,37,37,4,4,37,37,181,181,40,40,40,40,40,40,178,178,40,178,178,40,40,40,40,40,178,178,40,178,178,40,40,40,40,178,178,145,36,36,40,40,4,4,40,147,35,150,40,152,4,4,40,40,40,40,40,40,40,40,40,178,178,40,36,36,145,36,36,40,40,52,59,147,35,150,152,37,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,178,178,40,40,40,167,167,160,160,39,39,39,39,39,39,27,40,40,40,40,40,40,40,178,178,40,40,178,178,40,178,178,40,40,24,180,200,200,200,200,40,178,178,40,200,200,171,171,178,178,40,178,178,40,40,40,178,178,33,149,149,178,178,40,40,40,40,87,40,40,40,40,40,40,178,178,40,40,119,119,119,200,200,200,200,200,200,40,40,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,86,40,40,40,40,178,178,34,111,116,36,36,36,40,178,35,178,40,200,200,200,200,200,200,200,200,200,200,200,200,200,200,178,178,40,40,40,40,53,53,40,40,184,188,200,200,199,185,190,199,166,166,40,178,178,40,200,200,200,200,40,40,40,40,40,40,50,39,183,183,191,39,189,197,186,186,161,161,177,177,197,193,10,189,40,40,178,178,40,40,178,178,40,40,166,166,204,98,200,200,40,178,178,40,40,40,188,188,40,40,170,170,170,170,170,170,170,170,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,9,40,40,34,33,35,35,181,181,166,41,41,171,40,40,200,200,200,200,46,46,46,46,46,46,46,46,40,40,40,40,40,178,178,40,40,40,40,40,40,40,178,178,200,200,200,200,40,40,40,40,40,40,40,40,40,183,183,186,186,177,177,200,200,45,50,45,40,40,40,40,40,40,172,167,167,174,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,192,192,192,192,192,192,39,39,40,40,40,40,40,40,40,40,40,40,168,168,167,167,168,168,167,167,168,168,167,167,40,40,200,200,40,40,40,40,40,40,40,40,186,186,186,186,186,186,186,186,164,164,186,186,186,186,186,186,167,167,186,186,186,186,185,185,186,186,186,186,186,186,186,186,166,50,40,178,178,40,40,178,178,40,200,200,40,40,40,178,178,40,178,178,143,150,150,152,81,81,142,140,158,151,151,152,156,156,200,200,200,200,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,178,178,40,99,99,167,167,194,200,200,40,40,178,178,169,169,169,169,40,178,178,40,36,36,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,33,35,191,191,185,185,191,191,40,40,40,178,178,40,200,200,200,200,39,39,155,155,155,38,38,40,178,178,40,40,40,200,200,200,200,40,40,40,40,145,36,36,40,40,4,4,87,90,147,35,150,152,69,4,4,200,200,200,200,200,200,200,200,200,200,200,200,200,200,40,40,200,200,200,200,40,40,38,38,198,50,50,50,198,40,178,178,40,178,178,40,40,40,40,40,40,11,40,40,40,178,178,40,40,40,161,38,181,181,40,184,184,96,96,96,96,9,38,38,39,200,200,40,40,40,40,40,40,40,40,178,178,40,40,40,40,40,40,40,40,40,40,200,200,40,40,40,178,178,40,40,40,40,40,40,40,40,178,178,40,189,50,87,87,41,41,190,178,178,40,40,40,40,200,200,200,200,200,200,200,200,200,200,200,200,135,135,40,40,200,200,200,200,200,200,192,192,192,192,38,38,38,38,204,204,201,201,192,192,203,203,201,168,168,176,175,178,129,159,159,178,197,197,171,171,159,159,40,40,40,40,40,40,200,200,200,200,200,200,200,200,200,200,40,178,178,40,40,40,40,178,178,40,40,40,200,200,117,117,117,117,117,117,117,117,117,117,117,117,117,117,117,117,57,50,50,40,178,178,40,175,175,200,200,40,40,40,40,40,40,40,40,194,196,39,39,191,39,39,40,40,40,178,178,40,40,40,40,40,40,40,40,40,178,200,200,36,36,35,40,40,145,40,40,4,4,79,85,147,150,152,172,4,4,172,40,40,178,178,40,178,178,40,111,112,40,178,178,40,40,40,40,40,40,40,91,40,178,178,40,40,178,178,40,40,40,40,40,40,178,178,40,40,178,178,40,40,40,40,40,40,40,40,40,90,168,168,167,167,189,189,178,135,94,4,96,147,150,152,82,4,145,171,172,184,185,149,149,166,32,32,32,32,40,40,91,200,200,40,40,200,200,200,200,40,40,40,178,178,40,40,178,178,40,200,200,200,200,200,200],["Female","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Female","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Male","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Pooled","Pooled","Pooled","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Pooled","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Male","Female","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Male","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Male","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Male","Male","Male","Female","Female","Male","Female","Male","Female","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Male","Female","Male","Female","Male","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Pooled","Male","Female","Pooled","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Male","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Female","Male","Male","Female","Male","Pooled","Male","Male","Female","Female","Female","Female","Female","Female","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Female","Male","Male","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Pooled","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Male","Male","Male","Female","Female","Male","Male","Male","Female","Pooled","Male","Female","Male","Male","Male","Female","Male","Male","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Male","Male","Female","Male","Pooled","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Male","Male","Male","Male","Pooled","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Pooled","Male","Female","Pooled","Male","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Pooled","Pooled","Female","Female","Female","Female","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Female","Male","Male","Male","Male","Female","Male","Female","Female","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Pooled","Pooled","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Pooled","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Female","Male","Female","Female","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Male","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Male","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Pooled","Pooled","Female","Male","Female","Male","Female","Male","Male","Female","Male","Pooled","Male","Male","Male","Male","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Pooled","Female","Female","Female","Female","Male","Female","Female","Female","Male","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Pooled","Pooled","Male","Male","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Male","Male","Male","Male","Male","Pooled","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Female","Female","Pooled","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Female","Female","Female","Male","Female","Male","Male","Pooled","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Male","Male","Female","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Male","Male","Male","Pooled","Female","Male","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Pooled","Female","Male","Male","Female","Male","Male","Male","Male","Male","Male","Male","Male","Female","Male","Male","Female","Female","Male","Female","Female","Female","Female","Female","Male","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Female","Female","Female","Male","Male","Male","Male","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male"],["Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Behavioural","Morphological","Morphological","Morphological","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Activity","Activity","Behavioural","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Behavioural","Behavioural","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Pathogen response","Pathogen response","Pathogen response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Stress response","Stress response","Metabolome","Metabolome","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Behavioural","Metabolome","Metabolome","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Insecticide response","Microbiome","Microbiome","Metabolome","Metabolome","Drug response","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","Temperature related","Temperature related","Temperature related","Temperature related","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Temperature related","Temperature related","Activity","Activity","Activity","Activity","Activity","Activity","Metabolome","Metabolome","Metabolome","Metabolome","Physiological","Physiological","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Stress response","Stress response","Activity","Activity","Life history","Life history","Activity","Activity","Activity","Activity","Activity","Activity","Microbiome","Microbiome","Metabolome","Metabolome","Physiological","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Behavioural","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Insecticide response","Insecticide response","Metabolome","Metabolome","Insecticide response","Metabolome","Metabolome","Stress response","Stress response","Temperature related","Temperature related","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Physiological","Drug response","Drug response","Microbiome","Microbiome","Pathogen response","Morphological","Morphological","Life history","Microbiome","Pathogen response","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Drug response","Life history","Life history","Life history","Life history","Life history","Life history","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Metabolome","Metabolome","Activity","Activity","Microbiome","Microbiome","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Temperature related","Temperature related","Microbiome","Microbiome","Insecticide response","Life history","Life history","Life history","Life history","Metabolome","Metabolome","Activity","Activity","Pathogen response","Behavioural","Behavioural","Behavioural","Behavioural","Behavioural","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Behavioural","Genomic","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Metabolome","Sensory","Sensory","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Sensory","Sensory","Metabolome","Metabolome","Sensory","Sensory","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Behavioural","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Physiological","Physiological","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Temperature related","Stress response","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Insecticide response","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Life history","Morphological","Morphological","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Pathogen response","Pathogen response","Pathogen response","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Pathogen response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Pathogen response","Pathogen response","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Metabolome","Metabolome","Stress response","Stress response","Stress response","Stress response","Stress response","Life history","Life history","Stress response","Activity","Activity","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Behavioural","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Pathogen response","Pathogen response","Metabolome","Metabolome","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Behavioural","Behavioural","Behavioural","Stress response","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Life history","Life history","Life history","Life history","Life history","Life history","Microbiome","Microbiome","Life history","Life history","Life history","Physiological","Physiological","Physiological","Physiological","Physiological","Physiological","Drug response","Drug response","Drug response","Genomic","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Stress response","Stress response","Life history","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Behavioural","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Drug response","Drug response","Drug response","Drug response","Insecticide response","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Life history","Life history","Morphological","Morphological","Morphological","Life history","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Behavioural","Behavioural","Morphological","Morphological","Morphological","Morphological","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Physiological","Behavioural","Behavioural","Behavioural","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Pathogen response","Behavioural","Behavioural","Behavioural","Behavioural","Behavioural","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Drug response","Drug response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Microbiome","Behavioural","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Drug response","Drug response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Life history","Activity","Activity","Activity","Activity","Life history","Life history","Life history","Life history","Microbiome","Morphological","Microbiome","Morphological","Morphological","Morphological","Microbiome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Behavioural","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Pathogen response","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome"],["Mean relative 1/3-Methylhistidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1/3-Methylhistidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 1-Methyladenosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methyladenosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methylguanosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methylguanosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 13-HODE concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 13-HODE concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 2-Hydroxyglutarate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Hydroxyglutarate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 2-Phosphoglyceric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Phosphoglyceric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 3-Hydroxybutyric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxybutyric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 3-Hydroxykynurenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxykynurenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 4-Pyridoxic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 4-Pyridoxic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative 5-Aminovalerate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 5-Aminovalerate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is A. tropicalis; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Incidence of DGRP males abdomen curling in the presence of an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of abdomen curling.","Mean adult abdomen length, when reared at 18C. Higher values indicate larger flies.","Mean adult abdomen length, when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult abdomen lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean relative acetoacetate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetoacetate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative acetylcarnitine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcarnitine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative acetylcholine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcholine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.","An overall measure of activity level obtained by summing the total number of tapping, orientating, following, licking, singing and abdomen curling for a male housed in a vial with a single additional male. Higher values indicate greater activity.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative adenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative adenosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative ADP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ADP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of aggressive encounters between 8 males in an interaction vial over a two minute duration. Higher values indicate more aggressive males.","Mean number of aggressive lunges made in 20 min window following removal of a divider, by pairs of DGRP males of same genotype. Trials were run after males were kept isolated for five days. Higher values indicate more aggressive males.","Mean relative agmatine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative agmatine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative alanine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative alanine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative allantoin concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative allantoin concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean number of flies exposed to alpha-amanitin, at concentration 0.2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean number of flies exposed to alpha-amanitin, at concentration 2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean alpha-amanitin concentration required for 50% of larvae to die prior to reaching adulthood (LC50). Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a higher LC50 and a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative Alpha-Ketoglutaric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative Alpha-Ketoglutaric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean change in amino acid diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the amino acid mixture after deprivation.","Mean change in amino acid diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the amino acid mixture after deprivation.","Mean relative aminoisobutyrate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aminoisobutyrate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative AMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative AMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean survival (larvae that pupated) for late third instar larvae, 24 hours after exposure to one hour of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean relative anthranilate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative anthranilate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative arachidonate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arachidonate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative arginine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arginine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative asparagine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative asparagine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative aspartic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aspartic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.25 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.5 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 1 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 2 ug/ml. Higher values indicate greater mortality.","Mean dosage (ug/ml) of azinphos-methyl (an organophosphate insecticide) required for 50% of larvae to die. Higher values indicate greater resistance.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative betaine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative betaine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative biotin concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative biotin concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative cadaverine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cadaverine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative carnitine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative carnitine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative cGMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cGMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Preference is measured as the difference between the amount of cocaine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Preference is measured as the difference between the amount of cocaine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Preference is measured as the difference between the amount of methamphetamine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Preference is measured as the difference between the amount of methamphetamine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 0.5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 1 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 2 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 3 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 4 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean chlorantraniliprole dosage required for 50% of first instar larvae to die before eclosion. Chlorantraniliprole is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative choline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative choline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean citrate synthase activity per adult fly. It is involved in the first step of the TCA cycle, and is therefore considered a proxy for mitochondrial aerobic capacity. Higher values indicate higher citrate synthase activity.","Mean citrate synthase activity per adult fly. It is involved in the first step of the TCA cycle, and is therefore considered a proxy for mitochondrial aerobic capacity. Higher values indicate higher citrate synthase activity.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative citrulline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative citrulline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean climbing index score - calculated using a design that takes advantage of negative geotaxis - measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Mean climbing index score - calculated using a design that takes advantage of negative geotaxis - following a three-day exercise regime. Higher values indicate greater climbing ability.","Mean climbing index score - calculated using a design that takes advantage of negative geotaxis - prior to a three-day exercise regime. Higher values indicate greater climbing ability.","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 5% yeast. Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.","Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.","Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.","Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.","Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.","Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative CMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative CMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean CO2 production (uL per hour, per mg dry weight). The authors use this trait as a proxy for metabolic rate. Higher values indicate a faster metabolism.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean proportion of successfully eclosed individuals following an hour of cold shock (-5°C) at 5 days post oviposition (third instar). Higher values indicate larvae that are more resistant to cold shock.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.","Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.","Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving adults 24 hours after an hour of exposure to -1 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -2 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -3 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -4 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -5 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -6 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -7 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean time (mins) from a male flies introduction to a female until the onset of mating. Lower values indicate shorter copulation latencies and potentially more attractive males","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly ceases activity. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly ceases activity. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum (CT min), which is defined as the temperature at which a complete cessation of movement is observed as a result of the fly entering chill coma. Egg-to-adult development occurred at 23C. Higher values indicate a higher CT min.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Developmental stability across environments is the consistency of within-environment variation in CT min across environments. The authors calculate this by taking the highest coefficient of variation (CV) in CT min at any rearing temperature, minus the lowest CV at any other given rearing temperature. Lower values indicate greater developmental stability across rearing temperatures.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums were measured (CT min). Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The absolute value of the slope provides a measure of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The standard error of the slope provides a measure of the extent to which the same genotype can produce the same plastic response consistently. Higher values indicate greater variance in thermal plasticity.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums were measured. This trait captures how consistently a genotype produces the same phenotype at a given rearing temperature. Higher values indicate greater variance in phenotype within a rearing temperature.","Mean cumulative cold tolerance (CCT) score. This metric represents the shape of cold-survival curves calculated between -1°C and -7°C. Higher values indicate greater area under the curve, and therefore flies that are more resistant to cold shock.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative cystathionine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystathionine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative cystine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative cytidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative cytosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative D-GA3P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-GA3P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-Leucic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-Leucic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean knockdown of adults, four hours after vials were inoculated with 0.5 ug/ml of DDT. Knockdown is defined as either flies permanently seen to be in a prone position with jolting movement of legs or wings, or a prolonged inability to right themselves from a prone position. DDT is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean survival of adults, 24 hours after vials were inoculated with 0.5 ug/ml of DDT. This is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean change in proportion of adult flies surviving one hours of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean change in proportion of adult flies surviving one hour of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean Dopamine content (pg/ng tissue) extracted from the heads of two flies. Higher values indicate a greater dopamine content. Note that the 20 lines were chosen for extreme paraquat resistance or susceptibility values.","Mean Dopamine content (pg/ng tissue) extracted from the heads of two flies, after flies had been exposed to sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate a greater dopamine content. Note that the 20 lines were chosen for extreme paraquat resistance or susceptibility values.","Mean difference in Dopamine content (pg/ng tissue) extracted from the heads of two flies, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress on Dopamine content.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean days survived after infection with Drosophila C Virus (DCV) - a positive sense RNA virus that naturally infects multiple Drosophila species. Higher values indicate a longer lifespan, and therefore a greater resistance to DCV.","Dry weight of 24-hour old adult virgins. Higher values indicate heavier flies. Note that this data has not been published, but was collected and provided by G. Tollin, N. Elfiky, E. Davis, N. Morano, and H. Karp of the A. Fiumera lab.","Dry weight of 24-hour old adult virgins. Higher values indicate heavier flies. Note that this data has not been published, but was collected and provided by G. Tollin, N. Elfiky, E. Davis, N. Morano, and H. Karp of the A. Fiumera lab.","Mean duration of embryogenesis (DOE) - from fertilisation of an egg to hatching as a larva. Higher values indicate a longer period of embryogenesis.","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is E.faecalis; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis","Estimated mean time from eclosion to laying an egg (days), when reared under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Females cohabited a food vial with two males from the same DGRP line as the female. Calculated as T= Σ(lxmx)/R0, where lx = the proportion of females surviving each day, mx = the number of eggs laid each day and R0 is the net reproductive rate (or replacment rate). Higher values indicate a longer generation time. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Estimated mean time from eclosion to laying an egg (days), when reared under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Females cohabited a food vial with two males from the same DGRP line as the female. Calculated as T= Σ(lxmx)/R0, where lx = the proportion of females surviving each day, mx = the number of eggs laid each day and R0 is the net reproductive rate (or replacment rate). Higher values indicate a longer generation time. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Egg-to-adult development time when reared at 20C, recorded to the nearest 8 hours. Higher values indicate a longer development time.","Egg-to-adult development time when reared at 25C, recorded to the nearest 8 hours. Higher values indicate a longer development time.","Egg-to-adult development time when reared at 30C, recorded to the nearest 8 hours. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a standard diet. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a diet supplemented with 10% EtOH (ethanol). Higher values indicate a longer development time.","Proportion of 75 eggs in a vial that successfully survived to adulthood at 25C. Higher values indicate greater survival.","Mean proportion of flies surviving to adulthood, when reared at 25C on control food medium. Higher values indicate greater survival.","Mean proportion of eggs that survived to adulthood, when reared at 25C. Higher values indicate greater viability.","Percentage of flies surviving to eclosion, after development on standard food. Higher values indicate greater viability.","Proportion of 75 eggs in a vial that successfully survived to adulthood at 20C. Higher values indicate greater survival.","Mean proportion of eggs that survived to adulthood, when reared at 28C. Higher values indicate greater viability.","Proportion of 75 eggs in a vial that successfully survived to adulthood at 30C. Higher values indicate greater survival.","Percentage of flies surviving to eclosion, after development on food supplemented with 10% EtOH (ethanol). Higher values indicate greater viability.","Mean proportion of eggs that hatched. Higher values indicate a greater proportion of viable eggs.","Proportion of 75 eggs in a vial that successfully survived to pupation at 20C. Higher values indicate greater survival.","Proportion of 75 eggs in a vial that successfully survived to pupation at 25C. Higher values indicate greater survival.","Proportion of 75 eggs in a vial that successfully survived to pupation at 30C. Higher values indicate greater survival.","Average number of eggs laid per day across a female’s lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Higher values indicate greater fecundity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Average number of eggs laid per day across a female’s lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Higher values indicate greater fecundity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative epinephrine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative epinephrine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","The mean difference in egg-to-adult development time in days, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Higher values indicate a longer development time for ethanol-supplemented flies.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean difference in the amount of time adult flies remained mobile across a 60 second period following mechanical disturbance, between those grown on 10% EtOH (ethanol) supplemented food and those that developed on standard medium. Lower values indicate a greater negative effect on startle response.","Mean difference in the amount of time adult flies remained mobile across a 60 second period following mechanical disturbance, between those grown on 10% EtOH (ethanol) supplemented food and those that developed on standard medium. Lower values indicate a greater negative effect on startle response.","The mean difference in the percentage of flies surviving to eclosion, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Lower values indicate a greater negative effect of ethanol on egg to adult viability.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of eggs produced by single females, per day. Higher values indicate more fecund females.","Mean fertilised egg retention score. An estimation of the progression of embryo development taken two hours after a one hour egg laying period. Higher values indicate more developed embryos which signifies a greater degree of egg retention. Note that both males and femlaes were sourced from the DGRP line, therefore variation could be explained by either the male or female genotype.","Mean proportion of females that produced viable larvae. Females were reared at 25C. Higher values indicates greater fertility.","Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.","Mean proportion of females that produced viable larvae. Females were reared at 27C. Higher values indicates greater fertility","Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility","Mean proportion of females that produced viable larvae. Females were reared at 29C. Higher values indicates greater fertility","Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility","Mean slope of the change in female fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.","Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Larval to adult survival on food media supplemented with 0.5PPM fipronil – an insecticide from the phenylpyrazole family, released for commercial use in 1993. Higher values indicate greater resistance.","Mean number of L1 larvae produced by females aged 5-6 days over a 24h period. For each replicate measurement, 5 DGRP females were housed with standardised mates and opposite-sex competitors (15 males and 10 females, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae sired by males in competition for matings/fertilisations with same-aged competitor males over 3 days, beginning when all males were aged 2-3 days. For each replicate measurement, 5 DGRP males were housed with standardised mates and opposite-sex competitors (15 females and 10 males, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae produced by females aged 17-18 days over a 24h period. For each replicate measurement, 5 DGRP females were housed with standardised mates and opposite-sex competitors (15 males and 10 females, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae sired by males in competition for matings/fertilisations with same-aged competitor males over 3 days, beginning when all males were aged 14-15 days. For each replicate measurement, 5 DGRP males were housed with standardised mates and opposite-sex competitors (15 females and 10 males, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean height (m) at which flies catch themselves from freefall after vial inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean height (m) at which flies catch themselves from freefall after rapid inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean days survived after infection with Flock House Virus (FHV) - a nodavirus virus that typically infects beetles. Higher values indicate a longer lifespan, and therefore a greater resistance to FHV.","Incidence of DGRP males following an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of following.","Mean quantity of 10% sucrose, 5% yeast nutrient solution consumed by five virgin flies over an 8 hour period, measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate greater food consumption.","Mean quantity of 4% sucrose solution consumed per fly iover a 24 hour period. Higher values indicate greater food intake.","Mean quantity of 4% sucrose solution consumed per fly over a 24 hour period. Higher values indicate greater food intake.","Mean quantity of 10% sucrose, 5% yeast nutrient solution consumed by five virgin flies over an 8 hour period, following a three-day exercise regime. Higher values indicate greater food consumption.","Adult mean survival when flies were provided access to food over a 5 day period. Higher values indicate greater survival and assumed greater foraging ability.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative G1P, G6P, F6P and F1P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolites. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative G1P, G6P, F6P and F1P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolites. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.","Mean size of the genome in mb, estimated using flow cytometry. Higher values indicate a larger genome.","Mean relative geranyl pyrophosphate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative geranyl pyrophosphate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative glucoronate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glucoronate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Estimated mass of free glucose per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean quantity of glucose per adult fly. Higher values indicate a higher glucose content.","Mean quantity of glucose per adult fly. Higher values indicate a higher glucose content.","Mean free glucose (mg/ul) extracted from 10 flies. Higher values indicate greater levels of free glucose.","Mean free glucose (mg/ul) extracted from 10 flies. Higher values indicate greater levels of free glucose.","Mean abdomen glucose content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen glucose content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean preference for glucose when given a choice between glucose and fructose. Higher values indicate a greater preference for glucose consumption.","Mean preference for glucose when given a choice between glucose and fructose. Higher values indicate a greater preference for glucose consumption.","Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of free glucose.","Mean relative glucose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of glucose per third instar larvae. Higher values indicate a higher glucose content.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of free glucose.","Mean relative glucose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of free glucose.","Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.","Mean preference for glucose when given a choice between glucose and sucrose. Higher values indicate a greater preference for glucose consumption.","Mean preference for glucose when given a choice between glucose and sucrose. Higher values indicate a greater preference for glucose consumption.","Mean thorax glucose content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax glucose content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean preference for glucose when given a choice between glucose and trehalose. Higher values indicate a greater preference for glucose consumption.","Mean preference for glucose when given a choice between glucose and trehalose. Higher values indicate a greater preference for glucose consumption.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher glucose levels.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher glucose levels.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative glutamic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative glutamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative glyceraldehyde concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glyceraldehyde concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative glycerate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative glycerol-3-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerol-3-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Estimated mass of free glycerol per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean quantity of glycerol per adult fly. Higher values indicate a higher glycerol content.","Mean quantity of glycerol per adult fly. Higher values indicate a higher glycerol content.","Mean free glycerol (mg/ul) extracted from 10 flies. Higher values indicate greater levels of free glycerol.","Mean free glycerol (mg/ul) extracted from 10 flies. Higher values indicate greater levels of free glycerol.","Mean abdomen glycerol content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen glycerol content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycerol.","Mean quantity of glycerol per third instar larvae. Higher values indicate a higher glycerol content.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycerol.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycerol.","Mean thorax glycerol content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax glycerol content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative glycine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean change in glycine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the glycine supplemented medium after deprivation.","Mean change in glycine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the glycine supplemented medium after deprivation.","Estimated mass of glycogen stores per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean quantity of glycogen per adult fly. Higher values indicate a higher glycogen content.","Mean quantity of glycogen per adult fly. Higher values indicate a higher glycogen content.","Mean glycogen (mg/ul) extracted from 10 flies. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies. Higher values indicate greater levels of glycogen.","Mean combined glycogen content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glycogen content.","Mean combined glycogen content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glycogen content.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycogen.","Mean quantity of glycogen per third instar larvae. Higher values indicate a higher glycogen content.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycogen.","Mean change in the combined glycogen content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glycogen content. Note that the presence of micriobiota on average increased glycogen content.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative GMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative GMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean seconds spent grooming in a five minute interval. Higher values indicate more frequent grooming.","Mean seconds spent grooming in a five minute interval. Higher values indicate more frequent grooming.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative guanosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative guanosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","The mean proportion of 19 day-old flies with permeable guts, following no exposure to radiation. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","The mean proportion of 19 day-old flies with permeable guts, following exposure to 100 gray units of ionising radiation when flies were 5 days old. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.","Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.","Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.","Mean time elapsed within a 37C waterbath before the fly was comatose. Higher values indicate greater resistance to heat stress.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative histamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative histidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative homogentisate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative homogentisate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean expression of the heat shock protein Hsp70, after exposure to 35C for one hour and an additiol hour of recovery at 25C. Higher values indicate greater expression of Hsp70.","Lifespan on a diet supplemented with 2-3% hydrogen peroxide (H2O2). Data are corrected for line body mass. Higher values indicate greater resistance to hydrogen peroxide.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative hypoxanthine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative hypoxanthine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean wiggle index for third instar larvae 1 hour into exposure to 100ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean wiggle index for third instar larvae 1 hour into exposure to 25ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean relative IMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative IMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative inosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative inositol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inositol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of 10 females that a single male successfully inseminated in 48 hours. Insemination was verified by identification of viable pupal progeny. Higher values indicate males with a greater insemination capacity.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean relative iso-leucine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative iso-leucine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in isolation for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative isoValeric acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative isoValeric acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean survival after infection with the Kunjin subtype of the West Nile Virus. Calculated as the log(hazard ratio). Higher values indicate greater susceptibility to the virus.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative L-Kynurenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative L-Kynurenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is L.bevis; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","The median number  of colony forming units (originally log transformed) 48 hrs after initial infection with the bacterial pathogen L.monocytogenes. Higher values indicate a greater bacterial load and a more severe infection.","Median days till death after injection with 1000 colony forming units of the bacterial pathogen L. monocytogenes. Higher values indicate greater resistance to the L. monocytogenes.","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is L.plantarum; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Mean quantity of lactate per adult fly. Higher values indicate a higher lactate content.","Mean quantity of lactate per adult fly. Higher values indicate a higher lactate content.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative lactate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of lactate per third instar larvae. Higher values indicate a higher lactate content.","Mean relative lactate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative lactose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lactose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Larval size in mm 7 days post-embryonic development in protein poor conditions (lacking yeast). Larvae were purged of a microbiome and left to develop. Larger values indicate longer larvae.","Larval size in mm 7 days post-embryonic development in protein poor conditions (lacking yeast). Larvae were purged of a microbiome and subsequently associated with the commensal gut bacteria Lactobacillus plantarum. Larger values indicate longer larvae.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean difference in egg-to-adult viability between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a smaller negative effect of lead on viability.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean proportion of flies surviving to adulthood, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater survival.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative leucine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative leucine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Incidence of DGRP males licking an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of licking.","Mean days survived, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Higher values indicate greater longevity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days. Higher values indicate a longer lifespan.","Mean days survived, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Higher values indicate greater longevity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean adult lifespan, measured as the number of days from eclosion to death, when living with a single  standard male, at 25C. Higher values indicate longer lived flies.","Adult lifespan, measured in days, for virgin females. Higher values indicate greater longevity.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 5% yeast. Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days for virgin females following diapause induction (10 ̊C for 35 days, moved to 18 ̊C for a day of acclimation, remainder of their lives at 25 ̊C). Higher values indicate a longer lifespan.","Mean lifespan of 20 females maintained on a high sugar diet. Means are adjusted for block effects. Higher values indicate longer lived flies.","Mean number of days at which 50% of 20 adult flies had died in a vial filled with standard medium. Higher values indicate longer lived flies.","Fecundity was estimated every other week of the female’s lifespan as a 2-day total of eggs laid, when living with a single standard male, at 25C. Lifetime fecundity was estimated as the sum of all egg counts across the female's lifespan. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative linoleic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linoleic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative linolenic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linolenic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean total distance traveled by individual flies during a five minute interval. Higher values indicate more active lines.","The proportion of flies engaging in a ‘walking bout’ at a given time point. Higher values indicate more active lines.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative lysine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lysine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative malondialdehyde concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative malondialdehyde concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative margaric acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative margaric acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean mass of 4–6 females weighed together in milligrams. Higher values indicate heavier females.","Mean mass (mg) of 10 flies. Higher values indicate heavier flies.","Mean mass (mg) of 10 flies. Higher values indicate heavier flies.","Mean mass (mg). Higher values indicate heavier flies.","Mean mass (mg). Higher values indicate heavier flies.","Mean mass (mg) following forced exercise via rotational stimulation for 2h a day for 5 days. Higher values indicate heavier flies.","Mean mass (mg) following forced exercise via rotational stimulation for 2h a day for 5 days. Higher values indicate heavier flies.","Mean virgin female body mass for groups of three flies, after rearing on food with 0.5% yeast content (dietary restricted). Higher values indicate heavier flies.","Mean virgin female body mass for groups of three flies, after rearing on food with 5% yeast content (ab libitum diet). Higher values indicate heavier flies.","Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.","Mean time from the start of cohabitation until a male from the Canton S line successfully initiated copulation with a DGRP female (unclear whether these flies were virgin). Higher values indicate a longer mating latency.","Mean time from the start of cohabitation until a male from a DGRP line successfully initiated copulation with a Canton S female (unclear whether these flies were virgin). Higher values indicate a longer mating latency.","Methylmercury (MeHg) is a persistent environmental toxin present in seafood that can compromise the developing nervous system in humans. To assess overall developmental tolerance on MeHg media, the authors calculate an eclosion index – higher values indicate greater relative egg-to-adult viability when reared on MeHg supplemented media.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative methionine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methionine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative methylmalonate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methylmalonate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","The time elapsed from when eggs were placed into the vial until observation of the first pupal case at an 8 hour tempo resolution, at 20C. Higher values indicate a longer development time.","The time elapsed from when eggs were placed into the vial until observation of the first pupal case at an 8 hour tempo resolution, at 25C. Higher values indicate a longer development time.","The time elapsed from when eggs were placed into the vial until observation of the first pupal case at an 8 hour tempo resolution, at 30C. Higher values indicate a longer development time.","Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.","Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.","Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.","Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.","Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.","Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.","Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean copy number of mitochondrial genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more mitochondrial genomes, which the authors suggest is a proxy for mitochondrial density.","% of male brains dissected that had abnormally formed alpha lobes in their mushroom bodies. Higher values indicate a greater % of abnormally formed lobes.","% of male brains dissected that had abnormally formed beta lobes in their mushroom bodies. Higher values indicate a greater % of abnormally formed lobes.","% of male brains dissected that had fused beta lobes in their mushroom bodies. Higher values indicate a greater % of fused lobes.","% of male brains dissected that had missing alpha lobes in their mushroom bodies. Higher values indicate a greater % of missing lobes.","% of male brains dissected that had missing beta lobes in their mushroom bodies. Higher values indicate a greater % of missing lobes.","% of male brains dissected that had short alpha lobes in their mushroom bodies. Higher values indicate a greater % of short lobes.","% of male brains dissected that had short beta lobes in their mushroom bodies. Higher values indicate a greater % of short lobes.","% of male brains dissected that had thin alpha lobes in their mushroom bodies. Higher values indicate a greater % of thin lobes.","% of male brains dissected that had thin beta lobes in their mushroom bodies. Higher values indicate a greater % of thin lobes.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative myristic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative myristic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative N-AcetylGlycine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative N-AcetylGlycine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean number of eggs laid over a female's entire lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Calculated as R0= Σ(lxmx), where lx = the proportion of females surviving each day and mx = the number of eggs laid each day. Higher values indicate a higher net reproductive rate. This trait was measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean number of eggs laid over a female's entire lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Calculated as R0= Σ(lxmx), where lx = the proportion of females surviving each day and mx = the number of eggs laid each day. Higher values indicate a higher net reproductive rate. This trait was measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.","Incidence of DGRP males orientating towards an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of orientating.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative ornithine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ornithine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative orotate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative orotate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean difference between the number of ovarioles in the right and left ovary. Higher values indicate greater asymmetry.","Mean number of ovarioles in the right and left ovary. Higher values indicate a greater number of ovarioles.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative oxaloacetate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxaloacetate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative oxypurinol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxypurinol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean difference in larval viability between control and p-xylene-treated (concentration 1.068mM) first instar larvae after 14 days of development. P-xylene is an aromatic hydrocarbon associated with gasoline and diesel-powered engine emissions. Higher values indicate a greater negative effect of P-xylene on larval viability.","Mean number of fecal spots detected after 20h for 10 flies, after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate more defecation.","Mean pH of fecal matter for flies after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a higher pH.","Mean amount of intestinal mitosis, five days after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a greater number of mitotic itensitinal cells.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of flies in a vial to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean percentage of flies dying after 3 days post infection with P.entomophila L48, an entomopathogenic bacterium. Higher values indicate greater mortality.","Mean colony forming units (CFU) produced from the homogenate of three pooled flies, 20 hours after infection with P. rettgeri, a bacterial pathogen. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 10% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 2.5% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 5% glucose. Higher values indicate a greater bacterial load.","Mean proportion of flies surviving infection, two days after inoculation with the bacterial pathogen P. rettgeri. Higher values indicate greater resistance to P. rettgeri.","A general, nonlinear relationship between the proportion of flies surviving two days post infection and sustained pathogen load, after injection with the bacterial pathogen P. rettgeri, was predicted. The data presented here are the line deviations from the predicted relationship between survival and bacterial load. Positive values indicate a greater tolerance to P. rettgeri.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative pantothenate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pantothenate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative PEP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative PEP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative phenylalanine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative phenylalanine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean change in phenylalanine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the phenylalanine supplemented medium after deprivation.","Mean change in phenylalanine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the phenylalanine supplemented medium after deprivation.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean phototaxis score measured in 7 day old adults, without a light source. This measure was taken to approximate the effect of fly activity on the phototaxis response. Higher values indicate more active flies.","Mean phototaxis score measured in 7 day old adults, without a light source. This measure was taken to approximate the effect of fly activity on the phototaxis response. Higher values indicate more active flies.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative pipecolate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pipecolate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Lambda. Population growth rate or intrinsic rate of increase. Defined as e^r, where r = ln(net reproductive rate) / generation time. Higher values indicate faster population growth rates. This trait was measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Populations consisted of one female and two males from the same DGRP line. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Lambda. Population growth rate or intrinsic rate of increase. Defined as e^r, where r = ln(net reproductive rate) / generation time. Higher values indicate faster population growth rates. This trait was measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Populations consisted of one female and two males from the same DGRP line. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean PC1 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 78% of the variation in shape, and describes the degree of beak-like protrusion at the tip of the posterior lobes. Higher values indicate a greater degree of beak-like postrusion.","Mean PC2 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 10% of the variation in shape, and describes the width of the posterior lobes. Higher values indicate wider lobes.","Mean posterior lobe size, measured in pixels. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. Higher values indicate larger posterior lobes.","Mean number of progeny produced in 5 vials over 10 days by twice-mated females.","P1 is the proportion of progeny sired by the first male to mate with a twice-mated female. In this study, a DGRP female mated with two standard males. Hence, in this study, the variation in P1 is attributed to the female genotype. Higher values indicate a greater proportion of offspring sired by the male that mated first.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative proline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative proline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Estimated mass of soluble protein per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean quantity of protein per adult fly. Higher values indicate a higher protein content.","Mean quantity of protein per adult fly. Higher values indicate a higher protein content.","Mean protein (mg/ul) extracted from 10 flies. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies. Higher values indicate greater levels of protein.","Mean abdomen protein content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen protein content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.","Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of protein.","Mean quantity of protein per third instar larvae. Higher values indicate a higher protein content.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of protein.","Mean change in the combined protein content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher protein content. Note that the presence of micriobiota on average increased protein content.","Mean thorax protein content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax protein content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with 'punishment'. Higher values also indicate a higher learning score","Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with 'punishment'. Higher values also indicate a greater learning score.","Mean pupal case length in mm. Measurements were taken with image analysis software from at least 6 vials with a minimum of 15 pupa per vial, and were converted from pixels to mm based on the ratio to the diameter measurement of 1 euro cent coin (16.25 mm diameter)","Average female pupal weight, reared at 20C. Higher vales indicate heavier pupa.","Average female pupal weight, reared at 25C. Higher vales indicate heavier pupa.","Average female pupal weight, reared at 30C. Higher vales indicate heavier pupa.","Mean height at which larvae pupated in a 95mm Drosophila vial, after accounting for food height. Higher values indicate higher pupation sites.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative putrescine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative putrescine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative pyridoxal-5-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyridoxal-5-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Larval to adult survival on food media supplemented with 4PPM Pyripole – an insecticide from the phenylpyrazole family, released for commercial use in 2007. Higher values indicate greater resistance.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative pyruvate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyruvate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean survival 24 hours after exposure to radiation (Cs137, avg dose of 4.85 Gy/min. Total dose of 1382 Gy over the 4 hours and 45 minutes of continuous exposure). Higher values indicate greater radiation resistance.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature before exposing them to a cold temperature, while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared under fluctuating controlled temperatures, with a mean of 16.8C ± 6, prior to phenotyping. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature before exposing them to a cold temperature, while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared under fluctuating controlled temperatures, with a mean of 23.8C ± 6, prior to phenotyping. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Mean Reactive oxygen species (ROS) levels in the guts of flies. Higher values indicate greater levels of ROS.","Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with 'relief'. Higher values also indicate a greater learning score.","Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with 'relief'. Higher values also indicate a greater learning score.","Mean proportion of females that remated with a second male (genotype bwD), ~ 1 day after first mating with a male (genotype cn bw).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative ribose-5-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ribose-5-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative sarcosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sarcosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative serine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative serine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean proportion of adults infected with sigma virus (DmelSV), 13 days post infection. Sigma virus is a rhabdovirus that naturally occurs in D. melanogaster. It causes flies paralysis in flies exposed to CO2. Higher values indicate a higher (???) proportion of paralysed flies after exposure to CO2, and therefore greater sensitivity to sigma virus.","Incidence of courtship ‘song’ produced by a DGRP male when interacting with another male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of singing.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in a mixed Sex environment for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Mean difference in aggressive behavior between male flies raised in social and socially isolated environments. Higher values indicate that males raised in a social environment were more aggressive than males experiencing isolation (the authors adjusted line means to meet normality expectations, we present udjusted means for the social and isolation line means).","Mean nearest neighbour distances for a group of 25 flies, after one hour of acclimation in the environment. Higher values indicate greater spacing between individuals.","Mean nearest neighbour distances for a group of 25 flies, after one hour of acclimation in the environment. Higher values indicate greater spacing between individuals.","Median interpulse interval (IPI) in the male courtship 'song'. Males in many Drosophila species produce a series of auditory pulses (a pulse 'song') when courting females, and the pause between pulses (IPI) varies greatly among species. Higher values indicate a longer IPI.","Mean relative sorbitol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sorbitol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Note: this data is recoded from that recorded by Watanabe et al (2020).","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Higher values indicate that a greater proportion of activity was completed in the first 30 minutes, which corresponds to a 'sprinter' phenotype. Note: this data is recoded from that recorded by Watanabe et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on standard food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on standard food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on 10% EtOH (ethanol) supplemented food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on 10% EtOH (ethanol) supplemented food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Adult mean survival in the absence of food after two days is here used as a measure of starvation resistance. Higher values indicate greater survival and therefore higher starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high carbohydrate diet (carbs:494.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high fat diet (carbs:44.15g/L, protein:52g/L, fat:17.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high protein diet (carbs:44.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean hours survived for larvae subjected to nutrient poor starvation medium (20% sucrose in PBS solution). Higher values indicate greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on the laboratories standard diet (carbs:119.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 0.5% yeast content (dietary restricted).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 5% yeast content (ab libitum diet).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old.  Higher values indicate flies that exhibited greater starvation resistance.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative succinate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative succinate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative sucrose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sucrose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean change in combined triglyceride content after glycerol substraction of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher TAG content. Note though that the presence of micriobiota on average decreased TAG content.","Incidence of a male being tapped by an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of being tapped.","Male tapping; in this case when a male brings the tarsus of a foreleg into contact with another male. Higher values indicate a greater incidence of tapping.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative taurine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative taurine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean pigmentation scores for tergite 5. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean pigmentation scores for tergite 6. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean adult thorax length when reared at 18C. Higher values indicate larger flies.","Mean adult thorax length when reared at 28C. Higher values indicate larger flies.","Mean thorax length (mm). The authors use this as a proxy for body size. Higher values indicate a longer thorax.","Mean thorax length (mm). The authors use this as a proxy for body size. Higher values indicate a longer thorax.","Flies were reared at 18C and 28C, after which adult thorax lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Mean thorax width (mm). The authors use this as a proxy for body size. Higher values indicate a wider thorax.","Mean thorax width (mm). The authors use this as a proxy for body size. Higher values indicate a wider thorax.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative threonine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative threonine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","The mean percentage of flies that died within 24 hours, after suffering traumatic injury. Higher values indicate greater mortality.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean quantity of triacylglycerol per adult fly. Higher values indicate a higher triacylglycerol content.","Mean quantity of triacylglycerol per adult fly. Higher values indicate a higher triacylglycerol content.","Mean quantity of triacylglycerol per third instar larvae. Higher values indicate a higher triacylglycerol content.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Estimated mass of total trigylcerides per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean triglyceride (mg/ul) extracted from 10 flies. Higher values indicate greater levels of triglyceride.","Mean triglyceride (mg/ul) extracted from 10 flies. Higher values indicate greater levels of triglyceride.","Mean abdomen triglyceride content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen triglyceride content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher TAG content.","Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of triglycerides.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher triglyceride levels.","Mean thorax triglyceride content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax triglyceride content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher triglyceride levels.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative trimethylamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative trimethylamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative tryptophan concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tryptophan concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","The hazard ratio of the death rate of male flies exposed to tunicamycin - an endoplasmic reticulum (ER) stress inducing drug – with the control group, fed drug free media. Higher values indicate greater ER stress and lower survival of flies exposed to tunicamycin.","Mean time required for 50% of male flies exposed to tunicamycin (TM) media to die. TM is an antibiotic that causes endoplasmic reticulum (ER) stress. Higher values indicate longer survival times and greater resistance to ER stress.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative tyrosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tyrosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Proportion of unfertilised eggs produced by 3-4 flies that had been allowed to mate for at least two days beforehand. Note that flies were kept in population cages that contained both sexes from the same line during the mating period. This means that genetic variance in this trait variation could be explained by either the male or female genotype. Higher values indicate a greater proportion of unfertilised eggs.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative uracil concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uracil concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative urate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative urate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative uridine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uridine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative valine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative valine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean unfertilised egg retention time, measured as the number of days before a virgin female laid her first egg. Higher values indicate a longer retention time.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean fecundity in the first week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the third week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the fifth week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean fecundity in the seventh week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.","Mean wet weight of flies, measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate heavier flies.","Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 10% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 2.5% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 5% glucose. Higher values indicate heavier flies.","Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.","Mean wet weight of flies, following a three day exercise regime. Higher values indicate heavier flies.","Mean mass. Higher values indicate heavier flies.","Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.","Wing imaginal disk size in third instar larvae – measured as absolute centroid size. Larger values indicate larger wing disks.","Wing imaginal disk size in third instar larvae – measured as absolute centroid size. Larger values indicate larger wing disks.","Wing imaginal disk size in third instar larvae – measured as centroid size after controlling for body size. Larger values indicate larger wing disks relative to body size.","Wing imaginal disk size in third instar larvae – measured as centroid size after controlling for body size. Larger values indicate larger wing disks relative to body size.","Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.","Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.","Mean copy number of Wolbachia genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more Wolbachia genomes, which the authors suggest is a proxy for Wolbachia density. Note that only lines found to harbour Wolbachia are inlcuded in this dataset.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative xanthosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean relative xanthurenate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthurenate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species."],["Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Early et al (2017) PLOS One","Bailey and Hoskins (2014) Evolution","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Rohde et al (2019) Genetics","Rohde et al (2019) Genetics","Bailey and Hoskins (2014) Evolution","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Shorter et al (2015) PNAS","Chowdhury et al Nature Communications Biology","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Mitchell et al (2017) PLOS One","Mitchell et al (2017) PLOS One","Mitchell et al (2017) PLOS One","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Toshima et al (2014) Journal of Insect Physiology","Toshima et al (2014) Journal of Insect Physiology","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Campwell et al (2019) G3","Campwell et al (2019) G3","Campwell et al (2019) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Gaertner et al (2015) G3","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Gaertner et al (2015) G3","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Najarro, Hackett and Macdonald (2017) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Najarro et al (2015) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Mendez et al (2016) PLOS One","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Duun Rohde et al (2016) International Journal of Genomics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Freda et al (2017) Integrative and Comparative Biology","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2017) Integrative and Comparative Biology","Freda et al (2017) Integrative and Comparative Biology","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Dean et al (2020) Evolution","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Rolandi et al (2018) Ecology and Evolution","Rolandi et al (2018) Ecology and Evolution","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Teets and Hahn (2018) Journal of Evolutionary Biology","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Schmidt et al (2017) Genetics","Schmidt et al (2017) Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Duneau et al (2018) G3","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Rajpurohit et al (2018) Molecular Ecology","Rajpurohit et al (2018) Molecular Ecology","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Magwire et al (2012) PLOS Genetics","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Horvath, Betancourt and Kalinka (2016) BMC Evolutionary Biology","Early et al (2017) PLOS One","Chapman et al (2020) Genes","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Morozova (2018) G3","Morozova (2018) G3","Ellis et al (2014) PLOS Genetics","Zhou et al (2016) Environmental Health Perspectives","Duun Rohde et al (2016) International Journal of Genomics","Morozova (2018) G3","Ellis et al (2014) PLOS Genetics","Duun Rohde et al (2016) International Journal of Genomics","Ellis et al (2014) PLOS Genetics","Morozova (2018) G3","Horvath, Betancourt and Kalinka (2016) BMC Evolutionary Biology","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Morozova (2018) G3","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Evangelou et al (2019) G3","Horvath and Kalinka (2018) Evolution","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Remnant et al (2014) Insect Biochemistry and Molecular Biology","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Spierer et al (2021) PLOS Genetics","Spierer et al (2021) PLOS Genetics","Magwire et al (2012) PLOS Genetics","Bailey and Hoskins (2014) Evolution","Mendez et al (2016) PLOS One","Garlapow et al (2015) PLOS One","Garlapow et al (2015) PLOS One","Mendez et al (2016) PLOS One","Lee et al (2017) Genome Biology and Evolution","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Gaertner et al (2015) G3","Huang et al (2014) Genome Research &amp; Ellis et al (2014) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) G3","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Dobson et al (2015) Nature Communications","Zhou et al (2020) Genome Research","Uchizono and Tamimura (2017) Chemical Senses","Uchizono and Tamimura (2017) Chemical Senses","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Uchizono and Tamimura (2017) Chemical Senses","Uchizono and Tamimura (2017) Chemical Senses","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Uchizono and Tamimura (2017) Chemical Senses","Uchizono and Tamimura (2017) Chemical Senses","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Zhou et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Zhou et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Toshima et al (2014) Journal of Insect Physiology","Toshima et al (2014) Journal of Insect Physiology","Unckless, Rottschaefer and Lazzaro (2015) G3","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Hope et al (2019) Frontiers in Genetics","Hope et al (2019) Frontiers in Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Sharma et al (2020) Scientific reports","Sharma et al (2020) Scientific reports","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Duun Rohde et al (2016) International Journal of Genomics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Duun Rohde et al (2016) International Journal of Genomics","Harrison et al (2020) BMC Genomics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Denecke et al (2017) Scientific Reports","Denecke et al (2017) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Dean et al (2020) Evolution","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Rohde et al (2017) Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Ahlers et al (2019) Cell Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Early et al (2017) PLOS One","Hotson and Schneider (2015) G3","Hotson and Schneider (2015) G3","Early et al (2017) PLOS One","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Ma et al (2019) iScience","Ma et al (2019) iScience","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Bailey and Hoskins (2014) Evolution","Gerken, Eller and Morgan (2014) PhD thesis","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Gerken, Eller and Morgan (2014) PhD thesis","Durham et al (2014) Nature Communications","Ivanov et al (2015) The Journals of Gerontology: Series A","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Patel and Talbert (2021) Heliyon","Everman and Morgan (2017) Evolution","Durham et al (2014) Nature Communications","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Rohde et al (2018) G3","Maesani et al (2015) PLOS Computational Biology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Arbuthnott, Levin and Promislow (2016) Journal of Evolutionary Biology","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Watanabe and Riddle (2021) Scientific Reports","Watanabe and Riddle (2021) Scientific Reports","Watanabe and Riddle (2021) Scientific Reports","Watanabe and Riddle (2021) Scientific Reports","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Gaertner et al (2015) G3","Hope et al (2019) Frontiers in Genetics","Hope et al (2019) Frontiers in Genetics","Montgomery et al (2014) PLOS One","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Rohde et al (2019) Genetics","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Richardson (2012) PLOS Genetics","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Swarup et al (2013) PNAS","Swarup et al (2013) PNAS","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Gaertner et al (2015) G3","Bailey and Hoskins (2014) Evolution","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Lobell et al (2017) G3","Lobell et al (2017) G3","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Adebambo et al (2020) Frontiers in Genetics","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Evangelou et al (2019) G3","Bou Sleiman et al (2015) Nature Communications","Howick and Lazzaro (2017) Molecular Ecology","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Howick and Lazzaro (2017) Molecular Ecology","Howick and Lazzaro (2017) Molecular Ecology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Duneau et al (2018) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Toshima et al (2014) Journal of Insect Physiology","Toshima et al (2014) Journal of Insect Physiology","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Chow, Wolfner and Clark (2013) Genetics","Chow, Wolfner and Clark (2013) Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) G3","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Appel et al (2016) Biology Letters","Appel et al (2016) Biology Letters","Zhang et al (2021) Current Biology","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Zhang, Reeves and Tautz (2020) Molecular Ecology","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Remnant et al (2014) Insect Biochemistry and Molecular Biology","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Vaisnav et al (2014) PLOS One","Gerken, Eller and Morgan (2014) PhD thesis","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken, Eller and Morgan (2014) PhD thesis","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Early et al (2017) PLOS One","Appel et al (2016) Biology Letters","Appel et al (2016) Biology Letters","Chow, Wolfner and Clark (2013) Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Magwire et al (2011) PLOS Genetics","Bailey and Hoskins (2014) Evolution","Rohde et al (2017) Genetics","Rohde et al (2017) Genetics","Hope et al (2019) Frontiers in Genetics","Hope et al (2019) Frontiers in Genetics","Turner, Miller and Cochrane (2013) Molecular Biology and Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Riddle (2020) Journal of Experimental Biology","Riddle (2020) Journal of Experimental Biology","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Lee et al (2017) Genome Biology and Evolution","Everman et al (2019) Genetics","Everman et al (2019) Genetics","Francis et al (2021) G3","Francis et al (2021) G3","Francis et al (2021) G3","Akcay (2017) Thesis","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Francis et al (2021) G3","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Dobson et al (2015) Nature Communications","Bailey and Hoskins (2014) Evolution","Bailey and Hoskins (2014) Evolution","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Dembeck et al (2015) PLOS Genetics","Dembeck et al (2015) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Katzenberger et al (2015) eLife","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) G3","Zhou et al (2019) Genome Research","Zhou et al (2019) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Nelson et al (2016) BMC Genomics","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Nelson et al (2016) BMC Genomics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Chow, Wolfner and Clark (2013) PNAS","Chow, Wolfner and Clark (2013) PNAS","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Horvath and Kalinka (2018) Evolution","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Akhund-Zade et al (2017) Journal of Insect Science","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Dobson et al (2015) Nature Communications","Mendez et al (2016) PLOS One","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Mendez et al (2016) PLOS One","Unckless, Rottschaefer and Lazzaro (2015) G3","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Gaertner et al (2015) G3","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Muñoz-Muñoz et al (2016) Evolution","Muñoz-Muñoz et al (2016) Evolution","Richardson (2012) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Trait<\/th>\n      <th># lines measured<\/th>\n      <th>Sex<\/th>\n      <th>Trait guild<\/th>\n      <th>Trait description<\/th>\n      <th>Reference<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"autoWidth":true,"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":1000,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Trait_table"}],"pageLength":1214,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"orderClasses":false,"lengthMenu":[10,25,50,100,1214]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="trim-data-for-our-analysis" class="section level3">
<h3>Trim data for our analysis</h3>
<p>For the analysis presented in our manuscript, we remove the
metabolite traits measured in Jin <em>et al</em> (2020) and Zhou <em>et
al</em> (2020) and the microbiome traits measured in Everett <em>et
al</em> (2020).</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>selected.dgrp.phenos_unscaled <span class="ot">&lt;-</span> </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  all.dgrp.phenos_combined_subset <span class="sc">%&gt;%</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Reference <span class="sc">!=</span> <span class="st">&quot;Jin et al (2020) PLOS Genetics&quot;</span> <span class="sc">&amp;</span> Reference <span class="sc">!=</span> <span class="st">&quot;Zhou et al (2020) Genome Research&quot;</span> <span class="sc">&amp;</span> Reference <span class="sc">!=</span> <span class="st">&quot;Everett et al (2020) Genome Research&quot;</span>)</span></code></pre></div>
<p>Correlation estimates are imprecise for traits that have been
measured across a small number of lines. Therefore, we further trim the
dataset to only include traits that have been phenotyped across &gt;80
lines.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this lists traits in ascending order by the number of lines they are measured in</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>trait_test <span class="ot">&lt;-</span> </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  selected.dgrp.phenos_unscaled <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_lines =</span> <span class="fu">length</span>(line)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(n_lines)</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove traits that have been measured in &lt; 80 lines</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>selected.dgrp.phenos_unscaled <span class="ot">&lt;-</span> </span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>  selected.dgrp.phenos_unscaled <span class="sc">%&gt;%</span> </span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">length</span>(line) <span class="sc">&gt;</span> <span class="dv">79</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Reference)</span></code></pre></div>
<p>Save the data</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(selected.dgrp.phenos_unscaled, <span class="at">file =</span> <span class="st">&quot;data/derived/selected.dgrp.phenos_unscaled.csv&quot;</span>)</span></code></pre></div>
<p>This leaves us with 516 traits for analysis. They can be viewed
here:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>lines_measured <span class="ot">&lt;-</span> </span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  selected.dgrp.phenos_unscaled <span class="sc">%&gt;%</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="st">`</span><span class="at"># lines measured</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>table <span class="ot">&lt;-</span> selected.dgrp.phenos_unscaled <span class="sc">%&gt;%</span> </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(trait_value))</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>trimmed_table <span class="ot">&lt;-</span> <span class="fu">left_join</span>(lines_measured, table) <span class="sc">%&gt;%</span> </span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait)</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to build HTML searchable tables</span></span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>my_data_table <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">datatable</span>(</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>    df, <span class="at">rownames=</span><span class="cn">FALSE</span>,</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">autoHideNavigation =</span> <span class="cn">TRUE</span>,</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">extensions =</span> <span class="fu">c</span>(<span class="st">&quot;Scroller&quot;</span>,  <span class="st">&quot;Buttons&quot;</span>),</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">autoWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">dom =</span> <span class="st">&#39;Bfrtip&#39;</span>,</span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">deferRender=</span><span class="cn">TRUE</span>,</span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">scrollX=</span><span class="cn">TRUE</span>, <span class="at">scrollY=</span><span class="dv">1000</span>,</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">scrollCollapse=</span><span class="cn">TRUE</span>,</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">buttons =</span></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="st">&#39;pageLength&#39;</span>, <span class="st">&#39;colvis&#39;</span>, <span class="st">&#39;csv&#39;</span>, <span class="fu">list</span>(</span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">extend =</span> <span class="st">&#39;pdf&#39;</span>,</span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">pageSize =</span> <span class="st">&#39;A4&#39;</span>,</span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a>          <span class="at">orientation =</span> <span class="st">&#39;landscape&#39;</span>,</span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a>          <span class="at">filename =</span> <span class="st">&#39;Analysis_trait_table&#39;</span>)),</span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">pageLength =</span> <span class="dv">516</span></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a><span class="fu">my_data_table</span>(trimmed_table)</span></code></pre></div>
<div id="htmlwidget-b8169d79e38dab13872b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b8169d79e38dab13872b">{"x":{"filter":"none","vertical":false,"extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["2.3-butanedione.olfactory.attraction.f","2.3-butanedione.olfactory.attraction.m","2.3-butanedione.olfactory.latency.30.percentile.f","2.3-butanedione.olfactory.latency.30.percentile.m","2.3-butanedione.olfactory.latency.90.percentile.f","2.3-butanedione.olfactory.latency.90.percentile.m","2.3-butanedione.olfactory.response.intensity.f","2.3-butanedione.olfactory.response.intensity.m","2.3-butanedione.olfactory.strength.attraction.f","2.3-butanedione.olfactory.strength.attraction.m","abdomen.length.18C.f","abdomen.length.28C.f","abdomen.length.thermal.plasticity.f","acclimated.acute.cold.exposure.resistance.18C.f","acclimated.acute.cold.exposure.resistance.18C.m","acclimated.acute.cold.exposure.resistance.25C.f","acclimated.acute.cold.exposure.resistance.25C.m","activity.m","activity.MPH.m","acute.cold.exposure.resistance.18C.-6.f","acute.cold.exposure.resistance.18C.-6.m","acute.cold.exposure.resistance.18C.-8.f","acute.cold.exposure.resistance.18C.-8.m","acute.cold.exposure.resistance.25C.f","acute.cold.exposure.resistance.25C.m","aggression.2015.m","alpha-amanitin.resistance.0.2","alpha-amanitin.resistance.2","anoxia.tolerance.adult.f","anoxia.tolerance.adult.m","anoxia.tolerance.larval","approach.female.m","attempted.copulation.frequency.m","azinphos.methyl.resistance.0.25","azinphos.methyl.resistance.0.5","azinphos.methyl.resistance.1","azinphos.methyl.resistance.2","azinphos.methyl.resistance.LD50","basal.activity.f","basal.activity.m","boric.acid.resistance.f","caffeine.resistance.f","CHC.11 &amp; 9-Me-C23.f","CHC.13- &amp; 11-Me-C25.f","CHC.2-Me-C22.f","CHC.2-Me-C22.m","CHC.2-Me-C24.f","CHC.2-Me-C24.m","CHC.2-Me-C25.f","CHC.2-Me-C25.m","CHC.2-Me-C26.m","CHC.2-Me-C27.m","CHC.2-Me-C28 &amp; 7,11-C29:2.f","CHC.2-Me-C28.m","CHC.2-Me-C29 &amp; 8,12-C30:2.f","CHC.2-Me-C29.m","CHC.2-Me-C30.f","CHC.3-Me-C23.m","CHC.3-Me-C25 &amp; 9-C26:1.f","CHC.3-Me-C25.m","CHC.3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1.f","CHC.5-C23:1.f","CHC.5-C23:1.m","CHC.5-C24:1.f","CHC.5-C24:1.m","CHC.5-C25:1.f","CHC.5-C25:1.m","CHC.5-C27:1.f","CHC.5-Me-C25.f","CHC.5,9-C25:2 &amp; 9-C25:1.f","CHC.5,9-C27:2 &amp; 9-C27:1.f","CHC.6-C23:1.f","CHC.6-C23:1.m","CHC.6-C24:1.f","CHC.6-C24:1.m","CHC.6-C26:1 &amp; i.m","CHC.6,10-C25:2.f","CHC.6,10-C25:2.m","CHC.6,10-C26:2.f","CHC.7-C23:1.f","CHC.7-C23:1.m","CHC.7-C24:1.m","CHC.7-C25:1.f","CHC.7-C25:1.m","CHC.7-C26:1.f","CHC.7-C26:1.m","CHC.7-C27:1.f","CHC.7-C27:1.m","CHC.7-C29:1.f","CHC.7-C29:1.m","CHC.7,11-C23:2.f","CHC.7,11-C24:2.f","CHC.7,11-C25:2.f","CHC.7,11-C26:2.f","CHC.7,11-C27:2 &amp; 2-Me-C26.f","CHC.7,11-C28:2 &amp; 2-Me-C27.f","CHC.7,11-C31:2.f","CHC.8-C25:1.f","CHC.8,12-C26:2.f","CHC.8,12-C28:2.f","CHC.9-C23:1.f","CHC.9-C23:1.m","CHC.9-C25:1.m","CHC.9-C27:1.m","CHC.9-C29:1 &amp; 5,9-C29:2.f","CHC.9,13-C25:2.f","CHC.9,13-C27:2.f","CHC.n-C21.f","CHC.n-C21.m","CHC.n-C22.f","CHC.n-C22.m","CHC.n-C23.f","CHC.n-C23.m","CHC.n-C24.f","CHC.n-C24.m","CHC.n-C25.m","CHC.n-C26.f","CHC.n-C26.m","CHC.n-C27.f","CHC.n-C27.m","CHC.n-C28.f","CHC.n-C28.m","CHC.n-C29.f","CHC.n-C29.m","CHC.n-C31.f","CHC.n-C31.m","CHC.NI.20.f","CHC.NI.33.f","CHC.NI.34.f","CHC.NI.54.f","CHC.NI.54.m","CHC.NI.7.m","CHC.x-C22:1a.m","CHC.x-C22:1b.m","CHC.x-C23:1.f","CHC.x,y-C24:2.f","CHC.x,y-C24:2.m","chill.coma.recovery.f","chill.coma.recovery.m","chill.coma.recovery.variance.f","chill.coma.recovery.variance.m","chlorantraniliprole.resistance.0.5","chlorantraniliprole.resistance.1","chlorantraniliprole.resistance.2","chlorantraniliprole.resistance.3","chlorantraniliprole.resistance.4","chlorantraniliprole.resistance.5","chlorantraniliprole.resistance.LC50","chronic.cold.exposure.resistance.f","chronic.cold.exposure.resistance.m","circadian.period.chi.f","circadian.period.chi.m","circadian.period.mesa.f","circadian.period.mesa.m","circadian.rythmicity.index.f","circadian.rythmicity.index.m","climbing.ability.f","climbing.ability.m","climbing.ability.paraquat.f","climbing.ability.paraquat.m","climbing.age.decline.high.yeast.f","climbing.age.decline.low.yeast.f","cold.hardiness.-5C.larvae","cold.hardiness.f","cold.hardiness.m","cold.shock.acclimated.old.f","cold.shock.acclimated.old.m","cold.shock.acclimated.young.f","cold.shock.acclimated.young.m","cold.shock.old.f","cold.shock.old.m","cold.shock.young.f","cold.shock.young.m","critical.thermal.max.2020.f","critical.thermal.max.2020.m","critical.thermal.min.2020.f","critical.thermal.min.2020.m","critical.thermal.min.23C.m","critical.thermal.min.developmental.stability.m","critical.thermal.min.plasticity.m","critical.thermal.min.plasticity.variation.m","critical.thermal.min.within.environment.variation.plasticity.m","day.sleep.bout.length.f","day.sleep.bout.length.m","day.sleep.bout.length.SD.f","day.sleep.bout.length.SD.m","day.sleep.bouts.f","day.sleep.bouts.m","day.sleep.bouts.SD.f","day.sleep.bouts.SD.m","day.sleep.f","day.sleep.m","day.sleep.SD.f","day.sleep.SD.m","DDT.resistance.knockdown.f","DDT.resistance.mortality.f","deltamethrin.resistance.m","dessication.tolerance.f","dessication.tolerance.m","developmental.acclimation.cold.exposure.f","developmental.acclimation.cold.exposure.m","dopamine.response.to.paraquat.2021.m","drosophila.C.virus.resistance.m","E.faecalis.resistance.m","egg.adult.development.time.2016.f","egg.adult.development.time.2016.m","egg.adult.development.time.2018","egg.adult.development.time.EtOH.exposure","egg.adult.viability.2016","egg.adult.viability.2018","egg.adult.viability.ethanol.exposure","ethanol.sensitivity.development.time","ethanol.sensitivity.first.exposure.f","ethanol.sensitivity.first.exposure.m","ethanol.sensitivity.second.exposure.f","ethanol.sensitivity.second.exposure.m","ethanol.sensitivity.viability","ethanol.tolerance.f","ethanol.tolerance.m","exercise.induced.activity.f","exercise.induced.activity.m","fecundity.f","fertilised.egg.retention.score.f","fertility.25C.m","fertility.27C.m","fertility.29C.m","fertility.thermal.susceptibility.m","fitness.early.life.f","fitness.early.life.m","fitness.late.life.f","fitness.late.life.m","flight.performance.f","flight.performance.m","flock.house.virus.resistance.m","food.intake.f","food.intake.m","foraging.survival.m","genital.licking.frequency.m","genome.size","glucose.2015.m","glucose.anexic.m","glucose.gnotobiotic.m","glucose.high.glucose.diet.m","glucose.low.glucose.diet.m","glucose.medium.glucose.diet.m","glucose.microbiome.effect.m","glucose.yeast.poor.diet.f","glucose.yeast.rich.diet.f","glycerol.2015.m","glycerol.high.glucose.diet.m","glycerol.low.glucose.diet.m","glycerol.medium.glucose.diet.m","glycogen.2015.m","glycogen.high.glucose.diet.m","glycogen.low.glucose.diet.m","glycogen.medium.glucose.diet.m","gustatory.response.ethanol.f","gustatory.response.ethanol.m","gut.permeability.f","gut.permeability.radiation.f","hydrogen.peroxide.resistance.f","imidacloprid.resistance.100ppm","imidacloprid.resistance.25ppm","interocular.distance.f","interocular.distance.m","isolated.aggression.m","kallithea.resistance.LT50.f","kallithea.resistance.LT50.m","kallithea.viral.load.f","kujin.virus.resistance.f","L.monocytogenes.load.m","L.monocytogenes.MTD.m","lead.effect.development.time.f","lead.effect.development.time.m","lead.effect.locomotor.activity.f","lead.effect.locomotor.activity.m","lead.effect.viability","lead.egg.adult.development.time.2016.f","lead.egg.adult.development.time.2016.m","lead.egg.adult.viability.2016","lead.locomotor.activity.2016.f","lead.locomotor.activity.2016.m","lifespan.18C.f","lifespan.18C.m","lifespan.2019.f","lifespan.25C.2014.f","lifespan.25C.2015.f","lifespan.25C.2020.f","lifespan.25C.2020.m","lifespan.25C.high.yeast.f","lifespan.25C.low.yeast.f","lifespan.28C.f","lifespan.28C.m","lifespan.following.diapause.2019.f","lifespan.high.sugar.diet.f","lifetime.fecundity.25C.f","locomotor.activity.2016.f","locomotor.activity.2016.m","locomotor.activity.m","locomotor.frequency.f","M.anisopliae.resistance.f","M.anisopliae.resistance.m","malathion.mortality.12hr.f","malathion.mortality.12hr.m","malathion.mortality.24hr.f","malathion.mortality.24hr.m","malathion.mortality.3hr.f","malathion.mortality.3hr.m","malathion.mortality.6hr.f","malathion.mortality.6hr.m","mass.yeast.poor.diet.f","mass.yeast.rich.diet.f","mating.latency.2015.m","MeHg.tolerance","micro.environmental.plasticity.lifespan.18C.f","micro.environmental.plasticity.lifespan.18C.m","micro.environmental.plasticity.lifespan.25C.f","micro.environmental.plasticity.lifespan.25C.m","micro.environmental.plasticity.lifespan.28C.f","micro.environmental.plasticity.lifespan.28C.m","MPH.effect.on.activity.m","MSB.resistance.f","MSB.resistance.m","mtDNA.copy.number","negative.geotaxis.f","negative.geotaxis.m","negative.geotaxis.msb.f","negative.geotaxis.msb.m","negative.geotaxis.sensitivity.msb.f","negative.geotaxis.sensitivity.msb.m","night.sleep.bout.length.f","night.sleep.bout.length.m","night.sleep.bout.length.SD.f","night.sleep.bout.length.SD.m","night.sleep.bouts.f","night.sleep.bouts.m","night.sleep.bouts.SD.f","night.sleep.bouts.SD.m","night.sleep.f","night.sleep.m","night.sleep.SD.f","night.sleep.SD.m","olfactory.response.1-hexanol.f","olfactory.response.1-hexanol.m","olfactory.response.2-heptanone.f","olfactory.response.2-heptanone.m","olfactory.response.2-phenyl.ethyl.alcohol.f","olfactory.response.2-phenyl.ethyl.alcohol.m","olfactory.response.acetophenone.f","olfactory.response.acetophenone.m","olfactory.response.benzaldehyde.2013.f","olfactory.response.benzaldehyde.2013.m","olfactory.response.benzaldehyde.2015.f","olfactory.response.benzaldehyde.2015.m","olfactory.response.citral.f","olfactory.response.citral.m","olfactory.response.D-carvone.f","olfactory.response.D-carvone.m","olfactory.response.ethanol.f","olfactory.response.ethanol.m","olfactory.response.ethylacetate.f","olfactory.response.ethylacetate.m","olfactory.response.ethylbutyrate.f","olfactory.response.ethylbutyrate.m","olfactory.response.eugenol.f","olfactory.response.eugenol.m","olfactory.response.heliol.f","olfactory.response.heliol.m","olfactory.response.hexal.f","olfactory.response.hexal.m","olfactory.response.L-carvone.f","olfactory.response.L-carvone.m","olfactory.response.methylSalicylate.f","olfactory.response.methylSalicylate.m","orient.towards.female.m","ovariole.asymmetry.f","ovariole.number.f","p-xylene.susceptibility","P.aeruginosa.defecation.f","P.aeruginosa.fecal.pH.f","P.aeruginosa.intestinal.mitosis.f","P.aeruginosa.resistance.2017.f","P.aeruginosa.resistance.2017.m","P.aeruginosa.resistance.2019.f","P.entomophila.resistance.f","P.rettgeri.load.f","P.rettgeri.load.high.glucose.diet.m","P.rettgeri.load.low.glucose.diet.m","P.rettgeri.load.medium.glucose.diet.m","P.rettgeri.resistance.f","P.rettgeri.tolerance.f","paraquat.resistance.2021.f","paraquat.resistance.2021.m","paraquat.resistance.f","paraquat.resistance.m","parathion.resistance.m","permethrin.knockdown.f","permethrin.knockdown.m","permethrin.mortality.24hr.f","permethrin.mortality.24hr.m","phototaxis.mid.age.f","phototaxis.mid.age.m","phototaxis.old.f","phototaxis.old.m","phototaxis.young.f","phototaxis.young.m","posterior.lobe.shape.PC1.m","posterior.lobe.shape.PC2.m","posterior.lobe.size.m","protein.2015.m","protein.anexic.m","protein.gnotobiotic.m","protein.high.glucose.diet.m","protein.low.glucose.diet.m","protein.medium.glucose.diet.m","pupal.case.length","pupation.height","radiation.resistance.m","rapid.cold.hardening.18C.f","rapid.cold.hardening.18C.m","rapid.cold.hardening.25C.f","rapid.cold.hardening.25C.m","rapid.cold.hardening.old.f","rapid.cold.hardening.old.m","rapid.cold.hardening.young.f","rapid.cold.hardening.young.m","sigma.virus.sensitivity.f","social.aggression.m","social.effect.aggression.m","song.interpulse.interval.m","sprinter.marathoner.f","sprinter.marathoner.m","startle.response.2012.f","startle.response.2012.m","startle.response.2012.msb.f","startle.response.2012.msb.m","startle.response.f","startle.response.m","startle.response.variance.f","startle.response.variance.m","startle.sensitivity.msb.f","startle.sensitivity.msb.m","starvation.resistance.2012.f","starvation.resistance.2012.m","starvation.resistance.2017.m","starvation.resistance.2019.f","starvation.resistance.2019.m","starvation.resistance.high.carb.diet.m","starvation.resistance.high.fat.diet.m","starvation.resistance.high.protein.diet.m","starvation.resistance.larvae","starvation.resistance.old.f","starvation.resistance.old.m","starvation.resistance.standard.diet.m","starvation.resistance.variance.f","starvation.resistance.variance.m","starvation.resistance.yeast.poor.diet.f","starvation.resistance.yeast.rich.diet.f","starvation.resistance.young.f","starvation.resistance.young.m","T1.femur.f","T1.femur.m","T1.leg.length.f","T1.leg.length.m","T1.tarsus.f","T1.tarsus.m","T1.tibia.f","T1.tibia.m","T2.femur.f","T2.femur.m","T2.leg.length.f","T2.leg.length.m","T2.tarsus.f","T2.tarsus.m","T2.tibia.f","T2.tibia.m","tergite5.f","tergite6.f","thorax.length.18C.f","thorax.length.28C.f","thorax.length.thermal.plasticity.f","traumatic.injury.mortality","triglyceride.2015.m","triglyceride.gnotobiotic.m","triglyceride.high.glucose.diet.m","triglyceride.low.glucose.diet.m","triglyceride.medium.glucose.diet.m","triglyceride.poor.diet.f","triglyceride.yeast.rich.diet.f","tunicamycin.ER.stress.m","tunicamycin.LT50.m","unfertilised.eggs.post.mating","virgin.egg.retention.f","waking.activity.f","waking.activity.m","waking.activity.SD.f","waking.activity.SD.m","week1.fecundity.25C.f","week3.fecundity.25C.f","week5.fecundity.25C.f","week7.fecundity.25C.f","weight.anexic.m","weight.gnotobiotic.m","weight.high.glucose.diet.m","weight.low.glucose.diet.m","weight.medium.glucose.diet.m","weight.microbiome.effect.m","wet.weight.m","wing.centroid.size.2019_Dworkin.f","wing.centroid.size.2019_Dworkin.m","wing.centroid.size.2019_Houle.f","wing.centroid.size.2019_Houle.m","wing.centroid.size.f","wing.centroid.size.m","wing.display.frequency.m","wolbachia.copy.number"],[157,157,157,157,157,157,157,157,157,157,194,196,191,181,181,187,187,172,172,185,186,181,181,184,184,200,186,186,171,171,169,166,166,175,175,175,175,175,153,151,163,164,169,169,169,157,169,157,169,157,157,157,169,157,169,157,169,157,169,157,169,169,157,169,157,169,157,169,169,169,169,169,157,169,157,157,169,157,169,169,157,157,169,157,169,157,169,157,169,157,169,169,169,169,169,169,169,169,169,169,169,157,157,157,169,169,169,169,157,169,157,169,157,169,157,157,169,157,169,157,169,157,169,157,169,157,169,169,169,169,157,157,157,157,169,169,157,176,176,174,174,152,152,151,152,152,152,152,175,175,161,166,161,166,167,167,99,99,99,99,155,155,139,139,139,96,96,96,96,96,96,96,96,100,100,100,100,162,166,166,166,166,168,168,167,167,168,168,167,167,168,168,167,167,184,184,195,175,175,182,183,99,155,199,198,198,201,201,200,201,201,201,205,205,205,205,201,205,205,149,148,142,91,126,124,122,127,125,124,125,124,197,197,182,183,182,201,166,205,145,103,117,147,150,152,99,181,181,145,147,150,152,145,147,150,152,167,167,160,160,180,171,171,149,149,87,119,119,119,86,111,116,184,188,200,200,199,185,190,199,166,166,183,183,191,189,197,186,186,161,161,177,177,197,193,189,166,166,204,98,188,188,170,170,170,170,170,170,170,170,181,181,166,171,183,183,186,186,177,177,172,167,167,174,192,192,192,192,192,192,168,168,167,167,168,168,167,167,168,168,167,167,186,186,186,186,186,186,186,186,164,164,186,186,186,186,186,186,167,167,186,186,186,186,185,185,186,186,186,186,186,186,186,186,166,200,200,143,150,150,152,81,81,142,140,158,151,151,152,156,156,99,99,167,167,194,169,169,169,169,191,191,185,185,191,191,155,155,155,145,87,90,147,150,152,198,198,161,181,181,184,184,96,96,96,96,189,87,87,190,135,135,192,192,192,192,204,204,201,201,192,192,203,203,201,168,168,176,175,178,129,159,159,178,197,197,171,171,159,159,117,117,117,117,117,117,117,117,117,117,117,117,117,117,117,117,175,175,194,196,191,178,145,85,147,150,152,172,172,111,112,91,90,168,168,167,167,189,189,178,135,94,96,147,150,152,82,145,171,172,184,185,149,149,166,91],["Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Female","Male","Male","Male","Female","Male","Female","Male","Female","Male","Male","Pooled","Pooled","Female","Male","Pooled","Male","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Male","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Male","Male","Male","Female","Female","Male","Female","Male","Female","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Female","Male","Female","Male","Male","Male","Male","Female","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Female","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Female","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Male","Male","Pooled","Male","Male","Male","Male","Male","Male","Male","Female","Female","Male","Male","Male","Male","Male","Male","Male","Male","Female","Male","Female","Female","Female","Pooled","Pooled","Female","Male","Male","Female","Male","Female","Female","Male","Male","Female","Male","Female","Male","Pooled","Female","Male","Pooled","Female","Male","Female","Male","Female","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Female","Male","Male","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Male","Female","Male","Pooled","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Female","Pooled","Female","Female","Female","Female","Male","Female","Female","Female","Male","Male","Male","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Pooled","Pooled","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Male","Male","Male","Pooled","Female","Male","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Pooled","Male","Male","Male","Male","Male","Female","Female","Male","Male","Pooled","Female","Female","Male","Female","Male","Female","Female","Female","Female","Male","Male","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Male","Pooled"],["Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Morphological","Morphological","Morphological","Temperature related","Temperature related","Temperature related","Temperature related","Activity","Activity","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Behavioural","Pathogen response","Pathogen response","Stress response","Stress response","Stress response","Behavioural","Behavioural","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Activity","Activity","Insecticide response","Drug response","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","Temperature related","Temperature related","Temperature related","Temperature related","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Temperature related","Temperature related","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Stress response","Stress response","Life history","Life history","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Insecticide response","Insecticide response","Insecticide response","Stress response","Stress response","Temperature related","Temperature related","Drug response","Pathogen response","Pathogen response","Life history","Life history","Life history","Life history","Life history","Life history","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Activity","Activity","Life history","Life history","Life history","Life history","Life history","Temperature related","Life history","Life history","Life history","Life history","Activity","Activity","Pathogen response","Behavioural","Behavioural","Life history","Behavioural","Genomic","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Physiological","Physiological","Stress response","Insecticide response","Insecticide response","Morphological","Morphological","Behavioural","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Stress response","Stress response","Stress response","Stress response","Stress response","Life history","Life history","Stress response","Activity","Activity","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Activity","Activity","Activity","Activity","Pathogen response","Pathogen response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Morphological","Morphological","Behavioural","Stress response","Life history","Life history","Life history","Life history","Life history","Life history","Drug response","Drug response","Drug response","Genomic","Activity","Activity","Activity","Activity","Stress response","Stress response","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Behavioural","Morphological","Morphological","Stress response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Drug response","Drug response","Drug response","Drug response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Activity","Activity","Activity","Activity","Activity","Activity","Morphological","Morphological","Morphological","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Morphological","Life history","Stress response","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Pathogen response","Behavioural","Behavioural","Behavioural","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Drug response","Drug response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Stress response","Metabolome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Drug response","Drug response","Life history","Life history","Activity","Activity","Activity","Activity","Life history","Life history","Life history","Life history","Microbiome","Microbiome","Morphological","Morphological","Morphological","Microbiome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Behavioural","Pathogen response"],["Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Mean adult abdomen length, when reared at 18C. Higher values indicate larger flies.","Mean adult abdomen length, when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult abdomen lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean number of aggressive encounters between 8 males in an interaction vial over a two minute duration. Higher values indicate more aggressive males.","Mean number of flies exposed to alpha-amanitin, at concentration 0.2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean number of flies exposed to alpha-amanitin, at concentration 2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean survival (larvae that pupated) for late third instar larvae, 24 hours after exposure to one hour of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.","Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.25 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.5 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 1 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 2 ug/ml. Higher values indicate greater mortality.","Mean dosage (ug/ml) of azinphos-methyl (an organophosphate insecticide) required for 50% of larvae to die. Higher values indicate greater resistance.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.","Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 0.5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 1 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 2 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 3 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 4 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean chlorantraniliprole dosage required for 50% of first instar larvae to die before eclosion. Chlorantraniliprole is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 5% yeast. Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Mean proportion of successfully eclosed individuals following an hour of cold shock (-5°C) at 5 days post oviposition (third instar). Higher values indicate larvae that are more resistant to cold shock.","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum (CT min), which is defined as the temperature at which a complete cessation of movement is observed as a result of the fly entering chill coma. Egg-to-adult development occurred at 23C. Higher values indicate a higher CT min.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Developmental stability across environments is the consistency of within-environment variation in CT min across environments. The authors calculate this by taking the highest coefficient of variation (CV) in CT min at any rearing temperature, minus the lowest CV at any other given rearing temperature. Lower values indicate greater developmental stability across rearing temperatures.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums were measured (CT min). Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The absolute value of the slope provides a measure of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The standard error of the slope provides a measure of the extent to which the same genotype can produce the same plastic response consistently. Higher values indicate greater variance in thermal plasticity.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums were measured. This trait captures how consistently a genotype produces the same phenotype at a given rearing temperature. Higher values indicate greater variance in phenotype within a rearing temperature.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean knockdown of adults, four hours after vials were inoculated with 0.5 ug/ml of DDT. Knockdown is defined as either flies permanently seen to be in a prone position with jolting movement of legs or wings, or a prolonged inability to right themselves from a prone position. DDT is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean survival of adults, 24 hours after vials were inoculated with 0.5 ug/ml of DDT. This is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean change in proportion of adult flies surviving one hours of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean change in proportion of adult flies surviving one hour of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean difference in Dopamine content (pg/ng tissue) extracted from the heads of two flies, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress on Dopamine content.","Mean days survived after infection with Drosophila C Virus (DCV) - a positive sense RNA virus that naturally infects multiple Drosophila species. Higher values indicate a longer lifespan, and therefore a greater resistance to DCV.","Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a standard diet. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a diet supplemented with 10% EtOH (ethanol). Higher values indicate a longer development time.","Mean proportion of flies surviving to adulthood, when reared at 25C on control food medium. Higher values indicate greater survival.","Percentage of flies surviving to eclosion, after development on standard food. Higher values indicate greater viability.","Percentage of flies surviving to eclosion, after development on food supplemented with 10% EtOH (ethanol). Higher values indicate greater viability.","The mean difference in egg-to-adult development time in days, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Higher values indicate a longer development time for ethanol-supplemented flies.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","The mean difference in the percentage of flies surviving to eclosion, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Lower values indicate a greater negative effect of ethanol on egg to adult viability.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Mean number of eggs produced by single females, per day. Higher values indicate more fecund females.","Mean fertilised egg retention score. An estimation of the progression of embryo development taken two hours after a one hour egg laying period. Higher values indicate more developed embryos which signifies a greater degree of egg retention. Note that both males and femlaes were sourced from the DGRP line, therefore variation could be explained by either the male or female genotype.","Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.","Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility","Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility","Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.","Mean number of L1 larvae produced by females aged 5-6 days over a 24h period. For each replicate measurement, 5 DGRP females were housed with standardised mates and opposite-sex competitors (15 males and 10 females, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae sired by males in competition for matings/fertilisations with same-aged competitor males over 3 days, beginning when all males were aged 2-3 days. For each replicate measurement, 5 DGRP males were housed with standardised mates and opposite-sex competitors (15 females and 10 males, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae produced by females aged 17-18 days over a 24h period. For each replicate measurement, 5 DGRP females were housed with standardised mates and opposite-sex competitors (15 males and 10 females, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae sired by males in competition for matings/fertilisations with same-aged competitor males over 3 days, beginning when all males were aged 14-15 days. For each replicate measurement, 5 DGRP males were housed with standardised mates and opposite-sex competitors (15 females and 10 males, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean height (m) at which flies catch themselves from freefall after vial inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean height (m) at which flies catch themselves from freefall after rapid inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean days survived after infection with Flock House Virus (FHV) - a nodavirus virus that typically infects beetles. Higher values indicate a longer lifespan, and therefore a greater resistance to FHV.","Mean quantity of 4% sucrose solution consumed per fly iover a 24 hour period. Higher values indicate greater food intake.","Mean quantity of 4% sucrose solution consumed per fly over a 24 hour period. Higher values indicate greater food intake.","Adult mean survival when flies were provided access to food over a 5 day period. Higher values indicate greater survival and assumed greater foraging ability.","Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.","Mean size of the genome in mb, estimated using flow cytometry. Higher values indicate a larger genome.","Estimated mass of free glucose per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.","Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of free glucose.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of free glucose.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of free glucose.","Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher glucose levels.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher glucose levels.","Estimated mass of free glycerol per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycerol.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycerol.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycerol.","Estimated mass of glycogen stores per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycogen.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","The mean proportion of 19 day-old flies with permeable guts, following no exposure to radiation. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","The mean proportion of 19 day-old flies with permeable guts, following exposure to 100 gray units of ionising radiation when flies were 5 days old. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","Lifespan on a diet supplemented with 2-3% hydrogen peroxide (H2O2). Data are corrected for line body mass. Higher values indicate greater resistance to hydrogen peroxide.","Mean wiggle index for third instar larvae 1 hour into exposure to 100ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean wiggle index for third instar larvae 1 hour into exposure to 25ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in isolation for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.","Mean survival after infection with the Kunjin subtype of the West Nile Virus. Calculated as the log(hazard ratio). Higher values indicate greater susceptibility to the virus.","The median number  of colony forming units (originally log transformed) 48 hrs after initial infection with the bacterial pathogen L.monocytogenes. Higher values indicate a greater bacterial load and a more severe infection.","Median days till death after injection with 1000 colony forming units of the bacterial pathogen L. monocytogenes. Higher values indicate greater resistance to the L. monocytogenes.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean difference in egg-to-adult viability between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a smaller negative effect of lead on viability.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean proportion of flies surviving to adulthood, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater survival.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days. Higher values indicate a longer lifespan.","Mean adult lifespan, measured as the number of days from eclosion to death, when living with a single  standard male, at 25C. Higher values indicate longer lived flies.","Adult lifespan, measured in days, for virgin females. Higher values indicate greater longevity.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 5% yeast. Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days for virgin females following diapause induction (10 ̊C for 35 days, moved to 18 ̊C for a day of acclimation, remainder of their lives at 25 ̊C). Higher values indicate a longer lifespan.","Mean lifespan of 20 females maintained on a high sugar diet. Means are adjusted for block effects. Higher values indicate longer lived flies.","Fecundity was estimated every other week of the female’s lifespan as a 2-day total of eggs laid, when living with a single standard male, at 25C. Lifetime fecundity was estimated as the sum of all egg counts across the female's lifespan. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean total distance traveled by individual flies during a five minute interval. Higher values indicate more active lines.","The proportion of flies engaging in a ‘walking bout’ at a given time point. Higher values indicate more active lines.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean virgin female body mass for groups of three flies, after rearing on food with 0.5% yeast content (dietary restricted). Higher values indicate heavier flies.","Mean virgin female body mass for groups of three flies, after rearing on food with 5% yeast content (ab libitum diet). Higher values indicate heavier flies.","Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.","Methylmercury (MeHg) is a persistent environmental toxin present in seafood that can compromise the developing nervous system in humans. To assess overall developmental tolerance on MeHg media, the authors calculate an eclosion index – higher values indicate greater relative egg-to-adult viability when reared on MeHg supplemented media.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean copy number of mitochondrial genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more mitochondrial genomes, which the authors suggest is a proxy for mitochondrial density.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.","Mean difference between the number of ovarioles in the right and left ovary. Higher values indicate greater asymmetry.","Mean number of ovarioles in the right and left ovary. Higher values indicate a greater number of ovarioles.","Mean difference in larval viability between control and p-xylene-treated (concentration 1.068mM) first instar larvae after 14 days of development. P-xylene is an aromatic hydrocarbon associated with gasoline and diesel-powered engine emissions. Higher values indicate a greater negative effect of P-xylene on larval viability.","Mean number of fecal spots detected after 20h for 10 flies, after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate more defecation.","Mean pH of fecal matter for flies after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a higher pH.","Mean amount of intestinal mitosis, five days after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a greater number of mitotic itensitinal cells.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of flies in a vial to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean percentage of flies dying after 3 days post infection with P.entomophila L48, an entomopathogenic bacterium. Higher values indicate greater mortality.","Mean colony forming units (CFU) produced from the homogenate of three pooled flies, 20 hours after infection with P. rettgeri, a bacterial pathogen. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 10% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 2.5% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 5% glucose. Higher values indicate a greater bacterial load.","Mean proportion of flies surviving infection, two days after inoculation with the bacterial pathogen P. rettgeri. Higher values indicate greater resistance to P. rettgeri.","A general, nonlinear relationship between the proportion of flies surviving two days post infection and sustained pathogen load, after injection with the bacterial pathogen P. rettgeri, was predicted. The data presented here are the line deviations from the predicted relationship between survival and bacterial load. Positive values indicate a greater tolerance to P. rettgeri.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean PC1 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 78% of the variation in shape, and describes the degree of beak-like protrusion at the tip of the posterior lobes. Higher values indicate a greater degree of beak-like postrusion.","Mean PC2 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 10% of the variation in shape, and describes the width of the posterior lobes. Higher values indicate wider lobes.","Mean posterior lobe size, measured in pixels. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. Higher values indicate larger posterior lobes.","Estimated mass of soluble protein per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.","Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of protein.","Mean pupal case length in mm. Measurements were taken with image analysis software from at least 6 vials with a minimum of 15 pupa per vial, and were converted from pixels to mm based on the ratio to the diameter measurement of 1 euro cent coin (16.25 mm diameter)","Mean height at which larvae pupated in a 95mm Drosophila vial, after accounting for food height. Higher values indicate higher pupation sites.","Mean survival 24 hours after exposure to radiation (Cs137, avg dose of 4.85 Gy/min. Total dose of 1382 Gy over the 4 hours and 45 minutes of continuous exposure). Higher values indicate greater radiation resistance.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Mean proportion of adults infected with sigma virus (DmelSV), 13 days post infection. Sigma virus is a rhabdovirus that naturally occurs in D. melanogaster. It causes flies paralysis in flies exposed to CO2. Higher values indicate a higher (???) proportion of paralysed flies after exposure to CO2, and therefore greater sensitivity to sigma virus.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in a mixed Sex environment for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Mean difference in aggressive behavior between male flies raised in social and socially isolated environments. Higher values indicate that males raised in a social environment were more aggressive than males experiencing isolation (the authors adjusted line means to meet normality expectations, we present udjusted means for the social and isolation line means).","Median interpulse interval (IPI) in the male courtship 'song'. Males in many Drosophila species produce a series of auditory pulses (a pulse 'song') when courting females, and the pause between pulses (IPI) varies greatly among species. Higher values indicate a longer IPI.","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Note: this data is recoded from that recorded by Watanabe et al (2020).","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Higher values indicate that a greater proportion of activity was completed in the first 30 minutes, which corresponds to a 'sprinter' phenotype. Note: this data is recoded from that recorded by Watanabe et al (2020).","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Adult mean survival in the absence of food after two days is here used as a measure of starvation resistance. Higher values indicate greater survival and therefore higher starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high carbohydrate diet (carbs:494.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high fat diet (carbs:44.15g/L, protein:52g/L, fat:17.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high protein diet (carbs:44.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean hours survived for larvae subjected to nutrient poor starvation medium (20% sucrose in PBS solution). Higher values indicate greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on the laboratories standard diet (carbs:119.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 0.5% yeast content (dietary restricted).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 5% yeast content (ab libitum diet).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old.  Higher values indicate flies that exhibited greater starvation resistance.","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean pigmentation scores for tergite 5. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean pigmentation scores for tergite 6. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean adult thorax length when reared at 18C. Higher values indicate larger flies.","Mean adult thorax length when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult thorax lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","The mean percentage of flies that died within 24 hours, after suffering traumatic injury. Higher values indicate greater mortality.","Estimated mass of total trigylcerides per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of triglycerides.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher triglyceride levels.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher triglyceride levels.","The hazard ratio of the death rate of male flies exposed to tunicamycin - an endoplasmic reticulum (ER) stress inducing drug – with the control group, fed drug free media. Higher values indicate greater ER stress and lower survival of flies exposed to tunicamycin.","Mean time required for 50% of male flies exposed to tunicamycin (TM) media to die. TM is an antibiotic that causes endoplasmic reticulum (ER) stress. Higher values indicate longer survival times and greater resistance to ER stress.","Proportion of unfertilised eggs produced by 3-4 flies that had been allowed to mate for at least two days beforehand. Note that flies were kept in population cages that contained both sexes from the same line during the mating period. This means that genetic variance in this trait variation could be explained by either the male or female genotype. Higher values indicate a greater proportion of unfertilised eggs.","Mean unfertilised egg retention time, measured as the number of days before a virgin female laid her first egg. Higher values indicate a longer retention time.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean fecundity in the first week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the third week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the fifth week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean fecundity in the seventh week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.","Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 10% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 2.5% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 5% glucose. Higher values indicate heavier flies.","Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.","Mean mass. Higher values indicate heavier flies.","Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.","Mean copy number of Wolbachia genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more Wolbachia genomes, which the authors suggest is a proxy for Wolbachia density. Note that only lines found to harbour Wolbachia are inlcuded in this dataset."],["Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Rohde et al (2019) Genetics","Rohde et al (2019) Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Shorter et al (2015) PNAS","Mitchell et al (2017) PLOS One","Mitchell et al (2017) PLOS One","Campwell et al (2019) G3","Campwell et al (2019) G3","Campwell et al (2019) G3","Gaertner et al (2015) G3","Gaertner et al (2015) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Najarro, Hackett and Macdonald (2017) G3","Najarro et al (2015) PLOS Genetics","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Freda et al (2017) Integrative and Comparative Biology","Freda et al (2017) Integrative and Comparative Biology","Freda et al (2017) Integrative and Comparative Biology","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Schmidt et al (2017) Genetics","Schmidt et al (2017) Genetics","Duneau et al (2018) G3","Rajpurohit et al (2018) Molecular Ecology","Rajpurohit et al (2018) Molecular Ecology","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Lovejoy et al (2021) Molecular Ecology","Magwire et al (2012) PLOS Genetics","Chapman et al (2020) Genes","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Morozova (2018) G3","Morozova (2018) G3","Zhou et al (2016) Environmental Health Perspectives","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova (2018) G3","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Evangelou et al (2019) G3","Horvath and Kalinka (2018) Evolution","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Spierer et al (2021) PLOS Genetics","Spierer et al (2021) PLOS Genetics","Magwire et al (2012) PLOS Genetics","Garlapow et al (2015) PLOS One","Garlapow et al (2015) PLOS One","Lee et al (2017) Genome Biology and Evolution","Gaertner et al (2015) G3","Huang et al (2014) Genome Research &amp; Ellis et al (2014) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Sharma et al (2020) Scientific reports","Sharma et al (2020) Scientific reports","Harrison et al (2020) BMC Genomics","Denecke et al (2017) Scientific Reports","Denecke et al (2017) Scientific Reports","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Rohde et al (2017) Genetics","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Ahlers et al (2019) Cell Reports","Hotson and Schneider (2015) G3","Hotson and Schneider (2015) G3","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Durham et al (2014) Nature Communications","Ivanov et al (2015) The Journals of Gerontology: Series A","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Patel and Talbert (2021) Heliyon","Durham et al (2014) Nature Communications","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Rohde et al (2018) G3","Maesani et al (2015) PLOS Computational Biology","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Gaertner et al (2015) G3","Montgomery et al (2014) PLOS One","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Rohde et al (2019) Genetics","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Richardson (2012) PLOS Genetics","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Swarup et al (2013) PNAS","Swarup et al (2013) PNAS","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Gaertner et al (2015) G3","Lobell et al (2017) G3","Lobell et al (2017) G3","Adebambo et al (2020) Frontiers in Genetics","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Evangelou et al (2019) G3","Bou Sleiman et al (2015) Nature Communications","Howick and Lazzaro (2017) Molecular Ecology","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Howick and Lazzaro (2017) Molecular Ecology","Howick and Lazzaro (2017) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Duneau et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Unckless, Rottschaefer and Lazzaro (2015) G3","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Zhang et al (2021) Current Biology","Zhang, Reeves and Tautz (2020) Molecular Ecology","Vaisnav et al (2014) PLOS One","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Magwire et al (2011) PLOS Genetics","Rohde et al (2017) Genetics","Rohde et al (2017) Genetics","Turner, Miller and Cochrane (2013) Molecular Biology and Evolution","Riddle (2020) Journal of Experimental Biology","Riddle (2020) Journal of Experimental Biology","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Lee et al (2017) Genome Biology and Evolution","Everman et al (2019) Genetics","Everman et al (2019) Genetics","Francis et al (2021) G3","Francis et al (2021) G3","Francis et al (2021) G3","Akcay (2017) Thesis","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Francis et al (2021) G3","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Dembeck et al (2015) PLOS Genetics","Dembeck et al (2015) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Katzenberger et al (2015) eLife","Unckless, Rottschaefer and Lazzaro (2015) G3","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Chow, Wolfner and Clark (2013) PNAS","Chow, Wolfner and Clark (2013) PNAS","Horvath and Kalinka (2018) Evolution","Akhund-Zade et al (2017) Journal of Insect Science","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) G3","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Gaertner et al (2015) G3","Richardson (2012) PLOS Genetics"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Trait<\/th>\n      <th># lines measured<\/th>\n      <th>Sex<\/th>\n      <th>Trait guild<\/th>\n      <th>Trait description<\/th>\n      <th>Reference<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"autoWidth":true,"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":1000,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Analysis_trait_table"}],"pageLength":516,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"orderClasses":false,"lengthMenu":[10,25,50,100,516]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="standardise-line-means-for-each-trait" class="section level2">
<h2>Standardise line means for each trait</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Some of our traits have already been scaled, either when we have fit
models to raw data or by the authors that have provided line means. For
consistency and to aid comparison of variances between traits, we
standardise them all here.</p>
<p>We scale line means by the standard deviation of each phenotype. The
result is that the mean value for each phenotype is 0 with a standard
deviation of 1.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># We simply load the data if its already been saved. If not, we change the format of the data. Each row represents a single line, while each column represents a trait and contains the lines trait mean. We then scale the values for each trait specifically, so that trait has a mean of 0 and an SD of 1. </span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;data/derived/all.dgrp.phenos_scaled.csv&quot;</span>)){</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  all.dgrp.phenos_scaled <span class="ot">&lt;-</span> </span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">split</span>(all.dgrp.phenos_combined_subset, all.dgrp.phenos_combined_subset<span class="sc">$</span>Trait) <span class="sc">%&gt;%</span> <span class="co"># split divides the data into trait groups</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(<span class="cf">function</span>(x){</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>      trait_name <span class="ot">&lt;-</span> x<span class="sc">$</span>Trait[<span class="dv">1</span>]</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>      x <span class="ot">&lt;-</span> x[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">names</span>(x) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;line&quot;</span>, <span class="st">&quot;value&quot;</span>)</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>      x <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(value))</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">names</span>(x) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;line&quot;</span>, trait_name)</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>      x}) <span class="sc">%&gt;%</span> </span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">reduce</span>(full_join, <span class="at">by =</span> <span class="st">&quot;line&quot;</span>, <span class="at">match =</span> <span class="st">&quot;first&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>(<span class="st">&quot;line&quot;</span>)), <span class="sc">~</span> <span class="fu">as.numeric</span>(<span class="fu">scale</span>(.x))) <span class="co"># this is the scaling step</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv</span>(all.dgrp.phenos_scaled, <span class="at">file =</span> <span class="st">&quot;data/derived/all.dgrp.phenos_scaled.csv&quot;</span>)</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> all.dgrp.phenos_scaled <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/all.dgrp.phenos_scaled.csv&quot;</span>)</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Now the selected traits for analysis</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;data/derived/selected.dgrp.phenos_scaled.csv&quot;</span>)){</span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a>  selected.dgrp.phenos_scaled <span class="ot">&lt;-</span> </span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">split</span>(selected.dgrp.phenos_unscaled, selected.dgrp.phenos_unscaled<span class="sc">$</span>Trait) <span class="sc">%&gt;%</span> <span class="co"># split divides the data into trait groups</span></span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(<span class="cf">function</span>(x){</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a>      trait_name <span class="ot">&lt;-</span> x<span class="sc">$</span>Trait[<span class="dv">1</span>]</span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>      x <span class="ot">&lt;-</span> x[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)]</span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>      <span class="fu">names</span>(x) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;line&quot;</span>, <span class="st">&quot;value&quot;</span>)</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>      x <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(value))</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">names</span>(x) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;line&quot;</span>, trait_name)</span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>      x}) <span class="sc">%&gt;%</span> </span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">reduce</span>(full_join, <span class="at">by =</span> <span class="st">&quot;line&quot;</span>, <span class="at">match =</span> <span class="st">&quot;first&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>(<span class="st">&quot;line&quot;</span>)), <span class="sc">~</span> <span class="fu">as.numeric</span>(<span class="fu">scale</span>(.x))) <span class="co"># this is the scaling step</span></span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv</span>(selected.dgrp.phenos_scaled, <span class="at">file =</span> <span class="st">&quot;data/derived/selected.dgrp.phenos_scaled.csv&quot;</span>)</span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> selected.dgrp.phenos_scaled <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/selected.dgrp.phenos_scaled.csv&quot;</span>)</span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] osfr_0.2.8          broom_0.8.0         DT_0.23            
 [4] tidybayes_3.0.2     brms_2.17.0         Rcpp_1.0.8.3       
 [7] googlesheets4_1.0.0 forcats_0.5.1       stringr_1.4.0      
[10] dplyr_1.0.9         purrr_0.3.4         readr_2.1.2        
[13] tidyr_1.2.0         tibble_3.1.7        ggplot2_3.3.6      
[16] tidyverse_1.3.1     Matrix_1.4-1        workflowr_1.7.0    

loaded via a namespace (and not attached):
  [1] readxl_1.4.0         backports_1.4.1      plyr_1.8.7          
  [4] igraph_1.3.2         svUnit_1.0.6         crosstalk_1.2.0     
  [7] rstantools_2.2.0     inline_0.3.19        digest_0.6.29       
 [10] htmltools_0.5.2      fansi_1.0.3          magrittr_2.0.3      
 [13] checkmate_2.1.0      memoise_2.0.1        tzdb_0.3.0          
 [16] modelr_0.1.8         RcppParallel_5.1.5   matrixStats_0.62.0  
 [19] vroom_1.5.7          xts_0.12.1           prettyunits_1.1.1   
 [22] colorspace_2.0-3     rvest_1.0.2          ggdist_3.1.1        
 [25] haven_2.5.0          xfun_0.31            callr_3.7.0         
 [28] crayon_1.5.1         jsonlite_1.8.0       zoo_1.8-10          
 [31] glue_1.6.2           gtable_0.3.0         gargle_1.2.0        
 [34] distributional_0.3.0 pkgbuild_1.3.1       rstan_2.21.5        
 [37] abind_1.4-5          scales_1.2.0         mvtnorm_1.1-3       
 [40] DBI_1.1.3            miniUI_0.1.1.1       xtable_1.8-4        
 [43] diffobj_0.3.5        bit_4.0.4            stats4_4.2.1        
 [46] StanHeaders_2.21.0-7 htmlwidgets_1.5.4    httr_1.4.3          
 [49] threejs_0.3.3        arrayhelpers_1.1-0   posterior_1.2.2     
 [52] ellipsis_0.3.2       pkgconfig_2.0.3      loo_2.5.1           
 [55] farver_2.1.0         sass_0.4.1           dbplyr_2.2.1        
 [58] utf8_1.2.2           crul_1.3             tidyselect_1.1.2    
 [61] rlang_1.0.3          reshape2_1.4.4       later_1.3.0         
 [64] cachem_1.0.6         munsell_0.5.0        cellranger_1.1.0    
 [67] tools_4.2.1          cli_3.3.0            generics_0.1.2      
 [70] ggridges_0.5.3       evaluate_0.15        fastmap_1.1.0       
 [73] yaml_2.3.5           bit64_4.0.5          processx_3.6.1      
 [76] knitr_1.39           fs_1.5.2             nlme_3.1-158        
 [79] whisker_0.4          mime_0.12            xml2_1.3.3          
 [82] compiler_4.2.1       bayesplot_1.9.0      shinythemes_1.2.0   
 [85] rstudioapi_0.13      curl_4.3.2           reprex_2.0.1        
 [88] bslib_0.3.1          stringi_1.7.6        ps_1.7.1            
 [91] Brobdingnag_1.2-7    lattice_0.20-45      markdown_1.1        
 [94] shinyjs_2.1.0        tensorA_0.36.2       vctrs_0.4.1         
 [97] pillar_1.7.0         lifecycle_1.0.1      jquerylib_0.1.4     
[100] bridgesampling_1.1-2 httpuv_1.6.5         R6_2.5.1            
[103] promises_1.2.0.1     gridExtra_2.3        codetools_0.2-18    
[106] colourpicker_1.1.1   gtools_3.9.2.2       assertthat_0.2.1    
[109] rprojroot_2.0.3      withr_2.5.0          httpcode_0.3.0      
[112] shinystan_2.6.0      parallel_4.2.1       hms_1.1.1           
[115] grid_4.2.1           coda_0.19-4          rmarkdown_2.14      
[118] googledrive_2.0.0    git2r_0.30.1         getPass_0.2-2       
[121] shiny_1.7.1          lubridate_1.8.0      base64enc_0.1-3     
[124] dygraphs_1.1.1.6    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
