<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Estimating line means from raw data</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.0/jquery-3.5.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">About</a>
</li>
<li>
  <a href="How_to.html">Upload your data</a>
</li>
<li>
  <a href="Data_collation.html">Data collation</a>
</li>
<li>
  <a href="Trait_correlations.html">Calculating genetic trait correlations</a>
</li>
<li>
  <a href="SNP_heritabilities.html">Calculating 'SNP-heritability'</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Estimating line means from raw data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220705code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220705code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220705)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtree21408d4c914485d67dac9cc170dfc5a394d8bf00targetblank21408d4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tomkeaney/DGRP/tree/21408d4c914485d67dac9cc170dfc5a394d8bf00" target="_blank">21408d4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtree21408d4c914485d67dac9cc170dfc5a394d8bf00targetblank21408d4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tomkeaney/DGRP/tree/21408d4c914485d67dac9cc170dfc5a394d8bf00" target="_blank">21408d4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/input/.DS_Store
    Ignored:    fits/

Untracked files:
    Untracked:  analysis/exporting GWAS stats for the app.R
    Untracked:  analysis/luke_cluster_ideas.R
    Untracked:  analysis/luke_weighted_beta_ideas.R
    Untracked:  analysis/run_GWAS.Rmd
    Untracked:  code/plink
    Untracked:  data/derived/dgrp_phenos_calculated_from_raw_data.csv
    Untracked:  data/imputed_data.RData
    Untracked:  data/trait_information.csv
    Untracked:  gwas_data/
    Untracked:  raw_data.csv

Unstaged changes:
    Modified:   analysis/Data_collation.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/get_line_means_from_raw_data.Rmd</code>) and HTML
(<code>docs/get_line_means_from_raw_data.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/21408d4c914485d67dac9cc170dfc5a394d8bf00/analysis/get_line_means_from_raw_data.Rmd" target="_blank">21408d4</a>
</td>
<td>
lukeholman
</td>
<td>
2022-10-13
</td>
<td>
First commit of get_line_means_from_raw_data.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-in-necessary-packages" class="section level1">
<h1>Load in necessary packages</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms) <span class="co"># for bayesian models</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidybayes) <span class="co"># for more bayesian things</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom) <span class="co"># convert results of functions into tables</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;fits&quot;</span>)) <span class="fu">dir.create</span>(<span class="st">&quot;fits&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>my_scale <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">scale</span>(x))</span></code></pre></div>
<div id="estimating-line-means-from-raw-data" class="section level2">
<h2>Estimating line means from raw data</h2>
<p>Some studies/authors provided the raw data that was collected when
measuring DGRP phenotypes (e.g. measurements of individual flies),
although we need line mean data to calculate genetic correlations
between traits and run typical GWAS analyses. To get line means, we fit
Bayesian mixed effect models and use these to estimate mean trait values
for each line, using the model formula that made most sense to us given
the study’s experimental design (see annotations in the code below). Our
analyses of each paper’s data are presented in a separate tab below.</p>
<p>Where the response follows a Gaussian distribution, we standardise
the response to have a mean of 0 and a standard deviation of 1, in order
to make prior specification more straightforward; in these cases, the
estimated line means are given in standard units.</p>
<p>The Bayesian models won’t run unless you have <code>stan</code> and
the <code>brms</code> package installed. To install <code>stan</code>
follow these <a
href="https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started">instructions</a>.
Otherwise, you can skip to the data cleaning section and load in the csv
file <code>"data/input/all.dgrp.phenos_combined.csv"</code> as a
tibble/dataframe.</p>
</div>
<div id="analysis-of-raw-data" class="section level2 tabset">
<h2 class="tabset">Analysis of raw data</h2>
<div id="jumbo-lucioni-et-al-2012" class="section level3">
<h3>Jumbo-Lucioni et al 2012</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Jumbo_mito_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Jumbo_Lucioni_2012.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Gender),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Rep =</span> <span class="fu">as.factor</span>(Rep),</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">po_ratio =</span> <span class="fu">na_if</span>(<span class="st">`</span><span class="at">po ratio</span><span class="st">`</span>, <span class="st">&quot;.&quot;</span>),</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">po_ratio =</span> <span class="fu">as.numeric</span>(po_ratio)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">state_3 =</span> <span class="st">`</span><span class="at">st 3 (pmol/s*mg protein)</span><span class="st">`</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">state_4 =</span> <span class="st">`</span><span class="at">st 4 (pmol/s*mg protein)</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Sex, Rep, state_3, state_4, po_ratio) <span class="sc">%&gt;%</span> </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(line))</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># standardise </span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>Jumbo_mito_data <span class="ot">&lt;-</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  Jumbo_mito_data <span class="sc">%&gt;%</span> </span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">st_state_3 =</span> <span class="fu">my_scale</span>(state_3),</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_state_4 =</span> <span class="fu">my_scale</span>(state_4),</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_po_ratio =</span> <span class="fu">my_scale</span>(po_ratio)) </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co"># respiration state 3 model</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>state_3_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_state_3 <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Jumbo_mito_data, <span class="at">family =</span> gaussian,</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/state_3.model&quot;</span>)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(state_3_model)</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Jumbo_mito_data<span class="sc">$</span>line,</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>, <span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>fitted_state_3 <span class="ot">&lt;-</span> <span class="fu">fitted</span>(state_3_model,</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_3) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_3),</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_3) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_3),</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_3) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_3))</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>state_3_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_state_3) <span class="sc">%&gt;%</span> </span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding sex specific line means</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>Jumbo_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, y, Trait, description){</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> y) <span class="sc">%&gt;%</span> </span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Jumbo-Lucioni et al (2012) BMC Genomics&quot;</span>,</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Physiological&quot;</span>,</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a><span class="co"># create the state 3 tibbles</span></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>state_3_f_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_3_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;mitochondrial.state.3.respiration.f&quot;</span>, <span class="st">&quot;Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>state_3_m_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_3_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;mitochondrial.state.3.respiration.m&quot;</span>, <span class="st">&quot;Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a><span class="co"># respiration state 4 model</span></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>state_4_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_state_4 <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data =</span> Jumbo_mito_data, <span class="at">family =</span> gaussian,</span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>                     <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>                     <span class="at">iter =</span> <span class="dv">10000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>                     <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/state_4.model&quot;</span>)</span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(state_4_model)</span></span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>fitted_state_4 <span class="ot">&lt;-</span> <span class="fu">fitted</span>(state_4_model,</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_4) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_4),</span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_4) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_4),</span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_mito_data<span class="sc">$</span>state_4) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_mito_data<span class="sc">$</span>state_4))</span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a>state_4_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_state_4) <span class="sc">%&gt;%</span> </span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a><span class="co"># create the state 4 tibbles</span></span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a>state_4_f_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_4_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;mitochondrial.state.4.respiration.f&quot;</span>, <span class="st">&quot;Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a>state_4_m_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(state_4_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;mitochondrial.state.4.respiration.m&quot;</span>, <span class="st">&quot;Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.&quot;</span>)</span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a><span class="co"># P:O ratio model</span></span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a>PO_ratio_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_po_ratio <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Jumbo_mito_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(po_ratio <span class="sc">!=</span> <span class="st">&quot;NA&quot;</span>), <span class="at">family =</span> gaussian,</span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/PO_ratio.model&quot;</span>)</span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(PO_ratio_model)</span></span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a>Jumbo_PO_data <span class="ot">&lt;-</span> Jumbo_mito_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(po_ratio <span class="sc">!=</span> <span class="st">&quot;NA&quot;</span>)</span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a>fitted_PO_ratio <span class="ot">&lt;-</span> <span class="fu">fitted</span>(PO_ratio_model, <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio),</span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio),</span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio) <span class="sc">+</span> <span class="fu">mean</span>(Jumbo_PO_data<span class="sc">$</span>po_ratio))</span>
<span id="cb2-133"><a href="#cb2-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-134"><a href="#cb2-134" aria-hidden="true" tabindex="-1"></a>PO_ratio_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_PO_ratio) <span class="sc">%&gt;%</span> </span>
<span id="cb2-135"><a href="#cb2-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb2-136"><a href="#cb2-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-137"><a href="#cb2-137" aria-hidden="true" tabindex="-1"></a><span class="co"># create the P:O ratio tibbles</span></span>
<span id="cb2-138"><a href="#cb2-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-139"><a href="#cb2-139" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb2-140"><a href="#cb2-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-141"><a href="#cb2-141" aria-hidden="true" tabindex="-1"></a>PO_ratio_f_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(PO_ratio_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;mitochondrial.PO.ratio.f&quot;</span>, <span class="st">&quot;Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.&quot;</span>)</span>
<span id="cb2-142"><a href="#cb2-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-143"><a href="#cb2-143" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb2-144"><a href="#cb2-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-145"><a href="#cb2-145" aria-hidden="true" tabindex="-1"></a>PO_ratio_m_line_means <span class="ot">&lt;-</span> <span class="fu">Jumbo_line_means</span>(PO_ratio_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;mitochondrial.PO.ratio.m&quot;</span>, <span class="st">&quot;Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.&quot;</span>)</span>
<span id="cb2-146"><a href="#cb2-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-147"><a href="#cb2-147" aria-hidden="true" tabindex="-1"></a>Jumbo_traits <span class="ot">&lt;-</span> </span>
<span id="cb2-148"><a href="#cb2-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(state_3_f_line_means, </span>
<span id="cb2-149"><a href="#cb2-149" aria-hidden="true" tabindex="-1"></a>        state_3_m_line_means, </span>
<span id="cb2-150"><a href="#cb2-150" aria-hidden="true" tabindex="-1"></a>        state_4_f_line_means, </span>
<span id="cb2-151"><a href="#cb2-151" aria-hidden="true" tabindex="-1"></a>        state_4_m_line_means, </span>
<span id="cb2-152"><a href="#cb2-152" aria-hidden="true" tabindex="-1"></a>        PO_ratio_f_line_means, </span>
<span id="cb2-153"><a href="#cb2-153" aria-hidden="true" tabindex="-1"></a>        PO_ratio_m_line_means)</span></code></pre></div>
</div>
<div id="grubbs-et-al-2013" class="section level3">
<h3>Grubbs et al 2013</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Grubbs et al 2013</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>Grubbs_2013_leg_props <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Grubbs_2013_leg_props.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Femur =</span> <span class="fu">my_scale</span>(Femur),</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Tibia =</span> <span class="fu">my_scale</span>(Tibia),</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Tarsus =</span> <span class="fu">my_scale</span>(Tarsus), </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Total =</span> <span class="fu">my_scale</span>(Total))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>Grubbs_2013_T1_leg_props <span class="ot">&lt;-</span> Grubbs_2013_leg_props <span class="sc">%&gt;%</span> </span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(T1_T2 <span class="sc">==</span> <span class="st">&quot;T1&quot;</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>Grubbs_2013_T2_leg_props <span class="ot">&lt;-</span> Grubbs_2013_leg_props <span class="sc">%&gt;%</span> </span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(T1_T2 <span class="sc">==</span> <span class="st">&quot;T2&quot;</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="co"># mean total T1 leg length</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>T1_leg_length_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Total <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1_leg.length.model&quot;</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">Line =</span> Grubbs_2013_leg_props<span class="sc">$</span>Line,</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>, <span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Line, Sex)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>fitted_T1_leg_total <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_leg_length_model, <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total),</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total),</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Total))</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding sex specific line means</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>Grubbs_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, y, Trait, guild, description){</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> y) <span class="sc">%&gt;%</span> </span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(Line, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> Trait,</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Grubbs et al (2013) PLOS One&quot;</span>,</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> guild,</span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>T1_leg_length_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_leg_total) <span class="sc">%&gt;%</span> </span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>T1_leg_length_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_leg_length_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.leg.length.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>T1_leg_length_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_leg_length_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.leg.length.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a><span class="co"># mean total T2 leg length</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>T2_leg_length_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Total <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> <span class="st">&quot;Gaussian&quot;</span>,</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2_leg.length.model&quot;</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>fitted_T2_leg_total <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_leg_length_model,</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total),</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total),</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Total))</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>T2_leg_length_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_leg_total) <span class="sc">%&gt;%</span> </span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>T2_leg_length_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_leg_length_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.leg.length.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>T2_leg_length_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_leg_length_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.leg.length.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs&quot;</span>)</span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Femur length</span></span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>T1_femur_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Femur <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1.femur.model&quot;</span>, </span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a>fitted_T1_femur <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_femur_model,</span>
<span id="cb3-109"><a href="#cb3-109" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Femur))</span>
<span id="cb3-115"><a href="#cb3-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-116"><a href="#cb3-116" aria-hidden="true" tabindex="-1"></a>T1_femur_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_femur) <span class="sc">%&gt;%</span> </span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-119"><a href="#cb3-119" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb3-120"><a href="#cb3-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a>femur_T1_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_femur_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.femur.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a>femur_T1_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_femur_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.femur.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a>T2_femur_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Femur <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2.femur.model&quot;</span>, </span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-136"><a href="#cb3-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-137"><a href="#cb3-137" aria-hidden="true" tabindex="-1"></a>fitted_T2_femur <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_femur_model,</span>
<span id="cb3-138"><a href="#cb3-138" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-139"><a href="#cb3-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur),</span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Femur))</span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a>T2_femur_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_femur) <span class="sc">%&gt;%</span> </span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a>femur_T2_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_femur_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.femur.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-154"><a href="#cb3-154" aria-hidden="true" tabindex="-1"></a>femur_T2_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_femur_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.femur.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs&quot;</span>)</span>
<span id="cb3-155"><a href="#cb3-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-156"><a href="#cb3-156" aria-hidden="true" tabindex="-1"></a><span class="co"># Tibia length</span></span>
<span id="cb3-157"><a href="#cb3-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-158"><a href="#cb3-158" aria-hidden="true" tabindex="-1"></a>T1_tibia_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tibia <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-159"><a href="#cb3-159" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-160"><a href="#cb3-160" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-161"><a href="#cb3-161" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-162"><a href="#cb3-162" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-163"><a href="#cb3-163" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-164"><a href="#cb3-164" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1.tibia.model&quot;</span>, </span>
<span id="cb3-165"><a href="#cb3-165" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-166"><a href="#cb3-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-167"><a href="#cb3-167" aria-hidden="true" tabindex="-1"></a>fitted_T1_tibia <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_tibia_model,</span>
<span id="cb3-168"><a href="#cb3-168" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-169"><a href="#cb3-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-170"><a href="#cb3-170" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-171"><a href="#cb3-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb3-172"><a href="#cb3-172" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb3-173"><a href="#cb3-173" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tibia))</span>
<span id="cb3-174"><a href="#cb3-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-175"><a href="#cb3-175" aria-hidden="true" tabindex="-1"></a>T1_tibia_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_tibia) <span class="sc">%&gt;%</span> </span>
<span id="cb3-176"><a href="#cb3-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-177"><a href="#cb3-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-178"><a href="#cb3-178" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb3-179"><a href="#cb3-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-180"><a href="#cb3-180" aria-hidden="true" tabindex="-1"></a>tibia_T1_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tibia_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.tibia.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb3-181"><a href="#cb3-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-182"><a href="#cb3-182" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb3-183"><a href="#cb3-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-184"><a href="#cb3-184" aria-hidden="true" tabindex="-1"></a>tibia_T1_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tibia_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.tibia.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb3-185"><a href="#cb3-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-186"><a href="#cb3-186" aria-hidden="true" tabindex="-1"></a>T2_tibia_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tibia <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-187"><a href="#cb3-187" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-188"><a href="#cb3-188" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-189"><a href="#cb3-189" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-190"><a href="#cb3-190" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-191"><a href="#cb3-191" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">10000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-192"><a href="#cb3-192" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2.tibia.model&quot;</span>, </span>
<span id="cb3-193"><a href="#cb3-193" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-194"><a href="#cb3-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-195"><a href="#cb3-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-196"><a href="#cb3-196" aria-hidden="true" tabindex="-1"></a>fitted_T2_tibia <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_tibia_model,</span>
<span id="cb3-197"><a href="#cb3-197" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-198"><a href="#cb3-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-199"><a href="#cb3-199" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-200"><a href="#cb3-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb3-201"><a href="#cb3-201" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia),</span>
<span id="cb3-202"><a href="#cb3-202" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tibia))</span>
<span id="cb3-203"><a href="#cb3-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-204"><a href="#cb3-204" aria-hidden="true" tabindex="-1"></a>T2_tibia_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_tibia) <span class="sc">%&gt;%</span> </span>
<span id="cb3-205"><a href="#cb3-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-206"><a href="#cb3-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-207"><a href="#cb3-207" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb3-208"><a href="#cb3-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-209"><a href="#cb3-209" aria-hidden="true" tabindex="-1"></a>tibia_T2_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tibia_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.tibia.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb3-210"><a href="#cb3-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-211"><a href="#cb3-211" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb3-212"><a href="#cb3-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-213"><a href="#cb3-213" aria-hidden="true" tabindex="-1"></a>tibia_T2_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tibia_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.tibia.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;</span>)</span>
<span id="cb3-214"><a href="#cb3-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-215"><a href="#cb3-215" aria-hidden="true" tabindex="-1"></a><span class="co"># Tarsus length</span></span>
<span id="cb3-216"><a href="#cb3-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-217"><a href="#cb3-217" aria-hidden="true" tabindex="-1"></a>T1_tarsus_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tarsus <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-218"><a href="#cb3-218" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T1_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-219"><a href="#cb3-219" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-220"><a href="#cb3-220" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-221"><a href="#cb3-221" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-222"><a href="#cb3-222" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-223"><a href="#cb3-223" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T1.tarsus.model&quot;</span>, </span>
<span id="cb3-224"><a href="#cb3-224" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-225"><a href="#cb3-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-226"><a href="#cb3-226" aria-hidden="true" tabindex="-1"></a>fitted_T1_tarsus <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T1_tarsus_model,</span>
<span id="cb3-227"><a href="#cb3-227" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-228"><a href="#cb3-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-229"><a href="#cb3-229" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-230"><a href="#cb3-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb3-231"><a href="#cb3-231" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb3-232"><a href="#cb3-232" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T1_leg_props<span class="sc">$</span>Tarsus))</span>
<span id="cb3-233"><a href="#cb3-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-234"><a href="#cb3-234" aria-hidden="true" tabindex="-1"></a>T1_tarsus_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T1_tarsus) <span class="sc">%&gt;%</span> </span>
<span id="cb3-235"><a href="#cb3-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-236"><a href="#cb3-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-237"><a href="#cb3-237" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 females</span></span>
<span id="cb3-238"><a href="#cb3-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-239"><a href="#cb3-239" aria-hidden="true" tabindex="-1"></a>tarsus_T1_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tarsus_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T1.tarsus.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb3-240"><a href="#cb3-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-241"><a href="#cb3-241" aria-hidden="true" tabindex="-1"></a><span class="co"># T1 males</span></span>
<span id="cb3-242"><a href="#cb3-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-243"><a href="#cb3-243" aria-hidden="true" tabindex="-1"></a>tarsus_T1_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T1_tarsus_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T1.tarsus.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb3-244"><a href="#cb3-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-245"><a href="#cb3-245" aria-hidden="true" tabindex="-1"></a>T2_tarsus_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Tarsus <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>Line),</span>
<span id="cb3-246"><a href="#cb3-246" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Grubbs_2013_T2_leg_props, <span class="at">family =</span> gaussian,</span>
<span id="cb3-247"><a href="#cb3-247" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb3-248"><a href="#cb3-248" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb3-249"><a href="#cb3-249" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb3-250"><a href="#cb3-250" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb3-251"><a href="#cb3-251" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/T2.tarsus.model&quot;</span>, </span>
<span id="cb3-252"><a href="#cb3-252" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb3-253"><a href="#cb3-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-254"><a href="#cb3-254" aria-hidden="true" tabindex="-1"></a>fitted_T2_tarsus <span class="ot">&lt;-</span> <span class="fu">fitted</span>(T2_tarsus_model,</span>
<span id="cb3-255"><a href="#cb3-255" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb3-256"><a href="#cb3-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-257"><a href="#cb3-257" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb3-258"><a href="#cb3-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb3-259"><a href="#cb3-259" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus),</span>
<span id="cb3-260"><a href="#cb3-260" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus) <span class="sc">+</span> <span class="fu">mean</span>(Grubbs_2013_T2_leg_props<span class="sc">$</span>Tarsus))</span>
<span id="cb3-261"><a href="#cb3-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-262"><a href="#cb3-262" aria-hidden="true" tabindex="-1"></a>T2_tarsus_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_T2_tarsus) <span class="sc">%&gt;%</span> </span>
<span id="cb3-263"><a href="#cb3-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>))</span>
<span id="cb3-264"><a href="#cb3-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-265"><a href="#cb3-265" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 females</span></span>
<span id="cb3-266"><a href="#cb3-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-267"><a href="#cb3-267" aria-hidden="true" tabindex="-1"></a>tarsus_T2_f_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tarsus_estimates, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;T2.tarsus.f&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb3-268"><a href="#cb3-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-269"><a href="#cb3-269" aria-hidden="true" tabindex="-1"></a><span class="co"># T2 males</span></span>
<span id="cb3-270"><a href="#cb3-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-271"><a href="#cb3-271" aria-hidden="true" tabindex="-1"></a>tarsus_T2_m_line_means <span class="ot">&lt;-</span> <span class="fu">Grubbs_line_means</span>(T2_tarsus_estimates, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;T2.tarsus.m&quot;</span>, <span class="st">&quot;Morphological&quot;</span>, <span class="st">&quot;Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus&#39;.&quot;</span>)</span>
<span id="cb3-272"><a href="#cb3-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-273"><a href="#cb3-273" aria-hidden="true" tabindex="-1"></a>Grubbs_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(T1_leg_length_f_line_means, </span>
<span id="cb3-274"><a href="#cb3-274" aria-hidden="true" tabindex="-1"></a>                       T1_leg_length_m_line_means, </span>
<span id="cb3-275"><a href="#cb3-275" aria-hidden="true" tabindex="-1"></a>                       T2_leg_length_f_line_means, </span>
<span id="cb3-276"><a href="#cb3-276" aria-hidden="true" tabindex="-1"></a>                       T2_leg_length_m_line_means,</span>
<span id="cb3-277"><a href="#cb3-277" aria-hidden="true" tabindex="-1"></a>                       femur_T1_f_line_means, </span>
<span id="cb3-278"><a href="#cb3-278" aria-hidden="true" tabindex="-1"></a>                       femur_T1_m_line_means, </span>
<span id="cb3-279"><a href="#cb3-279" aria-hidden="true" tabindex="-1"></a>                       femur_T2_f_line_means, </span>
<span id="cb3-280"><a href="#cb3-280" aria-hidden="true" tabindex="-1"></a>                       femur_T2_m_line_means,</span>
<span id="cb3-281"><a href="#cb3-281" aria-hidden="true" tabindex="-1"></a>                       tibia_T1_f_line_means, </span>
<span id="cb3-282"><a href="#cb3-282" aria-hidden="true" tabindex="-1"></a>                       tibia_T1_m_line_means, </span>
<span id="cb3-283"><a href="#cb3-283" aria-hidden="true" tabindex="-1"></a>                       tibia_T2_f_line_means, </span>
<span id="cb3-284"><a href="#cb3-284" aria-hidden="true" tabindex="-1"></a>                       tibia_T2_m_line_means,</span>
<span id="cb3-285"><a href="#cb3-285" aria-hidden="true" tabindex="-1"></a>                       tarsus_T1_f_line_means, </span>
<span id="cb3-286"><a href="#cb3-286" aria-hidden="true" tabindex="-1"></a>                       tarsus_T1_m_line_means, </span>
<span id="cb3-287"><a href="#cb3-287" aria-hidden="true" tabindex="-1"></a>                       tarsus_T2_f_line_means, </span>
<span id="cb3-288"><a href="#cb3-288" aria-hidden="true" tabindex="-1"></a>                       tarsus_T2_m_line_means)</span></code></pre></div>
</div>
<div id="dobson-et-al-2015" class="section level3">
<h3>Dobson et al 2015</h3>
<p>From <strong>Supplementary Table 1</strong>’s caption: “In the second
approach (c), simpler models were fitted, in which microbiota, weight
and Wolbachia status were additive, with genotype nested in experimental
block as a random effect. The statistical analysis presented in Fig.1b
uses these models simplified by removal of non-significant factors.”</p>
<p>We follow this approach to best recapitulate the study’s analysis,
except that we allow the intercept for the random effect line to vary
with microbiota treatment. This is needed in our modelling approach to
estimate line-specific effects of microbiota. We also include day as a
blocking factor, which represents a block of data collection involving a
subset of lines. This controls for micro-environmental variance between
blocks of phenotyping.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># weight</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>Dobson_NI_microbiome.dryweight <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.dw.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_weight.5.flies =</span> <span class="fu">my_scale</span>(weight.<span class="fl">5.</span>flies)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(PC1, PC2, PC3, Code, RAL, ...<span class="dv">1</span>))</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_NI_microbiome.dryweight <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_NI_microbiome.dryweight <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measurements in both treatments </span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>weight_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding specific line means</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>Dobson_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, Trait, guild, description){</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Dobson et al (2015) Nature Communications&quot;</span>,</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> guild,</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>weight_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_weight.<span class="fl">5.</span>flies <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_NI_microbiome.dryweight, <span class="at">family =</span> gaussian,</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/weight.microbiome.model&quot;</span>)</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(weight_microbiome_model)</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_NI_microbiome.dryweight<span class="sc">$</span>line,</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_NI_microbiome.dryweight<span class="sc">$</span>trt,</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day)</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>fitted_dryweight <span class="ot">&lt;-</span> <span class="fu">fitted</span>(weight_microbiome_model, <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies),</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies),</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_NI_microbiome.dryweight<span class="sc">$</span>weight.<span class="fl">5.</span>flies))</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>dry_weight_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_dryweight) <span class="sc">%&gt;%</span> </span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>weight.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(dry_weight_estimates_axenic, <span class="st">&quot;weight.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.&quot;</span>)</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>dry_weight_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_dryweight) <span class="sc">%&gt;%</span> </span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>weight.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(dry_weight_estimates_gnotobiotic, <span class="st">&quot;weight.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.&quot;</span>)</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>axenic <span class="ot">&lt;-</span> </span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(weight_microbiome_effect_lines, dry_weight_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>gnoto <span class="ot">&lt;-</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(weight_microbiome_effect_lines, dry_weight_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>dry_weight_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic, gnoto) <span class="sc">%&gt;%</span> </span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_weight_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(dry_weight_estimates_diff, <span class="st">&quot;weight.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.&quot;</span>)</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Glucose</span></span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a>Dobson_glucose_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.glucose.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_glucose =</span> <span class="fu">my_scale</span>(glucose)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, wolb, glucose, trt, day, weight.<span class="fl">5.</span>flies, st_glucose))</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_glucose_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_glucose_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a>glucose_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that we include weight as a moderator variable. This means we only estimate the direct effect of line on glucose levels, instead of the total effect. This is because line likely also affects weight which is likely to have a flow-on effect on glucose levels. </span></span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>glucose_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_glucose <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_glucose_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/glucose.microbiome.model&quot;</span>)</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(glucose_microbiome_model)</span></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_glucose_microbiome<span class="sc">$</span>line,</span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_glucose_microbiome<span class="sc">$</span>trt,</span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>fitted_glucose <span class="ot">&lt;-</span> <span class="fu">fitted</span>(glucose_microbiome_model,</span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data, <span class="at">allow_new_levels =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="co"># Why do we need to allow new levels? Might be an issue</span></span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose),</span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose),</span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glucose_microbiome<span class="sc">$</span>glucose))</span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>glucose_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glucose) <span class="sc">%&gt;%</span> </span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>glucose.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glucose_estimates_axenic, <span class="st">&quot;glucose.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.&quot;</span>)</span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a>glucose_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glucose) <span class="sc">%&gt;%</span> </span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a>glucose.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glucose_estimates_gnotobiotic, <span class="st">&quot;glucose.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.&quot;</span>)</span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>axenic_glucose <span class="ot">&lt;-</span> </span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glucose_microbiome_effect_lines, glucose_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a>gnoto_glucose <span class="ot">&lt;-</span></span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glucose_microbiome_effect_lines, glucose_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a>glucose_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_glucose, gnoto_glucose) <span class="sc">%&gt;%</span> </span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_glucose_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glucose_estimates_diff, <span class="st">&quot;glucose.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.&quot;</span>)</span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a><span class="co"># Glycogen</span></span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>Dobson_glycogen_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.glycogen.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_glycogen =</span> <span class="fu">my_scale</span>(glycogen)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, wolb, glycogen, st_glycogen, trt, day, weight.<span class="fl">5.</span>flies))</span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_glycogen_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_glycogen_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a>glycogen_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>glycogen_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_glycogen <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_glycogen_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/glycogen.microbiome.model&quot;</span>)</span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(glycogen_microbiome_model)</span></span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_glycogen_microbiome<span class="sc">$</span>line,</span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_glycogen_microbiome<span class="sc">$</span>trt,</span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a>fitted_glycogen <span class="ot">&lt;-</span> <span class="fu">fitted</span>(glycogen_microbiome_model,</span>
<span id="cb4-207"><a href="#cb4-207" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span></span>
<span id="cb4-208"><a href="#cb4-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen),</span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen),</span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_glycogen_microbiome<span class="sc">$</span>glycogen))</span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a>glycogen_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glycogen) <span class="sc">%&gt;%</span> </span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-217"><a href="#cb4-217" aria-hidden="true" tabindex="-1"></a>glycogen.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glycogen_estimates_axenic, <span class="st">&quot;glycogen.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined glycogen content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glycogen content.&quot;</span>)</span>
<span id="cb4-218"><a href="#cb4-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a>glycogen_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_glycogen) <span class="sc">%&gt;%</span> </span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a>glycogen.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glycogen_estimates_gnotobiotic, <span class="st">&quot;glycogen.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>,  <span class="st">&quot;Mean combined glycogen content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glycogen content.&quot;</span>)</span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a>axenic_glycogen <span class="ot">&lt;-</span> </span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glycogen_microbiome_effect_lines, glycogen_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a>gnoto_glycogen <span class="ot">&lt;-</span></span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(glycogen_microbiome_effect_lines, glycogen_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a>glycogen_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_glycogen, gnoto_glycogen) <span class="sc">%&gt;%</span> </span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_glycogen_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(glycogen_estimates_diff, <span class="st">&quot;glycogen.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined glycogen content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glycogen content. Note that the presence of micriobiota on average increased glycogen content.&quot;</span>)</span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a><span class="co"># Protein</span></span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a>Dobson_protein_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.protein.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="fu">as.factor</span>(wolb),</span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_protein =</span> <span class="fu">my_scale</span>(protein)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-251"><a href="#cb4-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, wolb, protein, st_protein, trt, day, weight.<span class="fl">5.</span>flies))</span>
<span id="cb4-252"><a href="#cb4-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_protein_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_protein_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a>protein_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a>protein_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_protein <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> wolb <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_protein_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/protein.microbiome.model&quot;</span>)</span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(protein_microbiome_model)</span></span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_protein_microbiome<span class="sc">$</span>line,</span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_protein_microbiome<span class="sc">$</span>trt,</span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a>         <span class="at">wolb =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-278"><a href="#cb4-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, wolb, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb4-279"><a href="#cb4-279" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a>fitted_protein <span class="ot">&lt;-</span> <span class="fu">fitted</span>(protein_microbiome_model,</span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span></span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_protein_microbiome<span class="sc">$</span>protein) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_protein_microbiome<span class="sc">$</span>protein),</span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_protein_microbiome<span class="sc">$</span>protein) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_protein_microbiome<span class="sc">$</span>protein),</span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_protein_microbiome<span class="sc">$</span>protein) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_protein_microbiome<span class="sc">$</span>protein))</span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-289"><a href="#cb4-289" aria-hidden="true" tabindex="-1"></a>protein_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_protein) <span class="sc">%&gt;%</span> </span>
<span id="cb4-290"><a href="#cb4-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb4-291"><a href="#cb4-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-292"><a href="#cb4-292" aria-hidden="true" tabindex="-1"></a>protein.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(protein_estimates_axenic, <span class="st">&quot;protein.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.&quot;</span>)</span>
<span id="cb4-293"><a href="#cb4-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-294"><a href="#cb4-294" aria-hidden="true" tabindex="-1"></a>protein_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_protein) <span class="sc">%&gt;%</span> </span>
<span id="cb4-295"><a href="#cb4-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb4-296"><a href="#cb4-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-297"><a href="#cb4-297" aria-hidden="true" tabindex="-1"></a>protein.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(protein_estimates_gnotobiotic, <span class="st">&quot;protein.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.&quot;</span>)</span>
<span id="cb4-298"><a href="#cb4-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-299"><a href="#cb4-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb4-300"><a href="#cb4-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-301"><a href="#cb4-301" aria-hidden="true" tabindex="-1"></a>axenic_protein <span class="ot">&lt;-</span> </span>
<span id="cb4-302"><a href="#cb4-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_microbiome_effect_lines, protein_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-303"><a href="#cb4-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-304"><a href="#cb4-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb4-305"><a href="#cb4-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-306"><a href="#cb4-306" aria-hidden="true" tabindex="-1"></a>gnoto_protein <span class="ot">&lt;-</span></span>
<span id="cb4-307"><a href="#cb4-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_microbiome_effect_lines, protein_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-308"><a href="#cb4-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-309"><a href="#cb4-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb4-310"><a href="#cb4-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-311"><a href="#cb4-311" aria-hidden="true" tabindex="-1"></a>protein_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb4-312"><a href="#cb4-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_protein, gnoto_protein) <span class="sc">%&gt;%</span> </span>
<span id="cb4-313"><a href="#cb4-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb4-314"><a href="#cb4-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-315"><a href="#cb4-315" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_protein_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(protein_estimates_diff, <span class="st">&quot;protein.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in the combined protein content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher protein content. Note that the presence of micriobiota on average increased protein content.&quot;</span>)</span>
<span id="cb4-316"><a href="#cb4-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-317"><a href="#cb4-317" aria-hidden="true" tabindex="-1"></a><span class="co"># TAG glycerol</span></span>
<span id="cb4-318"><a href="#cb4-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-319"><a href="#cb4-319" aria-hidden="true" tabindex="-1"></a>Dobson_TAG_glycerol_microbiome <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dobson_2015/gwasB.filtered.TAG_glycerol.ab.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-320"><a href="#cb4-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="fu">str_remove</span>(RAL, <span class="st">&quot;RAL_&quot;</span>)),</span>
<span id="cb4-321"><a href="#cb4-321" aria-hidden="true" tabindex="-1"></a>         <span class="co">#wolb = as.factor(wolb),</span></span>
<span id="cb4-322"><a href="#cb4-322" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">as.factor</span>(trt),</span>
<span id="cb4-323"><a href="#cb4-323" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">as.factor</span>(day),</span>
<span id="cb4-324"><a href="#cb4-324" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb4-325"><a href="#cb4-325" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_TAG_glycerol =</span> <span class="fu">my_scale</span>(TAG_glycerol)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-326"><a href="#cb4-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(line, TAG_glycerol, st_TAG_glycerol, trt, day, weight.<span class="fl">5.</span>flies))</span>
<span id="cb4-327"><a href="#cb4-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-328"><a href="#cb4-328" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line</span></span>
<span id="cb4-329"><a href="#cb4-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-330"><a href="#cb4-330" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> Dobson_TAG_glycerol_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-331"><a href="#cb4-331" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> Dobson_TAG_glycerol_microbiome <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(line)</span>
<span id="cb4-332"><a href="#cb4-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-333"><a href="#cb4-333" aria-hidden="true" tabindex="-1"></a><span class="co"># inner_join selects lines that have measures for both treatments</span></span>
<span id="cb4-334"><a href="#cb4-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-335"><a href="#cb4-335" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_microbiome_effect_lines <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(a, b)</span>
<span id="cb4-336"><a href="#cb4-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-337"><a href="#cb4-337" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_microbiome_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_TAG_glycerol <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> trt <span class="sc">+</span> day <span class="sc">+</span> weight.<span class="fl">5.</span>flies <span class="sc">+</span> (trt<span class="sc">|</span>line),</span>
<span id="cb4-338"><a href="#cb4-338" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> Dobson_TAG_glycerol_microbiome, <span class="at">family =</span> gaussian,</span>
<span id="cb4-339"><a href="#cb4-339" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb4-340"><a href="#cb4-340" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb4-341"><a href="#cb4-341" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb4-342"><a href="#cb4-342" aria-hidden="true" tabindex="-1"></a>                               <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb4-343"><a href="#cb4-343" aria-hidden="true" tabindex="-1"></a>                               <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/TAG_glycerol.microbiome.model&quot;</span>)</span>
<span id="cb4-344"><a href="#cb4-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-345"><a href="#cb4-345" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(TAG_glycerol_microbiome_model)</span></span>
<span id="cb4-346"><a href="#cb4-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-347"><a href="#cb4-347" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb4-348"><a href="#cb4-348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dobson_TAG_glycerol_microbiome<span class="sc">$</span>line,</span>
<span id="cb4-349"><a href="#cb4-349" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> Dobson_TAG_glycerol_microbiome<span class="sc">$</span>trt,</span>
<span id="cb4-350"><a href="#cb4-350" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="st">&quot;a&quot;</span>,</span>
<span id="cb4-351"><a href="#cb4-351" aria-hidden="true" tabindex="-1"></a>         <span class="at">weight.5.flies =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-352"><a href="#cb4-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, trt, day, weight.<span class="fl">5.</span>flies)</span>
<span id="cb4-353"><a href="#cb4-353" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb4-354"><a href="#cb4-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-355"><a href="#cb4-355" aria-hidden="true" tabindex="-1"></a>fitted_TAG_glycerol <span class="ot">&lt;-</span> <span class="fu">fitted</span>(TAG_glycerol_microbiome_model,</span>
<span id="cb4-356"><a href="#cb4-356" aria-hidden="true" tabindex="-1"></a>                              <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span></span>
<span id="cb4-357"><a href="#cb4-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-358"><a href="#cb4-358" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb4-359"><a href="#cb4-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol),</span>
<span id="cb4-360"><a href="#cb4-360" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol),</span>
<span id="cb4-361"><a href="#cb4-361" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol) <span class="sc">+</span> <span class="fu">mean</span>(Dobson_TAG_glycerol_microbiome<span class="sc">$</span>TAG_glycerol))</span>
<span id="cb4-362"><a href="#cb4-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-363"><a href="#cb4-363" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_estimates_axenic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_TAG_glycerol) <span class="sc">%&gt;%</span> </span>
<span id="cb4-364"><a href="#cb4-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;a&quot;</span>)</span>
<span id="cb4-365"><a href="#cb4-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-366"><a href="#cb4-366" aria-hidden="true" tabindex="-1"></a>TAG_glycerol.anexic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(TAG_glycerol_estimates_axenic, <span class="st">&quot;triglyceride.anexic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher TAG content.&quot;</span>)</span>
<span id="cb4-367"><a href="#cb4-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-368"><a href="#cb4-368" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_estimates_gnotobiotic <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_TAG_glycerol) <span class="sc">%&gt;%</span> </span>
<span id="cb4-369"><a href="#cb4-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb4-370"><a href="#cb4-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-371"><a href="#cb4-371" aria-hidden="true" tabindex="-1"></a>TAG_glycerol.gnotobiotic_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(TAG_glycerol_estimates_gnotobiotic, <span class="st">&quot;triglyceride.gnotobiotic.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.&quot;</span>)</span>
<span id="cb4-372"><a href="#cb4-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-373"><a href="#cb4-373" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb4-374"><a href="#cb4-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-375"><a href="#cb4-375" aria-hidden="true" tabindex="-1"></a>axenic_TAG_glycerol <span class="ot">&lt;-</span> </span>
<span id="cb4-376"><a href="#cb4-376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(TAG_glycerol_microbiome_effect_lines, TAG_glycerol_estimates_axenic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-377"><a href="#cb4-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-378"><a href="#cb4-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">axenic_estimate =</span> Estimate)</span>
<span id="cb4-379"><a href="#cb4-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-380"><a href="#cb4-380" aria-hidden="true" tabindex="-1"></a>gnoto_TAG_glycerol <span class="ot">&lt;-</span></span>
<span id="cb4-381"><a href="#cb4-381" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(TAG_glycerol_microbiome_effect_lines, TAG_glycerol_estimates_gnotobiotic) <span class="sc">%&gt;%</span> </span>
<span id="cb4-382"><a href="#cb4-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-383"><a href="#cb4-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gnoto_estimate =</span> Estimate)</span>
<span id="cb4-384"><a href="#cb4-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-385"><a href="#cb4-385" aria-hidden="true" tabindex="-1"></a>TAG_glycerol_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb4-386"><a href="#cb4-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(axenic_TAG_glycerol, gnoto_TAG_glycerol) <span class="sc">%&gt;%</span> </span>
<span id="cb4-387"><a href="#cb4-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> gnoto_estimate <span class="sc">-</span> axenic_estimate)</span>
<span id="cb4-388"><a href="#cb4-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-389"><a href="#cb4-389" aria-hidden="true" tabindex="-1"></a>gnoto_anexic_diff_TAG_glycerol_line_means <span class="ot">&lt;-</span> <span class="fu">Dobson_line_means</span>(TAG_glycerol_estimates_diff, <span class="st">&quot;TAG.glycerol.microbiome.effect.m&quot;</span>, <span class="st">&quot;Microbiome&quot;</span>, <span class="st">&quot;Mean change in combined triglyceride content after glycerol substraction of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher TAG content. Note though that the presence of micriobiota on average decreased TAG content.&quot;</span>)</span>
<span id="cb4-390"><a href="#cb4-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-391"><a href="#cb4-391" aria-hidden="true" tabindex="-1"></a>Dobson_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(weight.anexic_line_means, </span>
<span id="cb4-392"><a href="#cb4-392" aria-hidden="true" tabindex="-1"></a>                       weight.gnotobiotic_line_means, </span>
<span id="cb4-393"><a href="#cb4-393" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_weight_line_means,</span>
<span id="cb4-394"><a href="#cb4-394" aria-hidden="true" tabindex="-1"></a>                       glucose.anexic_line_means, </span>
<span id="cb4-395"><a href="#cb4-395" aria-hidden="true" tabindex="-1"></a>                       glucose.gnotobiotic_line_means, </span>
<span id="cb4-396"><a href="#cb4-396" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_glucose_line_means,</span>
<span id="cb4-397"><a href="#cb4-397" aria-hidden="true" tabindex="-1"></a>                       glycogen.anexic_line_means, </span>
<span id="cb4-398"><a href="#cb4-398" aria-hidden="true" tabindex="-1"></a>                       glycogen.gnotobiotic_line_means, </span>
<span id="cb4-399"><a href="#cb4-399" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_glycogen_line_means,</span>
<span id="cb4-400"><a href="#cb4-400" aria-hidden="true" tabindex="-1"></a>                       protein.anexic_line_means, </span>
<span id="cb4-401"><a href="#cb4-401" aria-hidden="true" tabindex="-1"></a>                       protein.gnotobiotic_line_means, </span>
<span id="cb4-402"><a href="#cb4-402" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_protein_line_means,</span>
<span id="cb4-403"><a href="#cb4-403" aria-hidden="true" tabindex="-1"></a>                       TAG_glycerol.anexic_line_means, </span>
<span id="cb4-404"><a href="#cb4-404" aria-hidden="true" tabindex="-1"></a>                       TAG_glycerol.gnotobiotic_line_means, </span>
<span id="cb4-405"><a href="#cb4-405" aria-hidden="true" tabindex="-1"></a>                       gnoto_anexic_diff_TAG_glycerol_line_means)</span></code></pre></div>
</div>
<div id="gaertner-et-al-2015" class="section level3">
<h3>Gaertner et al 2015</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>Gaertner_2015_male_mating <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Gaertner_2015_male_mating.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Block, Day, Assay, Chamber, Line, Subject, SUM, orient_female, approach, wing, genital_lick, attempt_cop, copulate, sum_court) <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">MMP =</span> SUM) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Block =</span> <span class="fu">as.factor</span>(Block),</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">courtship_events =</span> orient_female <span class="sc">+</span> approach <span class="sc">+</span> wing <span class="sc">+</span> genital_lick <span class="sc">+</span> attempt_cop,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">mating_latency =</span> <span class="dv">31</span> <span class="sc">-</span> copulate, <span class="co"># we subtract from 31 to tell the model that flies did not start the assay mating </span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">censor =</span> <span class="fu">if_else</span>(mating_latency <span class="sc">==</span> <span class="dv">31</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding specific line means</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>Gaertner_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, Trait, guild, description){</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span> </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Gaertner et al (2015) G3&quot;</span>,</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> guild,</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co"># We will model the 5 courtship processes individually (note that the courtship data is very zero inflated)</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="co"># orientation towards females</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>orient_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(orient_female <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> zero_inflated_poisson,</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>                    <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>                    <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/orient.female.model&quot;</span>,</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Gaertner_2015_male_mating<span class="sc">$</span>Line,</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">Block =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Block)</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>fitted_orient <span class="ot">&lt;-</span> <span class="fu">fitted</span>(orient_model,</span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> orient_female <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>orient_to_female_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_orient) </span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>orient_to_female_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(orient_to_female_m_estimates, <span class="st">&quot;orient.towards.female.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a><span class="co"># approaching females</span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>approach_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(approach <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> zero_inflated_poisson,</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/approach.female.model&quot;</span>,</span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>fitted_approach <span class="ot">&lt;-</span> <span class="fu">fitted</span>(approach_model,</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> approach <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>approach_female_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_approach) </span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>approach_female_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(approach_female_m_estimates, <span class="st">&quot;approach.female.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a><span class="co"># wing display (singing)</span></span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>wing_display_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(wing <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>                          <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>                          <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>                          <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/wing.display.model&quot;</span>,</span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>                          <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>fitted_wing_display <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_display_model,</span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> wing <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>wing_display_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_display) </span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>wing_display_frequency_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(wing_display_m_estimates, <span class="st">&quot;wing.display.frequency.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a><span class="co"># genital licking</span></span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>genital_licking_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(genital_lick <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a>                             <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>                             <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a>                             <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/genital.lick.model&quot;</span>,</span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a>                             <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a>fitted_genital_licking <span class="ot">&lt;-</span> <span class="fu">fitted</span>(genital_licking_model,</span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> genital_lick <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>genital_licking_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_genital_licking) </span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a>genital_licking_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(genital_licking_m_estimates, <span class="st">&quot;genital.licking.frequency.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a><span class="co"># attempted copulations</span></span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a>attempted_copulations_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(attempt_cop <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">data =</span> Gaertner_2015_male_mating, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> b),</span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">cauchy</span>(<span class="dv">0</span>, <span class="fl">0.2</span>), <span class="at">class =</span> sd),</span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">beta</span>(<span class="dv">2</span>, <span class="dv">6</span>), <span class="at">class =</span> zi)),  <span class="co"># the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values</span></span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/attempt.cop.model&quot;</span>,</span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a>fitted_attempted_copulations <span class="ot">&lt;-</span> </span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fitted</span>(attempted_copulations_model,</span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>attempted_copulations_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_attempted_copulations) </span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>attempted_copulations_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(attempted_copulations_m_estimates, <span class="st">&quot;attempted.copulation.frequency.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.&quot;</span>)</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a><span class="co"># Mating latency (like a no choice trial)</span></span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a><span class="co"># The data is right censored for males that did not mate</span></span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>mating_latency_model <span class="ot">&lt;-</span></span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(mating_latency <span class="sc">|</span> <span class="fu">cens</span>(censor) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Gaertner_2015_male_mating, </span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> <span class="fu">weibull</span>(), <span class="at">inits =</span> <span class="dv">0</span>,</span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">class =</span> Intercept), <span class="co"># the intercept in this model is the rate of decline in un-mated males</span></span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a>      <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/mating.latency.model&quot;</span>,</span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a>fitted_mating_latency <span class="ot">&lt;-</span> <span class="fu">fitted</span>(mating_latency_model,</span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> mating_latency <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a>mating_latency_m_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_mating_latency) </span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a>mating_latency_m_line_means <span class="ot">&lt;-</span> <span class="fu">Gaertner_line_means</span>(mating_latency_m_estimates, <span class="st">&quot;mating.latency.2015.m&quot;</span>, <span class="st">&quot;Behavioural&quot;</span>, <span class="st">&quot;Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.&quot;</span>)</span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a>Gaertner_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(orient_to_female_m_line_means, </span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a>                         approach_female_m_line_means, </span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a>                         wing_display_frequency_m_line_means, </span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a>                         genital_licking_m_line_means, </span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a>                         attempted_copulations_m_line_means, </span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a>                         mating_latency_m_line_means)</span></code></pre></div>
</div>
<div id="najarro-et-al-2015" class="section level3">
<h3>Najarro et al 2015</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Line 34061 doesn&#39;t exist on Bloomington or flybase. There is a 34062 though (line 49). Currently removed</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>Najarro_2015_caff_resistance <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Najarro_et_al_2015_caffeine_resistance.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove unidentified line</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(line <span class="sc">!=</span> <span class="dv">34061</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_CaffeineResistance =</span> <span class="fu">my_scale</span>(CaffeineResistance))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a> <span class="co"># mean caffeine resistance model</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>caffeine_resistance_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_CaffeineResistance <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> Najarro_2015_caff_resistance, <span class="at">family =</span> <span class="st">&quot;Gaussian&quot;</span>,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/caffeine.resistance.model&quot;</span>)</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Najarro_2015_caff_resistance<span class="sc">$</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line)</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>fitted_caffeine_resistance <span class="ot">&lt;-</span> <span class="fu">fitted</span>(caffeine_resistance_model,</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance),</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance),</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2015_caff_resistance<span class="sc">$</span>CaffeineResistance))</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>caffeine_resistance_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_caffeine_resistance) <span class="sc">%&gt;%</span> </span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;caffeine.resistance.f&quot;</span>,</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Najarro et al (2015) PLOS Genetics&quot;</span>,</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Drug response&quot;</span>,</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine&quot;</span>,</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="appel-et-al-2016" class="section level3">
<h3>Appel et al 2016</h3>
<p>We need to do some wrangling to fit an appropriate model for these
data. The response is a learning score that is bounded between -100 to
100. It is the percentage of flies that approached a trained target
compared with a control target. Negative values indicate that the flies
preferred the control target; positive values indicate a preference for
the trained target. This data generating process can be approximated
using the beta distribution. However, we need to re-express the data so
that it falls between 0 and 1.</p>
<p>The authors calculate the learning score using the formula <span
class="math inline">\(learning.score =\frac{(Trained.odour -
Control.odour)*100}{Trained.odour + control.odour}\)</span>. A value of
0 represents an even split of flies between both odours. We can convert
this to the proportion scale, where this even split now = 0.5.</p>
<p>We can transform our data using the following formula:</p>
<p><span class="math inline">\(normalised.learning.score =
\frac{learning.score - min(learning.score)}{max(learing.score) -
min(learning.score)}\)</span></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Punishment learning</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>Appel_2016_memory_punishment <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Appel_2016_memory_learning.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Test <span class="sc">==</span> <span class="st">&quot;Punishment&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>         memory_score <span class="sc">!=</span> <span class="st">&quot;-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group_ID =</span> <span class="fu">as.factor</span>(Group_ID),</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">memory_score =</span> <span class="fu">as.numeric</span>(memory_score),</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>         <span class="co"># ok here&#39;s our normalising step. Note that a value of -100 now = 0</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>         <span class="co"># this means that lower values = greater learning i.e. a value of 0.1 means 90% of flies avoided the punishment associated odour</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">norm_mem_score =</span> (memory_score <span class="sc">-</span> <span class="fu">min</span>(memory_score)) <span class="sc">/</span> (<span class="fu">max</span>(memory_score) <span class="sc">-</span> <span class="fu">min</span>(memory_score))</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                           )</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co"># I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>punishment_memory_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(norm_mem_score <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Appel_2016_memory_punishment, <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>(),</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/punishment.memory.model&quot;</span>,</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Appel_2016_memory_punishment<span class="sc">$</span>line,</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>fitted_punishment_memory <span class="ot">&lt;-</span> <span class="fu">fitted</span>(punishment_memory_model,</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>punishment_memory_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_punishment_memory) <span class="sc">%&gt;%</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;punishment.memory.learning.f&quot;</span>,</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with &#39;punishment&#39;. Higher values also indicate a higher learning score&quot;</span>,</span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>punishment_memory_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_punishment_memory) <span class="sc">%&gt;%</span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;punishment.memory.learning.m&quot;</span>,</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with &#39;punishment&#39;. Higher values also indicate a greater learning score.&quot;</span>,</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="do">## Relief learning</span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>Appel_2016_memory_relief <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Appel_2016_memory_learning.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Test <span class="sc">==</span> <span class="st">&quot;Relief&quot;</span>,</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>         memory_score <span class="sc">!=</span> <span class="st">&quot;-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group_ID =</span> <span class="fu">as.factor</span>(Group_ID),</span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>         <span class="at">memory_score =</span> <span class="fu">as.numeric</span>(memory_score),</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>          <span class="co"># ok here&#39;s our normalising step. Note that a value of -100 now = 0</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>         <span class="co"># this means that lower values = greater learning i.e. a value of 0.1 means 90% of flies avoided the punishment associated odour</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>         <span class="at">norm_mem_score =</span> (memory_score <span class="sc">-</span> <span class="fu">min</span>(memory_score)) <span class="sc">/</span> (<span class="fu">max</span>(memory_score) <span class="sc">-</span> <span class="fu">min</span>(memory_score)))</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a><span class="co"># I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>relief_memory_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(norm_mem_score <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>                           <span class="at">data =</span> Appel_2016_memory_relief, <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>(),</span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/relief.memory.model&quot;</span>,</span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>                           <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(relief_memory_model)</span></span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Appel_2016_memory_relief<span class="sc">$</span>line,</span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>fitted_relief_memory <span class="ot">&lt;-</span> <span class="fu">fitted</span>(relief_memory_model,</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>relief_memory_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_relief_memory) <span class="sc">%&gt;%</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;relief.memory.learning.f&quot;</span>,</span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with &#39;relief&#39;. Higher values also indicate a greater learning score.&quot;</span>,</span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a>relief_memory_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_relief_memory) <span class="sc">%&gt;%</span></span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;relief.memory.learning.m&quot;</span>,</span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Appel et al (2016) Biology Letters&quot;</span>,</span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with &#39;relief&#39;. Higher values also indicate a greater learning score.&quot;</span>,</span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a>Appel_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(punishment_memory_f_line_means,</span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a>                      punishment_memory_m_line_means,</span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a>                      relief_memory_f_line_means,</span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a>                      relief_memory_m_line_means)</span></code></pre></div>
</div>
<div id="munoz-munoz-et-al-2016" class="section level3">
<h3>Munoz-Munoz et al 2016</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>Munoz_2016_wing_size <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/munoz_munoz_2016.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Strain),</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Repl =</span> (<span class="fu">as.factor</span>(Repl)),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">CS =</span> <span class="fu">as.double</span>(CS),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_CS =</span> <span class="fu">my_scale</span>(CS))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>wing_size_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_CS <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Munoz_2016_wing_size, <span class="at">family =</span> gaussian,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/wing.size.model&quot;</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Munoz_2016_wing_size<span class="sc">$</span>line,</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>, <span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>fitted_wing_size <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_size_model,</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Munoz_2016_wing_size<span class="sc">$</span>CS) <span class="sc">+</span> <span class="fu">mean</span>(Munoz_2016_wing_size<span class="sc">$</span>CS),</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Munoz_2016_wing_size<span class="sc">$</span>CS) <span class="sc">+</span> <span class="fu">mean</span>(Munoz_2016_wing_size<span class="sc">$</span>CS),</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Munoz_2016_wing_size<span class="sc">$</span>CS) <span class="sc">+</span> <span class="fu">mean</span>(Munoz_2016_wing_size<span class="sc">$</span>CS))</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>wing_size_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_size) <span class="sc">%&gt;%</span></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.size.f&quot;</span>,</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Muñoz-Muñoz et al (2016) Evolution&quot;</span>,</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>wing_size_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_size) <span class="sc">%&gt;%</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.size.m&quot;</span>,</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Muñoz-Muñoz et al (2016) Evolution&quot;</span>,</span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>Munoz_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(wing_size_f_line_means,</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>                      wing_size_m_line_means)</span></code></pre></div>
</div>
<div id="freda-2017" class="section level3">
<h3>Freda, 2017</h3>
<p>We have line mean data for larvae, but raw data for adults.
Therefore, we only need to model the adult data here.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>Freda_2017_cold_hardiness <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2017_ThermalHardiness.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>cold_hardiness_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(Proportion_alive <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">data =</span> Freda_2017_cold_hardiness, <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>(),</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/cold_hardiness.model&quot;</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(cold_hardiness_model)</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Freda_2017_cold_hardiness<span class="sc">$</span>line,</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>fitted_cold_hardiness <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_model,</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>cold_hardiness_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cold_hardiness) <span class="sc">%&gt;%</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.f&quot;</span>,</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2017) Integrative and Comparative Biology&quot;</span>,</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.&quot;</span>,</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>cold_hardiness_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cold_hardiness) <span class="sc">%&gt;%</span></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.m&quot;</span>,</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2017) Integrative and Comparative Biology&quot;</span>,</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.&quot;</span>,</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>Freda_2017_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cold_hardiness_f_line_means,</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>                      cold_hardiness_m_line_means)</span></code></pre></div>
</div>
<div id="najarro-hackett-and-macdonald-2017" class="section level3">
<h3>Najarro, Hackett and Macdonald, 2017</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>Najarro_2017_boric_acid <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/najarro_2017.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">lifespan =</span> <span class="st">`</span><span class="at">life-span (hr) on 1.5% boric acid</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_lifespan =</span> <span class="fu">my_scale</span>(lifespan))</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>boric_acid_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_lifespan <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">data =</span> Najarro_2017_boric_acid, <span class="at">family =</span> <span class="st">&quot;Gaussian&quot;</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/boric.acid.model&quot;</span>)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Najarro_2017_boric_acid<span class="sc">$</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>fitted_boric_acid <span class="ot">&lt;-</span> <span class="fu">fitted</span>(boric_acid_model,</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan),</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan),</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan) <span class="sc">+</span> <span class="fu">mean</span>(Najarro_2017_boric_acid<span class="sc">$</span>lifespan))</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>boric_acid_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_boric_acid) <span class="sc">%&gt;%</span> </span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL&quot;</span>),</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="st">&quot;boric.acid.resistance.f&quot;</span>,</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Najarro, Hackett and Macdonald (2017) G3&quot;</span>,</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Insecticide response&quot;</span>,</span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.&quot;</span>,</span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="dean-et-al-2018" class="section level3">
<h3>Dean et al 2018</h3>
<p>To model copulation latency Dean <em>et al</em> log transform the
data and fit a linear model with a gaussian distribution. This model
struggles to converge using <code>brms</code>, so we treat the data as
time to event with right censoring for the flies that failed to mate
within a 2 hour window (coded as 120 in the dataset). To model this data
we fit a survival model, specifying a weibull distribution.</p>
<p>To model insemination capacity Dean <em>et al</em> specify a
quasipoisson distribution family because they treat the data as
underdispersed counts. There is no family like this in
<code>brms</code>. However, we can treat the data as binomial, with each
female treated as a YES/NO for mating success. The posterior predictive
check suggests this is a good fit to the data. Our predictions are also
a good match with those of Dean et al.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>Dean_2018_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Dean_2018.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">Rep =</span> <span class="fu">as.factor</span>(Rep),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">Block =</span> <span class="fu">as.factor</span>(Block),</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">censor =</span> <span class="fu">if_else</span>(LatCop <span class="sc">==</span> <span class="dv">120</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>copulation_latency_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(LatCop <span class="sc">|</span> <span class="fu">cens</span>(censor) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Dean_2018_data, <span class="at">family =</span> <span class="fu">weibull</span>(),</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">class =</span> b),</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/copulation_latency.model&quot;</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(copulation_latency_model)</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dean_2018_data<span class="sc">$</span>line,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">Block =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Block)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>fitted_cop_lat <span class="ot">&lt;-</span> <span class="fu">fitted</span>(copulation_latency_model,</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>cop_lat_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cop_lat) <span class="sc">%&gt;%</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;copulation.latency.m&quot;</span>,</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Dean et al (2020) Evolution&quot;</span>,</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean time (mins) from a male flies introduction to a female until the onset of mating. Lower values indicate shorter copulation latencies and potentially more attractive males&quot;</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Dean et al fit a quasipoisson model because they treat the data as underdispersed counts. There is no family like this in brms. However, I think we can treat the data as binomial with each female treated as a YES/NO for mating success. The posterior predictive check suggests this is a good fit to the data.</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>Dean_2018_data <span class="ot">&lt;-</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>  Dean_2018_data <span class="sc">%&gt;%</span> </span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Cohabiting_females =</span> <span class="dv">10</span>)</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>insemination_capacity_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(promiscuity <span class="sc">|</span> <span class="fu">trials</span>(Cohabiting_females) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Block <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">data =</span> Dean_2018_data, <span class="at">family =</span> binomial,</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/insemination_capacity.model&quot;</span>,</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.8</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(insemination_capacity_model)</span></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Dean_2018_data<span class="sc">$</span>line,</span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">Cohabiting_females =</span> <span class="dv">1</span>,</span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>         <span class="at">Block =</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Cohabiting_females, Block)</span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>fitted_insem_capac <span class="ot">&lt;-</span> <span class="fu">fitted</span>(insemination_capacity_model,</span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>insem_capac_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_insem_capac) <span class="sc">%&gt;%</span></span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;insemination.capacity.m&quot;</span>,</span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Dean et al (2020) Evolution&quot;</span>,</span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Life history&quot;</span>,</span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of 10 females that a single male successfully inseminated in 48 hours. Insemination was verified by identification of viable pupal progeny. Higher values indicate males with a greater insemination capacity.&quot;</span>,</span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>Dean_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cop_lat_line_means,</span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>                     insem_capac_line_means)</span></code></pre></div>
</div>
<div id="duneau-et-al-2018" class="section level3">
<h3>Duneau et al 2018</h3>
<p>We fit a slightly different parathion model to that presented in the
manuscript. This is because some lines are resistant to parathion,
leading to heavy sensoring. This results in two groups of lines: those
that are susceptible and those that are resistant. The resistant lines
have estimated mean survival times of 400+ days, which is not realistic
for D. melanogaster. Therefore, I instead model resistance to parathion
using a binomial response, which describes whether the line had
surviving flies 48 hours after exposure to parathion.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Deltamethrin</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>Duneau_2018_deltamethrin_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Duneau_2018/Duneau_2018_deltamethrin.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(DGRP_lines, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">Experiment =</span> <span class="fu">as.factor</span>(Experiment),</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">line =</span> <span class="fu">as.factor</span>(line)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample.size <span class="sc">!=</span> <span class="st">&quot;NA&quot;</span>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>deltamethrin_model <span class="ot">&lt;-</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Delta_Alive.at<span class="fl">.48</span>h <span class="sc">|</span> <span class="fu">trials</span>(Sample.size) <span class="sc">~</span> <span class="dv">1</span>  <span class="sc">+</span> Experiment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Duneau_2018_deltamethrin_data,</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">iter =</span> <span class="dv">10000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/deltamethrin.model&quot;</span>,</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(deltamethrin_model)</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Duneau_2018_deltamethrin_data<span class="sc">$</span>line,</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Experiment =</span> <span class="st">&quot;Exp 1&quot;</span>,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sample.size =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Experiment, Sample.size)</span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>fitted_deltamethrin <span class="ot">&lt;-</span> <span class="fu">fitted</span>(deltamethrin_model,</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>deltamethrin_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_deltamethrin) <span class="sc">%&gt;%</span></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;deltamethrin.resistance.m&quot;</span>,</span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Duneau et al (2018) G3&quot;</span>,</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Insecticide response&quot;</span>,</span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.&quot;</span>,</span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Parathion</span></span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a><span class="co"># set up data for right censoring in brms - it is the opposite to what the survival frequentist package uses. We don&#39;t use this but it is useful to know if we do choose to fit a time to event model</span></span>
<span id="cb12-47"><a href="#cb12-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-48"><a href="#cb12-48" aria-hidden="true" tabindex="-1"></a>Duneau_2018_parathion_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Duneau_2018/Duneau_2018_parathion.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-49"><a href="#cb12-49" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(DGRP_lines, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb12-50"><a href="#cb12-50" aria-hidden="true" tabindex="-1"></a>          <span class="at">Experiment =</span> <span class="fu">as.factor</span>(Experiment)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-51"><a href="#cb12-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb12-52"><a href="#cb12-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">Alive.at.48hr =</span> <span class="fu">if_else</span>(Censor <span class="sc">==</span> <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb12-53"><a href="#cb12-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">Censor =</span> <span class="fu">if_else</span>(Censor <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;none&quot;</span>, <span class="st">&quot;right&quot;</span>)) </span>
<span id="cb12-54"><a href="#cb12-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-55"><a href="#cb12-55" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the new binomial model</span></span>
<span id="cb12-56"><a href="#cb12-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-57"><a href="#cb12-57" aria-hidden="true" tabindex="-1"></a>parathion_binomial_model <span class="ot">&lt;-</span></span>
<span id="cb12-58"><a href="#cb12-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Alive.at<span class="fl">.48</span>hr <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Experiment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb12-59"><a href="#cb12-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Duneau_2018_parathion_data,</span>
<span id="cb12-60"><a href="#cb12-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb12-61"><a href="#cb12-61" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb12-62"><a href="#cb12-62" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb12-63"><a href="#cb12-63" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> bernoulli,</span>
<span id="cb12-64"><a href="#cb12-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb12-65"><a href="#cb12-65" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb12-66"><a href="#cb12-66" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/parathion_binomial.model&quot;</span>)</span>
<span id="cb12-67"><a href="#cb12-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-68"><a href="#cb12-68" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(parathion_binomial_model) </span></span>
<span id="cb12-69"><a href="#cb12-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-70"><a href="#cb12-70" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb12-71"><a href="#cb12-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Duneau_2018_parathion_data<span class="sc">$</span>line,</span>
<span id="cb12-72"><a href="#cb12-72" aria-hidden="true" tabindex="-1"></a>         <span class="at">Experiment =</span> <span class="st">&quot;Experiment_1&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-73"><a href="#cb12-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Experiment)</span>
<span id="cb12-74"><a href="#cb12-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-75"><a href="#cb12-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-76"><a href="#cb12-76" aria-hidden="true" tabindex="-1"></a>fitted_parathion_binomial <span class="ot">&lt;-</span> <span class="fu">fitted</span>(parathion_binomial_model,</span>
<span id="cb12-77"><a href="#cb12-77" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb12-78"><a href="#cb12-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb12-79"><a href="#cb12-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-80"><a href="#cb12-80" aria-hidden="true" tabindex="-1"></a>parathion_binomial_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_parathion_binomial) <span class="sc">%&gt;%</span></span>
<span id="cb12-81"><a href="#cb12-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;parathion.resistance.m&quot;</span>,</span>
<span id="cb12-82"><a href="#cb12-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb12-83"><a href="#cb12-83" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Duneau et al (2018) G3&quot;</span>,</span>
<span id="cb12-84"><a href="#cb12-84" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Insecticide response&quot;</span>,</span>
<span id="cb12-85"><a href="#cb12-85" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.&quot;</span>,</span>
<span id="cb12-86"><a href="#cb12-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb12-87"><a href="#cb12-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb12-88"><a href="#cb12-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb12-89"><a href="#cb12-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-90"><a href="#cb12-90" aria-hidden="true" tabindex="-1"></a>Duneau_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(deltamethrin_line_means,</span>
<span id="cb12-91"><a href="#cb12-91" aria-hidden="true" tabindex="-1"></a>                       parathion_binomial_line_means)</span></code></pre></div>
</div>
<div id="palmer-et-al-2018" class="section level3">
<h3>Palmer et al 2018</h3>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>Palmer_2018_KV_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Palmer_2018_kallithea.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Injection.date) <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">injection_date =</span> <span class="fu">as_factor</span>(<span class="fu">cur_group_id</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">value =</span> <span class="fu">as.numeric</span>(value))</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>Palmer_2018_LT50 <span class="ot">&lt;-</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  Palmer_2018_KV_data <span class="sc">%&gt;%</span> </span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait.column <span class="sc">==</span> <span class="st">&quot;Mortality&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_value =</span> <span class="fu">my_scale</span>(value))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># LT50 model</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>kv_LT50_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>injection_date),</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Palmer_2018_LT50, <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">8000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/kv.LT50.model&quot;</span>,</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.97</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(kv_LT50_model)</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Palmer_2018_LT50<span class="sc">$</span>line,</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>fitted_kv_LT50 <span class="ot">&lt;-</span> <span class="fu">fitted</span>(kv_LT50_model,</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> value <span class="sc">~</span> (Sex<span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_LT50<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_LT50<span class="sc">$</span>value),</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_LT50<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_LT50<span class="sc">$</span>value),</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_LT50<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_LT50<span class="sc">$</span>value))</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>kv_LT50_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_kv_LT50) <span class="sc">%&gt;%</span></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;kallithea.resistance.LT50.f&quot;</span>,</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Palmer et al (2018) PLOS Pathogens&quot;</span>,</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.&quot;</span>,</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>kv_LT50_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_kv_LT50) <span class="sc">%&gt;%</span></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;kallithea.resistance.LT50.m&quot;</span>,</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Palmer et al (2018) PLOS Pathogens&quot;</span>,</span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.&quot;</span>,</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a><span class="co"># titre model</span></span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>Palmer_2018_KV_titre_data <span class="ot">&lt;-</span></span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a>  Palmer_2018_KV_data <span class="sc">%&gt;%</span> </span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait.column <span class="sc">==</span> <span class="st">&quot;titre&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_value =</span> <span class="fu">my_scale</span>(value))</span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>kv_titre_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>injection_date),</span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Palmer_2018_KV_titre_data, <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/kv.titre.model&quot;</span>,</span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Palmer_2018_KV_titre_data<span class="sc">$</span>line,</span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a>fitted_kv_titre <span class="ot">&lt;-</span> <span class="fu">fitted</span>(kv_titre_model,</span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> value <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value),</span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value),</span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value) <span class="sc">+</span> <span class="fu">mean</span>(Palmer_2018_KV_titre_data<span class="sc">$</span>value))</span>
<span id="cb13-99"><a href="#cb13-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-100"><a href="#cb13-100" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-102"><a href="#cb13-102" aria-hidden="true" tabindex="-1"></a>kv_titre_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_kv_titre) <span class="sc">%&gt;%</span></span>
<span id="cb13-103"><a href="#cb13-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;kallithea.viral.load.f&quot;</span>,</span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Palmer et al (2018) PLOS Pathogens&quot;</span>,</span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.&quot;</span>,</span>
<span id="cb13-109"><a href="#cb13-109" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb13-110"><a href="#cb13-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb13-111"><a href="#cb13-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a>Palmer_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(kv_LT50_f_line_means,</span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a>                       kv_LT50_m_line_means,</span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a>                       kv_titre_f_line_means)</span></code></pre></div>
</div>
<div id="teets-and-hahn-2018" class="section level3">
<h3>Teets and Hahn 2018</h3>
<p>Cumulative cold tolerance data was supplied by the authors in line
mean form, so we do not model it here.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>Teets_2018_cold_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Teets_Hahn_2018_data.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(<span class="st">`</span><span class="at">Line (DGRP)</span><span class="st">`</span>),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temp =</span> <span class="fu">as.factor</span>(Temp))</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>cold_survival_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(Live <span class="sc">|</span> <span class="fu">trials</span>(Total) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Temp <span class="sc">+</span> (Temp<span class="sc">|</span>line),</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Teets_2018_cold_data, <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/cold.survival.model&quot;</span>,</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(cold_survival_model)</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Teets_2018_cold_data<span class="sc">$</span>line,</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">Temp =</span> Teets_2018_cold_data<span class="sc">$</span>Temp,</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">Total =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Temp, Total)</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>fitted_cold_survival <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_survival_model,</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>cold_survival_line_means_data <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_cold_survival) <span class="sc">%&gt;%</span> </span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>)</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create tibbles holding line and temp specific line means</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>Teets_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, value, Trait, description){</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temp <span class="sc">==</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Teets and Hahn (2018) Journal of Evolutionary Biology&quot;</span>,</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span><span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>neg1_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;cold.tolerance.-1.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -1 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>neg2_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-2&quot;</span>, <span class="st">&quot;cold.tolerance.-2.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -2 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>neg3_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-3&quot;</span>, <span class="st">&quot;cold.tolerance.-3.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -3 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>neg4_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-4&quot;</span>, <span class="st">&quot;cold.tolerance.-4.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -4 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>neg5_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-5&quot;</span>, <span class="st">&quot;cold.tolerance.-5.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -5 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>neg6_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-6&quot;</span>, <span class="st">&quot;cold.tolerance.-6.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -6 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a>neg7_cold_tolerance_line_means <span class="ot">&lt;-</span> <span class="fu">Teets_line_means</span>(cold_survival_line_means_data, <span class="at">value =</span> <span class="st">&quot;-7&quot;</span>, <span class="st">&quot;cold.tolerance.-7.f&quot;</span>, <span class="st">&quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -7 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;</span>)</span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>Teets_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(neg1_cold_tolerance_line_means,</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>                      neg2_cold_tolerance_line_means,</span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>                      neg3_cold_tolerance_line_means,</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>                      neg4_cold_tolerance_line_means,</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>                      neg5_cold_tolerance_line_means,</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a>                      neg6_cold_tolerance_line_means,</span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a>                      neg7_cold_tolerance_line_means)</span></code></pre></div>
</div>
<div id="everman-et-al-2019" class="section level3">
<h3>Everman et al 2019</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>Everman_2019_starvation_resistance <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Everman_2019_starvation_res.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex),</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Lifespan_hrs =</span> <span class="fu">as.numeric</span>(Lifespan_hrs),</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_Lifespan_hrs =</span> <span class="fu">my_scale</span>(Lifespan_hrs))</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>starvation_resistance_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_Lifespan_hrs <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Everman_2019_starvation_resistance, <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/starvation.res.model&quot;</span>,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(starvation_resistance_model)</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_grid</span>(<span class="at">line =</span> Everman_2019_starvation_resistance<span class="sc">$</span>line,</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">Sex =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>fitted_starvation_resistance <span class="ot">&lt;-</span> <span class="fu">fitted</span>(starvation_resistance_model,</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs) <span class="sc">+</span> <span class="fu">mean</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs),</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs) <span class="sc">+</span> <span class="fu">mean</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs),</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs) <span class="sc">+</span> <span class="fu">mean</span>(Everman_2019_starvation_resistance<span class="sc">$</span>Lifespan_hrs))</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>starvation_resistance_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_starvation_resistance) <span class="sc">%&gt;%</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;starvation.resistance.2019.f&quot;</span>,</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Everman et al (2019) Genetics&quot;</span>,</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Stress response&quot;</span>,</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.&quot;</span>,</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>starvation_resistance_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_starvation_resistance) <span class="sc">%&gt;%</span></span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;starvation.resistance.2019.m&quot;</span>,</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Everman et al (2019) Genetics&quot;</span>,</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Stress response&quot;</span>,</span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.&quot;</span>,</span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>Everman_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(starvation_resistance_f_line_means,</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>                        starvation_resistance_m_line_means)</span></code></pre></div>
</div>
<div id="freda-2019" class="section level3">
<h3>Freda 2019</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>Freda_2019a <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Cold18.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline))</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>Freda_2019b <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Cold25.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>Freda_2019c <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Heat18.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline))</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>Freda_2019d <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Freda_2019/Heat25.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(dgrpline))</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 18C for larvae model</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_18_l.model&quot;</span>,</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_18_larvae_model)</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>Freda_2019a_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>  Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019a_larvae_data<span class="sc">$</span>line,</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>fitted_ch_18_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_18_larvae_model,</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a><span class="co"># larvae</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_18_l) <span class="sc">%&gt;%</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.18C.larvae&quot;</span>,</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 18C for females model</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_18_f.model&quot;</span>,</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_18_f_model)</span></span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>Freda_2019a_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>  Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019a_larvae_data<span class="sc">$</span>line,</span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a>fitted_ch_18_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_18_f_model,</span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a><span class="co"># female</span></span>
<span id="cb16-76"><a href="#cb16-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-77"><a href="#cb16-77" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_18_f) <span class="sc">%&gt;%</span></span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.18C.f&quot;</span>,</span>
<span id="cb16-79"><a href="#cb16-79" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb16-80"><a href="#cb16-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-81"><a href="#cb16-81" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-82"><a href="#cb16-82" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb16-83"><a href="#cb16-83" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-84"><a href="#cb16-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-85"><a href="#cb16-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-86"><a href="#cb16-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-87"><a href="#cb16-87" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 18C for males model</span></span>
<span id="cb16-88"><a href="#cb16-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-89"><a href="#cb16-89" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-90"><a href="#cb16-90" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-91"><a href="#cb16-91" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-92"><a href="#cb16-92" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-93"><a href="#cb16-93" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-94"><a href="#cb16-94" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_18_m.model&quot;</span>,</span>
<span id="cb16-95"><a href="#cb16-95" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-96"><a href="#cb16-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-97"><a href="#cb16-97" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_18_m_model)</span></span>
<span id="cb16-98"><a href="#cb16-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-99"><a href="#cb16-99" aria-hidden="true" tabindex="-1"></a>Freda_2019a_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb16-100"><a href="#cb16-100" aria-hidden="true" tabindex="-1"></a>  Freda_2019a <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-101"><a href="#cb16-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-102"><a href="#cb16-102" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-103"><a href="#cb16-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019a_larvae_data<span class="sc">$</span>line,</span>
<span id="cb16-104"><a href="#cb16-104" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-105"><a href="#cb16-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb16-106"><a href="#cb16-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-107"><a href="#cb16-107" aria-hidden="true" tabindex="-1"></a>fitted_ch_18_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_18_m_model,</span>
<span id="cb16-108"><a href="#cb16-108" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-109"><a href="#cb16-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-110"><a href="#cb16-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-111"><a href="#cb16-111" aria-hidden="true" tabindex="-1"></a>cold_hardiness_18_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_18_m) <span class="sc">%&gt;%</span></span>
<span id="cb16-112"><a href="#cb16-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.18C.m&quot;</span>,</span>
<span id="cb16-113"><a href="#cb16-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb16-114"><a href="#cb16-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-115"><a href="#cb16-115" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-116"><a href="#cb16-116" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb16-117"><a href="#cb16-117" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-118"><a href="#cb16-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-119"><a href="#cb16-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-120"><a href="#cb16-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-121"><a href="#cb16-121" aria-hidden="true" tabindex="-1"></a><span class="do">#################</span></span>
<span id="cb16-122"><a href="#cb16-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-123"><a href="#cb16-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 25C for larvae model</span></span>
<span id="cb16-124"><a href="#cb16-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-125"><a href="#cb16-125" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-126"><a href="#cb16-126" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-127"><a href="#cb16-127" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-128"><a href="#cb16-128" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-129"><a href="#cb16-129" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-130"><a href="#cb16-130" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_25_l.model&quot;</span>,</span>
<span id="cb16-131"><a href="#cb16-131" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-132"><a href="#cb16-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-133"><a href="#cb16-133" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_25_larvae_model)</span></span>
<span id="cb16-134"><a href="#cb16-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-135"><a href="#cb16-135" aria-hidden="true" tabindex="-1"></a>Freda_2019b_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb16-136"><a href="#cb16-136" aria-hidden="true" tabindex="-1"></a>  Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb16-137"><a href="#cb16-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-138"><a href="#cb16-138" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-139"><a href="#cb16-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019b_larvae_data<span class="sc">$</span>line,</span>
<span id="cb16-140"><a href="#cb16-140" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-141"><a href="#cb16-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb16-142"><a href="#cb16-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-143"><a href="#cb16-143" aria-hidden="true" tabindex="-1"></a>fitted_ch_25_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_25_larvae_model,</span>
<span id="cb16-144"><a href="#cb16-144" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-145"><a href="#cb16-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-146"><a href="#cb16-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-147"><a href="#cb16-147" aria-hidden="true" tabindex="-1"></a><span class="co"># larvae</span></span>
<span id="cb16-148"><a href="#cb16-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-149"><a href="#cb16-149" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_l) <span class="sc">%&gt;%</span></span>
<span id="cb16-150"><a href="#cb16-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.25C.larvae&quot;</span>,</span>
<span id="cb16-151"><a href="#cb16-151" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb16-152"><a href="#cb16-152" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-153"><a href="#cb16-153" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-154"><a href="#cb16-154" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb16-155"><a href="#cb16-155" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-156"><a href="#cb16-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-157"><a href="#cb16-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-158"><a href="#cb16-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-159"><a href="#cb16-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-160"><a href="#cb16-160" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 25C for females model</span></span>
<span id="cb16-161"><a href="#cb16-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-162"><a href="#cb16-162" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-163"><a href="#cb16-163" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-164"><a href="#cb16-164" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-165"><a href="#cb16-165" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-166"><a href="#cb16-166" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-167"><a href="#cb16-167" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_25_f.model&quot;</span>,</span>
<span id="cb16-168"><a href="#cb16-168" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-169"><a href="#cb16-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-170"><a href="#cb16-170" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_25_f_model)</span></span>
<span id="cb16-171"><a href="#cb16-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-172"><a href="#cb16-172" aria-hidden="true" tabindex="-1"></a>Freda_2019b_f_data <span class="ot">&lt;-</span></span>
<span id="cb16-173"><a href="#cb16-173" aria-hidden="true" tabindex="-1"></a>  Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-174"><a href="#cb16-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-175"><a href="#cb16-175" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-176"><a href="#cb16-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019b_f_data<span class="sc">$</span>line,</span>
<span id="cb16-177"><a href="#cb16-177" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-178"><a href="#cb16-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb16-179"><a href="#cb16-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-180"><a href="#cb16-180" aria-hidden="true" tabindex="-1"></a>fitted_ch_25_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_25_f_model,</span>
<span id="cb16-181"><a href="#cb16-181" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-182"><a href="#cb16-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-183"><a href="#cb16-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-184"><a href="#cb16-184" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_f) <span class="sc">%&gt;%</span></span>
<span id="cb16-185"><a href="#cb16-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.25C.f&quot;</span>,</span>
<span id="cb16-186"><a href="#cb16-186" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb16-187"><a href="#cb16-187" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-188"><a href="#cb16-188" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-189"><a href="#cb16-189" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb16-190"><a href="#cb16-190" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-191"><a href="#cb16-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-192"><a href="#cb16-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-193"><a href="#cb16-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-194"><a href="#cb16-194" aria-hidden="true" tabindex="-1"></a><span class="co"># Cold hardiness at 25C for males model</span></span>
<span id="cb16-195"><a href="#cb16-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-196"><a href="#cb16-196" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-197"><a href="#cb16-197" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-198"><a href="#cb16-198" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-199"><a href="#cb16-199" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-200"><a href="#cb16-200" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-201"><a href="#cb16-201" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/ch_25_m.model&quot;</span>,</span>
<span id="cb16-202"><a href="#cb16-202" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-203"><a href="#cb16-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-204"><a href="#cb16-204" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(cold_hardiness_25_m_model)</span></span>
<span id="cb16-205"><a href="#cb16-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-206"><a href="#cb16-206" aria-hidden="true" tabindex="-1"></a>Freda_2019b_m_data <span class="ot">&lt;-</span></span>
<span id="cb16-207"><a href="#cb16-207" aria-hidden="true" tabindex="-1"></a>  Freda_2019b <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-208"><a href="#cb16-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-209"><a href="#cb16-209" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-210"><a href="#cb16-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019b_m_data<span class="sc">$</span>line,</span>
<span id="cb16-211"><a href="#cb16-211" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-212"><a href="#cb16-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb16-213"><a href="#cb16-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-214"><a href="#cb16-214" aria-hidden="true" tabindex="-1"></a>fitted_ch_25_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(cold_hardiness_25_m_model,</span>
<span id="cb16-215"><a href="#cb16-215" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-216"><a href="#cb16-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-217"><a href="#cb16-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-218"><a href="#cb16-218" aria-hidden="true" tabindex="-1"></a>cold_hardiness_25_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_m) <span class="sc">%&gt;%</span></span>
<span id="cb16-219"><a href="#cb16-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;cold.hardiness.25C.m&quot;</span>,</span>
<span id="cb16-220"><a href="#cb16-220" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb16-221"><a href="#cb16-221" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-222"><a href="#cb16-222" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-223"><a href="#cb16-223" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.&quot;</span>,</span>
<span id="cb16-224"><a href="#cb16-224" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-225"><a href="#cb16-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-226"><a href="#cb16-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-227"><a href="#cb16-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-228"><a href="#cb16-228" aria-hidden="true" tabindex="-1"></a><span class="do">##############</span></span>
<span id="cb16-229"><a href="#cb16-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-230"><a href="#cb16-230" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 18C for larvae model</span></span>
<span id="cb16-231"><a href="#cb16-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-232"><a href="#cb16-232" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-233"><a href="#cb16-233" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-234"><a href="#cb16-234" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-235"><a href="#cb16-235" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-236"><a href="#cb16-236" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-237"><a href="#cb16-237" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_18_l.model&quot;</span>,</span>
<span id="cb16-238"><a href="#cb16-238" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-239"><a href="#cb16-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-240"><a href="#cb16-240" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_18_larvae_model)</span></span>
<span id="cb16-241"><a href="#cb16-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-242"><a href="#cb16-242" aria-hidden="true" tabindex="-1"></a>Freda_2019c_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb16-243"><a href="#cb16-243" aria-hidden="true" tabindex="-1"></a>  Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb16-244"><a href="#cb16-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-245"><a href="#cb16-245" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-246"><a href="#cb16-246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019c_larvae_data<span class="sc">$</span>line,</span>
<span id="cb16-247"><a href="#cb16-247" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-248"><a href="#cb16-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb16-249"><a href="#cb16-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-250"><a href="#cb16-250" aria-hidden="true" tabindex="-1"></a>fitted_hh_18_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_18_larvae_model,</span>
<span id="cb16-251"><a href="#cb16-251" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-252"><a href="#cb16-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-253"><a href="#cb16-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-254"><a href="#cb16-254" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_18_l) <span class="sc">%&gt;%</span></span>
<span id="cb16-255"><a href="#cb16-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.18C.larvae&quot;</span>,</span>
<span id="cb16-256"><a href="#cb16-256" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb16-257"><a href="#cb16-257" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-258"><a href="#cb16-258" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-259"><a href="#cb16-259" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb16-260"><a href="#cb16-260" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-261"><a href="#cb16-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-262"><a href="#cb16-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-263"><a href="#cb16-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-264"><a href="#cb16-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-265"><a href="#cb16-265" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 18C for females model</span></span>
<span id="cb16-266"><a href="#cb16-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-267"><a href="#cb16-267" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-268"><a href="#cb16-268" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-269"><a href="#cb16-269" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-270"><a href="#cb16-270" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-271"><a href="#cb16-271" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-272"><a href="#cb16-272" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_18_f.model&quot;</span>,</span>
<span id="cb16-273"><a href="#cb16-273" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-274"><a href="#cb16-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-275"><a href="#cb16-275" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_18_f_model)</span></span>
<span id="cb16-276"><a href="#cb16-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-277"><a href="#cb16-277" aria-hidden="true" tabindex="-1"></a>Freda_2019c_f_data <span class="ot">&lt;-</span></span>
<span id="cb16-278"><a href="#cb16-278" aria-hidden="true" tabindex="-1"></a>  Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-279"><a href="#cb16-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-280"><a href="#cb16-280" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-281"><a href="#cb16-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019c_f_data<span class="sc">$</span>line,</span>
<span id="cb16-282"><a href="#cb16-282" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-283"><a href="#cb16-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb16-284"><a href="#cb16-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-285"><a href="#cb16-285" aria-hidden="true" tabindex="-1"></a>fitted_hh_18_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_18_f_model,</span>
<span id="cb16-286"><a href="#cb16-286" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-287"><a href="#cb16-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-288"><a href="#cb16-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-289"><a href="#cb16-289" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_18_f) <span class="sc">%&gt;%</span></span>
<span id="cb16-290"><a href="#cb16-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.18C.f&quot;</span>,</span>
<span id="cb16-291"><a href="#cb16-291" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb16-292"><a href="#cb16-292" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-293"><a href="#cb16-293" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-294"><a href="#cb16-294" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb16-295"><a href="#cb16-295" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-296"><a href="#cb16-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-297"><a href="#cb16-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-298"><a href="#cb16-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-299"><a href="#cb16-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 18C for males model</span></span>
<span id="cb16-300"><a href="#cb16-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-301"><a href="#cb16-301" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-302"><a href="#cb16-302" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-303"><a href="#cb16-303" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-304"><a href="#cb16-304" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-305"><a href="#cb16-305" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-306"><a href="#cb16-306" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_18_m.model&quot;</span>,</span>
<span id="cb16-307"><a href="#cb16-307" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-308"><a href="#cb16-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-309"><a href="#cb16-309" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_18_m_model)</span></span>
<span id="cb16-310"><a href="#cb16-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-311"><a href="#cb16-311" aria-hidden="true" tabindex="-1"></a>Freda_2019c_m_data <span class="ot">&lt;-</span></span>
<span id="cb16-312"><a href="#cb16-312" aria-hidden="true" tabindex="-1"></a>  Freda_2019c <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-313"><a href="#cb16-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-314"><a href="#cb16-314" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-315"><a href="#cb16-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019c_m_data<span class="sc">$</span>line,</span>
<span id="cb16-316"><a href="#cb16-316" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-317"><a href="#cb16-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb16-318"><a href="#cb16-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-319"><a href="#cb16-319" aria-hidden="true" tabindex="-1"></a>fitted_hh_18_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_18_m_model,</span>
<span id="cb16-320"><a href="#cb16-320" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-321"><a href="#cb16-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-322"><a href="#cb16-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-323"><a href="#cb16-323" aria-hidden="true" tabindex="-1"></a>heat_hardiness_18_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_18_m) <span class="sc">%&gt;%</span></span>
<span id="cb16-324"><a href="#cb16-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.18C.m&quot;</span>,</span>
<span id="cb16-325"><a href="#cb16-325" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb16-326"><a href="#cb16-326" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-327"><a href="#cb16-327" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-328"><a href="#cb16-328" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb16-329"><a href="#cb16-329" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-330"><a href="#cb16-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-331"><a href="#cb16-331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-332"><a href="#cb16-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-333"><a href="#cb16-333" aria-hidden="true" tabindex="-1"></a><span class="do">#################</span></span>
<span id="cb16-334"><a href="#cb16-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-335"><a href="#cb16-335" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 25C for larvae model</span></span>
<span id="cb16-336"><a href="#cb16-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-337"><a href="#cb16-337" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_larvae_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(alive <span class="sc">|</span> <span class="fu">trials</span>(n) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-338"><a href="#cb16-338" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-339"><a href="#cb16-339" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-340"><a href="#cb16-340" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-341"><a href="#cb16-341" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-342"><a href="#cb16-342" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_25_l.model&quot;</span>,</span>
<span id="cb16-343"><a href="#cb16-343" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-344"><a href="#cb16-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-345"><a href="#cb16-345" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_25_larvae_model)</span></span>
<span id="cb16-346"><a href="#cb16-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-347"><a href="#cb16-347" aria-hidden="true" tabindex="-1"></a>Freda_2019d_larvae_data <span class="ot">&lt;-</span></span>
<span id="cb16-348"><a href="#cb16-348" aria-hidden="true" tabindex="-1"></a>  Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb16-349"><a href="#cb16-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-350"><a href="#cb16-350" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-351"><a href="#cb16-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019d_larvae_data<span class="sc">$</span>line,</span>
<span id="cb16-352"><a href="#cb16-352" aria-hidden="true" tabindex="-1"></a>         <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-353"><a href="#cb16-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, n)</span>
<span id="cb16-354"><a href="#cb16-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-355"><a href="#cb16-355" aria-hidden="true" tabindex="-1"></a>fitted_hh_25_l <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_25_larvae_model,</span>
<span id="cb16-356"><a href="#cb16-356" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-357"><a href="#cb16-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-358"><a href="#cb16-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-359"><a href="#cb16-359" aria-hidden="true" tabindex="-1"></a><span class="co"># females</span></span>
<span id="cb16-360"><a href="#cb16-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-361"><a href="#cb16-361" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_l_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_ch_25_l) <span class="sc">%&gt;%</span></span>
<span id="cb16-362"><a href="#cb16-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.25C.larvae&quot;</span>,</span>
<span id="cb16-363"><a href="#cb16-363" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb16-364"><a href="#cb16-364" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-365"><a href="#cb16-365" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-366"><a href="#cb16-366" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb16-367"><a href="#cb16-367" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-368"><a href="#cb16-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-369"><a href="#cb16-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-370"><a href="#cb16-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-371"><a href="#cb16-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-372"><a href="#cb16-372" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 25C for females model</span></span>
<span id="cb16-373"><a href="#cb16-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-374"><a href="#cb16-374" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_f_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_female <span class="sc">|</span> <span class="fu">trials</span>(s_female) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-375"><a href="#cb16-375" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-376"><a href="#cb16-376" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-377"><a href="#cb16-377" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-378"><a href="#cb16-378" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-379"><a href="#cb16-379" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_25_f.model&quot;</span>,</span>
<span id="cb16-380"><a href="#cb16-380" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-381"><a href="#cb16-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-382"><a href="#cb16-382" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_25_f_model)</span></span>
<span id="cb16-383"><a href="#cb16-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-384"><a href="#cb16-384" aria-hidden="true" tabindex="-1"></a>Freda_2019d_f_data <span class="ot">&lt;-</span></span>
<span id="cb16-385"><a href="#cb16-385" aria-hidden="true" tabindex="-1"></a>  Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-386"><a href="#cb16-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-387"><a href="#cb16-387" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-388"><a href="#cb16-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019d_f_data<span class="sc">$</span>line,</span>
<span id="cb16-389"><a href="#cb16-389" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_female =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-390"><a href="#cb16-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_female)</span>
<span id="cb16-391"><a href="#cb16-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-392"><a href="#cb16-392" aria-hidden="true" tabindex="-1"></a>fitted_hh_25_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_25_f_model,</span>
<span id="cb16-393"><a href="#cb16-393" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-394"><a href="#cb16-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-395"><a href="#cb16-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-396"><a href="#cb16-396" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_25_f) <span class="sc">%&gt;%</span></span>
<span id="cb16-397"><a href="#cb16-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.25C.f&quot;</span>,</span>
<span id="cb16-398"><a href="#cb16-398" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb16-399"><a href="#cb16-399" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-400"><a href="#cb16-400" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-401"><a href="#cb16-401" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb16-402"><a href="#cb16-402" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-403"><a href="#cb16-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-404"><a href="#cb16-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-405"><a href="#cb16-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-406"><a href="#cb16-406" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat hardiness at 25C for males model</span></span>
<span id="cb16-407"><a href="#cb16-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-408"><a href="#cb16-408" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_m_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(f_male <span class="sc">|</span> <span class="fu">trials</span>(s_male) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb16-409"><a href="#cb16-409" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>), <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb16-410"><a href="#cb16-410" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb16-411"><a href="#cb16-411" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb16-412"><a href="#cb16-412" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb16-413"><a href="#cb16-413" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/hh_25_m.model&quot;</span>,</span>
<span id="cb16-414"><a href="#cb16-414" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb16-415"><a href="#cb16-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-416"><a href="#cb16-416" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(heat_hardiness_25_m_model)</span></span>
<span id="cb16-417"><a href="#cb16-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-418"><a href="#cb16-418" aria-hidden="true" tabindex="-1"></a>Freda_2019d_m_data <span class="ot">&lt;-</span></span>
<span id="cb16-419"><a href="#cb16-419" aria-hidden="true" tabindex="-1"></a>  Freda_2019d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;A&quot;</span>)</span>
<span id="cb16-420"><a href="#cb16-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-421"><a href="#cb16-421" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb16-422"><a href="#cb16-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Freda_2019d_m_data<span class="sc">$</span>line,</span>
<span id="cb16-423"><a href="#cb16-423" aria-hidden="true" tabindex="-1"></a>         <span class="at">s_male =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-424"><a href="#cb16-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, s_male)</span>
<span id="cb16-425"><a href="#cb16-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-426"><a href="#cb16-426" aria-hidden="true" tabindex="-1"></a>fitted_hh_25_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(heat_hardiness_25_m_model,</span>
<span id="cb16-427"><a href="#cb16-427" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb16-428"><a href="#cb16-428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb16-429"><a href="#cb16-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-430"><a href="#cb16-430" aria-hidden="true" tabindex="-1"></a>heat_hardiness_25_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_hh_25_m) <span class="sc">%&gt;%</span></span>
<span id="cb16-431"><a href="#cb16-431" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;heat.hardiness.25C.m&quot;</span>,</span>
<span id="cb16-432"><a href="#cb16-432" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb16-433"><a href="#cb16-433" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Freda et al (2019) Heredity&quot;</span>,</span>
<span id="cb16-434"><a href="#cb16-434" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb16-435"><a href="#cb16-435" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.&quot;</span>,</span>
<span id="cb16-436"><a href="#cb16-436" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb16-437"><a href="#cb16-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb16-438"><a href="#cb16-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb16-439"><a href="#cb16-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-440"><a href="#cb16-440" aria-hidden="true" tabindex="-1"></a>Freda_2019_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cold_hardiness_18_l_line_means,</span>
<span id="cb16-441"><a href="#cb16-441" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_18_f_line_means,</span>
<span id="cb16-442"><a href="#cb16-442" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_18_m_line_means,</span>
<span id="cb16-443"><a href="#cb16-443" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_25_l_line_means,</span>
<span id="cb16-444"><a href="#cb16-444" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_25_f_line_means,</span>
<span id="cb16-445"><a href="#cb16-445" aria-hidden="true" tabindex="-1"></a>                           cold_hardiness_25_m_line_means,</span>
<span id="cb16-446"><a href="#cb16-446" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_18_l_line_means,</span>
<span id="cb16-447"><a href="#cb16-447" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_18_f_line_means,</span>
<span id="cb16-448"><a href="#cb16-448" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_18_m_line_means,</span>
<span id="cb16-449"><a href="#cb16-449" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_25_l_line_means,</span>
<span id="cb16-450"><a href="#cb16-450" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_25_f_line_means,</span>
<span id="cb16-451"><a href="#cb16-451" aria-hidden="true" tabindex="-1"></a>                           heat_hardiness_25_m_line_means)</span></code></pre></div>
</div>
<div id="pitchers-et-al-2019" class="section level3">
<h3>Pitchers et al 2019</h3>
<p>This study measures wing shape and area. They then run a multivariate
GWA. Here, we just model the mean centroid size for each line.</p>
<p>For whatever reason, this is a hard trait to model.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>Pitchers_2019_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Pitchers_2019_wing_multivariate.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Linenum),</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Lab =</span> <span class="fu">as.factor</span>(Lab),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Perp =</span> <span class="fu">as.factor</span>(Perp),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">block =</span> <span class="fu">as.factor</span>(block),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex))</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>Houle_data <span class="ot">&lt;-</span> Pitchers_2019_data <span class="sc">%&gt;%</span> </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Lab <span class="sc">==</span> <span class="st">&quot;Hou&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardise</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_csmm =</span> <span class="fu">my_scale</span>(csmm))</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>Dworkin_data <span class="ot">&lt;-</span> Pitchers_2019_data <span class="sc">%&gt;%</span> </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Lab <span class="sc">==</span> <span class="st">&quot;Dwo&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardise</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_csmm =</span> <span class="fu">my_scale</span>(csmm))</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Houle lab model</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>wing_centroid_size_model_Hou <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_csmm <span class="sc">~</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> gaussian, <span class="at">data =</span> Houle_data,</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/Houle_cs_wing.model&quot;</span>,</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(wing_centroid_size_model_Hou)</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Houle_data<span class="sc">$</span>line,</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> Houle_data<span class="sc">$</span>Sex) <span class="sc">%&gt;%</span> </span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>fitted_wing_houle <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_centroid_size_model_Hou,</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Houle_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Houle_data<span class="sc">$</span>csmm),</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Houle_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Houle_data<span class="sc">$</span>csmm),</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Houle_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Houle_data<span class="sc">$</span>csmm))</span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>centroid_size_Houle_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_houle) <span class="sc">%&gt;%</span></span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Houle.f&quot;</span>,</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the &#39;Wingmachine&#39; system. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>centroid_size_Houle_m_line_means <span class="ot">&lt;-</span> </span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(new_data, fitted_wing_houle) <span class="sc">%&gt;%</span></span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Houle.m&quot;</span>,</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the &#39;Wingmachine&#39; system. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Dworkin lab model</span></span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>wing_centroid_size_model_Dwo_f <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_csmm <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> student, <span class="at">data =</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>),</span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#prior(normal(0, 0.5), class = b),</span></span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">20000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/Dwo_cs_wing.f.model&quot;</span>,</span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(wing_centroid_size_model_Dwo_f)</span></span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a>dworkin_f <span class="ot">&lt;-</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>)</span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> dworkin_f<span class="sc">$</span>line,</span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;F&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a>fitted_wing_dworkin_f <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_centroid_size_model_Dwo_f,</span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm))</span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a>centroid_size_Dworkin_f_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_dworkin_f) <span class="sc">%&gt;%</span></span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Dworkin.f&quot;</span>,</span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb17-103"><a href="#cb17-103" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-104"><a href="#cb17-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a>wing_centroid_size_model_Dwo_m <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_csmm <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> student, <span class="at">data =</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>),</span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb17-110"><a href="#cb17-110" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#prior(normal(0, 0.5), class = b),</span></span>
<span id="cb17-111"><a href="#cb17-111" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb17-112"><a href="#cb17-112" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">20000</span>, <span class="at">warmup =</span> <span class="dv">4000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-113"><a href="#cb17-113" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/Dwo_cs_wing.m.model&quot;</span>,</span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a>dworkin_m <span class="ot">&lt;-</span> Dworkin_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>)</span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> dworkin_m<span class="sc">$</span>line,</span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;M&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Sex)</span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-124"><a href="#cb17-124" aria-hidden="true" tabindex="-1"></a>fitted_wing_dworkin_m <span class="ot">&lt;-</span> <span class="fu">fitted</span>(wing_centroid_size_model_Dwo_m,</span>
<span id="cb17-125"><a href="#cb17-125" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm),</span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Dworkin_data<span class="sc">$</span>csmm) <span class="sc">+</span> <span class="fu">mean</span>(Dworkin_data<span class="sc">$</span>csmm))</span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a>centroid_size_Dworkin_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_wing_dworkin_m) <span class="sc">%&gt;%</span></span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;wing.centroid.size.2019_Dworkin.m&quot;</span>,</span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Pitchers et al (2019) Evolution&quot;</span>,</span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Morphological&quot;</span>,</span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.&quot;</span>,</span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a>Pitchers_traits <span class="ot">&lt;-</span> </span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(centroid_size_Houle_f_line_means,</span>
<span id="cb17-144"><a href="#cb17-144" aria-hidden="true" tabindex="-1"></a>        centroid_size_Houle_m_line_means,</span>
<span id="cb17-145"><a href="#cb17-145" aria-hidden="true" tabindex="-1"></a>        centroid_size_Dworkin_f_line_means,</span>
<span id="cb17-146"><a href="#cb17-146" aria-hidden="true" tabindex="-1"></a>        centroid_size_Dworkin_m_line_means)</span></code></pre></div>
</div>
<div id="rohde-et-al-2019" class="section level3">
<h3>Rohde et al 2019</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>Rohde_2019_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Rohde_2019.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(DGRP_id),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">arena_id =</span> <span class="fu">as.factor</span>(arena_id),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> <span class="fu">as.factor</span>(treatment),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">day_id =</span> <span class="fu">as.factor</span>(day_id),</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">neighbour =</span> <span class="fu">as.factor</span>(neighbour),</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">plate_id =</span> <span class="fu">as.factor</span>(plate_id),</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>         <span class="co"># standardise</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">st_distance_cm =</span> <span class="fu">my_scale</span>(distance_cm))</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co"># distance travelled</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>activity_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(st_distance_cm <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>day_id) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>plate_id) <span class="sc">+</span> (treatment<span class="sc">|</span>line),</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">family =</span> gaussian, <span class="at">data =</span> Rohde_2019_data,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>                      <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                      <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/activity.model&quot;</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(activity_model)</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Rohde_2019_data<span class="sc">$</span>line,</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> Rohde_2019_data<span class="sc">$</span>treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, treatment)</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>fitted_activity <span class="ot">&lt;-</span> <span class="fu">fitted</span>(activity_model,</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> distance_cm <span class="sc">~</span> (treatment<span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Rohde_2019_data<span class="sc">$</span>distance_cm) <span class="sc">+</span> <span class="fu">mean</span>(Rohde_2019_data<span class="sc">$</span>distance_cm),</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Rohde_2019_data<span class="sc">$</span>distance_cm) <span class="sc">+</span> <span class="fu">mean</span>(Rohde_2019_data<span class="sc">$</span>distance_cm),</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Rohde_2019_data<span class="sc">$</span>distance_cm) <span class="sc">+</span> <span class="fu">mean</span>(Rohde_2019_data<span class="sc">$</span>distance_cm))</span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>activity_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_activity) <span class="sc">%&gt;%</span> </span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, treatment, Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">treatment =</span> <span class="fu">if_else</span>(treatment <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;MPH&quot;</span>, <span class="st">&quot;SUC&quot;</span>))</span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a><span class="co"># first lets get line means for activity in the control sucrose treatment</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>activity_line_means <span class="ot">&lt;-</span> activity_line_estimates <span class="sc">%&gt;%</span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">&quot;SUC&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;activity.m&quot;</span>,</span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Rohde et al (2019) Genetics&quot;</span>,</span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.&quot;</span>,</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a><span class="co"># now get line means for activity in the mph treatment</span></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a>activity_mph_line_means <span class="ot">&lt;-</span> activity_line_estimates <span class="sc">%&gt;%</span></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">&quot;MPH&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;activity.MPH.m&quot;</span>,</span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Rohde et al (2019) Genetics&quot;</span>,</span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.&quot;</span>,</span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-68"><a href="#cb18-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-69"><a href="#cb18-69" aria-hidden="true" tabindex="-1"></a><span class="co"># now lets get means for the effect of mph on movement by getting the difference between treatments</span></span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a>SUC_activity <span class="ot">&lt;-</span> </span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a>  activity_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">SUC_estimate =</span> trait_value)</span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a>mph_activity <span class="ot">&lt;-</span> </span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a>  activity_mph_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">mph_estimate =</span> trait_value)</span>
<span id="cb18-80"><a href="#cb18-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-81"><a href="#cb18-81" aria-hidden="true" tabindex="-1"></a>activity_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb18-82"><a href="#cb18-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(SUC_activity, mph_activity) <span class="sc">%&gt;%</span> </span>
<span id="cb18-83"><a href="#cb18-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> mph_estimate <span class="sc">-</span> SUC_estimate)</span>
<span id="cb18-84"><a href="#cb18-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-85"><a href="#cb18-85" aria-hidden="true" tabindex="-1"></a>mph_effect_line_means <span class="ot">&lt;-</span> activity_estimates_diff <span class="sc">%&gt;%</span></span>
<span id="cb18-86"><a href="#cb18-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;MPH.effect.on.activity.m&quot;</span>,</span>
<span id="cb18-87"><a href="#cb18-87" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb18-88"><a href="#cb18-88" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Rohde et al (2019) Genetics&quot;</span>,</span>
<span id="cb18-89"><a href="#cb18-89" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Drug response&quot;</span>,</span>
<span id="cb18-90"><a href="#cb18-90" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.&quot;</span>,</span>
<span id="cb18-91"><a href="#cb18-91" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb18-92"><a href="#cb18-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb18-93"><a href="#cb18-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb18-94"><a href="#cb18-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-95"><a href="#cb18-95" aria-hidden="true" tabindex="-1"></a>Rohde_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(activity_line_means,</span>
<span id="cb18-96"><a href="#cb18-96" aria-hidden="true" tabindex="-1"></a>                      activity_mph_line_means,</span>
<span id="cb18-97"><a href="#cb18-97" aria-hidden="true" tabindex="-1"></a>                      mph_effect_line_means)</span></code></pre></div>
</div>
<div id="everett-et-al-2020" class="section level3">
<h3>Everett et al 2020</h3>
<p>This is mainly data wrangling. We find the mean from two data points
for each line simply via the <code>mean()</code> function.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>Everett_2020_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Everett_microbiome_2020.csv&quot;</span>) </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>microbiome_line_means <span class="ot">&lt;-</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  Everett_2020_data <span class="sc">%&gt;%</span> </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="st">`</span><span class="at">Microbial Species</span><span class="st">`</span>, <span class="at">names_to =</span> <span class="st">&quot;line&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;trait_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(line, <span class="st">&quot;F&quot;</span>), <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>),</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_F1&quot;</span>),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_F2&quot;</span>),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_M1&quot;</span>),</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;_M2&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Microbial Species</span><span class="st">`</span>, line, Sex) <span class="sc">%&gt;%</span> </span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">trait_value =</span> <span class="fu">mean</span>(trait_value)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>( <span class="at">Reference =</span> <span class="st">&quot;Everett et al (2020) Genome Research&quot;</span>,</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Microbiome&quot;</span>,</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>          <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.&quot;</span>,</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">Trait =</span> <span class="fu">str_replace</span>(<span class="st">`</span><span class="at">Microbial Species</span><span class="st">`</span>, <span class="st">&quot; &quot;</span>, <span class="st">&quot;.&quot;</span>),</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">Trait_1 =</span> <span class="fu">if_else</span>(<span class="fu">str_detect</span>(Sex, <span class="st">&quot;Female&quot;</span>), <span class="st">&quot;.f&quot;</span>, <span class="st">&quot;.m&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Trait, Trait, Trait_1, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait, line, Sex)</span></code></pre></div>
</div>
<div id="chapman-et-al-2020" class="section level3">
<h3>Chapman et al 2020</h3>
<p><strong>Note that I haven’t included Wolbachia status in the model
yet, which they do in the paper…</strong></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>Chapman_2020_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Chapman_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line),</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">Treatment =</span> <span class="fu">as.factor</span>(Treatment),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">Date =</span> <span class="fu">as.factor</span>(Date))</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>E.faecalis_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(Alive_at_day5 <span class="sc">|</span> <span class="fu">trials</span>(number_infected) <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Infector <span class="sc">+</span> Date <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Vial) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Chapman_2020_data, <span class="at">family =</span> <span class="fu">binomial</span>(),</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/E.faecalis.model&quot;</span>,</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(E.faecalis_model)</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="co"># set the fixed effects in the new data for which we will derive predictions to the reference levels in the model i.e. Infector = jo and Date = `5/3/17`</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Chapman_2020_data<span class="sc">$</span>line,</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">number_infected =</span> <span class="dv">1</span>,</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">Infector =</span> <span class="st">&quot;Jo&quot;</span>,</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Date =</span> <span class="st">&#39;5/3/17&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, number_infected, Infector, Date) </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>fitted_E.faecalis <span class="ot">&lt;-</span> <span class="fu">fitted</span>(E.faecalis_model,</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data,</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">re_formula =</span> Alive_at_day5 <span class="sc">|</span> <span class="fu">trials</span>(number_infected) <span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>line)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>E.faecalis_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_E.faecalis) <span class="sc">%&gt;%</span></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;E.faecalis.resistance.m&quot;</span>,</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Chapman et al (2020) Genes&quot;</span>,</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Pathogen response&quot;</span>,</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis&quot;</span>,</span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="zhou-et-al-2020" class="section level3">
<h3>Zhou et al 2020</h3>
<p>Line means are provided for the non-metabolome traits in the
supplementary material. Raw data are provided for the metabolome traits,
which we model below.</p>
<p>There are 453 metabolites. We need a quick and easy way to model
these and find line means. To do this, we fit relatively simple gaussian
models with Sex as a fixed effect (where this is applicable) and line as
a random effect, with a sex-specific intercept.</p>
<p>Five traits show no genetic variation in one of the sexes; that is,
each line has the same value. We do not model these, but still need to
model the trait for the sex that shows genetic variation. We fit a very
similar model, except without the sex fixed effect.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>Zhou_data <span class="ot">&lt;-</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Zhou_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">SUB PATHWAY</span><span class="st">`</span>, <span class="st">`</span><span class="at">METABOLITE  ID</span><span class="st">`</span>, PLATFORM, <span class="st">`</span><span class="at">CHEMICAL ID</span><span class="st">`</span>,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">RETENTION INDEX</span><span class="st">`</span>, MASS, CAS, PUBCHEM, KEGG, <span class="st">`</span><span class="at">CLIENT IDENTIFIER</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">242</span>, <span class="at">names_to =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Trait_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(ID, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;Line&quot;</span>, <span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Rep&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;-&quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(Line), <span class="at">Sex =</span> <span class="fu">as.factor</span>(Sex)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Line) <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># standardise</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(METABOLITE, Sex) <span class="sc">%&gt;%</span> </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">st_trait_value =</span> <span class="fu">my_scale</span>(Trait_value)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># some traits have the same value for every line; these can&#39;t be standardised because the SD = 0. We remove these traits from the dataset</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(st_trait_value <span class="sc">!=</span> <span class="st">&quot;NaN&quot;</span>)</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="co"># this is our model outline - pick your favourite metabolite! We can then update this iteratively with new filtered data to model every trait. We do this because we don&#39;t want brms to recompile every time. With the model already compiled we can simply update, which saves a load of time.</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> Zhou_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(METABOLITE <span class="sc">==</span> <span class="st">&quot;xanthurenate&quot;</span>)</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>model_outline <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="at">data =</span> data,</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>             st_trait_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> Sex <span class="sc">+</span> (Sex<span class="sc">|</span>line),</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>             <span class="at">family =</span> gaussian,</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>             <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, </span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>             <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.95</span>, </span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">max_treedepth =</span> <span class="dv">12</span>))</span></code></pre></div>
<pre><code>Running /Library/Frameworks/R.framework/Resources/bin/R CMD SHLIB foo.c
clang -mmacosx-version-min=10.13 -I&quot;/Library/Frameworks/R.framework/Resources/include&quot; -DNDEBUG   -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/unsupported&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include&quot; -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/src/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppParallel/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/rstan/include&quot; -DEIGEN_NO_DEBUG  -DBOOST_DISABLE_ASSERTS  -DBOOST_PENDING_INTEGER_LOG2_HPP  -DSTAN_THREADS  -DBOOST_NO_AUTO_PTR  -include &#39;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp&#39;  -D_REENTRANT -DRCPP_PARALLEL_USE_TBB=1   -I/usr/local/include   -fPIC  -Wall -g -O2  -c foo.c -o foo.o
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Dense:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Core:88:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:613:1: error: unknown type name &#39;namespace&#39;
namespace Eigen {
^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:613:16: error: expected &#39;;&#39; after top level declarator
namespace Eigen {
               ^
               ;
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Dense:1:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Core:96:10: fatal error: &#39;complex&#39; file not found
#include &lt;complex&gt;
         ^~~~~~~~~
3 errors generated.
make: *** [foo.o] Error 1</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Now make a function to run a model for each trait and get the posterior line means for each sex </span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>metabolite_modeller <span class="ot">&lt;-</span> <span class="cf">function</span>(selected_trait){</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> Zhou_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(METABOLITE <span class="sc">==</span> selected_trait)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">update</span>(model_outline, <span class="at">newdata =</span> data) <span class="co"># same model, but with new data each time</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>fitted_new_data <span class="ot">&lt;-</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> data<span class="sc">$</span>line,</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> data<span class="sc">$</span>Sex,</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> data<span class="sc">$</span>METABOLITE,</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolome&quot;</span>,</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.&quot;</span>,</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zhou et al (2020) Genome Research&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Trait, Sex, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference)</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>fitted_metabolite <span class="ot">&lt;-</span> <span class="fu">fitted</span>(model,</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> fitted_new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">trait_value =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value))</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>metabolite_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(fitted_new_data, fitted_metabolite) <span class="sc">%&gt;%</span> </span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>),</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, <span class="fu">paste</span>(Trait, <span class="st">&quot;f&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>), <span class="fu">paste</span>(Trait, <span class="st">&quot;m&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>)))</span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a><span class="co"># create a list of all trait to iterate our function over</span></span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a>Both_sexes <span class="ot">&lt;-</span></span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a>  Zhou_data <span class="sc">%&gt;%</span> </span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(METABOLITE, Sex) <span class="sc">%&gt;%</span></span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(METABOLITE) <span class="sc">%&gt;%</span> </span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb23-41"><a href="#cb23-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true" tabindex="-1"></a>trait_list_both_sexes <span class="ot">&lt;-</span> Both_sexes<span class="sc">$</span>METABOLITE</span>
<span id="cb23-44"><a href="#cb23-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-45"><a href="#cb23-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Now make a function to run a model for each trait and get the posterior line means for each sex </span></span>
<span id="cb23-46"><a href="#cb23-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-47"><a href="#cb23-47" aria-hidden="true" tabindex="-1"></a>metabolite_modeller_single_sex <span class="ot">&lt;-</span> <span class="cf">function</span>(selected_trait){</span>
<span id="cb23-48"><a href="#cb23-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-49"><a href="#cb23-49" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> Zhou_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(METABOLITE <span class="sc">==</span> selected_trait)</span>
<span id="cb23-50"><a href="#cb23-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-51"><a href="#cb23-51" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">update</span>(model_outline, <span class="at">newdata =</span> data,</span>
<span id="cb23-52"><a href="#cb23-52" aria-hidden="true" tabindex="-1"></a>                  st_trait_value <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb23-53"><a href="#cb23-53" aria-hidden="true" tabindex="-1"></a>                  <span class="at">family =</span> gaussian,</span>
<span id="cb23-54"><a href="#cb23-54" aria-hidden="true" tabindex="-1"></a>                  <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb23-55"><a href="#cb23-55" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd))) <span class="co"># same model, but with new data each time</span></span>
<span id="cb23-56"><a href="#cb23-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-57"><a href="#cb23-57" aria-hidden="true" tabindex="-1"></a>  fitted_new_data <span class="ot">&lt;-</span></span>
<span id="cb23-58"><a href="#cb23-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tibble</span>(<span class="at">line =</span> data<span class="sc">$</span>line,</span>
<span id="cb23-59"><a href="#cb23-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">Sex =</span> data<span class="sc">$</span>Sex,</span>
<span id="cb23-60"><a href="#cb23-60" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> data<span class="sc">$</span>METABOLITE,</span>
<span id="cb23-61"><a href="#cb23-61" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolome&quot;</span>,</span>
<span id="cb23-62"><a href="#cb23-62" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.&quot;</span>,</span>
<span id="cb23-63"><a href="#cb23-63" aria-hidden="true" tabindex="-1"></a>           <span class="at">Reference =</span> <span class="st">&quot;Zhou et al (2020) Genome Research&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-64"><a href="#cb23-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(line, Trait, Sex, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference)</span>
<span id="cb23-65"><a href="#cb23-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-66"><a href="#cb23-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-67"><a href="#cb23-67" aria-hidden="true" tabindex="-1"></a>  fitted_metabolite <span class="ot">&lt;-</span> <span class="fu">fitted</span>(model,</span>
<span id="cb23-68"><a href="#cb23-68" aria-hidden="true" tabindex="-1"></a>                              <span class="at">newdata =</span> fitted_new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb23-69"><a href="#cb23-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-70"><a href="#cb23-70" aria-hidden="true" tabindex="-1"></a>    <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb23-71"><a href="#cb23-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">trait_value =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb23-72"><a href="#cb23-72" aria-hidden="true" tabindex="-1"></a>           <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value),</span>
<span id="cb23-73"><a href="#cb23-73" aria-hidden="true" tabindex="-1"></a>           <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(data<span class="sc">$</span>Trait_value) <span class="sc">+</span> <span class="fu">mean</span>(data<span class="sc">$</span>Trait_value))</span>
<span id="cb23-74"><a href="#cb23-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-75"><a href="#cb23-75" aria-hidden="true" tabindex="-1"></a>  metabolite_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(fitted_new_data, fitted_metabolite) <span class="sc">%&gt;%</span> </span>
<span id="cb23-76"><a href="#cb23-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb23-77"><a href="#cb23-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>),</span>
<span id="cb23-78"><a href="#cb23-78" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> <span class="fu">if_else</span>(Sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, <span class="fu">paste</span>(Trait, <span class="st">&quot;f&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>), <span class="fu">paste</span>(Trait, <span class="st">&quot;m&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>)))</span>
<span id="cb23-79"><a href="#cb23-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-80"><a href="#cb23-80" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-81"><a href="#cb23-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-82"><a href="#cb23-82" aria-hidden="true" tabindex="-1"></a><span class="co"># create trait list to run function over</span></span>
<span id="cb23-83"><a href="#cb23-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-84"><a href="#cb23-84" aria-hidden="true" tabindex="-1"></a>Single_sex <span class="ot">&lt;-</span></span>
<span id="cb23-85"><a href="#cb23-85" aria-hidden="true" tabindex="-1"></a>  Zhou_data <span class="sc">%&gt;%</span> </span>
<span id="cb23-86"><a href="#cb23-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(METABOLITE, Sex) <span class="sc">%&gt;%</span></span>
<span id="cb23-87"><a href="#cb23-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(METABOLITE) <span class="sc">%&gt;%</span> </span>
<span id="cb23-88"><a href="#cb23-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb23-89"><a href="#cb23-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb23-90"><a href="#cb23-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-91"><a href="#cb23-91" aria-hidden="true" tabindex="-1"></a>trait_list_single_sex <span class="ot">&lt;-</span> Single_sex<span class="sc">$</span>METABOLITE</span>
<span id="cb23-92"><a href="#cb23-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-93"><a href="#cb23-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the models</span></span>
<span id="cb23-94"><a href="#cb23-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-95"><a href="#cb23-95" aria-hidden="true" tabindex="-1"></a>Run_function <span class="ot">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># Change this to TRUE to run the models</span></span>
<span id="cb23-96"><a href="#cb23-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-97"><a href="#cb23-97" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(Run_function){</span>
<span id="cb23-98"><a href="#cb23-98" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-99"><a href="#cb23-99" aria-hidden="true" tabindex="-1"></a>  models_both_sexes <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(trait_list_both_sexes, metabolite_modeller) <span class="co"># map_dfr returns a data frame created by row-binding each output</span></span>
<span id="cb23-100"><a href="#cb23-100" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-101"><a href="#cb23-101" aria-hidden="true" tabindex="-1"></a>  models_single_sex <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(trait_list_single_sex, metabolite_modeller_single_sex)</span>
<span id="cb23-102"><a href="#cb23-102" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-103"><a href="#cb23-103" aria-hidden="true" tabindex="-1"></a>  Zhou_line_means <span class="ot">&lt;-</span></span>
<span id="cb23-104"><a href="#cb23-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbind</span>(models_both_sexes,</span>
<span id="cb23-105"><a href="#cb23-105" aria-hidden="true" tabindex="-1"></a>          models_single_sex)</span>
<span id="cb23-106"><a href="#cb23-106" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-107"><a href="#cb23-107" aria-hidden="true" tabindex="-1"></a>  Zhou_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb23-108"><a href="#cb23-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_csv</span>(<span class="st">&quot;data/derived/Zhou_line_means.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-109"><a href="#cb23-109" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">str_replace_all</span>(Trait, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>),</span>
<span id="cb23-110"><a href="#cb23-110" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> <span class="fu">str_remove</span>(Trait, <span class="st">&quot;[*]&quot;</span>))  <span class="co"># replace invalid names</span></span>
<span id="cb23-111"><a href="#cb23-111" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb23-112"><a href="#cb23-112" aria-hidden="true" tabindex="-1"></a>  Zhou_line_means <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/Zhou_line_means.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-113"><a href="#cb23-113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">str_replace_all</span>(Trait, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>),</span>
<span id="cb23-114"><a href="#cb23-114" aria-hidden="true" tabindex="-1"></a>           <span class="at">Trait =</span> <span class="fu">str_remove</span>(Trait, <span class="st">&quot;[*]&quot;</span>))  <span class="co"># replace invalid names</span></span>
<span id="cb23-115"><a href="#cb23-115" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="zwoinska-et-al-2020" class="section level3">
<h3>Zwoinska et al 2020</h3>
<p>The authors treat temperature as a continuous variable. But they only
have measures at 3 temperatures, so we choose to consider temperature as
an ordered predictor, following Solomon Kurz’s <a
href="https://bookdown.org/content/4857/monsters-and-mixtures.html#ordered-categorical-predictors">translation</a>
of McElreath (2020) to the <code>brms</code> framework. This approach
results in an improved model fit.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>Zwoinska_male_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Zwoinska_2020/Zwoinska_males_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">line =</span> DGRP) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="fu">as.factor</span>(Batch),</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> <span class="fu">recode</span>(Temperature,</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;25&quot;</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;27&quot;</span> <span class="ot">=</span> <span class="dv">2</span>,</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;29&quot;</span> <span class="ot">=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.integer</span>())</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>fertility_male_model_ordered <span class="ot">&lt;-</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Larvae <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> <span class="fu">mo</span>(Temperature_new) <span class="sc">+</span> (<span class="fu">mo</span>(Temperature_new)<span class="sc">|</span>line),</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Zwoinska_male_data,</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">dirichlet</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">class =</span> simo, <span class="at">coef =</span> moTemperature_new1),</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> bernoulli,</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">4000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/fertility_male.model_ordered&quot;</span>)</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Zwoinska_male_data<span class="sc">$</span>line,</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> Zwoinska_male_data<span class="sc">$</span>Temperature_new,</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Temperature_new, Batch)</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>fitted_male_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_male_model_ordered,</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>fertility_m_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_male_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">.keep =</span> <span class="st">&quot;unused&quot;</span>, </span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">Temperature =</span> <span class="fu">recode</span>(Temperature_new,</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="dv">25</span>,</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;2&quot;</span> <span class="ot">=</span> <span class="dv">27</span>,</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;3&quot;</span> <span class="ot">=</span> <span class="dv">29</span>)) </span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create the line means tibbles</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>Zwoinska_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, value, Trait, sex, description){</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span></span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature <span class="sc">==</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> sex,</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Life history&quot;</span>,</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>fertility_25C_m_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_m_line_estimates, <span class="st">&quot;25&quot;</span>, <span class="st">&quot;fertility.25C.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.&quot;</span>)</span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a>fertility_27C_m_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_m_line_estimates, <span class="st">&quot;27&quot;</span>, <span class="st">&quot;fertility.27C.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>fertility_29C_m_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_m_line_estimates, <span class="st">&quot;29&quot;</span>, <span class="st">&quot;fertility.29C.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a><span class="co"># we need a plasticity measure</span></span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a><span class="do">## The code below uses the slope estimated by the model for each line as the plasticity measure.</span></span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). The line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).</span></span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a>fertility_m_temp_effect_line_means <span class="ot">&lt;-</span> <span class="fu">as_draws_df</span>(fertility_male_model_ordered) <span class="sc">%&gt;%</span> </span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;r_line&quot;</span>) <span class="sc">&amp;</span> <span class="fu">contains</span>(<span class="st">&quot;moTemperature&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(key) <span class="sc">%&gt;%</span></span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the trait in question is plasticity, which is measured by the line specific random slope between treatments - we can get this straight from our posterior samples, without adding anything else </span></span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(value) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(key, <span class="st">&quot;r_line&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;,moTemperature_new&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">[|</span><span class="sc">\\</span><span class="st">]&quot;</span>, <span class="st">&quot;&quot;</span>, line),</span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.m&quot;</span>,</span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.&quot;</span>,</span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> mean</span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Slopes are hard to interpret, so we also provide this data in a different format. Below we calculate the change in fertility between 25C and 29C for each line. Greater values indicate a greater loss in fertility. Note this approach produces a very similar line ordering to the bove approach</span></span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a>  new_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">paste0</span>(<span class="st">&quot;V&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">372</span>))</span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a>fitted_male_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_male_model_ordered,</span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data, <span class="at">summary =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Estimate&quot;</span>)</span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a>fitted_male_fertility <span class="ot">&lt;-</span> <span class="fu">right_join</span>(new_data, fitted_male_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>  as_tibble <span class="sc">%&gt;%</span> </span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ID)</span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-105"><a href="#cb24-105" aria-hidden="true" tabindex="-1"></a><span class="co"># Different numbers of lines have been phenotyped for fertility in the 25C and 29C conditions. The code below finds the lines that are measured in one temp but not the other and removes them. We then find the difference in fertility between 25 and 29C conditions</span></span>
<span id="cb24-106"><a href="#cb24-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-107"><a href="#cb24-107" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> </span>
<span id="cb24-108"><a href="#cb24-108" aria-hidden="true" tabindex="-1"></a>  fitted_male_fertility <span class="sc">%&gt;%</span> </span>
<span id="cb24-109"><a href="#cb24-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">1</span>, line <span class="sc">!=</span> <span class="st">&quot;38&quot;</span> <span class="sc">&amp;</span> line <span class="sc">!=</span> <span class="st">&quot;83&quot;</span> <span class="sc">&amp;</span> line <span class="sc">!=</span> <span class="st">&quot;395&quot;</span>, line <span class="sc">!=</span> <span class="st">&quot;707&quot;</span> <span class="sc">&amp;</span> line <span class="sc">!=</span> <span class="st">&quot;727&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-110"><a href="#cb24-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp25 =</span> Estimate)</span>
<span id="cb24-111"><a href="#cb24-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-112"><a href="#cb24-112" aria-hidden="true" tabindex="-1"></a>c <span class="ot">&lt;-</span> </span>
<span id="cb24-113"><a href="#cb24-113" aria-hidden="true" tabindex="-1"></a>  fitted_male_fertility <span class="sc">%&gt;%</span></span>
<span id="cb24-114"><a href="#cb24-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">3</span>, line <span class="sc">!=</span> <span class="st">&quot;911&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-115"><a href="#cb24-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp29 =</span> Estimate) <span class="sc">%&gt;%</span></span>
<span id="cb24-116"><a href="#cb24-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Temperature_new, Batch, line))</span>
<span id="cb24-117"><a href="#cb24-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-118"><a href="#cb24-118" aria-hidden="true" tabindex="-1"></a><span class="co">#a_line &lt;- a %&gt;% distinct(line) </span></span>
<span id="cb24-119"><a href="#cb24-119" aria-hidden="true" tabindex="-1"></a><span class="co">#c_line &lt;- c %&gt;% distinct(line) # we remove the line column in the above code to make binding possible. This line of code will therefore not work, but it has already done its job.</span></span>
<span id="cb24-120"><a href="#cb24-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-121"><a href="#cb24-121" aria-hidden="true" tabindex="-1"></a><span class="co">#anti_join(a_line, c_line) # this line finds the lines present in the a_line vector but not in the b_line vector</span></span>
<span id="cb24-122"><a href="#cb24-122" aria-hidden="true" tabindex="-1"></a><span class="co">#anti_join(c_line, a_line) # this line finds the lines present in the b_line vector but not in the a_line vector</span></span>
<span id="cb24-123"><a href="#cb24-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-124"><a href="#cb24-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-125"><a href="#cb24-125" aria-hidden="true" tabindex="-1"></a>backup_approach <span class="ot">&lt;-</span> <span class="fu">cbind</span>(a, c) <span class="sc">%&gt;%</span> </span>
<span id="cb24-126"><a href="#cb24-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-127"><a href="#cb24-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fertility_diff =</span> Temp25 <span class="sc">-</span> Temp29) <span class="sc">%&gt;%</span> </span>
<span id="cb24-128"><a href="#cb24-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb24-129"><a href="#cb24-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">trait_value =</span> <span class="fu">mean</span>(fertility_diff)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-130"><a href="#cb24-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.m&quot;</span>,</span>
<span id="cb24-131"><a href="#cb24-131" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb24-132"><a href="#cb24-132" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb24-133"><a href="#cb24-133" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb24-134"><a href="#cb24-134" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in male fertility (the proportion of males that produced viable larvae), between 25C and 29C rearing conditions. Higher values indicate a greater decrease in fertility. This is a measure of thermal susceptibility or thermal plasticity in fertility.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-135"><a href="#cb24-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-136"><a href="#cb24-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>trait_value)</span>
<span id="cb24-137"><a href="#cb24-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-138"><a href="#cb24-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-139"><a href="#cb24-139" aria-hidden="true" tabindex="-1"></a><span class="co"># female fertility</span></span>
<span id="cb24-140"><a href="#cb24-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-141"><a href="#cb24-141" aria-hidden="true" tabindex="-1"></a>Zwoinska_female_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Zwoinska_2020/Zwoinska_females_2020.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-142"><a href="#cb24-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">line =</span> DGRP) <span class="sc">%&gt;%</span> </span>
<span id="cb24-143"><a href="#cb24-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb24-144"><a href="#cb24-144" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="fu">as.factor</span>(Batch),</span>
<span id="cb24-145"><a href="#cb24-145" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> <span class="fu">recode</span>(Temperature,</span>
<span id="cb24-146"><a href="#cb24-146" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;25&quot;</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb24-147"><a href="#cb24-147" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;27&quot;</span> <span class="ot">=</span> <span class="dv">2</span>,</span>
<span id="cb24-148"><a href="#cb24-148" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;29&quot;</span> <span class="ot">=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-149"><a href="#cb24-149" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.integer</span>())</span>
<span id="cb24-150"><a href="#cb24-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-151"><a href="#cb24-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-152"><a href="#cb24-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-153"><a href="#cb24-153" aria-hidden="true" tabindex="-1"></a>fertility_female_model_ordered <span class="ot">&lt;-</span></span>
<span id="cb24-154"><a href="#cb24-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(Larvae <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> <span class="fu">mo</span>(Temperature_new) <span class="sc">+</span> (<span class="fu">mo</span>(Temperature_new)<span class="sc">|</span>line),</span>
<span id="cb24-155"><a href="#cb24-155" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Zwoinska_female_data,</span>
<span id="cb24-156"><a href="#cb24-156" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb24-157"><a href="#cb24-157" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb24-158"><a href="#cb24-158" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">dirichlet</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">class =</span> simo, <span class="at">coef =</span> moTemperature_new1),</span>
<span id="cb24-159"><a href="#cb24-159" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb24-160"><a href="#cb24-160" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> bernoulli,</span>
<span id="cb24-161"><a href="#cb24-161" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">4000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb24-162"><a href="#cb24-162" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb24-163"><a href="#cb24-163" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/fertility_female.model_ordered&quot;</span>)</span>
<span id="cb24-164"><a href="#cb24-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-165"><a href="#cb24-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-166"><a href="#cb24-166" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb24-167"><a href="#cb24-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Zwoinska_female_data<span class="sc">$</span>line,</span>
<span id="cb24-168"><a href="#cb24-168" aria-hidden="true" tabindex="-1"></a>         <span class="at">Temperature_new =</span> Zwoinska_female_data<span class="sc">$</span>Temperature_new,</span>
<span id="cb24-169"><a href="#cb24-169" aria-hidden="true" tabindex="-1"></a>         <span class="at">Batch =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-170"><a href="#cb24-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Temperature_new, Batch)</span>
<span id="cb24-171"><a href="#cb24-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-172"><a href="#cb24-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-173"><a href="#cb24-173" aria-hidden="true" tabindex="-1"></a>fitted_female_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_female_model_ordered,</span>
<span id="cb24-174"><a href="#cb24-174" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb24-175"><a href="#cb24-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb24-176"><a href="#cb24-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-177"><a href="#cb24-177" aria-hidden="true" tabindex="-1"></a>fertility_f_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_female_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb24-178"><a href="#cb24-178" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">.keep =</span> <span class="st">&quot;unused&quot;</span>, </span>
<span id="cb24-179"><a href="#cb24-179" aria-hidden="true" tabindex="-1"></a>            <span class="at">Temperature =</span> <span class="fu">recode</span>(Temperature_new,</span>
<span id="cb24-180"><a href="#cb24-180" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="dv">25</span>,</span>
<span id="cb24-181"><a href="#cb24-181" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;2&quot;</span> <span class="ot">=</span> <span class="dv">27</span>,</span>
<span id="cb24-182"><a href="#cb24-182" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;3&quot;</span> <span class="ot">=</span> <span class="dv">29</span>)) </span>
<span id="cb24-183"><a href="#cb24-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-184"><a href="#cb24-184" aria-hidden="true" tabindex="-1"></a>fertility_25C_f_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_f_line_estimates, <span class="st">&quot;25&quot;</span>, <span class="st">&quot;fertility.25C.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean proportion of females that produced viable larvae. Females were reared at 25C. Higher values indicates greater fertility.&quot;</span>)</span>
<span id="cb24-185"><a href="#cb24-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-186"><a href="#cb24-186" aria-hidden="true" tabindex="-1"></a>fertility_27C_f_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_f_line_estimates, <span class="st">&quot;27&quot;</span>, <span class="st">&quot;fertility.27C.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean proportion of females that produced viable larvae. Females were reared at 27C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb24-187"><a href="#cb24-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-188"><a href="#cb24-188" aria-hidden="true" tabindex="-1"></a>fertility_29C_f_line_means <span class="ot">&lt;-</span> <span class="fu">Zwoinska_line_means</span>(fertility_f_line_estimates, <span class="st">&quot;29&quot;</span>, <span class="st">&quot;fertility.29C.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean proportion of females that produced viable larvae. Females were reared at 29C. Higher values indicates greater fertility&quot;</span>)</span>
<span id="cb24-189"><a href="#cb24-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-190"><a href="#cb24-190" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). Once again, the line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).</span></span>
<span id="cb24-191"><a href="#cb24-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-192"><a href="#cb24-192" aria-hidden="true" tabindex="-1"></a><span class="co"># we need a plasticity measure</span></span>
<span id="cb24-193"><a href="#cb24-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-194"><a href="#cb24-194" aria-hidden="true" tabindex="-1"></a><span class="do">## The code below uses the slope estimated by the model for each line as the plasticity measure.</span></span>
<span id="cb24-195"><a href="#cb24-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-196"><a href="#cb24-196" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). The line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).</span></span>
<span id="cb24-197"><a href="#cb24-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-198"><a href="#cb24-198" aria-hidden="true" tabindex="-1"></a>fertility_f_temp_effect_line_means <span class="ot">&lt;-</span> <span class="fu">as_draws_df</span>(fertility_female_model_ordered) <span class="sc">%&gt;%</span> </span>
<span id="cb24-199"><a href="#cb24-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;r_line&quot;</span>) <span class="sc">&amp;</span> <span class="fu">contains</span>(<span class="st">&quot;moTemperature&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-200"><a href="#cb24-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-201"><a href="#cb24-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(key) <span class="sc">%&gt;%</span></span>
<span id="cb24-202"><a href="#cb24-202" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the trait in question is plasticity, which is measured by the line specific random slope between treatments - we can get this straight from our posterior samples, without adding anything else </span></span>
<span id="cb24-203"><a href="#cb24-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(value) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-204"><a href="#cb24-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(key, <span class="st">&quot;r_line&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-205"><a href="#cb24-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;,moTemperature_new&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-206"><a href="#cb24-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">[|</span><span class="sc">\\</span><span class="st">]&quot;</span>, <span class="st">&quot;&quot;</span>, line),</span>
<span id="cb24-207"><a href="#cb24-207" aria-hidden="true" tabindex="-1"></a>         <span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.f&quot;</span>,</span>
<span id="cb24-208"><a href="#cb24-208" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb24-209"><a href="#cb24-209" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb24-210"><a href="#cb24-210" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb24-211"><a href="#cb24-211" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean slope of the change in female fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.&quot;</span>,</span>
<span id="cb24-212"><a href="#cb24-212" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> mean</span>
<span id="cb24-213"><a href="#cb24-213" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb24-214"><a href="#cb24-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-215"><a href="#cb24-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trait_value)</span>
<span id="cb24-216"><a href="#cb24-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-217"><a href="#cb24-217" aria-hidden="true" tabindex="-1"></a><span class="co"># Slopes are hard to interpret, so we also provide this data in a different format. Below we calculate the change in fertility between 25C and 29C for each line. Greater values indicate a greater loss in fertility. Note this approach produces a very similar line ordering to the bove approach</span></span>
<span id="cb24-218"><a href="#cb24-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-219"><a href="#cb24-219" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb24-220"><a href="#cb24-220" aria-hidden="true" tabindex="-1"></a>  new_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-221"><a href="#cb24-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">paste0</span>(<span class="st">&quot;V&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">120</span>))</span>
<span id="cb24-222"><a href="#cb24-222" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-223"><a href="#cb24-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-224"><a href="#cb24-224" aria-hidden="true" tabindex="-1"></a>fitted_female_fertility <span class="ot">&lt;-</span> <span class="fu">fitted</span>(fertility_female_model_ordered,</span>
<span id="cb24-225"><a href="#cb24-225" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data, <span class="at">summary =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb24-226"><a href="#cb24-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-227"><a href="#cb24-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Estimate&quot;</span>)</span>
<span id="cb24-228"><a href="#cb24-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-229"><a href="#cb24-229" aria-hidden="true" tabindex="-1"></a>fitted_female_fertility <span class="ot">&lt;-</span> <span class="fu">right_join</span>(new_data, fitted_female_fertility) <span class="sc">%&gt;%</span> </span>
<span id="cb24-230"><a href="#cb24-230" aria-hidden="true" tabindex="-1"></a>  as_tibble <span class="sc">%&gt;%</span> </span>
<span id="cb24-231"><a href="#cb24-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ID)</span>
<span id="cb24-232"><a href="#cb24-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-233"><a href="#cb24-233" aria-hidden="true" tabindex="-1"></a><span class="co"># phenotypes for all 40 lines are measured so no fancy steps for binding are needed</span></span>
<span id="cb24-234"><a href="#cb24-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-235"><a href="#cb24-235" aria-hidden="true" tabindex="-1"></a>a_female <span class="ot">&lt;-</span> </span>
<span id="cb24-236"><a href="#cb24-236" aria-hidden="true" tabindex="-1"></a>  fitted_female_fertility <span class="sc">%&gt;%</span> </span>
<span id="cb24-237"><a href="#cb24-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-238"><a href="#cb24-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp25 =</span> Estimate)</span>
<span id="cb24-239"><a href="#cb24-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-240"><a href="#cb24-240" aria-hidden="true" tabindex="-1"></a>c_female <span class="ot">&lt;-</span> </span>
<span id="cb24-241"><a href="#cb24-241" aria-hidden="true" tabindex="-1"></a>  fitted_female_fertility <span class="sc">%&gt;%</span></span>
<span id="cb24-242"><a href="#cb24-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Temperature_new <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-243"><a href="#cb24-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Temp29 =</span> Estimate) <span class="sc">%&gt;%</span></span>
<span id="cb24-244"><a href="#cb24-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Temperature_new, Batch, line))</span>
<span id="cb24-245"><a href="#cb24-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-246"><a href="#cb24-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-247"><a href="#cb24-247" aria-hidden="true" tabindex="-1"></a>backup_approach <span class="ot">&lt;-</span> <span class="fu">cbind</span>(a_female, c_female) <span class="sc">%&gt;%</span> </span>
<span id="cb24-248"><a href="#cb24-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-249"><a href="#cb24-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fertility_diff =</span> Temp25 <span class="sc">-</span> Temp29) <span class="sc">%&gt;%</span> </span>
<span id="cb24-250"><a href="#cb24-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(line) <span class="sc">%&gt;%</span> </span>
<span id="cb24-251"><a href="#cb24-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">trait_value =</span> <span class="fu">mean</span>(fertility_diff)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-252"><a href="#cb24-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;fertility.thermal.susceptibility.f&quot;</span>,</span>
<span id="cb24-253"><a href="#cb24-253" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Zwoinska et al (2020) Frontiers in Genetics&quot;</span>,</span>
<span id="cb24-254"><a href="#cb24-254" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Temperature related&quot;</span>,</span>
<span id="cb24-255"><a href="#cb24-255" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb24-256"><a href="#cb24-256" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in female fertility (the proportion of males that produced viable larvae), between 25C and 29C rearing conditions. Higher values indicate a greater decrease in fertility. This is a measure of thermal susceptibility or thermal plasticity in fertility.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-257"><a href="#cb24-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-258"><a href="#cb24-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>trait_value)</span>
<span id="cb24-259"><a href="#cb24-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-260"><a href="#cb24-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-261"><a href="#cb24-261" aria-hidden="true" tabindex="-1"></a><span class="co"># combine all the data from this study into a simple dataframe</span></span>
<span id="cb24-262"><a href="#cb24-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-263"><a href="#cb24-263" aria-hidden="true" tabindex="-1"></a>Zwoinska_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(fertility_25C_m_line_means,</span>
<span id="cb24-264"><a href="#cb24-264" aria-hidden="true" tabindex="-1"></a>                         fertility_27C_m_line_means,</span>
<span id="cb24-265"><a href="#cb24-265" aria-hidden="true" tabindex="-1"></a>                         fertility_29C_m_line_means,</span>
<span id="cb24-266"><a href="#cb24-266" aria-hidden="true" tabindex="-1"></a>                         fertility_m_temp_effect_line_means,</span>
<span id="cb24-267"><a href="#cb24-267" aria-hidden="true" tabindex="-1"></a>                         fertility_25C_f_line_means,</span>
<span id="cb24-268"><a href="#cb24-268" aria-hidden="true" tabindex="-1"></a>                         fertility_27C_f_line_means,</span>
<span id="cb24-269"><a href="#cb24-269" aria-hidden="true" tabindex="-1"></a>                         fertility_29C_f_line_means,</span>
<span id="cb24-270"><a href="#cb24-270" aria-hidden="true" tabindex="-1"></a>                         fertility_f_temp_effect_line_means)</span></code></pre></div>
</div>
<div id="chowdhury-et-al-2021" class="section level3">
<h3>Chowdhury et al 2021</h3>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>Chowdhury_2021_aggressive_lunges <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Chowdhury_2021_aggression.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">as.factor</span>(line))</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>aggression_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(agressive.lunges.per<span class="fl">.20</span>mins.m <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>line),</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> Chowdhury_2021_aggressive_lunges, <span class="at">family =</span> <span class="fu">zero_inflated_poisson</span>(),</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">3</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">seed =</span> <span class="dv">2</span>, <span class="at">file =</span> <span class="st">&quot;fits/aggression.model&quot;</span>,</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>))</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a> <span class="co">#pp_check(aggression_model)</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Chowdhury_2021_aggressive_lunges<span class="sc">$</span>line) <span class="sc">%&gt;%</span> </span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>fitted_aggression <span class="ot">&lt;-</span> <span class="fu">fitted</span>(aggression_model,</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>aggression_m_line_means <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_aggression) <span class="sc">%&gt;%</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;aggression.2021.m&quot;</span>,</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Chowdhury et al Nature Communications Biology&quot;</span>,</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Behavioural&quot;</span>,</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean number of aggressive lunges made in 20 min window following removal of a divider, by pairs of DGRP males of same genotype. Trials were run after males were kept isolated for five days. Higher values indicate more aggressive males.&quot;</span>,</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;RAL-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span></code></pre></div>
</div>
<div id="watanabe-and-riddle-2021" class="section level3">
<h3>Watanabe and Riddle 2021</h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>Watanabe_climbing_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/Raw_data_files/Watanabe_Riddle_2021.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">line =</span> Line) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">str_remove</span>(line, <span class="st">&quot;line_&quot;</span>),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">line =</span> <span class="fu">as.factor</span>(line),</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Vial =</span> <span class="fu">as.factor</span>(Vial))</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Females</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>Watanabe_female_data <span class="ot">&lt;-</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  Watanabe_climbing_data <span class="sc">%&gt;%</span> </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;F&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_Climbing =</span> <span class="fu">my_scale</span>(Climbing))</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>climbing_female_model <span class="ot">&lt;-</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(st_Climbing <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span>  Treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Vial) <span class="sc">+</span> (Treatment<span class="sc">|</span>line),</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Watanabe_female_data,</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> gaussian,</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.9</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/climbing_female.model&quot;</span>)</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Watanabe_female_data<span class="sc">$</span>line,</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Treatment =</span> Watanabe_female_data<span class="sc">$</span>Treatment,</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Vial =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Treatment, Vial)</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>fitted_female_climbing <span class="ot">&lt;-</span> <span class="fu">fitted</span>(climbing_female_model,</span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Watanabe_female_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_female_data<span class="sc">$</span>Climbing),</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_female_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_female_data<span class="sc">$</span>Climbing),</span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_female_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_female_data<span class="sc">$</span>Climbing))</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>climbing_f_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_female_climbing)</span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a><span class="co"># build a function to create the line means tibbles</span></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>Watanabe_line_means <span class="ot">&lt;-</span> <span class="cf">function</span>(estimates, value, Trait, sex, description){</span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  estimates <span class="sc">%&gt;%</span></span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Treatment <span class="sc">==</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> Trait,</span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> sex,</span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;</span>,</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span><span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a>climbing_f_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_f_line_estimates, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;climbing.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a>climbing_post_exercise_f_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_f_line_estimates, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;climbing.post.exercise.f&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a><span class="co"># get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). Once again, the line rankings produced here are very similar to the slopes estimates presented in Watanabe and Riddle (2021).</span></span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a>control_climbing_f <span class="ot">&lt;-</span> </span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a>  climbing_f_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">control_estimate =</span> trait_value)</span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a>exercise_climbing_f <span class="ot">&lt;-</span> </span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a>  climbing_post_exercise_f_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">exercise_estimate =</span> trait_value)</span>
<span id="cb26-75"><a href="#cb26-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-76"><a href="#cb26-76" aria-hidden="true" tabindex="-1"></a>climbing_estimates_diff <span class="ot">&lt;-</span></span>
<span id="cb26-77"><a href="#cb26-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(control_climbing_f, exercise_climbing_f) <span class="sc">%&gt;%</span> </span>
<span id="cb26-78"><a href="#cb26-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> exercise_estimate <span class="sc">-</span> control_estimate)</span>
<span id="cb26-79"><a href="#cb26-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-80"><a href="#cb26-80" aria-hidden="true" tabindex="-1"></a>climbing_f_exercise_effect_line_means <span class="ot">&lt;-</span> climbing_estimates_diff <span class="sc">%&gt;%</span></span>
<span id="cb26-81"><a href="#cb26-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;climbing.exercise.effect.f&quot;</span>,</span>
<span id="cb26-82"><a href="#cb26-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Female&quot;</span>,</span>
<span id="cb26-83"><a href="#cb26-83" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;</span>,</span>
<span id="cb26-84"><a href="#cb26-84" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb26-85"><a href="#cb26-85" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.&quot;</span>,</span>
<span id="cb26-86"><a href="#cb26-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb26-87"><a href="#cb26-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb26-88"><a href="#cb26-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb26-89"><a href="#cb26-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-90"><a href="#cb26-90" aria-hidden="true" tabindex="-1"></a><span class="co"># males</span></span>
<span id="cb26-91"><a href="#cb26-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-92"><a href="#cb26-92" aria-hidden="true" tabindex="-1"></a>Watanabe_male_data <span class="ot">&lt;-</span></span>
<span id="cb26-93"><a href="#cb26-93" aria-hidden="true" tabindex="-1"></a>  Watanabe_climbing_data <span class="sc">%&gt;%</span> </span>
<span id="cb26-94"><a href="#cb26-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sex <span class="sc">==</span> <span class="st">&quot;M&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-95"><a href="#cb26-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="co"># standardise</span></span>
<span id="cb26-96"><a href="#cb26-96" aria-hidden="true" tabindex="-1"></a>    <span class="at">st_Climbing =</span> <span class="fu">my_scale</span>(Climbing))</span>
<span id="cb26-97"><a href="#cb26-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-98"><a href="#cb26-98" aria-hidden="true" tabindex="-1"></a>climbing_male_model <span class="ot">&lt;-</span></span>
<span id="cb26-99"><a href="#cb26-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brm</span>(st_Climbing <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span>  Treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Vial) <span class="sc">+</span> (Treatment<span class="sc">|</span>line),</span>
<span id="cb26-100"><a href="#cb26-100" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> Watanabe_male_data,</span>
<span id="cb26-101"><a href="#cb26-101" aria-hidden="true" tabindex="-1"></a>      <span class="at">prior =</span> <span class="fu">c</span>(<span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> Intercept),</span>
<span id="cb26-102"><a href="#cb26-102" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> b),</span>
<span id="cb26-103"><a href="#cb26-103" aria-hidden="true" tabindex="-1"></a>                <span class="fu">prior</span>(<span class="fu">exponential</span>(<span class="dv">1</span>), <span class="at">class =</span> sd)),</span>
<span id="cb26-104"><a href="#cb26-104" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> gaussian,</span>
<span id="cb26-105"><a href="#cb26-105" aria-hidden="true" tabindex="-1"></a>      <span class="at">cores =</span> <span class="dv">4</span>, <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">6000</span>, <span class="at">warmup =</span> <span class="dv">2000</span>,</span>
<span id="cb26-106"><a href="#cb26-106" aria-hidden="true" tabindex="-1"></a>      <span class="at">control =</span> <span class="fu">list</span>(<span class="at">adapt_delta =</span> <span class="fl">0.99</span>, <span class="at">max_treedepth =</span> <span class="dv">10</span>),</span>
<span id="cb26-107"><a href="#cb26-107" aria-hidden="true" tabindex="-1"></a>      <span class="at">seed =</span> <span class="dv">1</span>, <span class="at">file =</span> <span class="st">&quot;fits/climbing_male.model&quot;</span>)</span>
<span id="cb26-108"><a href="#cb26-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-109"><a href="#cb26-109" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check(climbing_male_model)</span></span>
<span id="cb26-110"><a href="#cb26-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-111"><a href="#cb26-111" aria-hidden="true" tabindex="-1"></a>new_data <span class="ot">&lt;-</span></span>
<span id="cb26-112"><a href="#cb26-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">line =</span> Watanabe_male_data<span class="sc">$</span>line,</span>
<span id="cb26-113"><a href="#cb26-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">Treatment =</span> Watanabe_male_data<span class="sc">$</span>Treatment,</span>
<span id="cb26-114"><a href="#cb26-114" aria-hidden="true" tabindex="-1"></a>         <span class="at">Vial =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-115"><a href="#cb26-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(line, Treatment, Vial)</span>
<span id="cb26-116"><a href="#cb26-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-117"><a href="#cb26-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-118"><a href="#cb26-118" aria-hidden="true" tabindex="-1"></a>fitted_male_climbing <span class="ot">&lt;-</span> <span class="fu">fitted</span>(climbing_male_model,</span>
<span id="cb26-119"><a href="#cb26-119" aria-hidden="true" tabindex="-1"></a>       <span class="at">newdata =</span> new_data) <span class="sc">%&gt;%</span> </span>
<span id="cb26-120"><a href="#cb26-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-121"><a href="#cb26-121" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now we need to convert back to response scale (note I don&#39;t convert Est.Error)</span></span>
<span id="cb26-122"><a href="#cb26-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> Estimate<span class="sc">*</span><span class="fu">sd</span>(Watanabe_male_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_male_data<span class="sc">$</span>Climbing),</span>
<span id="cb26-123"><a href="#cb26-123" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q2.5 =</span> Q2<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_male_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_male_data<span class="sc">$</span>Climbing),</span>
<span id="cb26-124"><a href="#cb26-124" aria-hidden="true" tabindex="-1"></a>         <span class="at">Q97.5 =</span> Q97<span class="fl">.5</span><span class="sc">*</span><span class="fu">sd</span>(Watanabe_male_data<span class="sc">$</span>Climbing) <span class="sc">+</span> <span class="fu">mean</span>(Watanabe_male_data<span class="sc">$</span>Climbing))</span>
<span id="cb26-125"><a href="#cb26-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-126"><a href="#cb26-126" aria-hidden="true" tabindex="-1"></a>climbing_m_line_estimates <span class="ot">&lt;-</span> <span class="fu">tibble</span>(new_data, fitted_male_climbing)</span>
<span id="cb26-127"><a href="#cb26-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-128"><a href="#cb26-128" aria-hidden="true" tabindex="-1"></a>climbing_m_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_m_line_estimates, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;climbing.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb26-129"><a href="#cb26-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-130"><a href="#cb26-130" aria-hidden="true" tabindex="-1"></a>climbing_post_exercise_m_line_means <span class="ot">&lt;-</span> <span class="fu">Watanabe_line_means</span>(climbing_m_line_estimates, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;climbing.post.exercise.m&quot;</span>, <span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.&quot;</span>)</span>
<span id="cb26-131"><a href="#cb26-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-132"><a href="#cb26-132" aria-hidden="true" tabindex="-1"></a><span class="co"># Now get the difference between treatments</span></span>
<span id="cb26-133"><a href="#cb26-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-134"><a href="#cb26-134" aria-hidden="true" tabindex="-1"></a>control_climbing_m <span class="ot">&lt;-</span> </span>
<span id="cb26-135"><a href="#cb26-135" aria-hidden="true" tabindex="-1"></a>  climbing_m_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb26-136"><a href="#cb26-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb26-137"><a href="#cb26-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">control_estimate =</span> trait_value)</span>
<span id="cb26-138"><a href="#cb26-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-139"><a href="#cb26-139" aria-hidden="true" tabindex="-1"></a>exercise_climbing_m <span class="ot">&lt;-</span> </span>
<span id="cb26-140"><a href="#cb26-140" aria-hidden="true" tabindex="-1"></a>  climbing_post_exercise_m_line_means <span class="sc">%&gt;%</span> </span>
<span id="cb26-141"><a href="#cb26-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb26-142"><a href="#cb26-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">exercise_estimate =</span> trait_value)</span>
<span id="cb26-143"><a href="#cb26-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-144"><a href="#cb26-144" aria-hidden="true" tabindex="-1"></a>climbing_estimates_diff_m <span class="ot">&lt;-</span></span>
<span id="cb26-145"><a href="#cb26-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(control_climbing_m, exercise_climbing_m) <span class="sc">%&gt;%</span> </span>
<span id="cb26-146"><a href="#cb26-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Estimate =</span> exercise_estimate <span class="sc">-</span> control_estimate)</span>
<span id="cb26-147"><a href="#cb26-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-148"><a href="#cb26-148" aria-hidden="true" tabindex="-1"></a>climbing_m_exercise_effect_line_means <span class="ot">&lt;-</span> climbing_estimates_diff_m <span class="sc">%&gt;%</span></span>
<span id="cb26-149"><a href="#cb26-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="st">&quot;climbing.exercise.effect.m&quot;</span>,</span>
<span id="cb26-150"><a href="#cb26-150" aria-hidden="true" tabindex="-1"></a>         <span class="at">Sex =</span> <span class="st">&quot;Male&quot;</span>,</span>
<span id="cb26-151"><a href="#cb26-151" aria-hidden="true" tabindex="-1"></a>         <span class="at">Reference =</span> <span class="st">&quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;</span>,</span>
<span id="cb26-152"><a href="#cb26-152" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Activity&quot;</span>,</span>
<span id="cb26-153"><a href="#cb26-153" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.&quot;</span>,</span>
<span id="cb26-154"><a href="#cb26-154" aria-hidden="true" tabindex="-1"></a>         <span class="at">trait_value =</span> Estimate) <span class="sc">%&gt;%</span> </span>
<span id="cb26-155"><a href="#cb26-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, Sex, trait_value, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb26-156"><a href="#cb26-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(line)</span>
<span id="cb26-157"><a href="#cb26-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-158"><a href="#cb26-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-159"><a href="#cb26-159" aria-hidden="true" tabindex="-1"></a>Watanabe_traits <span class="ot">&lt;-</span> <span class="fu">rbind</span>(climbing_f_line_means,</span>
<span id="cb26-160"><a href="#cb26-160" aria-hidden="true" tabindex="-1"></a>                         climbing_f_exercise_effect_line_means,</span>
<span id="cb26-161"><a href="#cb26-161" aria-hidden="true" tabindex="-1"></a>                         climbing_post_exercise_f_line_means,</span>
<span id="cb26-162"><a href="#cb26-162" aria-hidden="true" tabindex="-1"></a>                         climbing_m_line_means,</span>
<span id="cb26-163"><a href="#cb26-163" aria-hidden="true" tabindex="-1"></a>                         climbing_m_exercise_effect_line_means,</span>
<span id="cb26-164"><a href="#cb26-164" aria-hidden="true" tabindex="-1"></a>                         climbing_post_exercise_m_line_means)</span></code></pre></div>
</div>
</div>
<div id="combine-and-save-all-the-estimated-line-means"
class="section level2">
<h2>Combine and save all the estimated line means</h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>estimated_line_means <span class="ot">&lt;-</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(Jumbo_traits,</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>        Grubbs_traits,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>        Dobson_traits,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>        Gaertner_traits,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>        caffeine_resistance_line_means,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>        Appel_traits,</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>        Munoz_traits,</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>        Freda_2017_traits,</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>        boric_acid_line_means,</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>        Dean_traits,</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>        Duneau_traits,</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>        Palmer_traits,</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>        Teets_traits,</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>        Everman_traits,</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>        Freda_2019_traits,</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>        Pitchers_traits,</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>        Rohde_traits,</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>        microbiome_line_means,</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>        E.faecalis_line_means,</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>        Zhou_line_means,</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>        Zwoinska_traits,</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>        aggression_m_line_means,</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>        Watanabe_traits)</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(estimated_line_means, <span class="st">&quot;data/derived/dgrp_phenos_calculated_from_raw_data.csv&quot;</span>)</span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] broom_0.7.11    tidybayes_3.0.2 brms_2.16.3     Rcpp_1.0.4.6   
 [5] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.7     purrr_0.3.4    
 [9] readr_2.1.2     tidyr_1.1.4     tibble_3.1.6    ggplot2_3.3.6  
[13] tidyverse_1.3.2 Matrix_1.2-18   workflowr_1.7.0

loaded via a namespace (and not attached):
  [1] readxl_1.3.1         backports_1.1.7      plyr_1.8.6          
  [4] igraph_1.2.5         svUnit_1.0.3         splines_4.0.3       
  [7] crosstalk_1.1.0.1    TH.data_1.0-10       rstantools_2.1.1    
 [10] inline_0.3.15        digest_0.6.29        htmltools_0.5.2     
 [13] rsconnect_0.8.16     fansi_0.4.1          magrittr_2.0.1      
 [16] checkmate_2.0.0      googlesheets4_1.0.0  tzdb_0.1.2          
 [19] modelr_0.1.8         RcppParallel_5.0.1   matrixStats_0.56.0  
 [22] vroom_1.5.7          xts_0.12-0           sandwich_2.5-1      
 [25] prettyunits_1.1.1    colorspace_2.0-3     rvest_1.0.2         
 [28] ggdist_3.0.1         haven_2.5.0          xfun_0.31           
 [31] callr_3.7.0          crayon_1.4.2         jsonlite_1.8.0      
 [34] lme4_1.1-23          survival_3.2-7       zoo_1.8-10          
 [37] glue_1.4.2           gtable_0.3.0         gargle_1.2.0        
 [40] emmeans_1.4.7        V8_3.4.0             distributional_0.3.0
 [43] pkgbuild_1.0.8       rstan_2.21.2         abind_1.4-5         
 [46] scales_1.2.0         mvtnorm_1.1-0        DBI_1.1.0           
 [49] miniUI_0.1.1.1       xtable_1.8-4         bit_1.1-15.2        
 [52] stats4_4.0.3         StanHeaders_2.21.0-3 DT_0.13             
 [55] htmlwidgets_1.5.4    httr_1.4.2           threejs_0.3.3       
 [58] arrayhelpers_1.1-0   posterior_1.2.0      ellipsis_0.3.2      
 [61] pkgconfig_2.0.3      loo_2.3.1            farver_2.0.3        
 [64] dbplyr_2.1.1         utf8_1.1.4           tidyselect_1.1.0    
 [67] rlang_1.0.2          reshape2_1.4.4       later_1.0.0         
 [70] munsell_0.5.0        cellranger_1.1.0     tools_4.0.3         
 [73] cli_3.1.0            generics_0.0.2       ggridges_0.5.2      
 [76] evaluate_0.15        fastmap_1.1.0        yaml_2.3.5          
 [79] bit64_0.9-7          processx_3.5.2       knitr_1.39          
 [82] fs_1.4.1             nlme_3.1-149         projpred_2.0.2      
 [85] whisker_0.4          mime_0.9             xml2_1.3.3          
 [88] compiler_4.0.3       bayesplot_1.7.2      shinythemes_1.1.2   
 [91] rstudioapi_0.13      gamm4_0.2-6          curl_4.3            
 [94] reprex_2.0.1         statmod_1.4.34       stringi_1.5.3       
 [97] ps_1.3.3             Brobdingnag_1.2-6    lattice_0.20-41     
[100] nloptr_1.2.2.1       markdown_1.1         shinyjs_1.1         
[103] tensorA_0.36.1       vctrs_0.3.8          pillar_1.6.4.9003   
[106] lifecycle_1.0.1      jquerylib_0.1.1      bridgesampling_1.0-0
[109] estimability_1.3     httpuv_1.5.3.1       R6_2.4.1            
[112] promises_1.1.0       gridExtra_2.3        codetools_0.2-16    
[115] boot_1.3-25          MASS_7.3-53          colourpicker_1.0    
[118] gtools_3.8.2         assertthat_0.2.1     rprojroot_1.3-2     
[121] withr_2.4.3          shinystan_2.5.0      multcomp_1.4-13     
[124] mgcv_1.8-33          parallel_4.0.3       hms_1.1.1           
[127] grid_4.0.3           minqa_1.2.4          coda_0.19-3         
[130] rmarkdown_2.11       googledrive_2.0.0    git2r_0.27.1        
[133] getPass_0.2-2        shiny_1.4.0.2        lubridate_1.8.0     
[136] base64enc_0.1-3      dygraphs_1.1.1.6    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
