<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Running GWAS on the DGRP phenotypes</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.0/jquery-3.5.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">About</a>
</li>
<li>
  <a href="How_to.html">Upload your data</a>
</li>
<li>
  <a href="Data_collation.html">Data collation</a>
</li>
<li>
  <a href="Trait_correlations.html">Calculating genetic trait correlations</a>
</li>
<li>
  <a href="SNP_heritabilities.html">Calculating 'SNP-heritability'</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Running GWAS on the DGRP phenotypes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220705code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220705code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220705)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtreeb3a5ebea4d492b0c30762066a635b042a3bfa4a3targetblankb3a5ebea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tomkeaney/DGRP/tree/b3a5ebea4d492b0c30762066a635b042a3bfa4a3" target="_blank">b3a5ebe</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtreeb3a5ebea4d492b0c30762066a635b042a3bfa4a3targetblankb3a5ebea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tomkeaney/DGRP/tree/b3a5ebea4d492b0c30762066a635b042a3bfa4a3" target="_blank">b3a5ebe</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/derived/.DS_Store
    Ignored:    data/input/.DS_Store
    Ignored:    fits/
    Ignored:    gwas_data/.DS_Store
    Ignored:    gwas_data/derived/.DS_Store

Untracked files:
    Untracked:  analysis/exporting GWAS stats for the app.R
    Untracked:  analysis/luke_cluster_ideas.R
    Untracked:  analysis/luke_weighted_beta_ideas.R
    Untracked:  data/derived/correlation_SEs.rds
    Untracked:  data/imputed_data.RData
    Untracked:  gwas_data/README.md
    Untracked:  gwas_data/derived/bivariate_greml_results.tsv
    Untracked:  gwas_data/derived/delete_me.hsq
    Untracked:  gwas_data/derived/delete_me.log
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines.bed
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines.bim
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines.fam
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines.log
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_LDpruned.bed
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_LDpruned.bim
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_LDpruned.fam
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_LDpruned.log
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_LDpruned.rds
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.bed
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.bim
    Untracked:  gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.fam
    Untracked:  gwas_data/derived/gcta_GRM.grm.N.bin
    Untracked:  gwas_data/derived/gcta_GRM.grm.bin
    Untracked:  gwas_data/derived/gcta_GRM.grm.id
    Untracked:  gwas_data/derived/gcta_GRM.log
    Untracked:  gwas_data/derived/gwas_results/
    Untracked:  gwas_data/derived/luke_phenotype.txt
    Untracked:  gwas_data/derived/manhattan_plots/
    Untracked:  gwas_data/derived/phenotype.txt
    Untracked:  gwas_data/derived/phenotype_pair.txt
    Untracked:  gwas_data/derived/plink.frq
    Untracked:  gwas_data/derived/plink.log
    Untracked:  gwas_data/derived/plink.prune.in
    Untracked:  gwas_data/derived/test_null.log
    Untracked:  gwas_data/input/
    Untracked:  iris.csv
    Untracked:  test.pdf
    Untracked:  test2.pdf
    Untracked:  test3.pdf

Unstaged changes:
    Modified:   analysis/Trait_correlations.Rmd
    Modified:   analysis/cross_phenotype_meta_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/run_GWAS.Rmd</code>) and HTML
(<code>docs/run_GWAS.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/b3a5ebea4d492b0c30762066a635b042a3bfa4a3/analysis/run_GWAS.Rmd" target="_blank">b3a5ebe</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-25
</td>
<td>
Tweak GWAS summary tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/976dd522c94b3026593f0252399cad5bd232c804/docs/run_GWAS.html" target="_blank">976dd52</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/397bafec698144c3a28f284bf81639f0c77e6ef8/analysis/run_GWAS.Rmd" target="_blank">397bafe</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
wflow_publish("analysis/run_GWAS.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/8cdef9e1fd4c3af9c2e4f3f9796b607fe929664f/docs/run_GWAS.html" target="_blank">8cdef9e</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/1bb072c5ad91cfa15fd64da4f580a99764aa8016/analysis/run_GWAS.Rmd" target="_blank">1bb072c</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Add variant summary table
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/53ccdbf55cf6655e9bcb983697af89560c1b55a3/docs/run_GWAS.html" target="_blank">53ccdbf</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/3689cb480ffb0f978a386ba9a5107ad0a2689485/analysis/run_GWAS.Rmd" target="_blank">3689cb4</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Adding code to make master file of all signif SNPs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/328e480580fa4faacf7f7c1c307c28920af19508/docs/run_GWAS.html" target="_blank">328e480</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/2592a7be2265bb8acdaf9c65a10af932453abad0/analysis/run_GWAS.Rmd" target="_blank">2592a7b</a>
</td>
<td>
lukeholman
</td>
<td>
2022-11-24
</td>
<td>
Add Manhattan plot maker
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/9540b370e7b416b40144b6d6b2942049f2b854a9/analysis/run_GWAS.Rmd" target="_blank">9540b37</a>
</td>
<td>
lukeholman
</td>
<td>
2022-10-14
</td>
<td>
Commiting data files etc
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bigsnpr) <span class="co"># to install:   devtools::install_github(&quot;privefl/bigsnpr&quot;)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dbplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DBI)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Database of variant and gene annotations</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>db <span class="ot">&lt;-</span> DBI<span class="sc">::</span><span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(),</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;~/Rprojects/fitnessGWAS/data/derived/annotations.sqlite3&quot;</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>plink <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">getwd</span>(), <span class="st">&quot;code/plink&quot;</span>) <span class="co"># I used plink v1.9 for MacOS, available here: https://www.cog-genomics.org/plink/</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">readr.show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a helper function to pass commands to the terminal</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`,</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># ensuring that the Terminal output will be printed in this knitr report.</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>run_command <span class="ot">&lt;-</span> <span class="cf">function</span>(shell_command, <span class="at">wd =</span> <span class="fu">getwd</span>(), <span class="at">path =</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">system</span>(<span class="fu">glue</span>(<span class="st">&quot;cd &quot;</span>, wd, path, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, shell_command), <span class="at">intern =</span> <span class="cn">TRUE</span>), <span class="at">sep =</span> <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>save_manhattan_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(gwas_results, save_path){</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>  manhattan_data <span class="ot">&lt;-</span> gwas_results <span class="sc">%&gt;%</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">position =</span> <span class="fu">str_split</span>(SNP, <span class="st">&quot;_&quot;</span>),</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>           <span class="at">chr =</span> <span class="fu">map_chr</span>(position, <span class="sc">~</span> .x[<span class="dv">1</span>]),</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">position =</span> <span class="fu">as.numeric</span>(<span class="fu">map_chr</span>(position, <span class="sc">~</span> .x[<span class="dv">2</span>])),</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">pval =</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span> <span class="fu">log10</span>(P)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(chr, position, pval)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>  max_pos <span class="ot">&lt;-</span> manhattan_data <span class="sc">%&gt;%</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">max_pos =</span> <span class="fu">max</span>(position), </span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">middle_pos =</span> (max_pos <span class="sc">-</span> <span class="fu">min</span>(position)) <span class="sc">/</span> <span class="dv">2</span>,</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>  max_pos<span class="sc">$</span>max_pos <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">cumsum</span>(max_pos<span class="sc">$</span>max_pos[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]))</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>  max_pos<span class="sc">$</span>label_pos <span class="ot">&lt;-</span> max_pos<span class="sc">$</span>max_pos <span class="sc">+</span> max_pos<span class="sc">$</span>middle_pos</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>  manhattan_data <span class="ot">&lt;-</span> manhattan_data <span class="sc">%&gt;%</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(max_pos, <span class="at">by =</span> <span class="st">&quot;chr&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">position =</span> position <span class="sc">+</span> max_pos)</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>  y_max <span class="ot">&lt;-</span> <span class="fu">max</span>(manhattan_data<span class="sc">$</span>pval) <span class="sc">*</span> <span class="fl">1.08</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> manhattan_data <span class="sc">%&gt;%</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(position, pval, <span class="at">group =</span> chr, <span class="at">fill =</span> chr, <span class="at">stroke =</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># geom_hline(yintercept = 5, linetype = 2, colour = &quot;grey20&quot;) +</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.9</span>, <span class="at">colour=</span><span class="st">&quot;grey20&quot;</span>, <span class="at">pch =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Paired&quot;</span>, <span class="at">name =</span> <span class="st">&quot;Chromosome&quot;</span>) <span class="sc">+</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> max_pos<span class="sc">$</span>label_pos, <span class="at">labels =</span> max_pos<span class="sc">$</span>chr) <span class="sc">+</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, y_max)) <span class="sc">+</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, log[<span class="dv">10</span>], <span class="st">&quot; p value&quot;</span>))) <span class="sc">+</span> </span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Chromosome and position&quot;</span>) <span class="sc">+</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>()) </span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="at">filename =</span> save_path, <span class="at">plot =</span> plot, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(plot)</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div id="load-trait-data-to-be-used-in-gwas" class="section level2">
<h2>Load trait data to be used in GWAS</h2>
<p>Here we load the phenotype data to be used for the GWAS analyses
below. This step depends upon the file
<code>data/derived/all.dgrp.phenos_scaled.csv</code>, and so updating
that file with newly-received data will cause the GWAS to be re-run.</p>
<p>The other dependencies for this script are the files
<code>gwas_data/input/dgrp2.[XX]</code> where <code>XX</code> is
<code>bed</code>, <code>bim</code> and <code>fam</code> (obtained from
the <a href="http://dgrp2.gnets.ncsu.edu/">Mackay lab website</a>)), or
(if you’d like to skip several hours of imputing missing genotypes using
Beagle) the files
<code>gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.[XX]</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/meta_data_for_all_traits.csv&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of phenotypic traits that were measured in at least 80 DGRP lines</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>traits_for_gwas <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span> </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at"># lines measured</span><span class="st">`</span> <span class="sc">&gt;=</span> <span class="dv">80</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Trait)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the line mean phenotypes for all those traits</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>line_mean_phenotypes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/all.dgrp.phenos_scaled.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Trait <span class="sc">%in%</span> traits_for_gwas) <span class="sc">%&gt;%</span> </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">paste</span>(<span class="st">&quot;line&quot;</span>, line, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span></code></pre></div>
</div>
<div id="perform-snp-quality-control-and-imputation"
class="section level2">
<h2>Perform SNP quality control and imputation</h2>
<p>Before running GWAS analyses, we first cleaned up the DGRP’s
.bed/.bim/.fam files (available from the <a
href="http://dgrp2.gnets.ncsu.edu/">Mackay lab website</a>) as
follows:</p>
<ol style="list-style-type: decimal">
<li>Remove any SNPs for which genotypes are missing for &gt;10% of the
DGRP lines. We then use the software <a
href="https://faculty.washington.edu/browning/beagle/beagle.html">Beagle</a>
to impute the remaining missing genotypes.</li>
<li>Remove SNPs with a minor allele frequency of less than 5%.</li>
</ol>
<p>Note that in the PLINK-formatted genotype files, lines fixed for the
major allele are coded as 2, and lines fixed for the minor allele as 0.
This means that in the association tests we calculate below,
<em>negative</em> effect sizes mean that the minor allele is associated
with lower trait values, while <em>positive</em> effect sizes means that
the minor allele is associated with higher trait values.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>perform_SNP_QC_and_imputation <span class="ot">&lt;-</span> <span class="cf">function</span>(phenotypes){</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  beagle <span class="ot">&lt;-</span> bigsnpr<span class="sc">::</span><span class="fu">download_beagle</span>()</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use Plink to clean and subset the DGRP&#39;s SNP data as follows:</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only keep SNPs for which at least 90% of DGRP lines were successfully genotyped (--geno 0.1)</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only keep SNPs with a minor allele frequency of 0.05 or higher (--maf 0.05)</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, write the processed BIM/BED/FAM files to the data/derived directory</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_command</span>(<span class="fu">glue</span>(<span class="st">&quot;{plink} --bfile dgrp2&quot;</span>,</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --geno 0.1 --maf 0.05 --allow-no-sex&quot;</span>, </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --make-bed --out ../derived/dgrp2_QC_all_lines&quot;</span>), <span class="at">path =</span> <span class="st">&quot;/gwas_data/input/&quot;</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use the shell command &#39;sed&#39; to remove underscores from the DGRP line names in the .fam file (e.g. &#39;line_120&#39; becomes &#39;line120&#39;)</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Otherwise, these underscores cause trouble when we need to convert from PLINK to vcf format (vcf format uses underscore as a separator)</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) <span class="fu">run_command</span>(<span class="st">&quot;sed -i &#39;&#39; &#39;s/_//&#39; dgrp2_QC_all_lines.fam&quot;</span>, <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived/&quot;</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Now impute the missing genotypes using Beagle</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># This part uses the data for the full DGRP panel of &gt;200 lines, to infer missing genotypes as accurately as possible. </span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># This step uses a lot of memory (I set to 28MB max, and it used 26.5GB), but maybe it can also run on a less powerful computer?</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The bigsnpr package provides a helpful wrapper for Beagle called snp_beagleImpute(): it translates to a VCF file and back again using PLINK</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">snp_beagleImpute</span>(beagle, plink, </span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>                   <span class="at">bedfile.in =</span> <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines.bed&quot;</span>, </span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">bedfile.out =</span> <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.bed&quot;</span>,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncores =</span> <span class="dv">7</span>, </span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>                   <span class="at">memory.max =</span> <span class="dv">20</span>)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># assign a sex of &#39;female&#39; to all the DGRP lines (Beagle removes the sex, and it seems PLINK needs individuals to have a sex)</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_command</span>(<span class="st">&quot;sed -i &#39;&#39; &#39;s/    0   0   0/  0   0   2/&#39; dgrp2_QC_all_lines_imputed.fam&quot;</span>, <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived/&quot;</span>)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Re-write the .bed file, to make sure the MAF and genotyping thresholds are correctly assigned post-Beagle</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_command</span>(<span class="fu">glue</span>(<span class="st">&quot;{plink} --bfile dgrp2_QC_all_lines_imputed&quot;</span>,</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --geno 0.1 --maf 0.05&quot;</span>, </span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --make-bed --out dgrp2_QC_all_lines_imputed_correct&quot;</span>), <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived/&quot;</span>)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use PLINK to get the allele IDs and calculate the MAFs across the whole DGRP, for all SNPs that survived QC</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The file created is called data/derived/plink.frq</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_command</span>(<span class="st">&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct --freq&quot;</span>, <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived&quot;</span>)</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Clean up:</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlink</span>(<span class="fu">c</span>(<span class="st">&quot;gwas_data/derived/plink.log&quot;</span>,</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.bed&quot;</span>,</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.bim&quot;</span>,</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.fam&quot;</span>,</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.log&quot;</span>,</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.log&quot;</span>))</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co"># If the imputation is not already done, create the following 3 files of imputed genotype data: </span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a><span class="co"># dgrp2_QC_all_lines_imputed_correct.bed/bim/fam</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.bed&quot;</span>)) {</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">perform_SNP_QC_and_imputation</span>(<span class="at">phenotypes =</span> predicted_line_means)</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a><span class="co"># These are 205 genotyped DGRP lines. Double-check for incorrect line names in the phenotype data, and remove any</span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>genotyped_lines <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.fam&quot;</span>)[,<span class="dv">1</span>]</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>line_mean_phenotypes <span class="ot">&lt;-</span> line_mean_phenotypes <span class="sc">%&gt;%</span> </span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(line <span class="sc">%in%</span> genotyped_lines)</span></code></pre></div>
</div>
<div id="create-a-reduced-list-of-ld-pruned-snps-with-plink"
class="section level2">
<h2>Create a reduced list of LD-pruned SNPs with PLINK</h2>
<p>To keep the computation time and memory usage manageable, we do not
save the effect sizes for every variant (SNPs and indels) that passed
the MAF and missingness quality control step above (i.e. 1,646,652
variants), but rather we save the effect sizes for a subset of variants
that were approximately in linkage disequilibrium. We identified this
LD-pruned set of variants using the PLINK arguments
<code>--indep-pairwise 100 10 0.2</code>, i.e. pruning within 100kB
sliding windows, sliding 10 variants along with each step, and allowing
a maximum pairwise <span class="math inline">\(r^2\)</span> threshold of
0.2 between loci. With these parameters, 1,420,071 variants were
removed, leaving 226,581 for downstream analysis. Subsequent inspection
of the Manhattan plots suggests that this method removes most (but not
all) variants that are in complete or high LD.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># indep-pairwise arguments are: </span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 100kB window size, </span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># variant count to shift the window by 10 variants at the end of each step, </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># pairwise r^2 threshold of 0.2</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">run_command</span>(<span class="fu">glue</span>(<span class="st">&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct&quot;</span>,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot; --indep-pairwise 100 10 0.2&quot;</span>), <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived/&quot;</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="st">&quot;gwas_data/derived/plink.prune.out&quot;</span>)</span></code></pre></div>
</div>
<div id="run-all-the-gwas" class="section level2">
<h2>Run all the GWAS</h2>
<p>Note: because PLINK defines the minor allele as the alt allele (so,
lines fixed for the minor allele are scored as genotype: 2, and those
with the major allele as genotype: 0), a <em>positive</em> effect size
in these association tests means the <em>minor</em> allele is associated
with a <em>higher</em> value of the trait in question.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gwas_one_trait <span class="ot">&lt;-</span> <span class="cf">function</span>(focal_phenotype){</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  focal_data <span class="ot">&lt;-</span> line_mean_phenotypes <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Trait <span class="sc">==</span> focal_phenotype) <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">spread</span>(Trait, trait_value) </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(focal_data)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;focal_pheno&quot;</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Double check there are no missing values (shouldn&#39;t be any)</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  focal_data <span class="ot">&lt;-</span> focal_data <span class="sc">%&gt;%</span> </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(line) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(focal_pheno))</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make a list of the lines in our sample and save as a text file for passing to PLINK</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  lines_to_keep <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, focal_data<span class="sc">$</span>line) <span class="sc">%&gt;%</span> <span class="fu">cbind</span>(.,.)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(lines_to_keep, </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="st">&quot;gwas_data/derived/lines_to_keep.txt&quot;</span>, </span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Now cull the PLINK files to just the lines that we measured, and re-apply the </span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># MAF cut-off of 0.05 for the new smaller sample of DGRP lines</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note that this uses all the variants, not just LD-pruned set (via the -bfile argument)</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_command</span>(<span class="fu">glue</span>(<span class="st">&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct&quot;</span>,  </span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --keep-allele-order&quot;</span>, </span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --keep lines_to_keep.txt --geno 0.1 --maf 0.05&quot;</span>, </span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot; --make-bed --out dgrp2_QC_focal_lines&quot;</span>), </span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived/&quot;</span>)</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define a function to add our phenotype data to a .fam file, which is </span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># needed for GWAS analysis and to make sure PLINK includes these samples</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The &#39;phenotypes&#39; data frame needs to have a column called &#39;line&#39;</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  add_phenotypes_to_fam <span class="ot">&lt;-</span> <span class="cf">function</span>(filepath, focal_data){</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">read_delim</span>(filepath, <span class="at">col_names =</span> <span class="cn">FALSE</span>, <span class="at">delim =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(X1, X2, X3, X4, X5) <span class="sc">%&gt;%</span> <span class="co"># Get all the non-phenotype columns</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(focal_data, </span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>                <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;X1&quot;</span> <span class="ot">=</span> <span class="st">&quot;line&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write.table</span>(<span class="at">file =</span> <span class="st">&quot;gwas_data/derived/dgrp2_QC_focal_lines_NEW.fam&quot;</span>, </span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>                  <span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>                  <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlink</span>(<span class="st">&quot;gwas_data/derived/dgrp2_QC_focal_lines.fam&quot;</span>)</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">file.rename</span>(<span class="st">&quot;gwas_data/derived/dgrp2_QC_focal_lines_NEW.fam&quot;</span>, </span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;gwas_data/derived/dgrp2_QC_focal_lines.fam&quot;</span>)</span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_phenotypes_to_fam</span>(<span class="st">&quot;gwas_data/derived/dgrp2_QC_focal_lines.fam&quot;</span>, focal_data)</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># # Write a file with the line and phenotype data called phenotype.txt, for gcta64</span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pheno_data &lt;- focal_data %&gt;% </span></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   mutate(line_copy = line) %&gt;% </span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   select(line, line_copy, focal_pheno) %&gt;% as.matrix() </span></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pheno_data %&gt;% </span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   write.table(row.names = FALSE, col.names = FALSE, </span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#               file = &quot;gwas_data/derived/phenotype.txt&quot;, quote = FALSE)</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run mixed-model GWAS (in practice, the relatedness between almost all pairs of lines </span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># is sufficiently low that PLINK always instead chooses to run a linear model)</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(focal_phenotype)</span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_command</span>(<span class="st">&quot;{plink} --bfile dgrp2_QC_focal_lines  --assoc --maf 0.05 --out gwas_results/new&quot;</span>, </span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>              <span class="at">path =</span> <span class="st">&quot;/gwas_data/derived&quot;</span>)</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>  gwas_results <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;gwas_data/derived/gwas_results/new.qassoc&quot;</span>, </span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>                             <span class="at">header =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SNP, BETA, SE, P)</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save a Manhattan plot for this trait, using the all variants (not just the LD subset):</span></span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">save_manhattan_plot</span>(gwas_results,</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">glue</span>(<span class="st">&quot;gwas_data/derived/manhattan_plots/{focal_phenotype}.jpeg&quot;</span>))</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If the trait has a &quot;/&quot; in the name it cannot work as a file name. Change the &quot;/&quot; to &quot;_&quot;</span></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>  <span class="co"># focal_phenotype &lt;- str_replace_all(focal_phenotype, &quot;[/]&quot;, &quot;_&quot;)</span></span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save a file containing all SNPs with P &lt; 1e-5:</span></span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>  gwas_results <span class="sc">%&gt;%</span> </span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(P <span class="sc">&lt;</span> <span class="fl">1e-05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_tsv</span>(<span class="fu">glue</span>(<span class="st">&quot;gwas_data/derived/gwas_results/{focal_phenotype}_significant_SNPs.tsv.gz&quot;</span>))</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Rename and compress the GWAS summary stats file </span></span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The filter step means that only variants in the LD-pruned subset get saved to disk.</span></span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>  gwas_results <span class="sc">%&gt;%</span> </span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(SNP <span class="sc">%in%</span> (<span class="fu">pull</span>(<span class="fu">read_tsv</span>(<span class="st">&quot;gwas_data/derived/plink.prune.in&quot;</span>, <span class="at">col_names =</span> <span class="st">&quot;SNP&quot;</span>), SNP))) <span class="sc">%&gt;%</span> </span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_tsv</span>(<span class="fu">glue</span>(<span class="st">&quot;gwas_data/derived/gwas_results/{focal_phenotype}.tsv.gz&quot;</span>))</span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlink</span>(<span class="st">&quot;gwas_data/derived/gwas_results/new.qassoc&quot;</span>)</span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Rename the plink log file</span></span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">file.rename</span>(<span class="st">&quot;gwas_data/derived/gwas_results/new.log&quot;</span>,</span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>              <span class="fu">glue</span>(<span class="st">&quot;gwas_data/derived/gwas_results/{focal_phenotype}_log.txt&quot;</span>))</span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a><span class="co"># List the files in the gwas_results directory where association test results get stored:</span></span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;gwas_data/derived/gwas_results&quot;</span>, </span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pattern =</span> <span class="st">&quot;tsv.gz&quot;</span>, </span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a list of the phenotypic traits that have already been done:</span></span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>already_run_traits <span class="ot">&lt;-</span> <span class="fu">str_split</span>(files, <span class="st">&quot;/&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map_chr</span>(<span class="sc">~</span> .x[<span class="dv">4</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_remove_all</span>(<span class="st">&quot;.tsv.gz&quot;</span>)</span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a><span class="co"># List the phenotypic traits that have not yet been GWAS&#39;ed</span></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>traits_to_run_for_gwas <span class="ot">&lt;-</span> traits_for_gwas[</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>  <span class="sc">!</span>(traits_for_gwas <span class="sc">%in%</span> already_run_traits)]</span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a><span class="co"># run all the remaining gwas, one-by-one</span></span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(traits_to_run_for_gwas, gwas_one_trait)</span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a><span class="co"># Delete files no longer needed</span></span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">list.files</span>(<span class="st">&quot;gwas_data/derived&quot;</span>, </span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pattern =</span> <span class="st">&quot;focal_lines&quot;</span>, <span class="at">full.names =</span> T))</span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">c</span>(<span class="st">&quot;gwas_data/derived/lines_to_keep.txt&quot;</span>))</span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">c</span>(<span class="st">&quot;gwas_data/derived/gwas_results/new.qassoc&quot;</span>, <span class="st">&quot;gwas_data/derived/gwas_results/new.log&quot;</span>))</span></code></pre></div>
</div>
<div id="details-about-the-significant-variants" class="section level2">
<h2>Details about the significant variants</h2>
<div
id="create-a-table-of-information-about-all-the-significant-variants"
class="section level3">
<h3>Create a table of information about all the significant
variants</h3>
<p>The code below creates a large file (&gt;12MB) containing all the
significant variants (defined as <span class="math inline">\(P &lt;
10^{-5}\)</span>) for all the traits, alongside some annotations for
each variant (e.g. the gene associated with the variant, the site class
of the variant, and its alleles and minor allele frequencies) plus the
summary statistics from the GWAS (<code>BETA</code> is the effect size
of the association, where <em>negative</em> effect sizes mean that the
minor allele is associated with lower trait values, while
<em>positive</em> effect sizes means that the minor allele is associated
with higher trait values, SE is the standard error, and P is the
p-value). This information is saved in the file
<code>gwas_data/derived/all_traits_significant_SNPs.tsv</code>, which is
accessible on the Github repository.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(traits_to_run_for_gwas) <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  top_hits_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;gwas_data/derived/gwas_results&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">pattern =</span> <span class="st">&quot;significant_SNPs&quot;</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  add_annotation <span class="ot">&lt;-</span> <span class="cf">function</span>(filepath){</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    focal <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filepath) <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(P)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>    snp_order <span class="ot">&lt;-</span> focal<span class="sc">$</span>SNP</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>    trait <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(filepath, <span class="st">&quot;gwas_data[/]derived[/]gwas_results[/]&quot;</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    trait <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(trait, <span class="st">&quot;_significant_SNPs[.]tsv[.]gz&quot;</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(trait)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tbl</span>(db, <span class="st">&quot;variants&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(SNP <span class="sc">%in%</span> <span class="sc">!!</span>focal<span class="sc">$</span>SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>chr, <span class="sc">-</span>position) <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">collect</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(focal, <span class="at">by =</span> <span class="st">&quot;SNP&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">log10_P =</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span> <span class="fu">log10</span>(P),</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">SNP =</span> <span class="fu">factor</span>(SNP, snp_order)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rename_all</span>(<span class="sc">~</span> <span class="fu">str_replace_all</span>(.x, <span class="st">&quot;[.]&quot;</span>, <span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">trait =</span> trait, <span class="at">.before =</span> <span class="dv">1</span>)</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map_df</span>(top_hits_files, add_annotation)  <span class="sc">%&gt;%</span> </span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">tbl</span>(db, <span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(FBID, gene_name) <span class="sc">%&gt;%</span> <span class="fu">collect</span>(), <span class="at">by =</span> <span class="st">&quot;FBID&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(trait, SNP, FBID, gene_name, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene_name =</span> <span class="fu">replace</span>(gene_name, <span class="fu">is.na</span>(FBID), <span class="cn">NA</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_tsv</span>(<span class="st">&quot;gwas_data/derived/all_traits_significant_SNPs.tsv&quot;</span>)</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="various-summary-statistics-about-the-significant-variants"
class="section level3">
<h3>Various summary statistics about the significant variants</h3>
<div
id="version-of-the-table-that-is-correct-with-the-originally-published-dataset"
class="section level4">
<h4>Version of the table that is correct with the originally-published
dataset:</h4>
<p>The table shows various counts about the significant variants, as
well as the mean and standard error MAF (minor allele frequency) for all
the significant variants as well as the total set of variants that was
used in the GWAS (i.e. all those with minor allele frequency &gt;5% and
missingness &lt; 10%). Given that the choice of significance threshold
is arbitrary, we use two different definitions of statistical
significance: <span class="math inline">\(P &lt; 10^{-5}\)</span> (which
is used in many studies of the DGRP) and <span class="math inline">\(P
&lt; 10^{-8}\)</span> (which is commonly used in human GWAS).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>all_sig_snps <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;gwas_data/derived/all_traits_significant_SNPs.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta_data, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;trait&quot;</span> <span class="ot">=</span> <span class="st">&quot;Trait&quot;</span>))</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>make_full_tally_table <span class="ot">&lt;-</span> <span class="cf">function</span>(all_sig_snps){</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count number of GWAS that were run</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  gwas_results_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;gwas_data/derived/gwas_results/&quot;</span>)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  gwas_results_files <span class="ot">&lt;-</span> gwas_results_files[<span class="sc">!</span><span class="fu">str_detect</span>(gwas_results_files, <span class="st">&quot;_significant_SNPs&quot;</span>)]</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  gwas_results_files <span class="ot">&lt;-</span> gwas_results_files[<span class="sc">!</span><span class="fu">str_detect</span>(gwas_results_files, <span class="st">&quot;log[.]txt&quot;</span>)]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  num_gwas_run <span class="ot">&lt;-</span> <span class="fu">length</span>(gwas_results_files)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count number of unique studies that provided data for GWAS</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  num_unique_studies_used_for_gwas <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span> </span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Trait <span class="sc">%in%</span> <span class="fu">str_remove_all</span>(gwas_results_files, <span class="st">&quot;[.]tsv[.]gz&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Reference) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># A function to randomly select one trait per study, from among the traits measured in at least 80 DGRP lines</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  get_independent_traits <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(<span class="dv">1</span>) <span class="co"># Set the seed for reproducibility</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/meta_data_for_all_traits.csv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="st">`</span><span class="at"># lines measured</span><span class="st">`</span> <span class="sc">&gt;=</span> <span class="dv">80</span>) <span class="sc">%&gt;%</span> <span class="co"># Only analyse traits that measured at least 80 lines</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Reference) <span class="sc">%&gt;%</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">sample_n</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co"># Randomly sample one phenotype per study</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(Trait) <span class="sc">%&gt;%</span> <span class="fu">sort</span>() </span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  mean_and_SE <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    mean_x <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">mean</span>(x), <span class="dv">2</span>), <span class="at">nsmall =</span> <span class="dv">2</span>)</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    se_x <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">round</span>(<span class="fu">sd</span>(x) <span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">length</span>(x)), <span class="dv">3</span>), <span class="at">nsmall =</span> <span class="dv">3</span>)</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(se_x <span class="sc">==</span> <span class="st">&quot;0.000&quot;</span>) se_x <span class="ot">&lt;-</span> <span class="st">&quot;&lt;0.001&quot;</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(mean_x, <span class="st">&quot; (&quot;</span>, se_x, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>  maf_all_snps_tests <span class="ot">&lt;-</span> <span class="fu">tbl</span>(db, <span class="st">&quot;variants&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(MAF <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(MAF) <span class="sc">%&gt;%</span> <span class="fu">mean_and_SE</span>()</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>  make_tally_table <span class="ot">&lt;-</span> <span class="cf">function</span>(all_sig_snps){</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>    prettify <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">format</span>(x, <span class="at">big.mark=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>    num_traits_affected <span class="ot">&lt;-</span> all_sig_snps <span class="sc">%&gt;%</span> </span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(SNP, trait) <span class="sc">%&gt;%</span> </span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) </span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>    num_traits_affected_indep <span class="ot">&lt;-</span> all_sig_snps <span class="sc">%&gt;%</span> </span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(trait <span class="sc">%in%</span> <span class="fu">get_independent_traits</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(SNP, trait) <span class="sc">%&gt;%</span> </span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) </span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>    num_snps_sig_in_2_studies <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">pull</span>(num_traits_affected, n) <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>    num_snps_sig_in_3_studies <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">pull</span>(num_traits_affected, n) <span class="sc">&gt;</span> <span class="dv">2</span>)</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>    num_snps_sig_in_2_studies_indep <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">pull</span>(num_traits_affected_indep, n) <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>    num_snps_sig_in_3_studies_indep <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">pull</span>(num_traits_affected_indep, n) <span class="sc">&gt;</span> <span class="dv">2</span>)</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>    num_snps_sig_in_2_studies <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">prettify</span>(num_snps_sig_in_2_studies), <span class="st">&quot; (&quot;</span>, num_snps_sig_in_2_studies_indep, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>    num_snps_sig_in_3_studies <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">prettify</span>(num_snps_sig_in_3_studies), <span class="st">&quot; (&quot;</span>, num_snps_sig_in_3_studies_indep, <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tibble</span>(</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of studies that provided data for GWA tests</span><span class="st">`</span> <span class="ot">=</span> num_unique_studies_used_for_gwas,</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of traits analysed using GWA tests</span><span class="st">`</span> <span class="ot">=</span> num_gwas_run,</span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of studies with at least one significant SNP</span><span class="st">`</span> <span class="ot">=</span> all_sig_snps <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(Reference) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>(),</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of phenotypic traits with at least one significant SNP</span><span class="st">`</span> <span class="ot">=</span> all_sig_snps <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(trait) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>(),</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of variants significantly associated with at least one phenotypic trait</span><span class="st">`</span> <span class="ot">=</span> all_sig_snps <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(SNP) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>() <span class="sc">%&gt;%</span> <span class="fu">prettify</span>(),</span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of variants significantly associated with two phenotypic traits (which were measured in different studies)</span><span class="st">`</span> <span class="ot">=</span> num_snps_sig_in_2_studies,</span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of variants significantly associated with more than two phenotypic traits (which were measured in different studies)</span><span class="st">`</span> <span class="ot">=</span> num_snps_sig_in_3_studies,</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">Number of genes close to one of the significant variants</span><span class="st">`</span> <span class="ot">=</span> all_sig_snps <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(FBID)) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(FBID) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>() <span class="sc">%&gt;%</span> <span class="fu">prettify</span>(),</span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">MAF of the significant variants (mean and SE)</span><span class="st">`</span> <span class="ot">=</span> all_sig_snps <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(SNP, MAF) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(MAF) <span class="sc">%&gt;%</span> <span class="fu">mean_and_SE</span>(),</span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">MAF of all variants included in the GWAS (mean and SE)</span><span class="st">`</span> <span class="ot">=</span> maf_all_snps_tests) <span class="sc">%&gt;%</span> </span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>      <span class="fu">gather</span>(Quantity, n)</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>    all_sig_snps <span class="sc">%&gt;%</span> </span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a>      <span class="fu">make_tally_table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rename</span>(<span class="st">`</span><span class="at">Using p &lt; 10-5</span><span class="st">`</span> <span class="ot">=</span> n),</span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>    all_sig_snps <span class="sc">%&gt;%</span> </span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(log10_P <span class="sc">&gt;=</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>      <span class="fu">make_tally_table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rename</span>(<span class="st">`</span><span class="at">Using p &lt; 10-8</span><span class="st">`</span> <span class="ot">=</span> n), </span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">&quot;Quantity&quot;</span>) </span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>all_sig_snps <span class="sc">%&gt;%</span> </span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>  <span class="co"># restrict to only papers in the original database, i.e. not those added post-publication:</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Reference <span class="sc">%in%</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/input/references_in_original_database.csv&quot;</span>)<span class="sc">$</span>Reference) <span class="sc">%&gt;%</span> </span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_full_tally_table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Quantity
</th>
<th style="text-align:left;">
Using p &lt; 10-5
</th>
<th style="text-align:left;">
Using p &lt; 10-8
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of studies that provided data for GWA tests
</td>
<td style="text-align:left;">
92
</td>
<td style="text-align:left;">
92
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of traits analysed using GWA tests
</td>
<td style="text-align:left;">
974
</td>
<td style="text-align:left;">
974
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of studies with at least one significant SNP
</td>
<td style="text-align:left;">
92
</td>
<td style="text-align:left;">
46
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of phenotypic traits with at least one significant SNP
</td>
<td style="text-align:left;">
974
</td>
<td style="text-align:left;">
183
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of variants significantly associated with at least one phenotypic
trait
</td>
<td style="text-align:left;">
62,644
</td>
<td style="text-align:left;">
3,766
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of variants significantly associated with two phenotypic traits
(which were measured in different studies)
</td>
<td style="text-align:left;">
9,291 (42)
</td>
<td style="text-align:left;">
684 (0)
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of variants significantly associated with more than two
phenotypic traits (which were measured in different studies)
</td>
<td style="text-align:left;">
5,670 (0)
</td>
<td style="text-align:left;">
300 (0)
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes close to one of the significant variants
</td>
<td style="text-align:left;">
10,522
</td>
<td style="text-align:left;">
1,186
</td>
</tr>
<tr>
<td style="text-align:left;">
MAF of the significant variants (mean and SE)
</td>
<td style="text-align:left;">
0.15 (&lt;0.001)
</td>
<td style="text-align:left;">
0.14 (0.001)
</td>
</tr>
<tr>
<td style="text-align:left;">
MAF of all variants included in the GWAS (mean and SE)
</td>
<td style="text-align:left;">
0.22 (&lt;0.001)
</td>
<td style="text-align:left;">
0.22 (&lt;0.001)
</td>
</tr>
</tbody>
</table>
</div>
<div
id="version-of-the-table-that-is-correct-with-the-current-post-publication-version-of-the-dataset"
class="section level4">
<h4>Version of the table that is correct with the current,
post-publication version of the dataset</h4>
<p>The figures in this table will be updated when new DGRP data is added
to the database.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>all_sig_snps <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># do *not* restrict to only papers in the original database, i.e. not those added post-publication:</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_full_tally_table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
