<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Estimating ‘SNP-heritability’</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.11.3/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.11.3/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="Data_collation.html">Data collation</a>
</li>
<li>
  <a href="Analysis.html">Calculating genetic trait correlations</a>
</li>
<li>
  <a href="SNP_heritabilities.html">Calculating 'SNP-heritability'</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Estimating ‘SNP-heritability’</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220705code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220705code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220705)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtreeed527563b89e7462962afe6b2cfce58d27e2d809targetblanked52756a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tomkeaney/DGRP/tree/ed527563b89e7462962afe6b2cfce58d27e2d809" target="_blank">ed52756</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPtreeed527563b89e7462962afe6b2cfce58d27e2d809targetblanked52756a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tomkeaney/DGRP/tree/ed527563b89e7462962afe6b2cfce58d27e2d809" target="_blank">ed52756</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/Data_collation_cache/
    Ignored:    analysis/index_cache/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/derived/.DS_Store

Untracked files:
    Untracked:  code/LICENSE
    Untracked:  code/gcta64
    Untracked:  code/plink
    Untracked:  code/prettify
    Untracked:  code/run_mashr.R
    Untracked:  code/toy.map
    Untracked:  code/toy.ped
    Untracked:  data/DGRP_gwas_bfiles/
    Untracked:  data/Literature_table.csv
    Untracked:  data/Reported_heritability.xlsx
    Untracked:  data/derived/all.dgrp.phenos_combined.csv
    Untracked:  data/derived/all.dgrp.phenos_unscaled.csv
    Untracked:  data/derived/predicted_line_means.csv
    Untracked:  data/derived/rg_female.csv
    Untracked:  data/derived/rg_male.csv
    Untracked:  data/derived/selected.dgrp.phenos_unscaled.csv
    Untracked:  data/derived/variance_components.csv
    Untracked:  data/gwas_data/
    Untracked:  data/input/
    Untracked:  fits/

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/SNP_heritabilities.Rmd</code>)
and HTML (<code>docs/SNP_heritabilities.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/ed527563b89e7462962afe6b2cfce58d27e2d809/analysis/SNP_heritabilities.Rmd" target="_blank">ed52756</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
final (lol) version for analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/2ac2cbcefab1fa1941cc6c9c0e95b159e06ce9fe/docs/SNP_heritabilities.html" target="_blank">2ac2cbc</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/006deee4c5fd8db5e0fc02301d9005cc0addb93d/analysis/SNP_heritabilities.Rmd" target="_blank">006deee</a>
</td>
<td>
ausevo
</td>
<td>
2022-09-06
</td>
<td>
final version for analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/f924e99a74781b05a8a36e8b7d8b16a83930da65/docs/SNP_heritabilities.html" target="_blank">f924e99</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP/8b03a2f9a539a3555ff033a2ad4269c72c8e389c/docs/SNP_heritabilities.html" target="_blank">8b03a2f</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP/blob/5133de98630e74431cae4005a4f55365b1d59c76/analysis/SNP_heritabilities.Rmd" target="_blank">5133de9</a>
</td>
<td>
ausevo
</td>
<td>
2022-07-05
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="prepare-for-gcta-analysis" class="section level1">
<h1>Prepare for <code>GCTA</code> analysis</h1>
<p><span class="math inline">\(~\)</span></p>
<div id="load-packages-and-build-helper-functions"
class="section level2">
<h2>Load packages and build helper functions</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we load the necessary packages to build this document and code
in ‘tidy’ style. We also set the file path so that R can access the
<code>gcta64</code> software. Finally, we build a helper function to
pass <code>GCTA</code> commands to the terminal. This function allows us
to code entirely within R-studio, which is our preference.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>gcta <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">getwd</span>(), <span class="st">&quot;code/gcta64&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">readr.show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># helper function to pass commands to the terminal</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`,</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># ensuring that the Terminal output will be printed in this knitr report.</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>run_command <span class="ot">&lt;-</span> <span class="cf">function</span>(shell_command, <span class="at">wd =</span> <span class="fu">getwd</span>(), <span class="at">path =</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">system</span>(<span class="fu">glue</span>(<span class="st">&quot;cd &quot;</span>, wd, path, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,shell_command), <span class="at">intern =</span> <span class="cn">TRUE</span>), <span class="at">sep =</span> <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="load-and-clean-phenotype-data" class="section level2">
<h2>Load and clean phenotype data</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we load in the unscaled phenotype data we collated <a
href="Data_collation.html">here</a>.</p>
<ul>
<li><p>We then trim the dataset to only include traits that are measured
across &gt;80 lines.</p></li>
<li><p>Two traits cause GCTA to call an error:
<code>paraquat.resistance.2021.f</code> &amp;
<code>starvation.resistance.high.carb.diet.m</code>. We simply do not
estimate heritability for these two traits. The error is <em>Error: the
information matrix is not invertible.</em></p></li>
</ul>
<p>This takes the number of traits from 1214 to 972.</p>
<p>We then fix the names of traits and DGRP lines to conform with
<code>GCTA</code> and <code>Plink</code> e.g. <em>/</em> is not an
acceptable file name, and line is required in front of the line number
e.g. <code>line21</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/all.dgrp.phenos_unscaled.csv&quot;</span>) </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>traits_for_analysis <span class="ot">&lt;-</span> </span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  traits <span class="sc">%&gt;%</span> </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">lines_measured =</span> <span class="fu">length</span>(<span class="fu">unique</span>(line))) <span class="sc">%&gt;%</span> </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(lines_measured <span class="sc">&gt;</span> <span class="dv">79</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Trait <span class="sc">!=</span> <span class="st">&quot;paraquat.resistance.2021.f&quot;</span> <span class="sc">&amp;</span> Trait <span class="sc">!=</span> <span class="st">&quot;starvation.resistance.high.carb.diet.m&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># these traits call a gcta error, we remove them</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Trait) <span class="co"># find traits with 80+ replicates</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># replace any slashes in the trait name, e.g. &quot;1/3-Methylhistidine.high.yeast.f&quot;, as this is not ok to use as a file name</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>traits_for_analysis <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(traits_for_analysis, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>) </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>line_mean_phenotypes <span class="ot">&lt;-</span> traits <span class="sc">%&gt;%</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">str_replace_all</span>(Trait, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span>  <span class="co"># change to match `traits_for_analysis` list</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Trait <span class="sc">%in%</span> traits_for_analysis) <span class="sc">%&gt;%</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(line, Trait, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Trait, trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>line), <span class="sc">~</span> <span class="fu">as.numeric</span>(<span class="fu">scale</span>(.x))) <span class="sc">%&gt;%</span>  <span class="co"># scale the traits</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">line =</span> <span class="fu">paste</span>(<span class="st">&quot;line&quot;</span>, line, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="co"># add &quot;line&quot; in front of the line number to fit with plink/gcta formatting</span></span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div
id="use-gcta-to-estimate-the-genetic-variances-and-heritabilities-for-each-trait"
class="section level1">
<h1>Use <code>GCTA</code> to estimate the genetic variances and
heritabilities for each trait</h1>
<p><span class="math inline">\(~\)</span></p>
<p>Here we estimate ‘SNP-heritability’ for each phenotype using a
Genomic relatedness matrix REstricted Maximum Likelihood (GREML)
approach. GREML uses a set of SNPs <span
class="math inline">\(S\)</span> that are known across a population of
individuals (or DGRP lines, in our case) to estimate the relatedness
between random individuals (lines) and compares that with their
phenotypic similarity. This allows us to estimate the variation in
phenotype that is explained by the known set of SNPs in our population.
We can then divide this variance by the total variation found for the
phenotype, giving the ‘SNP-heritability’, <span
class="math inline">\(h^2_{SNP}\)</span>. Note that SNP-heritability
should be smaller than other estimates of heritability such as broad- or
narrow-sense heritability, as it is limited to additive effects from
<strong>known</strong> SNPs (and unknown causal variants in close
linkage with these), while the latter two estimates encompass a
more-complete range of genetic effects.</p>
<p>We use the <code>GCTA</code> software to run this analysis.</p>
<div id="make-a-genomic-relatedness-matrix-for-all-the-lines"
class="section level2">
<h2>Make a genomic relatedness matrix for all the lines</h2>
<p>This uses the <code>GCTA</code> command <code>--make-grm</code> to
make a genomic relatedness matrix (GRM) for the DGRP SNP data. This
matrix is needed to run the GREML models below.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_command</span>(<span class="fu">glue</span>(<span class="st">&quot;{gcta} --bfile dgrp2_QC_all_lines&quot;</span>,</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot; --make-grm&quot;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot; --out gcta_GRM&quot;</span>), <span class="at">path =</span> <span class="st">&quot;/data/gwas_data/derived/&quot;</span>)</span></code></pre></div>
<pre><code>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version 1.93.2 beta Mac
* (C) 2010-present, Jian Yang, The University of Queensland
* Please report bugs to Jian Yang &lt;jian.yang.qt@gmail.com&gt;
*******************************************************************
Analysis started at 16:47:19 AEST on Tue Sep 06 2022.
Hostname: 6300D-135452-M.local

Options: 
 
--bfile dgrp2_QC_all_lines 
--make-grm 
--out gcta_GRM 

Note: GRM is computed using the SNPs on the autosome.
Reading PLINK FAM file from [dgrp2_QC_all_lines.fam]...
205 individuals to be included from FAM file.
205 individuals to be included. 0 males, 205 females, 0 unknown.
Reading PLINK BIM file from [dgrp2_QC_all_lines.bim]...
1648723 SNPs to be included from BIM file(s).
Computing the genetic relationship matrix (GRM) v2 ...
Subset 1/1, no. subject 1-205
  205 samples, 1648723 markers, 21115 GRM elements
IDs for the GRM file has been saved in the file [gcta_GRM.grm.id]
Computing GRM...
  100% finished in 5.0 sec
1648723 SNPs have been processed.
  Used 1648723 valid SNPs.
The GRM computation is completed.
Saving GRM...
GRM has been saved in the file [gcta_GRM.grm.bin]
Number of SNPs in each pair of individuals has been saved in the file [gcta_GRM.grm.N.bin]

Analysis finished at 16:47:26 AEST on Tue Sep 06 2022
Overall computational time: 7.87 sec.</code></pre>
</div>
<div id="run-univariate-greml-models-to-find-heritabilites"
class="section level2">
<h2>Run univariate GREML models to find heritabilites</h2>
<p>The below uses the <code>GCTA</code> command –reml to estimate the
genetic variance (V(G)), environmental variance (V(e)), their sum (Vp),
and the so-called “SNP heritability” (V(G)/Vp).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>do_univariate_greml <span class="ot">&lt;-</span> <span class="cf">function</span>(trait1){</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  focal_data <span class="ot">&lt;-</span> line_mean_phenotypes <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(line, <span class="sc">!!</span> trait1)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(focal_data)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;focal_phenotype&quot;</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write a file with the line and phenotype data called phenotype.txt, for gcta64</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  pheno_data <span class="ot">&lt;-</span> focal_data <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter(!(line %in% lines_to_prune)) %&gt;% </span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">line_copy =</span> line) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(line, line_copy, focal_phenotype) <span class="sc">%&gt;%</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.matrix</span>()</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  pheno_data <span class="sc">%&gt;%</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(<span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">file =</span> <span class="st">&quot;data/gwas_data/derived/phenotype.txt&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  console_output <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(<span class="fu">capture.output</span>(</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">run_command</span>(<span class="fu">glue</span>(<span class="st">&quot;{gcta}  --reml --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype.txt  --out delete_me&quot;</span>), </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">path =</span> <span class="st">&quot;/data/gwas_data/derived/&quot;</span>)))</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  errors <span class="ot">&lt;-</span> console_output[<span class="fu">str_detect</span>(console_output, <span class="st">&quot;[Ee]rror&quot;</span>)]</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(errors) <span class="sc">==</span> <span class="dv">0</span>) errors <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(errors) <span class="sc">&gt;</span> <span class="dv">1</span>) errors <span class="ot">&lt;-</span> <span class="fu">paste</span>(errors, <span class="at">sep =</span> <span class="st">&quot;; &quot;</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">read.delim</span>(<span class="st">&quot;data/gwas_data/derived/delete_me.hsq&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) , </span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">Source =</span> <span class="st">&quot;Errors&quot;</span>, <span class="at">Variance =</span> errors, <span class="at">SE =</span> <span class="cn">NA</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Trait =</span> trait1) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Trait, <span class="fu">everything</span>())</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Now run the model for every trait and create a large tibble using map_dfr</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;data/derived/variance_components.csv&quot;</span>)){</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>  variance_components <span class="ot">&lt;-</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map_dfr</span>(traits_for_analysis, do_univariate_greml) <span class="sc">%&gt;%</span> </span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Source <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;V(G)&quot;</span>, <span class="st">&quot;V(e)&quot;</span>, <span class="st">&quot;Vp&quot;</span>, <span class="st">&quot;V(G)/Vp&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Variance =</span> <span class="fu">as.numeric</span>(Variance)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">Parameter =</span> Source) <span class="sc">%&gt;%</span> </span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>()</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv</span>(variance_components, <span class="at">file =</span> <span class="st">&quot;data/derived/variance_components.csv&quot;</span>)</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> variance_components <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/derived/variance_components.csv&quot;</span>)</span></code></pre></div>
<p>For now we simply don’t run models for these traits, so there are no
variance estimates for these traits.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="wrangle-data-and-save" class="section level2">
<h2>Wrangle data and save</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>trait_info <span class="ot">&lt;-</span> traits <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">str_replace_all</span>(Trait, <span class="st">&quot;[/]&quot;</span>, <span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>trait_value) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait, Sex, <span class="st">`</span><span class="at">Trait guild</span><span class="st">`</span>, <span class="st">`</span><span class="at">Trait description</span><span class="st">`</span>, Reference)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>variance_components <span class="ot">&lt;-</span> <span class="fu">left_join</span>(variance_components, trait_info)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># wrangle so that each variance component is its own column</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>variance_components_wide <span class="ot">&lt;-</span> </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  variance_components <span class="sc">%&gt;%</span> </span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Parameter,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> <span class="fu">c</span>(Variance, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), round, <span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">V(G)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Variance_V(G)</span><span class="st">`</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">V(e)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Variance_V(e)</span><span class="st">`</span>,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Vp</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Variance_Vp</span><span class="st">`</span>,</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">V(G)/Vp</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Variance_V(G)/Vp</span><span class="st">`</span>)</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Below we remove the suffix that denotes the sex the trait was measured in</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>SNP_heritability <span class="ot">&lt;-</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  variance_components_wide <span class="sc">%&gt;%</span> </span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">case_when</span>(</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>           <span class="fu">str_ends</span>(Trait, <span class="st">&quot;.f&quot;</span>) <span class="sc">|</span> <span class="fu">str_ends</span>(Trait, <span class="st">&quot;.m&quot;</span>)  <span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">&#39;.{2}$&#39;</span>, <span class="st">&#39;&#39;</span>, Trait))) <span class="sc">%&gt;%</span> </span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait, Sex, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span>  <span class="co"># reorder the columns for the table %&gt;% </span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Trait)</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(SNP_heritability, <span class="at">file =</span> <span class="st">&quot;data/derived/SNP_heritability_cleaned.csv&quot;</span>)</span></code></pre></div>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="table-1" class="section level1">
<h1>Table 1</h1>
<p>Table 1. Phenotypic variance for the traits in our dataset (<span
class="math inline">\(V_p\)</span>), SNP variance (<span
class="math inline">\(V_{SNP}\)</span>), residual variance (<span
class="math inline">\(V_e\)</span>) and heritability (<span
class="math inline">\(h^2_{SNP}\)</span>).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to build HTML searchable tables</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>my_data_table <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">datatable</span>(</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    df, <span class="at">rownames=</span><span class="cn">FALSE</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">autoHideNavigation =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">extensions =</span> <span class="fu">c</span>(<span class="st">&quot;Scroller&quot;</span>,  <span class="st">&quot;Buttons&quot;</span>),</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">autoWidth =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">columnDefs =</span> <span class="fu">list</span>(<span class="fu">list</span>(<span class="at">width =</span> <span class="st">&#39;300px&#39;</span>, <span class="at">targets =</span> <span class="fu">c</span>(<span class="dv">4</span>))),</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">dom =</span> <span class="st">&#39;Bfrtip&#39;</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">deferRender=</span><span class="cn">TRUE</span>,</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">scrollX=</span><span class="cn">TRUE</span>, <span class="at">scrollY=</span><span class="dv">1000</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">scrollCollapse=</span><span class="cn">TRUE</span>,</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">buttons =</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="st">&#39;pageLength&#39;</span>, <span class="st">&#39;colvis&#39;</span>, <span class="st">&#39;csv&#39;</span>, <span class="fu">list</span>(</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">extend =</span> <span class="st">&#39;pdf&#39;</span>,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">pageSize =</span> <span class="st">&#39;A4&#39;</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">orientation =</span> <span class="st">&#39;landscape&#39;</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">filename =</span> <span class="st">&#39;Variance_table&#39;</span>)),</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">pageLength =</span> <span class="dv">1000</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="fu">my_data_table</span>(SNP_heritability)</span></code></pre></div>
<div id="htmlwidget-661d5ae1cbb6935a426f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-661d5ae1cbb6935a426f">{"x":{"filter":"none","vertical":false,"extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["1_3-Methylhistidine.high.yeast","1_3-Methylhistidine.low.yeast","1-Methyladenosine.high.yeast","1-Methyladenosine.low.yeast","1-Methylguanosine.high.yeast","1-Methylguanosine.low.yeast","13-HODE.high.yeast","13-HODE.low.yeast","2-Hydroxyglutarate.high.yeast","2-Hydroxyglutarate.low.yeast","2-Phosphoglyceric.acid.high.yeast","2-Phosphoglyceric.acid.low.yeast","2.3-butanedione.olfactory.attraction","2.3-butanedione.olfactory.attraction","2.3-butanedione.olfactory.latency.30.percentile","2.3-butanedione.olfactory.latency.30.percentile","2.3-butanedione.olfactory.latency.90.percentile","2.3-butanedione.olfactory.latency.90.percentile","2.3-butanedione.olfactory.response.intensity","2.3-butanedione.olfactory.response.intensity","2.3-butanedione.olfactory.strength.attraction","2.3-butanedione.olfactory.strength.attraction","3-Hydroxybutyric.acid.high.yeast","3-Hydroxybutyric.acid.low.yeast","3-Hydroxykynurenine.high.yeast","3-Hydroxykynurenine.low.yeast","4-Pyridoxic.acid.high.yeast","4-Pyridoxic.acid.low.yeast","5-Aminovalerate.high.yeast","5-Aminovalerate.low.yeast","abdomen.length.18C","abdomen.length.28C","abdomen.length.thermal.plasticity","acclimated.acute.cold.exposure.resistance.18C","acclimated.acute.cold.exposure.resistance.18C","acclimated.acute.cold.exposure.resistance.25C","acclimated.acute.cold.exposure.resistance.25C","acetoacetate.high.yeast","acetoacetate.low.yeast","Acetobacter.aceti","Acetobacter.aceti","Acetobacter.cibinongensis","Acetobacter.cibinongensis","Acetobacter.ghanensis","Acetobacter.ghanensis","Acetobacter.indonesiensis","Acetobacter.indonesiensis","Acetobacter.malorum","Acetobacter.malorum","Acetobacter.nitrogenifigens","Acetobacter.nitrogenifigens","Acetobacter.okinawensis","Acetobacter.okinawensis","Acetobacter.orientalis","Acetobacter.orientalis","Acetobacter.papayae","Acetobacter.papayae","Acetobacter.pasteurianus","Acetobacter.pasteurianus","Acetobacter.persici","Acetobacter.persici","Acetobacter.pomorum","Acetobacter.pomorum","Acetobacter.senegalensis","Acetobacter.senegalensis","Acetobacter.syzygii","Acetobacter.syzygii","Acetobacter.tropicalis","Acetobacter.tropicalis","acetylcarnitine.high.yeast","acetylcarnitine.low.yeast","acetylcholine.high.yeast","acetylcholine.low.yeast","Acidiphilium.angustum","Acidiphilium.angustum","Acidomonas.methanolica","Acidomonas.methanolica","Acidovorax.sp.","Acidovorax.sp.","Acidovorax.temperans","Acidovorax.temperans","Acinetobacter.gyllenbergii","Acinetobacter.gyllenbergii","Acinetobacter.johnsonii","Acinetobacter.johnsonii","Acinetobacter.sp.","Acinetobacter.sp.","activity","activity.MPH","acute.cold.exposure.resistance.18C.-6","acute.cold.exposure.resistance.18C.-6","acute.cold.exposure.resistance.18C.-8","acute.cold.exposure.resistance.18C.-8","acute.cold.exposure.resistance.25C","acute.cold.exposure.resistance.25C","adenine.high.yeast","adenine.low.yeast","adenosine.high.yeast","adenosine.low.yeast","ADP.high.yeast","ADP.low.yeast","aggression.2015","agmatine.high.yeast","agmatine.low.yeast","Agrobacterium.tumefaciens","Agrobacterium.tumefaciens","alanine.high.yeast","alanine.low.yeast","allantoin.high.yeast","allantoin.low.yeast","alpha-Ketoglutaric.acid.high.yeast","alpha-Ketoglutaric.acid.low.yeast","aminoisobutyrate.high.yeast","aminoisobutyrate.low.yeast","AMP.high.yeast","AMP.low.yeast","anoxia.tolerance.adult","anoxia.tolerance.adult","anthranilate.high.yeast","anthranilate.low.yeast","approach.female","arachidonate.high.yeast","arachidonate.low.yeast","arginine.high.yeast","arginine.low.yeast","Asaia.prunellae","Asaia.prunellae","Asaia.sp.","Asaia.sp.","asparagine.high.yeast","asparagine.low.yeast","aspartic.acid.high.yeast","aspartic.acid.low.yeast","Aspergillus.nidulans","Aspergillus.nidulans","Aspergillus.terreus,Malassezia globosa","Aspergillus.terreus,Malassezia globosa","attempted.copulation.frequency","Azospirillum.brasilense","Azospirillum.brasilense","Azospirillum.thiophilum","Azospirillum.thiophilum","basal.activity","basal.activity","betaine.high.yeast","betaine.low.yeast","biotin.high.yeast","biotin.low.yeast","boric.acid.resistance","Borrelia.coriaceae","Borrelia.coriaceae","cadaverine.high.yeast","cadaverine.low.yeast","caffeine.resistance","Campylobacter.mucosalis","Campylobacter.mucosalis","Cardinium.endosymbiont","Cardinium.endosymbiont","carnitine.high.yeast","carnitine.low.yeast","Caulobacter.henricii","Caulobacter.henricii","cGMP.high.yeast","cGMP.low.yeast","Chaetomium.globosum","Chaetomium.globosum","CHC.11 &amp; 9-Me-C23","CHC.13- &amp; 11-Me-C25","CHC.2-Me-C22","CHC.2-Me-C22","CHC.2-Me-C24","CHC.2-Me-C24","CHC.2-Me-C25","CHC.2-Me-C25","CHC.2-Me-C26","CHC.2-Me-C27","CHC.2-Me-C28","CHC.2-Me-C28 &amp; 7,11-C29:2","CHC.2-Me-C29","CHC.2-Me-C29 &amp; 8,12-C30:2","CHC.2-Me-C30","CHC.3-Me-C23","CHC.3-Me-C25","CHC.3-Me-C25 &amp; 9-C26:1","CHC.3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1","CHC.5-C23:1","CHC.5-C23:1","CHC.5-C24:1","CHC.5-C24:1","CHC.5-C25:1","CHC.5-C25:1","CHC.5-C27:1","CHC.5-Me-C25","CHC.5,9-C25:2 &amp; 9-C25:1","CHC.5,9-C27:2 &amp; 9-C27:1","CHC.6-C23:1","CHC.6-C23:1","CHC.6-C24:1","CHC.6-C24:1","CHC.6-C26:1 &amp; i","CHC.6,10-C25:2","CHC.6,10-C25:2","CHC.6,10-C26:2","CHC.7-C23:1","CHC.7-C23:1","CHC.7-C24:1","CHC.7-C25:1","CHC.7-C25:1","CHC.7-C26:1","CHC.7-C26:1","CHC.7-C27:1","CHC.7-C27:1","CHC.7-C29:1","CHC.7-C29:1","CHC.7,11-C23:2","CHC.7,11-C24:2","CHC.7,11-C25:2","CHC.7,11-C26:2","CHC.7,11-C27:2 &amp; 2-Me-C26","CHC.7,11-C28:2 &amp; 2-Me-C27","CHC.7,11-C31:2","CHC.8-C25:1","CHC.8,12-C26:2","CHC.8,12-C28:2","CHC.9-C23:1","CHC.9-C23:1","CHC.9-C25:1","CHC.9-C27:1","CHC.9-C29:1 &amp; 5,9-C29:2","CHC.9,13-C25:2","CHC.9,13-C27:2","CHC.n-C21","CHC.n-C21","CHC.n-C22","CHC.n-C22","CHC.n-C23","CHC.n-C23","CHC.n-C24","CHC.n-C24","CHC.n-C25","CHC.n-C26","CHC.n-C26","CHC.n-C27","CHC.n-C27","CHC.n-C28","CHC.n-C28","CHC.n-C29","CHC.n-C29","CHC.n-C31","CHC.n-C31","CHC.NI.20","CHC.NI.33","CHC.NI.34","CHC.NI.54","CHC.NI.54","CHC.NI.7","CHC.x-C22:1a","CHC.x-C22:1b","CHC.x-C23:1","CHC.x,y-C24:2","CHC.x,y-C24:2","chill.coma.recovery","chill.coma.recovery","chill.coma.recovery.variance","chill.coma.recovery.variance","choline.high.yeast","choline.low.yeast","chronic.cold.exposure.resistance","chronic.cold.exposure.resistance","circadian.period.chi","circadian.period.chi","circadian.period.mesa","circadian.period.mesa","circadian.rythmicity.index","circadian.rythmicity.index","citrulline.high.yeast","citrulline.low.yeast","climbing.ability","climbing.ability","climbing.ability.paraquat","climbing.ability.paraquat","climbing.age.decline.high.yeast","climbing.age.decline.low.yeast","Clostridium.lentocellum","Clostridium.lentocellum","CMP.high.yeast","CMP.low.yeast","cold.hardiness","cold.hardiness","cold.shock.acclimated.old","cold.shock.acclimated.old","cold.shock.acclimated.young","cold.shock.acclimated.young","cold.shock.old","cold.shock.old","cold.shock.young","cold.shock.young","Corynebacterium.minutissimum","Corynebacterium.minutissimum","Corynebacterium.pseudogenitalium","Corynebacterium.pseudogenitalium","Corynebacterium.sp.","Corynebacterium.sp.","Corynebacterium.terpenotabidum","Corynebacterium.terpenotabidum","CT.max.2020","CT.max.2020","CT.min.2020","CT.min.2020","CT.min.23C","CT.min.developmental.stability","CT.min.plasticity","CT.min.plasticity.variation","CT.min.within.environment.variation.plasticity","cystathionine.high.yeast","cystathionine.low.yeast","cystine.high.yeast","cystine.low.yeast","cytidine.high.yeast","cytidine.low.yeast","cytosine.high.yeast","cytosine.low.yeast","D-GA3P.high.yeast","D-GA3P.low.yeast","D-Leucic.acid.high.yeast","D-Leucic.acid.low.yeast","day.sleep","day.sleep","day.sleep.bout.length","day.sleep.bout.length","day.sleep.bout.length.SD","day.sleep.bout.length.SD","day.sleep.bouts","day.sleep.bouts","day.sleep.bouts.SD","day.sleep.bouts.SD","day.sleep.SD","day.sleep.SD","DCV.resistance","DDT.resistance.knockdown","DDT.resistance.mortality","deltamethrin.resistance","dessication.tolerance","dessication.tolerance","developmental.acclimation.cold.exposure","developmental.acclimation.cold.exposure","dopamine.response.to.paraquat.2021","Drosophila.A Virus","Drosophila.A Virus","E.faecalis.resistance","egg.adult.development.time.2016","egg.adult.development.time.2016","Enterococcus.hirae","Enterococcus.hirae","epinephrine.high.yeast","epinephrine.low.yeast","Escherichia.coli","Escherichia.coli","ethanol.sensitivity.first.exposure","ethanol.sensitivity.first.exposure","ethanol.sensitivity.second.exposure","ethanol.sensitivity.second.exposure","ethanol.tolerance","ethanol.tolerance","exercise.induced.activity","exercise.induced.activity","Exophiala.xenobiotica","Exophiala.xenobiotica","fecundity","fertilised.egg.retention.score","fertility.25C","fertility.27C","fertility.29C","fertility.thermal.susceptibility","FHV.resistance","Fimbriimonas.ginsengisoli","Fimbriimonas.ginsengisoli","fitness.early","fitness.early","fitness.late","fitness.late","flight.performance","flight.performance","food.intake","food.intake","foraging.survival","G1P.G6P.F6P.F1P.high.yeast","G1P.G6P.F6P.F1P.low.yeast","Gardnerella.vaginalis","Gardnerella.vaginalis","Gemmata.obscuriglobus","Gemmata.obscuriglobus","Gemmata.sp.","Gemmata.sp.","genital.licking.frequency","geranyl.pyrophosphate.high.yeast","geranyl.pyrophosphate.low.yeast","Gluconacetobacter.diazotrophicus","Gluconacetobacter.diazotrophicus","Gluconobacter.frateurii","Gluconobacter.frateurii","Gluconobacter.morbifer","Gluconobacter.morbifer","Gluconobacter.oxydans","Gluconobacter.oxydans","Gluconobacter.thailandicus","Gluconobacter.thailandicus","glucoronate.high.yeast","glucoronate.low.yeast","glucose.2015","glucose.anexic","glucose.gnotobiotic","glucose.high.glucose.diet","glucose.high.yeast","glucose.low.glucose.diet","glucose.low.yeast","glucose.medium.glucose.diet","glucose.microbiome.effect","glucose.yeast.poor.diet","glucose.yeast.rich.diet","glutamic.acid.high.yeast","glutamic.acid.low.yeast","glutamine.high.yeast","glutamine.low.yeast","glyceraldehyde.high.yeast","glyceraldehyde.low.yeast","glycerate.high.yeast","glycerate.low.yeast","glycerol-3-P.high.yeast","glycerol-3-P.low.yeast","glycerol.2015","glycerol.high.glucose.diet","glycerol.low.glucose.diet","glycerol.medium.glucose.diet","glycine.high.yeast","glycine.low.yeast","glycogen.2015","glycogen.high.glucose.diet","glycogen.low.glucose.diet","glycogen.medium.glucose.diet","GMP.high.yeast","GMP.low.yeast","guanosine.high.yeast","guanosine.low.yeast","gustatory.response.ethanol","gustatory.response.ethanol","gut.permeability","gut.permeability.radiation","histamine.high.yeast","histamine.low.yeast","histidine.high.yeast","histidine.low.yeast","homogentisate.high.yeast","homogentisate.low.yeast","hydrogen.peroxide.resistance","Hymenobacter.sp.","Hymenobacter.sp.","Hymenobacter.swuensis","Hymenobacter.swuensis","hypoxanthine.high.yeast","hypoxanthine.low.yeast","Ideonella.sp.","Ideonella.sp.","imidacloprid.resistance.100p","imidacloprid.resistance.25p","IMP.high.yeast","IMP.low.yeast","inosine.high.yeast","inosine.low.yeast","inositol.high.yeast","inositol.low.yeast","interocular.distance","interocular.distance","IPI","iso-leucine.high.yeast","iso-leucine.low.yeast","isolated.aggression","isoValeric.acid.high.yeast","isoValeric.acid.low.yeast","kallithea.resistance.LT50","kallithea.resistance.LT50","kallithea.viral.load","Klebsiella.pneumoniae","Klebsiella.pneumoniae","Kocuria.polaris","Kocuria.polaris","Kocuria.sp.","Kocuria.sp.","Komagataeibacter.europaeus","Komagataeibacter.europaeus","Komagataeibacter.hansenii","Komagataeibacter.hansenii","Komagataeibacter.intermedius","Komagataeibacter.intermedius","Komagataeibacter.kakiaceti","Komagataeibacter.kakiaceti","Komagataeibacter.medellinensis","Komagataeibacter.medellinensis","Komagataeibacter.oboediens","Komagataeibacter.oboediens","Komagataeibacter.rhaeticus","Komagataeibacter.rhaeticus","Komagataeibacter.xylinus","Komagataeibacter.xylinus","Kozakia.baliensis","Kozakia.baliensis","kujin.virus.resistance","L-Kynurenine.high.yeast","L-Kynurenine.low.yeast","L.monocytogenes.load","L.monocytogenes.MTD","lactate.high.yeast","lactate.low.yeast","Lactobacillus.brevis","Lactobacillus.brevis","Lactobacillus.fabifermentans","Lactobacillus.fabifermentans","Lactobacillus.fermentum","Lactobacillus.fermentum","Lactobacillus.kisonensis","Lactobacillus.kisonensis","Lactobacillus.plantarum","Lactobacillus.plantarum","Lactobacillus.rhamnosus","Lactobacillus.rhamnosus","Lactococcus.lactis","Lactococcus.lactis","lactose.high.yeast","lactose.low.yeast","lead.effect.development.time","lead.effect.development.time","lead.effect.locomotor.activity","lead.effect.locomotor.activity","lead.egg.adult.development.time.2016","lead.egg.adult.development.time.2016","lead.locomotor.activity.2016","lead.locomotor.activity.2016","leucine.high.yeast","leucine.low.yeast","Leuconostoc.carnosum","Leuconostoc.carnosum","Leuconostoc.pseudomesenteroides","Leuconostoc.pseudomesenteroides","lifespan.18C","lifespan.18C","lifespan.2019","lifespan.25C.2014","lifespan.25C.2015","lifespan.25C.2020","lifespan.25C.2020","lifespan.25C.high.yeast","lifespan.25C.low.yeast","lifespan.28C","lifespan.28C","lifespan.following.diapause.2019","lifespan.high.sugar.diet","lifetime.fecundity.25C","linoleic.acid.high.yeast","linoleic.acid.low.yeast","linolenic.acid.high.yeast","linolenic.acid.low.yeast","locomotor.activity","locomotor.activity.2016","locomotor.activity.2016","locomotor.frequency","Luteipulveratus.mongoliensis","Luteipulveratus.mongoliensis","lysine.high.yeast","lysine.low.yeast","M.anisopliae.resistance","M.anisopliae.resistance","malathion.mortality.12hr","malathion.mortality.12hr","malathion.mortality.24hr","malathion.mortality.24hr","malathion.mortality.3hr","malathion.mortality.3hr","malathion.mortality.6hr","malathion.mortality.6hr","malondialdehyde.high.yeast","malondialdehyde.low.yeast","margaric.acid.high.yeast","margaric.acid.low.yeast","mass.yeast.poor.diet","mass.yeast.rich.diet","mating.latency.2015","Mesorhizobium.sp.","Mesorhizobium.sp.","Metarhizium.brunneum","Metarhizium.brunneum","methionine.high.yeast","methionine.low.yeast","methylmalonate.high.yeast","methylmalonate.low.yeast","Methylobacterium.extorquens","Methylobacterium.extorquens","Methylobacterium.sp.","Methylobacterium.sp.","micro.environmental.plasticity.lifespan.18C","micro.environmental.plasticity.lifespan.18C","micro.environmental.plasticity.lifespan.25C","micro.environmental.plasticity.lifespan.25C","micro.environmental.plasticity.lifespan.28C","micro.environmental.plasticity.lifespan.28C","Microbacterium.sp.","Microbacterium.sp.","MPH.effect.on.activity","msb.resistance","msb.resistance","myristic.acid.high.yeast","myristic.acid.low.yeast","N-AcetylGlycine.high.yeast","N-AcetylGlycine.low.yeast","negative.geotaxis","negative.geotaxis","negative.geotaxis.msb","negative.geotaxis.msb","negative.geotaxis.sensitivity.msb","negative.geotaxis.sensitivity.msb","night.sleep","night.sleep","night.sleep.bout.length","night.sleep.bout.length","night.sleep.bout.length.SD","night.sleep.bout.length.SD","night.sleep.bouts","night.sleep.bouts","night.sleep.bouts.SD","night.sleep.bouts.SD","night.sleep.SD","night.sleep.SD","Nora.virus","Nora.virus","olfactory.response.1-hexanol","olfactory.response.1-hexanol","olfactory.response.2-heptanone","olfactory.response.2-heptanone","olfactory.response.2-phenyl.ethyl.alcohol","olfactory.response.2-phenyl.ethyl.alcohol","olfactory.response.acetophenone","olfactory.response.acetophenone","olfactory.response.benzaldehyde.2013","olfactory.response.benzaldehyde.2013","olfactory.response.benzaldehyde.2015","olfactory.response.benzaldehyde.2015","olfactory.response.citral","olfactory.response.citral","olfactory.response.D-carvone","olfactory.response.D-carvone","olfactory.response.ethanol","olfactory.response.ethanol","olfactory.response.ethylacetate","olfactory.response.ethylacetate","olfactory.response.ethylbutyrate","olfactory.response.ethylbutyrate","olfactory.response.eugenol","olfactory.response.eugenol","olfactory.response.heliol","olfactory.response.heliol","olfactory.response.hexal","olfactory.response.hexal","olfactory.response.L-carvone","olfactory.response.L-carvone","olfactory.response.methylSalicylate","olfactory.response.methylSalicylate","orient.towards.female","ornithine.high.yeast","ornithine.low.yeast","orotate.high.yeast","orotate.low.yeast","ovariole.asymmetry","ovariole.number","oxaloacetate.high.yeast","oxaloacetate.low.yeast","oxypurinol.high.yeast","oxypurinol.low.yeast","P.aeruginosa.defecation","P.aeruginosa.fecal.pH","P.aeruginosa.intestinal.mitosis","P.aeruginosa.resistance.2017","P.aeruginosa.resistance.2017","P.aeruginosa.resistance.2019","P.entomophila.resistance","P.rettgeri.load","P.rettgeri.load.high.glucose.diet","P.rettgeri.load.low.glucose.diet","P.rettgeri.load.medium.glucose.diet","P.rettgeri.resistance","P.rettgeri.tolerance","Paenibacillus.borealis","Paenibacillus.borealis","Paenibacillus.sp.","Paenibacillus.sp.","pantothenate.high.yeast","pantothenate.low.yeast","paraquat.resistance","paraquat.resistance","paraquat.resistance.2021","parathion.resistance","Pectobacterium.carotovorum","Pectobacterium.carotovorum","PEP.high.yeast","PEP.low.yeast","permethrin.knockdown","permethrin.knockdown","permethrin.mortality.24hr","permethrin.mortality.24hr","phenylalanine.high.yeast","phenylalanine.low.yeast","phototaxis.mid.age","phototaxis.mid.age","phototaxis.old","phototaxis.old","phototaxis.young","phototaxis.young","pipecolate.high.yeast","pipecolate.low.yeast","Plasmodium.vivax","Plasmodium.vivax","Podospora.anserina","Podospora.anserina","posterior.lobe.shape.PC1","posterior.lobe.shape.PC2","posterior.lobe.size","proline.high.yeast","proline.low.yeast","Propionibacterium.acidipropionici","Propionibacterium.acidipropionici","Propionibacterium.acnes","Propionibacterium.acnes","protein.2015","protein.anexic","protein.gnotobiotic","protein.high.glucose.diet","protein.low.glucose.diet","protein.medium.glucose.diet","Pseudodonghicola.xiamenensis","Pseudodonghicola.xiamenensis","Pseudomonas.fuscovaginae","Pseudomonas.fuscovaginae","Pseudomonas.moraviensis","Pseudomonas.moraviensis","Pseudomonas.putida","Pseudomonas.putida","Pseudomonas.sp.","Pseudomonas.sp.","Pseudomonas.stutzeri","Pseudomonas.stutzeri","Pseudomonas.syringae","Pseudomonas.syringae","Psychrobacter.glacincola","Psychrobacter.glacincola","Psychrobacter.sp.","Psychrobacter.sp.","putrescine.high.yeast","putrescine.low.yeast","pyridoxal-5-P.high.yeast","pyridoxal-5-P.low.yeast","pyruvate.high.yeast","pyruvate.low.yeast","radiation.resistance","RCH.18C","RCH.18C","RCH.25C","RCH.25C","RCH.old","RCH.old","RCH.young","RCH.young","Rhizobacter.sp.","Rhizobacter.sp.","ribose-5-P.high.yeast","ribose-5-P.low.yeast","Saccharomyces.cerevisiae","Saccharomyces.cerevisiae","sarcosine.high.yeast","sarcosine.low.yeast","serine.high.yeast","serine.low.yeast","sigma.virus.sensitivity","social.aggression","social.effect.aggression","sorbitol.high.yeast","sorbitol.low.yeast","Sphingomonas.hengshuiensis","Sphingomonas.hengshuiensis","Sphingomonas.parapaucimobilis","Sphingomonas.parapaucimobilis","Sphingomonas.phyllosphaerae","Sphingomonas.phyllosphaerae","Sphingomonas.sp.","Sphingomonas.sp.","Sphingomonas.wittichii","Sphingomonas.wittichii","Sporolactobacillus.terrae","Sporolactobacillus.terrae","sprinter.marathoner","sprinter.marathoner","Staphylococcus.aureus","Staphylococcus.aureus","Staphylococcus.epidermidis","Staphylococcus.epidermidis","Staphylococcus.gallinarum","Staphylococcus.gallinarum","startle.response","startle.response","startle.response.2012","startle.response.2012","startle.response.2012.msb","startle.response.2012.msb","startle.response.variance","startle.response.variance","startle.sensitivity.msb","startle.sensitivity.msb","starvation.resistance.2012","starvation.resistance.2012","starvation.resistance.2017","starvation.resistance.2019","starvation.resistance.2019","starvation.resistance.high.fat.diet","starvation.resistance.high.protein.diet","starvation.resistance.old","starvation.resistance.old","starvation.resistance.standard.diet","starvation.resistance.variance","starvation.resistance.variance","starvation.resistance.yeast.poor.diet","starvation.resistance.yeast.rich.diet","starvation.resistance.young","starvation.resistance.young","Streptococcus.pneumoniae","Streptococcus.pneumoniae","Streptococcus.salivarius","Streptococcus.salivarius","Streptomyces.griseoflavus","Streptomyces.griseoflavus","Streptomyces.pristinaespiralis","Streptomyces.pristinaespiralis","Streptomyces.sp.","Streptomyces.sp.","succinate.high.yeast","succinate.low.yeast","sucrose.high.yeast","sucrose.low.yeast","Sum.of Non-Wolbachia Species","Sum.of Non-Wolbachia Species","T1.femur","T1.femur","T1.leg.length","T1.leg.length","T1.tarsus","T1.tarsus","T1.tibia","T1.tibia","T2.femur","T2.femur","T2.leg.length","T2.leg.length","T2.tarsus","T2.tarsus","T2.tibia","T2.tibia","taurine.high.yeast","taurine.low.yeast","tergite5","tergite6","Terriglobus.roseus","Terriglobus.roseus","thorax.length.18C","thorax.length.28C","thorax.length.thermal.plasticity","threonine.high.yeast","threonine.low.yeast","Tremella.mesenterica","Tremella.mesenterica","triglyceride.2015","triglyceride.gnotobiotic","triglyceride.high.glucose.diet","triglyceride.low.glucose.diet","triglyceride.medium.glucose.diet","triglyceride.poor.diet","triglyceride.yeast.rich.diet","trimethylamine.high.yeast","trimethylamine.low.yeast","tryptophan.high.yeast","tryptophan.low.yeast","tunicamycin.ER.stress","tunicamycin.LT50","tyrosine.high.yeast","tyrosine.low.yeast","uracil.high.yeast","uracil.low.yeast","urate.high.yeast","urate.low.yeast","uridine.high.yeast","uridine.low.yeast","valine.high.yeast","valine.low.yeast","virgin.egg.retention","waking.activity","waking.activity","waking.activity.SD","waking.activity.SD","week1.fecundity.25C","week3.fecundity.25C","week5.fecundity.25C","week7.fecundity.25C","weight.anexic","weight.gnotobiotic","weight.high.glucose.diet","weight.low.glucose.diet","weight.medium.glucose.diet","weight.microbiome.effect","wet.weight","wing.centroid.size","wing.centroid.size","wing.centroid.size.2019_Dworkin","wing.centroid.size.2019_Dworkin","wing.centroid.size.2019_Houle","wing.centroid.size.2019_Houle","wing.display.frequency","Wolbachia.pipentis","Wolbachia.pipentis","Xanthomonas.sp.","Xanthomonas.sp.","Xanthomonas.vesicatoria","Xanthomonas.vesicatoria","xanthosine.high.yeast","xanthosine.low.yeast","xanthurenate.high.yeast","xanthurenate.low.yeast","Yersinia.mollaretii","Yersinia.mollaretii","Yersinia.pekkanenii","Yersinia.pekkanenii","Zymomonas.mobilis","Zymomonas.mobilis",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Female","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Male","Male","Male","Female","Male","Female","Female","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Male","Female","Male","Female","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Male","Male","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Male","Male","Male","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Male","Male","Male","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Female","Male","Female","Male","Female","Male","Male","Female","Male","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Male","Male","Male","Male","Female","Male","Female","Male","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Male","Male","Male","Female","Female","Male","Male","Male","Male","Female","Female","Female","Female","Female","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Pooled","Pooled","Female","Female","Female","Female","Female","Female","Female","Male","Male","Female","Female","Male","Female","Female","Female","Male","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Male","Female","Female","Female","Male","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Female","Female","Male","Male","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Male","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Male","Male","Male","Male","Female","Female","Female","Male","Female","Male","Male","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Male","Female","Female","Female","Female","Female","Male","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Male","Male","Female","Male","Male","Female","Male","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Female","Female","Female","Female","Female","Male","Male","Male","Male","Male","Male","Female","Female","Female","Female","Female","Female","Male","Male","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Female","Male","Female","Female","Female","Female","Male","Male","Male","Male","Male","Male","Male","Female","Male","Female","Male","Female","Male","Male","Female","Male","Female","Male","Female","Male","Female","Female","Female","Female","Female","Male","Female","Male","Female","Male","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled","Pooled"],["Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Morphological","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Behavioural","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Stress response","Metabolome","Metabolome","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Behavioural","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Metabolome","Metabolome","Metabolome","Metabolome","Insecticide response","Microbiome","Microbiome","Metabolome","Metabolome","Drug response","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","CHC","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Temperature related","Temperature related","Activity","Activity","Activity","Activity","Activity","Activity","Metabolome","Metabolome","Activity","Activity","Stress response","Stress response","Life history","Life history","Microbiome","Microbiome","Metabolome","Metabolome","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Pathogen response","Insecticide response","Insecticide response","Insecticide response","Stress response","Stress response","Temperature related","Temperature related","Drug response","Microbiome","Microbiome","Pathogen response","Life history","Life history","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Drug response","Drug response","Drug response","Drug response","Drug response","Drug response","Activity","Activity","Microbiome","Microbiome","Life history","Life history","Life history","Life history","Life history","Temperature related","Pathogen response","Microbiome","Microbiome","Life history","Life history","Life history","Life history","Activity","Activity","Behavioural","Behavioural","Life history","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Behavioural","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Microbiome","Micobiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Sensory","Sensory","Physiology","Physiology","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Insecticide response","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Behavioural","Metabolome","Metabolome","Behavioural","Metabolome","Metabolome","Pathogen response","Pathogen response","Pathogen response","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Pathogen response","Metabolome","Metabolome","Pathogen response","Pathogen response","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Stress response","Stress response","Stress response","Stress response","Life history","Life history","Activity","Activity","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Life history","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Microbiome","Microbiome","Metabolome","Metabolome","Pathogen response","Pathogen response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Behavioural","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Life history","Life history","Life history","Life history","Life history","Life history","Microbiome","Microbiome","Drug response","Drug response","Drug response","Metabolome","Metabolome","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Stress response","Stress response","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Microbiome","Microbiome","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Sensory","Behavioural","Metabolome","Metabolome","Metabolome","Metabolome","Morphological","Morphological","Metabolome","Metabolome","Metabolome","Metabolome","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Pathogen response","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Drug response","Drug response","Drug response","Insecticide response","Microbiome","Microbiome","Metabolome","Metabolome","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Metabolome","Metabolome","Activity","Activity","Activity","Activity","Activity","Activity","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Morphological","Morphological","Morphological","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Stress response","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Temperature related","Microbiome","Microbiome","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Pathogen response","Behavioural","Behavioural","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Activity","Drug response","Drug response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Stress response","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Metabolome","Metabolome","Morphological","Morphological","Microbiome","Microbiome","Morphological","Morphological","Morphological","Metabolome","Metabolome","Microbiome","Microbiome","Metabolome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Drug response","Drug response","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Metabolome","Life history","Activity","Activity","Activity","Activity","Life history","Life history","Life history","Life history","Microbiome","Microbiome","Morphological","Morphological","Morphological","Microbiome","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Morphological","Behavioural","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Metabolome","Metabolome","Metabolome","Metabolome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Microbiome","Pathogen response","Pathogen response","Stress response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Insecticide response","Temperature related","Life history","Life history","Life history","Life history","Drug response","Drug response","Drug response","Genomic","Stress response","Stress response","Stress response","Genomic","Stress response","Morphology","Life history","Stress response","Stress response","Life history","Pathogen response"],["Mean relative 1/3-Methylhistidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1/3-Methylhistidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methyladenosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methyladenosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methylguanosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methylguanosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 13-HODE concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 13-HODE concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Hydroxyglutarate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Hydroxyglutarate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Phosphoglyceric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Phosphoglyceric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Mean relative 3-Hydroxybutyric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxybutyric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxykynurenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxykynurenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 4-Pyridoxic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 4-Pyridoxic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 5-Aminovalerate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 5-Aminovalerate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean adult abdomen length, when reared at 18C. Higher values indicate larger flies.","Mean adult abdomen length, when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult abdomen lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean relative acetoacetate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetoacetate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative acetylcarnitine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcarnitine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcholine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcholine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean relative adenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ADP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ADP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of aggressive encounters between 8 males in an interaction vial over a two minute duration. Higher values indicate more aggressive males.","Mean relative agmatine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative agmatine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative alanine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative alanine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative allantoin concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative allantoin concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative Alpha-Ketoglutaric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative Alpha-Ketoglutaric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aminoisobutyrate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aminoisobutyrate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative AMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative AMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean relative anthranilate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative anthranilate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.","Mean relative arachidonate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arachidonate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arginine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arginine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative asparagine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative asparagine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aspartic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aspartic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Mean relative betaine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative betaine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative biotin concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative biotin concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative cadaverine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cadaverine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative carnitine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative carnitine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative cGMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cGMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean relative choline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative choline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","Mean relative citrulline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative citrulline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 5% yeast. Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative CMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative CMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum (CT min), which is defined as the temperature at which a complete cessation of movement is observed as a result of the fly entering chill coma. Egg-to-adult development occurred at 23C. Higher values indicate a higher CT min.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Developmental stability across environments is the consistency of within-environment variation in CT min across environments. The authors calculate this by taking the highest coefficient of variation (CV) in CT min at any rearing temperature, minus the lowest CV at any other given rearing temperature. Lower values indicate greater developmental stability across rearing temperatures.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums were measured (CT min). Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The absolute value of the slope provides a measure of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The standard error of the slope provides a measure of the extent to which the same genotype can produce the same plastic response consistently. Higher values indicate greater variance in thermal plasticity.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums were measured. This trait captures how consistently a genotype produces the same phenotype at a given rearing temperature. Higher values indicate greater variance in phenotype within a rearing temperature.","Mean relative cystathionine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystathionine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-GA3P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-GA3P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-Leucic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-Leucic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean days survived after infection with Drosophila C Virus (DCV) - a positive sense RNA virus that naturally infects multiple Drosophila species. Higher values indicate a longer lifespan, and therefore a greater resistance to DCV.","Mean knockdown of adults, four hours after vials were inoculated with 0.5 ug/ml of DDT. Knockdown is defined as either flies permanently seen to be in a prone position with jolting movement of legs or wings, or a prolonged inability to right themselves from a prone position. DDT is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean survival of adults, 24 hours after vials were inoculated with 0.5 ug/ml of DDT. This is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean change in proportion of adult flies surviving one hours of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean change in proportion of adult flies surviving one hour of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean difference in Dopamine content (pg/ng tissue) extracted from the heads of two flies, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress on Dopamine content.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative epinephrine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative epinephrine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of eggs produced by single females, per day. Higher values indicate more fecund females.","Mean fertilised egg retention score. An estimation of the progression of embryo development taken two hours after a one hour egg laying period. Higher values indicate more developed embryos which signifies a greater degree of egg retention. Note that both males and femlaes were sourced from the DGRP line, therefore variation could be explained by either the male or female genotype.","Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.","Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility","Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility","Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.","Mean days survived after infection with Flock House Virus (FHV) - a nodavirus virus that typically infects beetles. Higher values indicate a longer lifespan, and therefore a greater resistance to FHV.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean absolute number of first instar larvae produced when 2-5 day old focal DGRP females were placed in a vial with competitor females and standard males. Focal females and their same-sex competitors were not replaced when they died, such that the fitness assays depend upon mortality as well as offspring production. Higher values indicate higher fitness.","Mean proportion of first instar larvae produced when 2-5 day old focal DGRP males were placed in a vial with competitor males and standard females. Focal males and their same-sex competitors were not replaced when they died, such that the fitness assays depend upon mortality as well as offspring production. Higher values indicate higher fitness.","Mean absolute number of first instar larvae produced when 14-17 day old focal DGRP females were placed in a vial with competitor females and standard males. Focal females and their same-sex competitors were not replaced when they died, such that the fitness assays depend upon mortality as well as offspring production. Higher values indicate higher fitness.","Mean proportion of first instar larvae produced when 2-5 day old focal DGRP males were placed in a vial with competitor males and standard females. Focal males and their same-sex competitors were not replaced when they died, such that the fitness assays depend upon mortality as well as offspring production. Higher values indicate higher fitness.","Mean height (m) at which flies catch themselves from freefall after vial inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean height (m) at which flies catch themselves from freefall after rapid inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean quantity of 4% sucrose solution consumed per fly iover a 24 hour period. Higher values indicate greater food intake.","Mean quantity of 4% sucrose solution consumed per fly over a 24 hour period. Higher values indicate greater food intake.","Adult mean survival when flies were provided access to food over a 5 day period. Higher values indicate greater survival and assumed greater foraging ability.","Mean relative G1P, G6P, F6P and F1P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolites. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative G1P, G6P, F6P and F1P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolites. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.","Mean relative geranyl pyrophosphate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative geranyl pyrophosphate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative glucoronate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glucoronate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Estimated mass of free glucose per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.","Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of free glucose.","Mean relative glucose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of free glucose.","Mean relative glucose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of free glucose.","Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher glucose levels.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher glucose levels.","Mean relative glutamic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glyceraldehyde concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glyceraldehyde concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerol-3-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerol-3-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Estimated mass of free glycerol per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycerol.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycerol.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycerol.","Mean relative glycine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Estimated mass of glycogen stores per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycogen.","Mean relative GMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative GMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative guanosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative guanosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","The mean proportion of 19 day-old flies with permeable guts, following no exposure to radiation. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","The mean proportion of 19 day-old flies with permeable guts, following exposure to 100 gray units of ionising radiation when flies were 5 days old. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","Mean relative histamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative homogentisate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative homogentisate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Lifespan on a diet supplemented with 2-3% hydrogen peroxide (H2O2). Data are corrected for line body mass. Higher values indicate greater resistance to hydrogen peroxide.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative hypoxanthine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative hypoxanthine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean wiggle index for third instar larvae 1 hour into exposure to 100ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean wiggle index for third instar larvae 1 hour into exposure to 25ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean relative IMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative IMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inositol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inositol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Median interpulse interval (IPI) in the male courtship 'song'. Males in many Drosophila species produce a series of auditory pulses (a pulse 'song') when courting females, and the pause between pulses (IPI) varies greatly among species. Higher values indicate a longer IPI.","Mean relative iso-leucine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative iso-leucine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in isolation for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Mean relative isoValeric acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative isoValeric acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean survival after infection with the Kunjin subtype of the West Nile Virus. Calculated as the log(hazard ratio). Higher values indicate greater susceptibility to the virus.","Mean relative L-Kynurenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative L-Kynurenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The median number  of colony forming units (originally log transformed) 48 hrs after initial infection with the bacterial pathogen L.monocytogenes. Higher values indicate a greater bacterial load and a more severe infection.","Median days till death after injection with 1000 colony forming units of the bacterial pathogen L. monocytogenes. Higher values indicate greater resistance to the L. monocytogenes.","Mean relative lactate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lactate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative lactose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lactose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean relative leucine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative leucine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days. Higher values indicate a longer lifespan.","Mean adult lifespan, measured as the number of days from eclosion to death, when living with a single  standard male, at 25C. Higher values indicate longer lived flies.","Adult lifespan, measured in days, for virgin females. Higher values indicate greater longevity.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 5% yeast. Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days for viring females following diapause induction (10 ̊C for 35 days, moved to 18 ̊C for a day of acclimation, remainder of their lives at 25 ̊C). Higher values indicate a longer lifespan.","Mean lifespan of 20 females maintained on a high sugar diet. Means are adjusted for block effects. Higher values indicate longer lived flies.","Fecundity was estimated every other week of the female’s lifespan as a 2-day total of eggs laid, when living with a single standard male, at 25C. Lifetime fecundity was estimated as the sum of all egg counts across the female's lifespan. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean relative linoleic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linoleic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linolenic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linolenic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean total distance traveled by individual flies during a five minute interval. Higher values indicate more active lines.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","The proportion of flies engaging in a ‘walking bout’ at a given time point. Higher values indicate more active lines.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative lysine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lysine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean relative malondialdehyde concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative malondialdehyde concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative margaric acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative margaric acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean virgin female body mass for groups of three flies, after rearing on food with 0.5% yeast content (dietary restricted). Higher values indicate heavier flies.","Mean virgin female body mass for groups of three flies, after rearing on food with 5% yeast content (ab libitum diet). Higher values indicate heavier flies.","Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative methionine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methionine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methylmalonate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methylmalonate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean relative myristic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative myristic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative N-AcetylGlycine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative N-AcetylGlycine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.","Mean relative ornithine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ornithine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative orotate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative orotate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean difference between the number of ovarioles in the right and left ovary. Higher values indicate greater asymmetry.","Mean number of ovarioles in the right and left ovary. Higher values indicate a greater number of ovarioles.","Mean relative oxaloacetate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxaloacetate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxypurinol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxypurinol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of fecal spots detected after 20h for 10 flies, after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate more defecation.","Mean pH of fecal matter for flies after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a higher pH.","Mean amount of intestinal mitosis, five days after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a greater number of mitotic itensitinal cells.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of flies in a vial to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean percentage of flies dying after 3 days post infection with P.entomophila L48, an entomopathogenic bacterium. Higher values indicate greater mortality.","Mean colony forming units (CFU) produced from the homogenate of three pooled flies, 20 hours after infection with P. rettgeri, a bacterial pathogen. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 10% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 2.5% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 5% glucose. Higher values indicate a greater bacterial load.","Mean proportion of flies surviving infection, two days after inoculation with the bacterial pathogen P. rettgeri. Higher values indicate greater resistance to P. rettgeri.","A general, nonlinear relationship between the proportion of flies surviving two days post infection and sustained pathogen load, after injection with the bacterial pathogen P. rettgeri, was predicted. The data presented here are the line deviations from the predicted relationship between survival and bacterial load. Positive values indicate a greater tolerance to P. rettgeri.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative pantothenate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pantothenate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative PEP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative PEP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean relative phenylalanine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative phenylalanine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean relative pipecolate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pipecolate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean PC1 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 78% of the variation in shape, and describes the degree of beak-like protrusion at the tip of the posterior lobes. Higher values indicate a greater degree of beak-like postrusion.","Mean PC2 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 10% of the variation in shape, and describes the width of the posterior lobes. Higher values indicate wider lobes.","Mean posterior lobe size, measured in pixels. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. Higher values indicate larger posterior lobes.","Mean relative proline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative proline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Estimated mass of soluble protein per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.","Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of protein.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative putrescine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative putrescine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyridoxal-5-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyridoxal-5-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyruvate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyruvate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean survival 24 hours after exposure to radiation (Cs137, avg dose of 4.85 Gy/min. Total dose of 1382 Gy over the 4 hours and 45 minutes of continuous exposure). Higher values indicate greater radiation resistance.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative ribose-5-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ribose-5-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative sarcosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sarcosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative serine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative serine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of adults infected with sigma virus (DmelSV), 13 days post infection. Sigma virus is a rhabdovirus that naturally occurs in D. melanogaster. It causes flies paralysis in flies exposed to CO2. Higher values indicate a higher (???) proportion of paralysed flies after exposure to CO2, and therefore greater sensitivity to sigma virus.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in a mixed Sex environment for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Mean difference in aggressive behavior between male flies raised in social and socially isolated environments. Higher values indicate that males raised in a social environment were more aggressive than males experiencing isolation (the authors adjusted line means to meet normality expectations, we present udjusted means for the social and isolation line means).","Mean relative sorbitol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sorbitol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Note: this data is recoded from that recorded by Watanabe et al (2020).","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Higher values indicate that a greater proportion of activity was completed in the first 30 minutes, which corresponds to a 'sprinter' phenotype. Note: this data is recoded from that recorded by Watanabe et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Adult mean survival in the absence of food after two days is here used as a measure of starvation resistance. Higher values indicate greater survival and therefore higher starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high fat diet (carbs:44.15g/L, protein:52g/L, fat:17.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high protein diet (carbs:44.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on the laboratories standard diet (carbs:119.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 0.5% yeast content (dietary restricted).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 5% yeast content (ab libitum diet).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old.  Higher values indicate flies that exhibited greater starvation resistance.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative succinate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative succinate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sucrose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sucrose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean relative taurine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative taurine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean pigmentation scores for tergite 5. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean pigmentation scores for tergite 6. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean adult thorax length when reared at 18C. Higher values indicate larger flies.","Mean adult thorax length when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult thorax lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Mean relative threonine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative threonine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Estimated mass of total trigylcerides per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of triglycerides.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher triglyceride levels.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher triglyceride levels.","Mean relative trimethylamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative trimethylamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tryptophan concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tryptophan concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The hazard ratio of the death rate of male flies exposed to tunicamycin - an endoplasmic reticulum (ER) stress inducing drug – with the control group, fed drug free media. Higher values indicate greater ER stress and lower survival of flies exposed to tunicamycin.","Mean time required for 50% of male flies exposed to tunicamycin (TM) media to die. TM is an antibiotic that causes endoplasmic reticulum (ER) stress. Higher values indicate longer survival times and greater resistance to ER stress.","Mean relative tyrosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tyrosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uracil concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uracil concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative urate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative urate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uridine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uridine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative valine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative valine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean unfertilised egg retention time, measured as the number of days before a virgin female laid her first egg. Higher values indicate a longer retention time.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean fecundity in the first week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the third week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the fifth week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean fecundity in the seventh week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.","Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 10% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 2.5% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 5% glucose. Higher values indicate heavier flies.","Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.","Mean mass. Higher values indicate heavier flies.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean relative xanthosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthurenate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthurenate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean number of flies exposed to alpha-amanitin, at concentration 0.2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean number of flies exposed to alpha-amanitin, at concentration 2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean survival (larvae that pupated) for late third instar larvae, 24 hours after exposure to one hour of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.25 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.5 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 1 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 2 ug/ml. Higher values indicate greater mortality.","Mean dosage (ug/ml) of azinphos-methyl (an organophosphate insecticide) required for 50% of larvae to die. Higher values indicate greater resistance.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 0.5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 1 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 2 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 3 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 4 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean chlorantraniliprole dosage required for 50% of first instar larvae to die before eclosion. Chlorantraniliprole is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean proportion of successfully eclosed individuals following an hour of cold shock (-5°C) at 5 days post oviposition (third instar). Higher values indicate larvae that are more resistant to cold shock.","Mean development time in days from egg to adult eclosion, on a standard diet. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a diet supplemented with 10% EtOH (ethanol). Higher values indicate a longer development time.","Mean proportion of flies surviving to adulthood, when reared at 25C on control food medium. Higher values indicate greater survival.","Percentage of flies surviving to eclosion, after development on standard food. Higher values indicate greater viability.","Percentage of flies surviving to eclosion, after development on food supplemented with 10% EtOH (ethanol). Higher values indicate greater viability.","The mean difference in egg-to-adult development time in days, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Higher values indicate a longer development time for ethanol-supplemented flies.","The mean difference in the percentage of flies surviving to eclosion, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Lower values indicate a greater negative effect of ethanol on egg to adult viability.","Mean size of the genome in mb, estimated using flow cytometry. Higher values indicate a larger genome.","Mean difference in egg-to-adult viability between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a smaller negative effect of lead on viability.","Mean proportion of flies surviving to adulthood, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater survival.","Methylmercury (MeHg) is a persistent environmental toxin present in seafood that can compromise the developing nervous system in humans. To assess overall developmental tolerance on MeHg media, the authors calculate an eclosion index – higher values indicate greater relative egg-to-adult viability when reared on MeHg supplemented media.","Mean copy number of mitochondrial genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more mitochondrial genomes, which the authors suggest is a proxy for mitochondrial density.","Mean difference in larval viability between control and p-xylene-treated (concentration 1.068mM) first instar larvae after 14 days of development. P-xylene is an aromatic hydrocarbon associated with gasoline and diesel-powered engine emissions. Higher values indicate a greater negative effect of P-xylene on larval viability.","Mean pupal case length in mm. Measurements were taken with image analysis software from at least 6 vials with a minimum of 15 pupa per vial, and were converted from pixels to mm based on the ratio to the diameter measurement of 1 euro cent coin (16.25 mm diameter)","Mean height at which larvae pupated in a 95mm Drosophila vial, after accounting for food height. Higher values indicate higher pupation sites.","Mean hours survived for larvae subjected to nutrient poor starvation medium (20% sucrose in PBS solution). Higher values indicate greater starvation resistance.","The mean percentage of flies that died within 24 hours, after suffering traumatic injury. Higher values indicate greater mortality.","Proportion of unfertilised eggs produced by 3-4 flies that had been allowed to mate for at least two days beforehand. Note that flies were kept in population cages that contained both sexes from the same line during the mating period. This means that genetic variance in this trait variation could be explained by either the male or female genotype. Higher values indicate a greater proportion of unfertilised eggs.","Mean copy number of Wolbachia genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more Wolbachia genomes, which the authors suggest is a proxy for Wolbachia density. Note that only lines found to harbour Wolbachia are inlcuded in this dataset."],["Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Rohde et al (2019) Genetics","Rohde et al (2019) Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Shorter et al (2015) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Campwell et al (2019) G3","Campwell et al (2019) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Gaertner et al (2015) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Gaertner et al (2015) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Najarro, Hackett and Macdonald (2017) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Najarro et al (2015) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Freda et al (2017) Integrative and Comparative Biology","Freda et al (2017) Integrative and Comparative Biology","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Magwire et al (2012) PLOS Genetics","Schmidt et al (2017) Genetics","Schmidt et al (2017) Genetics","Duneau et al (2018) G3","Rajpurohit et al (2018) Molecular Ecology","Rajpurohit et al (2018) Molecular Ecology","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Lovejoy et al (2021) Molecular Ecology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Chapman et al (2020) Genes","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Evangelou et al (2019) G3","Horvath and Kalinka (2018) Evolution","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Magwire et al (2012) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Spierer et al (2021) PLOS Genetics","Spierer et al (2021) PLOS Genetics","Garlapow et al (2015) PLOS One","Garlapow et al (2015) PLOS One","Lee et al (2017) Genome Biology and Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Gaertner et al (2015) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jin et al (2020) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jin et al (2020) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Sharma et al (2020) Scientific reports","Sharma et al (2020) Scientific reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Harrison et al (2020) BMC Genomics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Denecke et al (2017) Scientific Reports","Denecke et al (2017) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Turner, Miller and Cochrane (2013) Molecular Biology and Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Rohde et al (2017) Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Ahlers et al (2019) Cell Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Hotson and Schneider (2015) G3","Hotson and Schneider (2015) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Durham et al (2014) Nature Communications","Ivanov et al (2015) The Journals of Gerontology: Series A","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Patel and Talbert (2021) Heliyon","Durham et al (2014) Nature Communications","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Rohde et al (2018) G3","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Maesani et al (2015) PLOS Computational Biology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Gaertner et al (2015) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Rohde et al (2019) Genetics","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Swarup et al (2013) PNAS","Swarup et al (2013) PNAS","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Gaertner et al (2015) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Lobell et al (2017) G3","Lobell et al (2017) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Evangelou et al (2019) G3","Bou Sleiman et al (2015) Nature Communications","Howick and Lazzaro (2017) Molecular Ecology","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Howick and Lazzaro (2017) Molecular Ecology","Howick and Lazzaro (2017) Molecular Ecology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Lovejoy et al (2021) Molecular Ecology","Duneau et al (2018) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) G3","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Vaisnav et al (2014) PLOS One","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Magwire et al (2011) PLOS Genetics","Rohde et al (2017) Genetics","Rohde et al (2017) Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Riddle (2020) Journal of Experimental Biology","Riddle (2020) Journal of Experimental Biology","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Lee et al (2017) Genome Biology and Evolution","Everman et al (2019) Genetics","Everman et al (2019) Genetics","Francis et al (2021) G3","Francis et al (2021) G3","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Francis et al (2021) G3","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Dembeck et al (2015) PLOS Genetics","Dembeck et al (2015) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Unckless, Rottschaefer and Lazzaro (2015) G3","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Chow, Wolfner and Clark (2013) PNAS","Chow, Wolfner and Clark (2013) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Akhund-Zade et al (2017) Journal of Insect Science","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Dobson et al (2015) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) G3","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Gaertner et al (2015) G3","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Mitchell et al (2017) PLOS One","Mitchell et al (2017) PLOS One","Campwell et al (2019) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Freda et al (2017) Integrative and Comparative Biology","Morozova (2018) G3","Morozova (2018) G3","Zhou et al (2016) Environmental Health Perspectives","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Huang et al (2014) Genome Research &amp; Ellis et al (2014) PLOS Genetics","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Montgomery et al (2014) PLOS One","Richardson (2012) PLOS Genetics","Adebambo et al (2020) Frontiers in Genetics","Zhang et al (2021) Current Biology","Zhang, Reeves and Tautz (2020) Molecular Ecology","Akcay (2017) Thesis","Katzenberger et al (2015) eLife","Horvath and Kalinka (2018) Evolution","Richardson (2012) PLOS Genetics"],[0.2,0.16,0.15,0.07,0.05,0.02,0.46,0.03,0.15,0,0,0,0,0.04,0,0.43,0.16,0.43,0.25,0.41,0.06,0.3,0.39,0,0.12,0.06,0,0.11,0,0.21,0.23,0,0,0,0.03,0,0,0,0,0.13,0.05,0.16,0.15,0.12,0.03,0.13,0.07,0.08,0.05,0.15,0.1,0.21,0.03,0.13,0.2,0.11,0.1,0.17,0.08,0.15,0.14,0.16,0.06,0.15,0.07,0.26,0.11,0.15,0.1,0.15,0.46,0.23,0.05,0.26,0.09,0.05,0.39,0.08,0.03,0.16,0,0.08,0,0.39,0,0.34,0.01,0.28,0.35,0.21,0,0.28,0,0.41,0.16,0.22,0,0.19,0.24,0.31,0.21,0,0.06,0.28,0.09,0.51,0.17,0.29,0,0,0.04,0.16,0,0,0,0.08,0.06,0,0.13,0.32,0.5,0.06,0,0.19,0,0.43,0.23,0.13,0.17,0.24,0.22,0.03,0.26,0.04,0.45,0,0.38,0,0,0,0,0,0.41,0.5,0.06,0.21,0.23,0.12,0.26,0.44,0.49,0.02,0,0,0,0,0.36,0.28,0.01,0,0.02,0.18,0.35,0,0.01,0.34,0,0.14,0.36,0.33,0.16,0.15,0.29,0.14,0.04,0.23,0.22,0.2,0,0,0.46,0,0.03,0.37,0.36,0.04,0.42,0.43,0.18,0.31,0.45,0.33,0.03,0.4,0.39,0.33,0.06,0.33,0.33,0.35,0.44,0.18,0.22,0.23,0.31,0,0.08,0.37,0.18,0.31,0,0.32,0.28,0.25,0.46,0.31,0.21,0.12,0.39,0.31,0.24,0.26,0.3,0.24,0.43,0.48,0.17,0.44,0.28,0.46,0.01,0.42,0,0.37,0.01,0.23,0.04,0.17,0.45,0.17,0.4,0.48,0.31,0.31,0.33,0.43,0.19,0.07,0,0.31,0.15,0,0.3,0,0.29,0.43,0.3,0.14,0.21,0.12,0.17,0.21,0,0.37,0.14,0.22,0.28,0.18,0.26,0.25,0.19,0.46,0.28,0.35,0.49,0.5,0.29,0.25,0,0.03,0,0,0.07,0.13,0.23,0,0.26,0,0,0.46,0.49,0.07,0.36,0.24,0.33,0.02,0,0,0.03,0,0.02,0,0,0.3,0,0,0,0.12,0.2,0.25,0.18,0.23,0.12,0.13,0.16,0.09,0.34,0,0.35,0.01,0.43,0.07,0.04,0.07,0.09,0.03,0.27,0.42,0.23,0.39,0.36,0.25,0.1,0.18,0,0.06,0,0,0.32,0.49,0,0.21,0.2,0,0.38,0.14,0.04,0.13,0.09,0.52,0.19,0.18,0,0.31,0.31,0.1,0,0.09,0.23,0.11,0.44,0.27,0.32,0.44,0.08,0,0.01,0.22,0.21,0,0,0,0,0,0.14,0.39,0.49,0.12,0.19,0.39,0,0.33,0,0.47,0.38,0.31,0,0.19,0,0.1,0.02,0.22,0,0.08,0.45,0,0.08,0.23,0,0.11,0.02,0.11,0.04,0.12,0.05,0.14,0.21,0.16,0.1,0.05,0.26,0.05,0.33,0.23,0.09,0.29,0.25,0,0.21,0.21,0.07,0,0.01,0.01,0.09,0.2,0.15,0.24,0,0,0.29,0.02,0.44,0.26,0,0.27,0.11,0.42,0.46,0.45,0,0.27,0.2,0,0.39,0.36,0.48,0,0.07,0.2,0.07,0.39,0.31,0.48,0.16,0.3,0.38,0.08,0.19,0,0.2,0.23,0,0.49,0.06,0.01,0.15,0.09,0.15,0.15,0,0.28,0.33,0.37,0.01,0.26,0.46,0.18,0.16,0,0,0,0.22,0.46,0.02,0.21,0,0,0.12,0.04,0.03,0.04,0.17,0,0.24,0.13,0.16,0.07,0.21,0.07,0.21,0,0.36,0.11,0.17,0,0,0.02,0.06,0.04,0.08,0.14,0,0.11,0,0.04,0.06,0.01,0,0,0,0.1,0.06,0.12,0,0,0,0.1,0,0,0,0,0,0,0.02,0.43,0.5,0,0.2,0,0,0,0.43,0.33,0.07,0.11,0.27,0.02,0.39,0.28,0,0.09,0.13,0.09,0.13,0.05,0.11,0.17,0.09,0.5,0.02,0.2,0.4,0.47,0,0.04,0,0.08,0.21,0,0.36,0.28,0.33,0.13,0.23,0.48,0.46,0.48,0.44,0.16,0.17,0.26,0.18,0,0,0,0.09,0.29,0.1,0.15,0.05,0.17,0.3,0,0.12,0.03,0.12,0.03,0.37,0,0.5,0.16,0.04,0.02,0,0.05,0.29,0.21,0.22,0,0.01,0.33,0.03,0.11,0.24,0.17,0.11,0,0.04,0.41,0.44,0.45,0.47,0.3,0.45,0.5,0.27,0.25,0.01,0,0.16,0,0,0.22,0.19,0.43,0.17,0.06,0,0.05,0.08,0.23,0.12,0.1,0,0.24,0.23,0.21,0.19,0.08,0.11,0.49,0.48,0.21,0.39,0.13,0.1,0,0.43,0.07,0.07,0.23,0.49,0.29,0.29,0,0.43,0.09,0.42,0.26,0.25,0.33,0.2,0.18,0.17,0.12,0.3,0.22,0.11,0,0,0,0.07,0.38,0.34,0.2,0.3,0.38,0,0.19,0.12,0.19,0.22,0.33,0.36,0.41,0,0.04,0.41,0,0.03,0.03,0.02,0.28,0.3,0.26,0.25,0.23,0.2,0.08,0.31,0.09,0.22,0.02,0.12,0.03,0.34,0.08,0.37,0,0.28,0.31,0.34,0.46,0.34,0,0,0.05,0.16,0,0.47,0,0.12,0.5,0.11,0.41,0.01,0.05,0.06,0.02,0.38,0.19,0.1,0.24,0.28,0,0.24,0.02,0.3,0,0.27,0.11,0,0.07,0.35,0.1,0,0.13,0.02,0,0.21,0.04,0,0.2,0.12,0.47,0.36,0,0.07,0.36,0,0.29,0,0,0,0.35,0.23,0.29,0.22,0.39,0.13,0.48,0,0.21,0,0.2,0,0,0,0.09,0,0,0.07,0,0.01,0.18,0.15,0,0,0,0,0,0.1,0.12,0.24,0.21,0.21,0.24,0.13,0.2,0.36,0.38,0.03,0.05,0.44,0.45,0.24,0,0,0,0.05,0.13,0.27,0.14,0.16,0.16,0,0.02,0.45,0.29,0.06,0.15,0.02,0,0.01,0.3,0.03,0.34,0.01,0.16,0.04,0,0.44,0.08,0,0,0.33,0.26,0.47,0.22,0.48,0.4,0,0.01,0.38,0.51,0.5,0.5,0.47,0.46,0.41,0.28,0.43,0.39,0.47,0.27,0.5,0.19,0.12,0,0.1,0,0.26,0.31,0.18,0.17,0.2,0,0,0,0.08,0.18,0.1,0,0.04,0.22,0.27,0.19,0.45,0.27,0.33,0.2,0.29,0.46,0.17,0,0.03,0.23,0.16,0.2,0.43,0,0.26,0.06,0.3,0,0,0.32,0.3,0.18,0.4,0.39,0,0.35,0.29,0.43,0.21,0.22,0.19,0.18,0,0.32,0.37,0.35,0.03,0,0.1,0.06,0.18,0.33,0.23,0,0.16,0.08,0.05,0.11,0.05,0.07,0.38,0.05,0.21,0,0.05,0,0,0.28,0.24,0.14,0.11,0.19,0.37,0.18,0.14,0.37,0.15,0.38,0.41,0.48,0.09,0.36,0.49,0.19,0.49,0.5,0.5,0.25,0.47,0.21,0.28,0.12,0.46,0],[0.59,0.68,0.71,0.85,0.9,0.97,0,0.94,0.71,1.25,1.03,1.11,1.2,0.93,1.24,0,0.7,0.13,0.51,0.15,0.91,0.37,0.25,1.04,0.75,0.88,1.16,0.78,1,0.57,0.55,1.1,1.03,1.03,0.97,1.02,1.19,1.14,1.05,0.75,0.9,0.69,0.69,0.75,0.93,0.74,0.85,0.85,0.89,0.71,0.8,0.58,0.94,0.74,0.6,0.77,0.81,0.65,0.84,0.71,0.72,0.67,0.89,0.7,0.86,0.48,0.77,0.7,0.79,0.7,0.09,0.54,0.91,0.48,0.82,0.9,0.25,0.84,0.94,0.67,1.05,0.84,1.15,0.22,1.09,0.33,0.98,0.42,0.28,0.61,1.48,0.45,0.99,0.18,0.67,0.55,1.07,0.61,0.51,0.39,0.56,1.01,0.88,0.44,0.81,0.01,0.67,0.42,1.26,1.01,0.91,0.68,1.02,1.09,1.27,0.83,0.88,1.01,0.73,0.35,0,0.88,1.25,0.63,1.06,0.16,0.56,0.74,0.67,0.53,0.56,0.94,0.47,0.93,0.13,1.02,0.27,1.13,1.03,1.2,1.09,1.11,0.19,0,0.88,0.57,0.53,0.76,0.46,0,0.01,0.96,1.03,1.02,1.05,1.26,0.26,0.46,0.98,1.04,0.96,0.64,0.31,1.2,0.99,0.33,1.01,0.71,0.26,0.32,0.68,0.69,0.41,0.72,0.92,0,0.55,0.59,1.16,1.14,0.11,1.01,0.95,0.22,0.25,0.92,0.17,0.14,0.64,0,0,0,0.94,0,0,0.32,0.89,0.32,0.29,0.28,0,0.64,0.55,0.54,0.34,1.18,0.84,0.2,0.64,0,1.04,0.35,0.42,0.49,0.09,0,0,0.76,0.22,0.36,0.5,0.49,0.4,0.51,0,0.02,0.65,0.12,0.42,0.09,0.97,0.12,1.09,0.22,0.99,0.51,0.93,0.65,0.1,0.66,0.16,0,0.37,0.37,0.33,0,0.61,0.86,1.41,0.38,0.71,1.17,0.39,1.15,0.41,0,0.39,0.72,0.58,0.75,0.64,0.58,1.16,0.27,0.73,0.55,0.45,0.64,0.48,0,0,0,0.43,0.29,0,0.02,0.42,0.51,1.37,0.94,1.39,1.06,0.86,0.75,0.53,1.06,0.49,1.06,1.42,0,0,0.85,0,0,0,0.97,1.06,1.21,0.93,1,0.97,1.01,1.08,0.4,1.01,1.1,1.19,0.76,0.59,0.51,0.63,0.54,0.77,0.74,0.68,0.82,0.32,1.22,0.31,0.98,0.16,0.86,0.92,0.85,0.81,0.94,0.46,0.15,0.53,0,0.27,0,0.81,0.62,1.11,0.89,1.03,1.35,0.35,0,1.14,0.58,0.59,1,0.23,0.72,0.92,0.74,0.82,0.01,0.61,0.63,1.04,0.4,0.38,0.8,1.16,0.82,0.55,0.79,0.17,0.46,0.33,0.07,0.84,1.19,0.98,0.56,0.58,1.2,1.39,1.2,1.35,1.31,0.77,0.24,0.06,0.76,0.63,0.22,1.05,0.35,1.04,0,0.22,0.34,1.07,0.62,1.07,0.8,0.96,0.56,1.02,0.85,0,1.1,0.84,0.54,1.06,0.78,0.96,0.78,0.93,0.76,0.91,0.71,0.58,0.67,0.8,0.91,0.48,0.9,0.3,0.53,0.84,0.42,0.48,1.01,0.58,0.58,0.85,1.41,0.98,0.98,0.81,0.61,0.7,0.5,1.05,1.15,0.4,0.94,0.12,0.48,1.09,0.47,0.77,0.18,0,0,1.15,0.46,0.61,1.16,0.21,0,0,1.15,0.86,0.59,0.86,0.2,0.38,0,0.69,0.41,0.25,0.84,0.62,1,0.61,0.54,1.13,0,0.88,0.99,0.7,0.81,0.69,0.69,1.01,0.42,0.32,0.22,0.99,0.48,0,0.65,0.67,1.24,1.01,1.07,0.54,0,0.96,0.58,1.08,1.02,0.75,0.93,0.94,0.93,0.65,1.05,0.53,0.74,0.67,0.87,0.58,0.86,0.58,1.12,0.3,0.77,0.66,1.16,1.23,0.96,0.88,0.93,0.85,0.72,1.06,0.79,1.03,0.91,0.89,0.97,1,1.1,1.11,0.8,0.88,0.75,1.07,1.08,1.09,0.8,1.04,1.24,1.05,1.28,1.17,1,0.96,0,0,1.05,0.59,1.03,1.24,1.18,0.12,0.33,0.86,0.77,0.48,0.96,0.24,0.43,1.03,0.82,0.74,0.83,0.74,0.89,0.79,0.67,0.83,0,0.97,0.58,0,0,1.04,0.92,1.02,0.84,0.57,1.02,0.29,0.43,0.31,0.73,0.52,0,0,0,0.12,0.68,0.66,0.47,0.64,1.02,1.04,1.13,0.81,0.43,0.8,0.7,0.9,0.65,0.39,1.47,0.75,0.93,0.75,0.93,0.26,1.03,0,0.67,0.91,0.95,1.1,0.91,0.41,0.58,0.56,1.01,0.99,0.32,0.93,0.78,0.48,0.66,0.78,1.22,0.92,0.15,0,0.11,0,0.39,0.12,0,0.44,0.5,0.98,1.06,0.68,1.09,1.18,0.55,0.62,0.13,0.65,0.89,1.16,0.91,0.84,0.52,0.75,0.8,1.01,0.52,0.54,0.58,0.61,0.83,0.78,0,0,0.57,0.23,0.74,0.8,1.1,0,0.86,0.85,0.52,0.04,0.41,0.43,1.08,0.13,0.82,0.17,0.49,0.5,0.34,0.6,0.64,0.66,0.75,0.4,0.56,0.78,1.18,1.53,1.2,0.86,0.25,0.33,0.63,0.43,0.26,1.01,0.61,0.76,0.59,0.53,0.31,0.28,0.19,1.04,0.92,0,1.07,0.93,0.94,0.96,0.43,0.38,0.48,0.49,0.55,0.59,0.85,0.35,0.82,0.55,0.96,0.76,0.94,0.31,0.84,0.27,1.02,0.45,0.38,0.31,0.07,0.32,1.16,1.2,0.91,0.67,1.16,0,1.42,0.76,0.03,0.77,0.18,0.98,0.9,0.88,0.96,0.24,0.62,0.79,0.54,0.44,1.01,0.52,0.96,0.4,1.14,0.46,0.79,1.03,0.86,0.31,0.79,1.16,0.73,0.96,1.56,0,0.92,0.98,0.63,0.76,0,0,1.28,0.85,0.3,1.01,0.44,1.12,1.09,1.11,0.32,0.52,0.42,0.55,0.22,0.74,0.02,1,0.57,1.05,0.59,1.03,1.24,1.15,0.81,1.06,1.19,0.86,1.3,0.97,0.64,0.7,1.07,1.59,1.53,1.04,1.17,0.8,0.76,0.52,0.57,0.58,0.51,0.71,0.58,0.28,0.24,0.92,0.86,0.13,0.09,0.5,1.02,1.01,1.15,0.89,0.74,0.47,0.73,0.68,0.68,1.09,0.96,0,0.42,0.89,0.71,0.95,1.01,0.98,0.41,0.93,0.33,0.97,0.68,0.91,1.07,0.14,0.85,1.08,1.31,0.34,0.48,0,0.56,0,0,1.01,0.99,0.25,0,0,0,0,0,0.18,0.45,0,0.24,0.07,0.46,0,0.62,0.75,1.03,0.81,1.4,0.48,0.4,0.64,0.63,0.61,1.26,0.96,1.03,0.84,0.64,0.81,1.06,0.91,0.55,0.48,0.64,0.12,0.45,0.35,0.6,0.41,0,0.65,1.03,0.94,0.52,0.68,0.59,0.17,1.17,0.49,0.9,0.42,1.06,1.31,0,0.39,0.65,0.18,0.2,1.21,0.29,0.43,0.14,0.58,0.54,0.61,0.64,1.33,0.33,0.24,0.31,0.94,1.06,0.8,0.87,0.62,0.36,0.56,1.04,0.69,0.85,0.91,0.78,0.91,0.81,0.28,0.9,0.59,1.01,0.9,1.09,1.04,0.46,0.51,0.71,0.79,0.63,0,0.64,0.71,0.26,0.7,0.24,0.18,0,0.81,0.29,0,0.61,0.01,0,0,0.5,0.07,0.56,0.45,0.76,0.07,1.02],[0.79,0.84,0.85,0.93,0.95,0.98,0.46,0.97,0.85,1.25,1.03,1.11,1.2,0.97,1.24,0.43,0.86,0.56,0.76,0.56,0.96,0.68,0.64,1.04,0.87,0.94,1.16,0.89,1,0.78,0.77,1.1,1.03,1.03,1,1.02,1.19,1.14,1.05,0.87,0.95,0.84,0.85,0.88,0.97,0.87,0.92,0.92,0.95,0.85,0.9,0.79,0.97,0.87,0.8,0.89,0.9,0.83,0.92,0.85,0.86,0.83,0.94,0.85,0.93,0.74,0.89,0.85,0.89,0.85,0.55,0.77,0.95,0.74,0.91,0.95,0.64,0.92,0.97,0.83,1.05,0.92,1.15,0.61,1.09,0.67,0.99,0.71,0.64,0.82,1.48,0.73,0.99,0.59,0.84,0.77,1.07,0.8,0.75,0.7,0.78,1.01,0.94,0.72,0.9,0.52,0.83,0.71,1.26,1.01,0.96,0.84,1.02,1.09,1.27,0.92,0.94,1.01,0.87,0.67,0.5,0.94,1.25,0.82,1.06,0.59,0.78,0.87,0.84,0.76,0.77,0.97,0.73,0.96,0.58,1.02,0.64,1.13,1.03,1.2,1.09,1.11,0.6,0.5,0.94,0.78,0.76,0.88,0.72,0.44,0.5,0.98,1.03,1.02,1.05,1.26,0.62,0.73,0.99,1.04,0.98,0.82,0.66,1.2,0.99,0.67,1.01,0.86,0.62,0.64,0.83,0.84,0.7,0.86,0.96,0.23,0.78,0.79,1.16,1.14,0.57,1.01,0.97,0.59,0.61,0.96,0.58,0.57,0.82,0.31,0.45,0.33,0.97,0.4,0.39,0.64,0.94,0.65,0.62,0.63,0.44,0.81,0.77,0.77,0.66,1.18,0.92,0.57,0.82,0.31,1.04,0.67,0.71,0.74,0.54,0.31,0.21,0.88,0.61,0.67,0.75,0.75,0.7,0.75,0.43,0.5,0.82,0.56,0.7,0.55,0.99,0.54,1.09,0.6,0.99,0.75,0.96,0.82,0.55,0.83,0.57,0.48,0.68,0.69,0.66,0.43,0.8,0.93,1.41,0.69,0.85,1.17,0.69,1.15,0.7,0.43,0.69,0.86,0.79,0.87,0.81,0.79,1.16,0.64,0.87,0.78,0.72,0.82,0.74,0.25,0.19,0.46,0.72,0.65,0.49,0.51,0.71,0.76,1.37,0.97,1.39,1.06,0.93,0.87,0.76,1.06,0.75,1.06,1.42,0.46,0.49,0.92,0.36,0.24,0.33,0.98,1.06,1.21,0.97,1,0.98,1.01,1.08,0.7,1.01,1.1,1.19,0.88,0.8,0.76,0.82,0.78,0.89,0.87,0.83,0.91,0.66,1.22,0.66,0.99,0.59,0.93,0.96,0.93,0.91,0.97,0.73,0.57,0.77,0.39,0.63,0.25,0.9,0.8,1.11,0.94,1.03,1.35,0.66,0.49,1.14,0.79,0.8,1,0.62,0.86,0.96,0.87,0.91,0.53,0.8,0.81,1.04,0.71,0.7,0.9,1.16,0.91,0.78,0.89,0.6,0.73,0.65,0.51,0.92,1.19,0.99,0.78,0.79,1.2,1.39,1.2,1.35,1.31,0.91,0.63,0.55,0.88,0.81,0.62,1.05,0.68,1.04,0.47,0.6,0.65,1.07,0.81,1.07,0.9,0.98,0.78,1.02,0.93,0.45,1.1,0.92,0.77,1.06,0.89,0.98,0.89,0.96,0.88,0.95,0.86,0.79,0.83,0.9,0.97,0.74,0.95,0.64,0.77,0.93,0.71,0.73,1.01,0.79,0.79,0.93,1.41,0.99,0.99,0.91,0.8,0.85,0.74,1.05,1.15,0.69,0.96,0.56,0.74,1.09,0.74,0.88,0.6,0.46,0.45,1.15,0.73,0.81,1.16,0.61,0.36,0.48,1.15,0.93,0.79,0.93,0.6,0.69,0.48,0.84,0.71,0.63,0.92,0.81,1,0.8,0.77,1.13,0.49,0.94,0.99,0.85,0.91,0.85,0.85,1.01,0.7,0.66,0.59,0.99,0.73,0.46,0.82,0.83,1.24,1.01,1.07,0.77,0.46,0.98,0.79,1.08,1.02,0.88,0.96,0.97,0.96,0.83,1.05,0.77,0.87,0.84,0.93,0.79,0.93,0.79,1.12,0.66,0.89,0.83,1.16,1.23,0.98,0.94,0.97,0.93,0.86,1.06,0.89,1.03,0.96,0.94,0.99,1,1.1,1.11,0.9,0.94,0.87,1.07,1.08,1.09,0.9,1.04,1.24,1.05,1.28,1.17,1,0.98,0.43,0.5,1.05,0.79,1.03,1.24,1.18,0.54,0.66,0.93,0.89,0.76,0.98,0.63,0.71,1.03,0.91,0.87,0.91,0.87,0.95,0.91,0.84,0.91,0.5,0.98,0.79,0.4,0.47,1.04,0.96,1.02,0.92,0.78,1.02,0.65,0.71,0.64,0.87,0.75,0.48,0.46,0.48,0.56,0.84,0.83,0.73,0.82,1.02,1.04,1.13,0.91,0.72,0.9,0.85,0.95,0.82,0.69,1.47,0.87,0.97,0.87,0.97,0.63,1.03,0.5,0.83,0.96,0.98,1.1,0.95,0.7,0.79,0.78,1.01,0.99,0.65,0.97,0.89,0.72,0.83,0.89,1.22,0.96,0.56,0.44,0.56,0.47,0.69,0.57,0.5,0.72,0.75,0.99,1.06,0.84,1.09,1.18,0.77,0.81,0.56,0.82,0.94,1.16,0.95,0.92,0.76,0.87,0.9,1.01,0.76,0.77,0.79,0.81,0.91,0.89,0.49,0.48,0.78,0.62,0.87,0.9,1.1,0.43,0.93,0.93,0.76,0.53,0.71,0.72,1.08,0.56,0.91,0.59,0.74,0.74,0.67,0.8,0.81,0.83,0.88,0.7,0.77,0.89,1.18,1.53,1.2,0.93,0.63,0.68,0.82,0.73,0.63,1.01,0.8,0.88,0.79,0.75,0.64,0.63,0.6,1.04,0.96,0.41,1.07,0.97,0.97,0.98,0.71,0.68,0.74,0.74,0.78,0.79,0.94,0.67,0.91,0.77,0.98,0.89,0.97,0.65,0.92,0.64,1.02,0.73,0.69,0.66,0.53,0.66,1.16,1.2,0.95,0.83,1.16,0.47,1.42,0.88,0.53,0.88,0.59,0.99,0.95,0.94,0.98,0.63,0.81,0.89,0.77,0.72,1.01,0.75,0.98,0.7,1.14,0.74,0.89,1.03,0.93,0.66,0.89,1.16,0.87,0.98,1.56,0.21,0.96,0.98,0.83,0.89,0.47,0.36,1.28,0.93,0.65,1.01,0.73,1.12,1.09,1.11,0.67,0.76,0.71,0.77,0.61,0.87,0.5,1,0.78,1.05,0.8,1.03,1.24,1.15,0.9,1.06,1.19,0.93,1.3,0.99,0.82,0.85,1.07,1.59,1.53,1.04,1.17,0.9,0.88,0.76,0.78,0.79,0.75,0.85,0.78,0.65,0.62,0.96,0.91,0.57,0.55,0.75,1.02,1.01,1.15,0.95,0.87,0.74,0.86,0.84,0.84,1.09,0.98,0.45,0.71,0.94,0.86,0.98,1.01,0.99,0.71,0.97,0.67,0.99,0.84,0.95,1.07,0.58,0.92,1.08,1.31,0.67,0.74,0.47,0.78,0.48,0.4,1.01,0.99,0.63,0.51,0.5,0.5,0.47,0.46,0.6,0.72,0.43,0.63,0.54,0.73,0.5,0.81,0.88,1.03,0.9,1.4,0.73,0.71,0.82,0.8,0.81,1.26,0.96,1.03,0.92,0.82,0.9,1.06,0.96,0.76,0.74,0.82,0.57,0.72,0.68,0.8,0.71,0.46,0.82,1.03,0.97,0.75,0.84,0.79,0.6,1.17,0.75,0.96,0.72,1.06,1.31,0.32,0.69,0.83,0.57,0.6,1.21,0.64,0.72,0.58,0.78,0.77,0.8,0.82,1.33,0.65,0.61,0.66,0.97,1.06,0.9,0.94,0.8,0.68,0.78,1.04,0.85,0.92,0.95,0.89,0.95,0.88,0.66,0.95,0.8,1.01,0.95,1.09,1.04,0.73,0.75,0.86,0.89,0.81,0.37,0.82,0.85,0.63,0.84,0.62,0.59,0.48,0.91,0.65,0.49,0.81,0.5,0.5,0.5,0.76,0.54,0.77,0.73,0.88,0.53,1.02],[0.26,0.19,0.17,0.08,0.05,0.02,1,0.03,0.17,0,0,0,0,0.04,0,1,0.18,0.76,0.33,0.73,0.06,0.45,0.6,0,0.14,0.06,0,0.12,0,0.27,0.29,0,0,0,0.03,0,0,0,0,0.15,0.05,0.18,0.18,0.14,0.03,0.15,0.08,0.08,0.06,0.17,0.11,0.26,0.03,0.15,0.25,0.13,0.11,0.21,0.09,0.17,0.17,0.2,0.06,0.17,0.08,0.36,0.13,0.18,0.11,0.17,0.84,0.3,0.05,0.35,0.1,0.05,0.61,0.09,0.03,0.2,0,0.08,0,0.64,0,0.51,0.01,0.4,0.56,0.25,0,0.39,0,0.69,0.19,0.29,0,0.24,0.32,0.44,0.27,0,0.06,0.39,0.1,0.97,0.2,0.41,0,0,0.04,0.19,0,0,0,0.09,0.06,0,0.15,0.48,1,0.06,0,0.23,0,0.73,0.29,0.15,0.2,0.31,0.28,0.03,0.36,0.04,0.77,0,0.59,0,0,0,0,0,0.69,1,0.06,0.27,0.3,0.14,0.36,1,0.98,0.02,0,0,0,0,0.58,0.38,0.01,0,0.02,0.22,0.53,0,0.01,0.51,0,0.17,0.58,0.51,0.19,0.17,0.41,0.16,0.04,1,0.29,0.25,0,0,0.81,0,0.03,0.63,0.59,0.04,0.72,0.76,0.22,1,1,1,0.03,1,1,0.51,0.06,0.51,0.53,0.55,1,0.22,0.29,0.3,0.48,0,0.09,0.65,0.22,1,0,0.47,0.4,0.34,0.84,1,1,0.13,0.63,0.47,0.32,0.35,0.42,0.32,1,0.96,0.21,0.79,0.4,0.84,0.01,0.78,0,0.63,0.01,0.31,0.04,0.21,0.82,0.21,0.71,1,0.46,0.45,0.5,1,0.24,0.07,0,0.45,0.17,0,0.44,0,0.42,1,0.43,0.16,0.26,0.14,0.21,0.26,0,0.58,0.16,0.29,0.38,0.22,0.35,1,1,1,0.4,0.55,1,0.97,0.41,0.33,0,0.03,0,0,0.07,0.14,0.31,0,0.34,0,0,1,1,0.08,1,1,1,0.02,0,0,0.04,0,0.02,0,0,0.43,0,0,0,0.14,0.26,0.33,0.23,0.3,0.13,0.15,0.19,0.1,0.52,0,0.53,0.01,0.73,0.07,0.04,0.08,0.1,0.03,0.37,0.74,0.3,1,0.57,1,0.11,0.23,0,0.06,0,0,0.48,1,0,0.26,0.25,0,0.62,0.16,0.04,0.15,0.1,0.98,0.24,0.23,0,0.44,0.45,0.11,0,0.1,0.29,0.12,0.72,0.37,0.49,0.86,0.09,0,0.01,0.28,0.27,0,0,0,0,0,0.15,0.62,0.89,0.13,0.23,0.64,0,0.49,0,1,0.64,0.48,0,0.23,0,0.11,0.02,0.28,0,0.08,1,0,0.09,0.3,0,0.12,0.02,0.12,0.04,0.13,0.05,0.17,0.27,0.2,0.11,0.05,0.35,0.05,0.52,0.31,0.09,0.41,0.34,0,0.27,0.27,0.08,0,0.01,0.01,0.1,0.25,0.17,0.32,0,0,0.42,0.02,0.79,0.35,0,0.37,0.13,0.69,1,1,0,0.37,0.24,0,0.65,1,1,0,0.07,0.25,0.08,0.66,0.45,1,0.19,0.42,0.6,0.09,0.23,0,0.24,0.3,0,1,0.06,0.01,0.18,0.1,0.18,0.18,0,0.4,0.51,0.63,0.01,0.35,1,0.22,0.2,0,0,0,0.29,1,0.02,0.27,0,0,0.14,0.04,0.03,0.04,0.21,0,0.31,0.15,0.2,0.07,0.26,0.07,0.27,0,0.54,0.13,0.21,0,0,0.02,0.06,0.05,0.08,0.16,0,0.12,0,0.05,0.06,0.01,0,0,0,0.11,0.06,0.14,0,0,0,0.11,0,0,0,0,0,0,0.02,1,1,0,0.25,0,0,0,0.78,0.5,0.08,0.13,0.36,0.02,0.62,0.39,0,0.1,0.15,0.09,0.15,0.06,0.13,0.2,0.09,1,0.02,0.26,1,1,0,0.04,0,0.09,0.27,0,0.56,0.4,0.52,0.15,0.31,1,1,1,0.79,0.19,0.2,0.35,0.22,0,0,0,0.1,0.4,0.11,0.18,0.05,0.21,0.43,0,0.14,0.04,0.14,0.04,0.59,0,1,0.2,0.04,0.02,0,0.05,0.41,0.27,0.28,0,0.01,0.51,0.03,0.12,0.33,0.21,0.13,0,0.04,0.73,1,0.8,1,0.44,0.79,1,0.38,0.33,0.01,0,0.19,0,0,0.29,0.24,0.77,0.21,0.06,0,0.05,0.09,0.31,0.14,0.11,0,0.32,0.3,0.26,0.24,0.09,0.13,1,1,0.27,0.63,0.15,0.11,0,1,0.08,0.08,0.31,0.92,0.42,0.4,0,0.76,0.1,0.72,0.35,0.33,0.49,0.25,0.22,0.21,0.14,0.43,0.28,0.12,0,0,0,0.07,0.6,0.51,0.24,0.41,0.59,0,0.24,0.13,0.25,0.3,0.51,0.56,0.68,0,0.04,1,0,0.04,0.03,0.02,0.4,0.44,0.35,0.34,0.29,0.26,0.09,0.47,0.1,0.29,0.02,0.14,0.03,0.52,0.09,0.57,0,0.39,0.45,0.52,0.87,0.51,0,0,0.05,0.2,0,1,0,0.14,0.95,0.13,0.69,0.01,0.06,0.07,0.02,0.61,0.23,0.12,0.31,0.39,0,0.31,0.02,0.43,0,0.37,0.12,0,0.08,0.53,0.12,0,0.15,0.02,0,1,0.04,0,0.24,0.14,1,1,0,0.08,0.55,0,0.4,0,0,0,0.52,0.31,0.41,0.28,0.64,0.15,0.96,0,0.27,0,0.25,0,0,0,0.1,0,0,0.07,0,0.01,0.22,0.17,0,0,0,0,0,0.11,0.14,0.32,0.27,0.26,0.32,0.16,0.25,0.56,0.61,0.04,0.06,0.77,0.83,0.33,0,0,0,0.06,0.15,0.36,0.16,0.19,0.19,0,0.02,1,0.41,0.06,0.17,0.02,0,0.01,0.42,0.03,0.51,0.01,0.19,0.05,0,0.76,0.08,0,0,0.49,0.35,1,0.28,1,1,0,0.01,0.6,1,1,1,1,1,0.7,0.38,1,0.62,0.88,0.37,1,0.24,0.14,0,0.11,0,0.35,0.44,0.22,0.21,0.24,0,0,0,0.09,0.22,0.11,0,0.04,0.28,0.36,0.23,0.79,0.38,0.48,0.25,0.41,1,0.21,0,0.03,0.31,0.19,0.25,0.72,0,0.35,0.06,0.42,0,0,1,0.43,0.22,0.69,0.66,0,0.55,0.4,0.75,0.26,0.29,0.24,0.22,0,0.5,0.61,0.53,0.03,0,0.11,0.07,0.23,0.48,0.29,0,0.18,0.08,0.05,0.12,0.05,0.08,0.57,0.05,0.26,0,0.05,0,0,0.38,0.32,0.17,0.12,0.23,1,0.22,0.17,0.59,0.18,0.62,0.7,1,0.1,0.55,1,0.24,0.97,1,1,0.33,0.87,0.27,0.39,0.14,0.87,0],[0.16,0.17,0.17,0.15,0.15,0.14,0.13,0.16,0.17,0.22,0.21,0.18,0.33,0.21,0.28,0.15,0.21,0.17,0.16,0.15,0.26,0.22,0.18,0.2,0.13,0.14,0.22,0.18,0.14,0.14,0.15,0.18,0.14,0.17,0.16,0.16,0.26,0.25,0.17,0.15,0.15,0.15,0.17,0.15,0.15,0.13,0.12,0.16,0.16,0.15,0.15,0.15,0.13,0.14,0.17,0.15,0.15,0.15,0.15,0.18,0.18,0.14,0.14,0.13,0.11,0.16,0.17,0.13,0.12,0.19,0.12,0.19,0.19,0.15,0.17,0.13,0.15,0.12,0.14,0.13,0.2,0.14,0.14,0.14,0.15,0.17,0.11,0.14,0.14,0.2,0.38,0.21,0.13,0.16,0.17,0.16,0.21,0.14,0.15,0.17,0.14,0.14,0.14,0.17,0.09,0.11,0.17,0.17,0.26,0.17,0.18,0.17,0.15,0.22,0.2,0.15,0.18,0.17,0.17,0.13,0.11,0.15,0.25,0.19,0.2,0.15,0.18,0.15,0.18,0.16,0.14,0.17,0.14,0.12,0.14,0.13,0.16,0.26,0.16,0.22,0.17,0.19,0.19,0.13,0.15,0.14,0.15,0.17,0.14,0.09,0.1,0.16,0.17,0.18,0.13,0.21,0.13,0.16,0.17,0.19,0.1,0.17,0.16,0.2,0.11,0.16,0.17,0.19,0.13,0.13,0.13,0.13,0.14,0.21,0.23,0.07,0.2,0.15,0.25,0.19,0.17,0.2,0.16,0.11,0.12,0.19,0.15,0.18,0.18,0.08,0.1,0.09,0.16,0.12,0.08,0.12,0.2,0.16,0.12,0.17,0.13,0.17,0.17,0.17,0.15,0.28,0.16,0.12,0.19,0.09,0.16,0.17,0.16,0.19,0.14,0.1,0.05,0.17,0.16,0.14,0.16,0.2,0.17,0.15,0.09,0.14,0.14,0.15,0.15,0.14,0.18,0.1,0.12,0.11,0.14,0.12,0.13,0.12,0.16,0.24,0.13,0.13,0.18,0.21,0.15,0.18,0.17,0.21,0.36,0.21,0.19,0.21,0.16,0.23,0.21,0.13,0.18,0.17,0.19,0.16,0.12,0.19,0.22,0.18,0.22,0.17,0.17,0.22,0.2,0.08,0.06,0.12,0.2,0.18,0.14,0.11,0.15,0.41,0.81,0.41,0.72,0.22,0.18,0.13,0.15,0.17,0.17,0.25,0.41,0.18,0.2,0.33,0.17,0.11,0.15,0.3,0.32,0.25,0.11,0.1,0.09,0.11,0.16,0.14,0.11,0.21,0.28,0.21,0.21,0.18,0.16,0.18,0.18,0.16,0.13,0.14,0.17,0.27,0.16,0.16,0.13,0.15,0.15,0.18,0.14,0.16,0.19,0.14,0.16,0.09,0.15,0.06,0.18,0.15,0.19,0.2,0.18,0.35,0.14,0.13,0.2,0.13,0.17,0.13,0.16,0.16,0.41,0.18,0.17,0.12,0.14,0.14,0.15,0.17,0.18,0.2,0.18,0.14,0.18,0.17,0.15,0.15,0.12,0.1,0.19,0.25,0.12,0.17,0.22,0.44,0.52,0.47,0.45,0.4,0.2,0.15,0.12,0.22,0.19,0.2,0.24,0.16,0.17,0.14,0.13,0.11,0.16,0.15,0.14,0.14,0.12,0.17,0.16,0.18,0.12,0.22,0.15,0.17,0.22,0.14,0.14,0.15,0.14,0.14,0.14,0.15,0.17,0.16,0.18,0.18,0.23,0.14,0.15,0.18,0.22,0.16,0.16,0.27,0.19,0.18,0.19,0.27,0.19,0.19,0.2,0.18,0.16,0.13,0.22,0.24,0.2,0.22,0.19,0.24,0.18,0.19,0.23,0.23,0.16,0.17,0.18,0.15,0.19,0.22,0.16,0.1,0.12,0.28,0.13,0.15,0.14,0.12,0.16,0.11,0.18,0.15,0.15,0.16,0.14,0.18,0.16,0.14,0.17,0.13,0.17,0.14,0.14,0.17,0.18,0.17,0.18,0.13,0.15,0.17,0.16,0.15,0.32,0.18,0.16,0.24,0.16,0.23,0.12,0.1,0.14,0.16,0.18,0.13,0.15,0.12,0.1,0.13,0.17,0.15,0.16,0.16,0.15,0.14,0.16,0.14,0.17,0.21,0.16,0.18,0.15,0.18,0.38,0.16,0.17,0.33,0.25,0.18,0.16,0.15,0.16,0.1,0.13,0.11,0.13,0.13,0.2,0.12,0.13,0.13,0.14,0.16,0.16,0.14,0.16,0.21,0.15,0.29,0.16,0.14,0.13,0.12,0.13,0.19,0.15,0.14,0.19,0.19,0.1,0.14,0.14,0.14,0.17,0.13,0.17,0.15,0.17,0.15,0.17,0.15,0.15,0.14,0.18,0.19,0.19,0.13,0.16,0.12,0.12,0.14,0.34,0.11,0.14,0.18,0.15,0.15,0.17,0.14,0.13,0.14,0.12,0.13,0.11,0.11,0.11,0.16,0.16,0.15,0.16,0.16,0.18,0.28,0.12,0.15,0.13,0.16,0.15,0.13,0.16,0.29,0.09,0.16,0.11,0.15,0.15,0.16,0.11,0.17,0.2,0.14,0.14,0.13,0.14,0.21,0.2,0.14,0.14,0.13,0.16,0.12,0.15,0.12,0.12,0.21,0.14,0.12,0.11,0.15,0.12,0.17,0.15,0.13,0.21,0.2,0.2,0.19,0.22,0.17,0.23,0.16,0.17,0.12,0.16,0.17,0.22,0.15,0.18,0.16,0.14,0.18,0.15,0.17,0.2,0.16,0.18,0.16,0.21,0.11,0.1,0.14,0.16,0.17,0.14,0.23,0.12,0.19,0.16,0.15,0.14,0.17,0.16,0.24,0.13,0.18,0.15,0.16,0.13,0.12,0.16,0.13,0.19,0.16,0.18,0.16,0.19,0.43,0.78,0.29,0.23,0.17,0.19,0.21,0.2,0.17,0.22,0.13,0.14,0.12,0.11,0.11,0.14,0.15,0.24,0.41,0.09,0.17,0.14,0.2,0.15,0.14,0.13,0.15,0.14,0.18,0.15,0.13,0.14,0.13,0.13,0.13,0.16,0.15,0.15,0.14,0.16,0.12,0.17,0.2,0.2,0.14,0.16,0.19,0.16,0.16,0.15,0.18,0.16,0.5,0.17,0.18,0.2,0.22,0.15,0.15,0.12,0.15,0.14,0.15,0.12,0.17,0.14,0.15,0.12,0.14,0.15,0.14,0.16,0.16,0.15,0.14,0.16,0.14,0.22,0.19,0.13,0.34,0.08,0.19,0.14,0.18,0.16,0.22,0.15,0.44,0.26,0.14,0.12,0.21,0.23,0.17,0.14,0.19,0.13,0.16,0.14,0.14,0.35,0.26,0.21,0.16,0.13,0.14,0.13,0.24,0.2,0.11,0.13,0.21,0.13,0.25,0.12,0.14,0.19,0.2,0.3,0.32,0.12,0.18,0.17,0.16,0.14,0.13,0.13,0.14,0.14,0.13,0.15,0.14,0.13,0.14,0.11,0.11,0.14,0.22,0.2,0.15,0.15,0.23,0.2,0.16,0.16,0.16,0.2,0.17,0.17,0.23,0.11,0.18,0.11,0.15,0.12,0.17,0.12,0.16,0.12,0.14,0.14,0.15,0.14,0.18,0.18,0.28,0.31,0.28,0.22,0.32,0.16,0.18,0.26,0.25,0.3,0.2,0.18,0.19,0.17,0.18,0.28,0.32,0.12,0.16,0.13,0.18,0.13,0.18,0.14,0.17,0.17,0.31,0.14,0.18,0.18,0.22,0.28,0.4,0.15,0.19,0.22,0.22,0.16,0.17,0.12,0.12,0.2,0.24,0.14,0.16,0.16,0.14,0.16,0.11,0.16,0.16,0.14,0.14,0.36,0.24,0.19,0.25,0.25,0.17,0.18,0.23,0.52,0.12,0.18,0.24,0.17,0.18,0.56,0.2,0.2,0.17,0.13,0.13,0.18,0.18,0.5,0.12,0.11,0.14,0.14,0.16,0.16,0.14,0.12,0.17,0.23,0.12,0.18,0.16,0.14,0.15,0.13,0.14,0.18,0.15,0.16,0.15,0.16,0.19,0.19,0.26,0.17,0.19,0.18,0.19,0.12,0.24,0.18,0.13,0.12,0.13,0.12,0.09,0.12,0.15,0.1,0.16,0.1,0.15,0.16,0.2,0.11,0.12,0.2,0.2,0.42,0.31],[0.32,0.33,0.33,0.31,0.31,0.3,0.22,0.34,0.33,0.44,0.43,0.37,0.68,0.44,0.58,0.26,0.42,0.3,0.31,0.28,0.52,0.43,0.33,0.41,0.27,0.3,0.45,0.36,0.29,0.27,0.28,0.37,0.29,0.36,0.33,0.33,0.54,0.52,0.35,0.29,0.3,0.29,0.34,0.29,0.32,0.27,0.24,0.33,0.33,0.29,0.3,0.3,0.28,0.27,0.33,0.3,0.31,0.29,0.31,0.35,0.35,0.28,0.29,0.27,0.23,0.3,0.34,0.26,0.24,0.37,0.19,0.36,0.39,0.28,0.34,0.27,0.26,0.25,0.29,0.26,0.4,0.28,0.29,0.25,0.31,0.3,0.25,0.25,0.25,0.38,0.75,0.4,0.29,0.29,0.33,0.32,0.43,0.28,0.29,0.32,0.27,0.3,0.28,0.31,0.19,0.16,0.34,0.31,0.53,0.35,0.37,0.33,0.31,0.45,0.4,0.3,0.36,0.35,0.34,0.25,0.18,0.32,0.5,0.37,0.42,0.25,0.34,0.29,0.36,0.31,0.28,0.36,0.27,0.26,0.23,0.28,0.29,0.54,0.34,0.45,0.36,0.38,0.34,0.21,0.32,0.27,0.28,0.33,0.27,0.13,0.15,0.33,0.36,0.39,0.28,0.43,0.23,0.31,0.36,0.4,0.22,0.34,0.29,0.42,0.25,0.3,0.35,0.38,0.23,0.24,0.26,0.26,0.27,0.42,0.47,0.14,0.39,0.29,0.52,0.4,0.29,0.43,0.34,0.18,0.21,0.39,0.26,0.31,0.35,0.15,0.15,0.15,0.34,0.2,0.13,0.22,0.42,0.29,0.22,0.31,0.21,0.34,0.34,0.33,0.28,0.58,0.33,0.21,0.38,0.16,0.33,0.31,0.3,0.37,0.23,0.18,0.1,0.35,0.29,0.26,0.3,0.38,0.32,0.29,0.14,0.24,0.29,0.26,0.29,0.24,0.37,0.15,0.26,0.2,0.29,0.23,0.27,0.25,0.27,0.48,0.22,0.21,0.33,0.4,0.28,0.33,0.34,0.43,0.71,0.39,0.38,0.42,0.3,0.48,0.4,0.23,0.34,0.34,0.37,0.33,0.25,0.36,0.46,0.32,0.43,0.33,0.33,0.43,0.37,0.15,0.11,0.2,0.39,0.33,0.23,0.16,0.28,0.8,1.62,0.82,1.44,0.46,0.36,0.27,0.28,0.35,0.32,0.52,0.83,0.32,0.34,0.67,0.31,0.21,0.29,0.62,0.66,0.5,0.24,0.22,0.2,0.25,0.33,0.25,0.25,0.43,0.57,0.42,0.4,0.33,0.31,0.35,0.36,0.31,0.27,0.29,0.31,0.54,0.29,0.33,0.23,0.31,0.31,0.36,0.29,0.33,0.36,0.25,0.31,0.16,0.27,0.12,0.37,0.3,0.39,0.42,0.38,0.71,0.26,0.22,0.4,0.25,0.33,0.29,0.29,0.33,0.83,0.36,0.35,0.18,0.27,0.27,0.31,0.31,0.33,0.41,0.37,0.29,0.35,0.34,0.25,0.28,0.21,0.16,0.39,0.51,0.26,0.32,0.42,0.9,1.03,0.93,0.88,0.78,0.39,0.27,0.18,0.44,0.38,0.36,0.5,0.29,0.36,0.24,0.23,0.2,0.33,0.3,0.29,0.29,0.26,0.32,0.33,0.37,0.21,0.45,0.32,0.33,0.45,0.28,0.29,0.3,0.29,0.29,0.29,0.31,0.32,0.32,0.36,0.37,0.44,0.3,0.28,0.34,0.44,0.3,0.32,0.55,0.37,0.35,0.38,0.54,0.4,0.4,0.41,0.34,0.32,0.25,0.45,0.5,0.38,0.45,0.34,0.45,0.38,0.36,0.46,0.42,0.28,0.31,0.38,0.29,0.36,0.45,0.3,0.19,0.19,0.56,0.28,0.29,0.29,0.21,0.29,0.17,0.35,0.28,0.26,0.33,0.27,0.38,0.31,0.27,0.36,0.19,0.34,0.3,0.28,0.34,0.36,0.35,0.38,0.24,0.26,0.3,0.33,0.28,0.59,0.35,0.31,0.49,0.34,0.46,0.24,0.17,0.3,0.31,0.36,0.29,0.3,0.25,0.23,0.28,0.33,0.31,0.3,0.31,0.3,0.3,0.3,0.29,0.33,0.43,0.28,0.35,0.3,0.36,0.77,0.33,0.35,0.67,0.5,0.36,0.35,0.3,0.33,0.22,0.27,0.25,0.27,0.28,0.41,0.25,0.27,0.26,0.29,0.33,0.33,0.29,0.34,0.43,0.31,0.57,0.32,0.3,0.28,0.2,0.21,0.39,0.29,0.3,0.38,0.39,0.15,0.26,0.29,0.28,0.32,0.27,0.31,0.28,0.37,0.31,0.33,0.31,0.29,0.29,0.37,0.37,0.38,0.22,0.34,0.24,0.2,0.23,0.7,0.24,0.3,0.36,0.29,0.32,0.31,0.27,0.22,0.28,0.22,0.2,0.17,0.17,0.18,0.32,0.31,0.28,0.32,0.35,0.38,0.58,0.24,0.29,0.27,0.31,0.31,0.26,0.3,0.56,0.19,0.33,0.23,0.31,0.27,0.34,0.17,0.33,0.4,0.29,0.29,0.27,0.26,0.41,0.38,0.31,0.3,0.24,0.33,0.25,0.29,0.25,0.26,0.42,0.3,0.2,0.18,0.26,0.2,0.32,0.27,0.22,0.41,0.39,0.41,0.39,0.44,0.35,0.46,0.31,0.33,0.21,0.32,0.34,0.45,0.31,0.37,0.31,0.29,0.36,0.32,0.32,0.38,0.31,0.35,0.33,0.42,0.18,0.15,0.28,0.29,0.34,0.29,0.47,0.21,0.39,0.33,0.29,0.22,0.31,0.31,0.49,0.22,0.37,0.26,0.31,0.25,0.22,0.32,0.26,0.38,0.32,0.34,0.31,0.39,0.87,1.53,0.58,0.47,0.31,0.35,0.41,0.37,0.31,0.46,0.26,0.28,0.23,0.22,0.2,0.25,0.26,0.5,0.83,0.14,0.36,0.3,0.41,0.31,0.26,0.23,0.28,0.26,0.35,0.29,0.28,0.25,0.26,0.26,0.28,0.31,0.32,0.27,0.28,0.28,0.26,0.31,0.37,0.38,0.24,0.28,0.4,0.33,0.32,0.29,0.37,0.28,0.99,0.35,0.3,0.41,0.41,0.32,0.31,0.26,0.31,0.25,0.3,0.24,0.33,0.26,0.32,0.23,0.29,0.27,0.29,0.3,0.33,0.31,0.29,0.29,0.28,0.45,0.39,0.29,0.65,0.15,0.4,0.29,0.35,0.33,0.39,0.28,0.88,0.54,0.25,0.26,0.4,0.47,0.36,0.3,0.35,0.26,0.3,0.27,0.24,0.71,0.47,0.43,0.32,0.27,0.26,0.29,0.49,0.41,0.22,0.28,0.42,0.26,0.49,0.26,0.27,0.39,0.43,0.58,0.62,0.25,0.37,0.34,0.32,0.26,0.26,0.25,0.26,0.27,0.25,0.26,0.24,0.27,0.29,0.19,0.18,0.27,0.44,0.42,0.31,0.32,0.46,0.39,0.32,0.3,0.32,0.42,0.36,0.3,0.44,0.24,0.36,0.24,0.32,0.25,0.31,0.25,0.3,0.27,0.29,0.3,0.31,0.24,0.37,0.36,0.57,0.59,0.54,0.4,0.64,0.28,0.33,0.54,0.53,0.56,0.35,0.31,0.33,0.3,0.33,0.52,0.63,0.2,0.29,0.21,0.34,0.21,0.35,0.28,0.36,0.34,0.62,0.27,0.33,0.35,0.43,0.55,0.8,0.32,0.4,0.44,0.43,0.33,0.36,0.26,0.23,0.39,0.47,0.23,0.31,0.29,0.28,0.29,0.18,0.31,0.33,0.31,0.27,0.72,0.47,0.33,0.52,0.48,0.35,0.34,0.48,1.04,0.22,0.33,0.47,0.32,0.33,1.13,0.36,0.37,0.29,0.26,0.25,0.36,0.35,1.01,0.21,0.2,0.24,0.3,0.33,0.32,0.29,0.24,0.32,0.43,0.26,0.34,0.32,0.29,0.3,0.28,0.3,0.33,0.31,0.3,0.33,0.32,0.39,0.39,0.5,0.33,0.38,0.36,0.38,0.22,0.47,0.37,0.23,0.23,0.23,0.21,0.14,0.24,0.27,0.16,0.3,0.14,0.26,0.27,0.37,0.17,0.22,0.36,0.41,0.8,0.63],[0.17,0.18,0.18,0.18,0.18,0.18,0.11,0.19,0.18,0.25,0.24,0.21,0.36,0.24,0.31,0.13,0.23,0.15,0.17,0.14,0.28,0.22,0.16,0.23,0.16,0.17,0.25,0.19,0.17,0.15,0.15,0.21,0.17,0.21,0.19,0.19,0.29,0.28,0.2,0.16,0.17,0.16,0.18,0.16,0.18,0.15,0.15,0.18,0.18,0.16,0.17,0.16,0.16,0.15,0.17,0.17,0.17,0.16,0.17,0.19,0.19,0.16,0.16,0.15,0.14,0.16,0.18,0.15,0.14,0.2,0.1,0.19,0.21,0.15,0.19,0.16,0.13,0.15,0.17,0.14,0.22,0.16,0.17,0.13,0.18,0.15,0.15,0.14,0.13,0.2,0.4,0.21,0.17,0.15,0.18,0.17,0.24,0.16,0.16,0.17,0.15,0.17,0.17,0.16,0.12,0.08,0.18,0.16,0.29,0.2,0.2,0.18,0.18,0.24,0.23,0.17,0.2,0.2,0.19,0.13,0.1,0.18,0.27,0.2,0.23,0.12,0.18,0.16,0.19,0.17,0.15,0.2,0.15,0.15,0.12,0.17,0.15,0.29,0.19,0.25,0.2,0.21,0.17,0.11,0.18,0.15,0.16,0.18,0.15,0.07,0.08,0.19,0.2,0.22,0.17,0.24,0.12,0.16,0.2,0.22,0.14,0.18,0.15,0.24,0.15,0.15,0.2,0.2,0.12,0.14,0.15,0.16,0.15,0.23,0.25,0.07,0.21,0.16,0.28,0.23,0.14,0.24,0.2,0.1,0.12,0.22,0.14,0.15,0.19,0.08,0.09,0.08,0.19,0.1,0.07,0.12,0.23,0.15,0.12,0.16,0.1,0.19,0.18,0.17,0.15,0.31,0.19,0.12,0.2,0.08,0.19,0.17,0.16,0.2,0.12,0.09,0.05,0.19,0.15,0.14,0.16,0.2,0.17,0.16,0.08,0.12,0.16,0.14,0.15,0.12,0.21,0.09,0.17,0.11,0.18,0.13,0.17,0.15,0.14,0.25,0.12,0.11,0.17,0.21,0.15,0.16,0.18,0.23,0.38,0.2,0.21,0.24,0.16,0.26,0.2,0.11,0.18,0.19,0.2,0.18,0.15,0.19,0.25,0.16,0.22,0.18,0.17,0.22,0.19,0.07,0.05,0.1,0.2,0.17,0.11,0.09,0.15,0.4,0.82,0.43,0.74,0.25,0.2,0.15,0.15,0.2,0.17,0.28,0.44,0.16,0.17,0.35,0.15,0.11,0.14,0.34,0.36,0.27,0.15,0.14,0.13,0.15,0.19,0.13,0.15,0.25,0.32,0.23,0.22,0.17,0.17,0.18,0.2,0.17,0.15,0.17,0.16,0.29,0.15,0.19,0.12,0.18,0.18,0.2,0.16,0.19,0.19,0.13,0.17,0.09,0.14,0.06,0.2,0.17,0.22,0.23,0.22,0.38,0.14,0.11,0.23,0.14,0.18,0.17,0.15,0.18,0.43,0.19,0.19,0.09,0.15,0.15,0.18,0.16,0.17,0.22,0.21,0.17,0.18,0.18,0.13,0.15,0.11,0.09,0.22,0.28,0.16,0.17,0.22,0.48,0.52,0.47,0.46,0.41,0.21,0.14,0.1,0.24,0.21,0.18,0.28,0.15,0.2,0.11,0.12,0.11,0.19,0.16,0.17,0.16,0.15,0.17,0.19,0.2,0.11,0.25,0.18,0.17,0.24,0.16,0.17,0.17,0.17,0.16,0.17,0.17,0.17,0.17,0.2,0.21,0.23,0.18,0.15,0.18,0.24,0.16,0.17,0.3,0.19,0.19,0.21,0.3,0.22,0.22,0.22,0.18,0.18,0.14,0.24,0.27,0.2,0.25,0.17,0.23,0.22,0.18,0.25,0.21,0.14,0.15,0.22,0.15,0.19,0.25,0.15,0.09,0.1,0.3,0.16,0.16,0.17,0.11,0.15,0.09,0.19,0.15,0.13,0.18,0.15,0.21,0.17,0.15,0.2,0.1,0.19,0.17,0.16,0.19,0.19,0.19,0.21,0.14,0.14,0.15,0.19,0.15,0.29,0.19,0.17,0.28,0.2,0.26,0.13,0.08,0.17,0.17,0.2,0.17,0.17,0.15,0.14,0.16,0.17,0.18,0.16,0.17,0.16,0.17,0.16,0.17,0.17,0.24,0.14,0.19,0.16,0.21,0.42,0.19,0.2,0.35,0.27,0.2,0.2,0.17,0.19,0.14,0.16,0.15,0.16,0.17,0.23,0.15,0.16,0.15,0.17,0.19,0.19,0.17,0.19,0.24,0.18,0.31,0.19,0.18,0.16,0.1,0.11,0.22,0.16,0.18,0.22,0.22,0.09,0.14,0.17,0.16,0.17,0.16,0.15,0.15,0.21,0.18,0.18,0.18,0.16,0.17,0.2,0.2,0.21,0.11,0.19,0.13,0.1,0.11,0.38,0.15,0.17,0.2,0.16,0.18,0.16,0.14,0.12,0.16,0.13,0.1,0.09,0.09,0.1,0.18,0.17,0.15,0.18,0.2,0.21,0.31,0.14,0.15,0.16,0.17,0.18,0.15,0.16,0.31,0.12,0.19,0.14,0.17,0.14,0.19,0.09,0.18,0.22,0.17,0.17,0.16,0.14,0.22,0.2,0.18,0.18,0.13,0.19,0.15,0.16,0.14,0.15,0.24,0.17,0.11,0.1,0.13,0.1,0.17,0.14,0.11,0.21,0.2,0.23,0.22,0.23,0.2,0.25,0.17,0.18,0.11,0.17,0.19,0.25,0.18,0.2,0.17,0.17,0.19,0.19,0.17,0.2,0.17,0.19,0.19,0.23,0.09,0.08,0.15,0.15,0.18,0.16,0.25,0.1,0.21,0.18,0.16,0.11,0.16,0.16,0.27,0.12,0.2,0.13,0.16,0.14,0.12,0.17,0.15,0.2,0.18,0.18,0.17,0.21,0.46,0.79,0.32,0.25,0.16,0.18,0.22,0.2,0.16,0.25,0.14,0.16,0.13,0.12,0.11,0.13,0.14,0.27,0.43,0.07,0.2,0.17,0.22,0.18,0.14,0.13,0.15,0.14,0.18,0.16,0.16,0.14,0.15,0.14,0.17,0.18,0.18,0.14,0.16,0.14,0.16,0.16,0.19,0.19,0.12,0.15,0.23,0.19,0.18,0.16,0.21,0.14,0.52,0.21,0.15,0.22,0.2,0.18,0.18,0.15,0.18,0.13,0.16,0.14,0.17,0.14,0.18,0.13,0.17,0.14,0.17,0.16,0.18,0.18,0.17,0.15,0.16,0.25,0.21,0.17,0.35,0.07,0.22,0.17,0.19,0.18,0.19,0.14,0.47,0.29,0.13,0.16,0.2,0.25,0.2,0.18,0.18,0.14,0.16,0.15,0.13,0.37,0.23,0.24,0.17,0.16,0.14,0.17,0.27,0.23,0.14,0.17,0.23,0.15,0.27,0.15,0.15,0.21,0.24,0.31,0.33,0.16,0.21,0.18,0.17,0.14,0.14,0.14,0.15,0.16,0.14,0.13,0.12,0.16,0.17,0.1,0.09,0.14,0.24,0.23,0.19,0.18,0.24,0.2,0.18,0.16,0.17,0.23,0.2,0.14,0.23,0.14,0.19,0.14,0.18,0.15,0.16,0.15,0.15,0.16,0.16,0.17,0.19,0.12,0.2,0.2,0.3,0.3,0.28,0.19,0.33,0.15,0.16,0.3,0.29,0.28,0.18,0.16,0.16,0.15,0.16,0.26,0.32,0.1,0.15,0.11,0.18,0.1,0.18,0.16,0.2,0.18,0.33,0.15,0.17,0.18,0.23,0.29,0.42,0.19,0.23,0.23,0.22,0.18,0.21,0.16,0.13,0.21,0.25,0.12,0.16,0.15,0.16,0.15,0.09,0.17,0.19,0.18,0.15,0.38,0.24,0.17,0.28,0.24,0.19,0.18,0.26,0.54,0.11,0.18,0.25,0.16,0.17,0.59,0.19,0.19,0.15,0.15,0.14,0.19,0.19,0.52,0.11,0.11,0.13,0.17,0.19,0.17,0.17,0.14,0.16,0.22,0.16,0.18,0.18,0.16,0.16,0.16,0.17,0.17,0.18,0.16,0.19,0.18,0.22,0.22,0.25,0.18,0.21,0.2,0.2,0.11,0.25,0.2,0.12,0.14,0.12,0.11,0.08,0.14,0.14,0.08,0.16,0.08,0.13,0.13,0.19,0.09,0.13,0.19,0.22,0.4,0.34],[0.25,0.23,0.23,0.17,0.16,0.15,0.48,0.17,0.22,0.17,0.2,0.16,0.28,0.23,0.23,0.6,0.29,0.47,0.27,0.43,0.28,0.46,0.42,0.19,0.17,0.16,0.19,0.22,0.14,0.22,0.24,0.16,0.13,0.17,0.16,0.15,0.22,0.22,0.16,0.19,0.16,0.2,0.24,0.19,0.16,0.17,0.14,0.19,0.18,0.2,0.18,0.24,0.14,0.18,0.26,0.19,0.18,0.21,0.18,0.24,0.24,0.2,0.15,0.18,0.13,0.28,0.21,0.18,0.14,0.25,0.32,0.31,0.21,0.26,0.2,0.14,0.33,0.14,0.15,0.18,0.19,0.16,0.12,0.34,0.14,0.34,0.12,0.26,0.31,0.29,0.25,0.39,0.13,0.42,0.24,0.27,0.2,0.21,0.26,0.34,0.22,0.14,0.15,0.31,0.11,0.31,0.24,0.32,0.21,0.16,0.2,0.23,0.14,0.2,0.16,0.17,0.2,0.17,0.23,0.27,0.36,0.17,0.2,0.28,0.19,0.37,0.28,0.19,0.26,0.27,0.23,0.18,0.26,0.13,0.36,0.13,0.37,0.23,0.16,0.19,0.16,0.17,0.48,0.43,0.17,0.22,0.24,0.21,0.26,0.3,0.3,0.16,0.17,0.18,0.13,0.17,0.3,0.29,0.18,0.18,0.11,0.25,0.34,0.17,0.11,0.34,0.17,0.26,0.29,0.29,0.18,0.18,0.28,0.28,0.25,0.58,0.32,0.23,0.22,0.17,0.46,0.2,0.17,0.25,0.28,0.21,0.39,0.48,0.26,0.48,0.34,0.46,0.17,0.5,0.34,0.25,0.23,0.34,0.27,0.39,0.47,0.25,0.28,0.28,0.32,0.24,0.19,0.31,0.29,0.5,0.15,0.35,0.3,0.34,0.39,0.57,0.5,0.22,0.4,0.29,0.27,0.35,0.33,0.26,0.33,0.47,0.21,0.42,0.29,0.41,0.18,0.25,0.11,0.27,0.14,0.2,0.14,0.18,0.45,0.35,0.34,0.43,0.36,0.43,0.33,0.76,0.26,0.24,0.25,0.42,0.26,0.18,0.32,0.2,0.4,0.53,0.36,0.23,0.3,0.21,0.18,0.29,0.19,0.4,0.29,0.28,0.32,0.32,0.35,0.57,0.58,0.43,0.38,0.4,0.46,0.31,0.28,0.71,0.59,0.43,0.52,0.21,0.2,0.17,0.24,0.16,0.29,0.24,0.29,0.7,0.7,0.38,0.86,0.88,0.87,0.31,0.31,0.2,0.12,0.1,0.09,0.11,0.15,0.26,0.11,0.19,0.23,0.27,0.32,0.3,0.23,0.3,0.23,0.21,0.19,0.17,0.36,0.22,0.33,0.16,0.34,0.17,0.16,0.21,0.17,0.17,0.35,0.38,0.27,0.42,0.34,0.47,0.22,0.23,0.17,0.23,0.18,0.26,0.29,0.45,0.17,0.2,0.26,0.13,0.39,0.22,0.45,0.23,0.21,0.34,0.2,0.21,0.14,0.32,0.35,0.25,0.16,0.17,0.29,0.21,0.37,0.26,0.24,0.3,0.22,0.21,0.12,0.27,0.34,0.37,0.38,0.39,0.33,0.3,0.25,0.36,0.32,0.28,0.29,0.49,0.23,0.32,0.16,0.5,0.32,0.23,0.15,0.23,0.13,0.18,0.13,0.27,0.15,0.21,0.47,0.2,0.18,0.28,0.21,0.17,0.14,0.19,0.15,0.18,0.15,0.21,0.26,0.23,0.22,0.19,0.41,0.16,0.34,0.29,0.26,0.3,0.29,0.27,0.3,0.28,0.22,0.19,0.2,0.2,0.25,0.27,0.22,0.22,0.21,0.21,0.39,0.23,0.55,0.42,0.17,0.33,0.29,0.6,0.61,0.68,0.16,0.27,0.28,0.19,0.41,0.52,0.39,0.24,0.15,0.23,0.16,0.29,0.31,0.36,0.25,0.29,0.34,0.19,0.21,0.18,0.24,0.23,0.16,0.4,0.19,0.14,0.19,0.2,0.25,0.24,0.18,0.24,0.31,0.43,0.16,0.26,1.27,0.26,0.22,0.2,0.16,0.21,0.2,0.36,0.15,0.25,0.16,0.13,0.2,0.13,0.11,0.15,0.24,0.14,0.26,0.21,0.22,0.16,0.24,0.16,0.27,0.19,0.34,0.22,0.22,0.15,0.31,0.17,0.19,0.36,0.29,0.24,0.16,0.19,0.15,0.11,0.15,0.12,0.12,0.12,0.18,0.15,0.15,0.17,0.13,0.14,0.14,0.17,0.15,0.17,0.14,0.22,0.13,0.14,0.14,0.48,0.42,0.18,0.23,0.14,0.15,0.16,0.26,0.31,0.16,0.18,0.29,0.13,0.41,0.28,0.17,0.18,0.22,0.18,0.19,0.16,0.22,0.27,0.23,0.44,0.17,0.19,0.5,0.5,0.33,0.12,0.14,0.21,0.24,0.15,0.38,0.27,0.27,0.18,0.19,0.42,0.36,0.36,0.29,0.23,0.23,0.26,0.24,0.16,0.17,0.25,0.14,0.29,0.16,0.21,0.16,0.19,0.32,0.19,0.11,0.17,0.14,0.16,0.34,0.16,0.34,0.24,0.21,0.14,0.12,0.14,0.27,0.33,0.32,0.14,0.14,0.28,0.17,0.15,0.27,0.17,0.16,0.17,0.15,0.32,0.41,0.42,0.44,0.34,0.43,0.44,0.4,0.35,0.2,0.18,0.31,0.16,0.19,0.26,0.25,0.33,0.24,0.19,0.19,0.17,0.22,0.27,0.19,0.22,0.15,0.28,0.33,0.25,0.27,0.19,0.26,0.37,0.32,0.23,0.39,0.22,0.17,0.21,0.48,0.22,0.19,0.25,0.41,0.32,0.3,0.22,0.35,0.22,0.39,0.28,0.23,0.25,0.25,0.2,0.28,0.2,0.35,0.25,0.24,0.37,0.51,0.24,0.27,0.4,0.4,0.31,0.36,0.39,0.22,0.2,0.18,0.18,0.19,0.24,0.32,0.37,0.23,0.45,0.35,0.16,0.15,0.21,0.15,0.26,0.25,0.26,0.24,0.3,0.23,0.15,0.28,0.15,0.22,0.14,0.2,0.16,0.32,0.16,0.35,0.12,0.3,0.39,0.44,0.43,0.33,0.17,0.13,0.17,0.21,0.16,0.59,0.35,0.22,0.56,0.26,0.59,0.16,0.17,0.14,0.16,0.33,0.23,0.15,0.28,0.26,0.15,0.21,0.14,0.28,0.12,0.29,0.2,0.14,0.16,0.35,0.17,0.19,0.26,0.14,0.22,0.72,0.21,0.14,0.26,0.21,0.83,0.77,0.34,0.3,0.3,0.12,0.39,0.2,0.16,0.13,0.41,0.22,0.3,0.23,0.33,0.47,0.92,0.21,0.26,0.12,0.21,0.13,0.2,0.17,0.13,0.13,0.17,0.15,0.19,0.12,0.2,0.26,0.19,0.19,0.21,0.11,0.15,0.21,0.2,0.23,0.21,0.2,0.23,0.18,0.2,0.32,0.31,0.14,0.16,0.29,0.31,0.24,0.21,0.2,0.13,0.17,0.3,0.36,0.21,0.22,0.23,0.19,0.18,0.68,0.44,0.13,0.25,0.11,0.15,0.12,0.31,0.12,0.34,0.13,0.2,0.15,0.14,0.36,0.21,0.16,0.22,0.67,0.49,0.85,0.53,0.6,0.8,0.26,0.26,0.72,0.7,0.61,0.65,0.64,0.71,0.75,0.6,0.45,0.37,0.37,0.33,0.42,0.27,0.18,0.17,0.2,0.22,0.25,0.34,0.26,0.33,0.43,0.32,0.16,0.19,0.26,0.32,0.2,0.16,0.13,0.19,0.36,0.35,0.37,0.29,0.33,0.22,0.3,0.4,0.22,0.15,0.15,0.24,0.51,0.37,0.49,0.22,0.43,0.19,0.34,0.22,0.4,0.7,0.35,0.35,0.47,0.46,0.46,0.45,0.36,0.45,0.21,0.21,0.28,0.26,0.38,0.24,0.27,0.29,0.15,0.15,0.19,0.16,0.18,0.35,0.36,0.11,0.24,0.18,0.15,0.18,0.14,0.18,0.4,0.17,0.24,0.15,0.17,0.17,0.18,0.47,0.29,0.26,0.22,0.29,0.59,0.35,0.25,0.3,0.16,0.3,0.31,0.29,0.14,0.33,0.33,0.23,0.27,0.52,0.53,0.34,0.3,0.18,0.35,0.26,1.42,0.3]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Trait<\/th>\n      <th>Sex<\/th>\n      <th>Trait guild<\/th>\n      <th>Trait description<\/th>\n      <th>Reference<\/th>\n      <th>V(G)<\/th>\n      <th>V(e)<\/th>\n      <th>Vp<\/th>\n      <th>V(G)/Vp<\/th>\n      <th>SE_V(G)<\/th>\n      <th>SE_V(e)<\/th>\n      <th>SE_Vp<\/th>\n      <th>SE_V(G)/Vp<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"autoWidth":true,"columnDefs":[{"width":"300px","targets":4},{"className":"dt-right","targets":[5,6,7,8,9,10,11,12]}],"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":1000,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Variance_table"}],"pageLength":1000,"order":[],"orderClasses":false,"lengthMenu":[10,25,50,100,1000]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="use-gcta-to-estimate-genetic-correlations-between-traits"
class="section level1">
<h1>Use GCTA to estimate genetic correlations between traits</h1>
<p><span class="math inline">\(~\)</span></p>
<p>Here we use the <code>gcta64</code> command <code>--reml-bivar</code>
to estimate the genetic covariance <code>(C(G)_tr12)</code>, i.e. the
co-variance explained by the known SNPs in the DGRP, and the genetic
correlation (<code>rG</code>), in addition to the univariate parameters
for trait 1 and trait 2 mentioned above. Note that for traits where the
genetic variance is estimated to be low, the model cannot reliably
estimate genetic covariance or correlations.</p>
<p><strong>I haven’t run this yet because it will take AGES</strong></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] DT_0.23          kableExtra_1.3.4 glue_1.6.2       forcats_0.5.1   
 [5] stringr_1.4.0    dplyr_1.0.9      purrr_0.3.4      readr_2.1.2     
 [9] tidyr_1.2.0      tibble_3.1.7     ggplot2_3.3.6    tidyverse_1.3.1 
[13] workflowr_1.7.0 

loaded via a namespace (and not attached):
 [1] httr_1.4.3        sass_0.4.1        bit64_4.0.5       vroom_1.5.7      
 [5] jsonlite_1.8.0    viridisLite_0.4.0 modelr_0.1.8      bslib_0.3.1      
 [9] assertthat_0.2.1  getPass_0.2-2     cellranger_1.1.0  yaml_2.3.5       
[13] pillar_1.7.0      backports_1.4.1   digest_0.6.29     promises_1.2.0.1 
[17] rvest_1.0.2       colorspace_2.0-3  htmltools_0.5.2   httpuv_1.6.5     
[21] pkgconfig_2.0.3   broom_0.8.0       haven_2.5.0       scales_1.2.0     
[25] webshot_0.5.3     processx_3.6.1    svglite_2.1.0     whisker_0.4      
[29] later_1.3.0       tzdb_0.3.0        git2r_0.30.1      generics_0.1.2   
[33] ellipsis_0.3.2    withr_2.5.0       cli_3.3.0         magrittr_2.0.3   
[37] crayon_1.5.1      readxl_1.4.0      evaluate_0.15     ps_1.7.1         
[41] fs_1.5.2          fansi_1.0.3       xml2_1.3.3        tools_4.2.1      
[45] hms_1.1.1         lifecycle_1.0.1   munsell_0.5.0     reprex_2.0.1     
[49] callr_3.7.0       compiler_4.2.1    jquerylib_0.1.4   systemfonts_1.0.4
[53] rlang_1.0.3       grid_4.2.1        rstudioapi_0.13   htmlwidgets_1.5.4
[57] crosstalk_1.2.0   rmarkdown_2.14    gtable_0.3.0      DBI_1.1.3        
[61] R6_2.5.1          lubridate_1.8.0   knitr_1.39        fastmap_1.1.0    
[65] bit_4.0.4         utf8_1.2.2        rprojroot_2.0.3   stringi_1.7.6    
[69] parallel_4.2.1    Rcpp_1.0.8.3      vctrs_0.4.1       dbplyr_2.2.1     
[73] tidyselect_1.1.2  xfun_0.31        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
