---
title: "Estimating SNP heritability and genetic correlations"
site: workflowr::wflow_site
editor_options:
  chunk_output_type: console
---

```{r, setup, include=FALSE}
knitr::opts_chunk$set(
  message = FALSE, warning = FALSE#, results='hide'
)
```

# Prepare for `GCTA` analysis

## Load packages and build helper functions

Here we load the necessary packages to build this document and code in 'tidy' style. We also set the file path so that R can access the `gcta64` software. Finally, we build a helper function to pass `GCTA` commands to the terminal. This function allows us to code entirely within R-studio, which is our strong preference.

```{r}
library(tidyverse)
library(glue)
library(kableExtra)
library(DT)
library(MetBrewer)
library(patchwork)
library(rcartocolor)

gcta <- file.path(getwd(), "code/gcta64")
options(readr.show_col_types = FALSE)


# helper function to pass commands to the terminal
# Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`,
# ensuring that the Terminal output will be printed in this knitr report.
run_command <- function(shell_command, wd = getwd(), path = ""){
  cat(system(glue("cd ", wd, path, "\n",shell_command), intern = TRUE), sep = '\n')
}


### useful code that I didn't end up needing

  #remove the suffix that denotes the sex the trait was measured in and replace the level 'both' with 'pooled'
 # mutate(Sex = recode(Sex, Both = "Pooled"),
  #       Trait = case_when(
   #        str_ends(Trait, ".f") | str_ends(Trait, ".m")  ~ gsub('.{2}$', '', Trait)),
    #     `Trait 2` = case_when(
     #      str_ends(`Trait 2`, ".f") | str_ends(`Trait 2`, ".m")  ~ gsub('.{2}$', '', `Trait 2`))) %>% 

```

## Load phenotype data

```{r}
traits <- read_csv("data/derived/all.dgrp.phenos_unscaled.csv") 

traits_for_analysis <- 
  traits %>% 
  # filter(!(`Trait guild` %in% allowed_guilds) | is.na(`Trait guild`)) %>% 
  group_by(Trait) %>% 
  summarise(lines_measured = length(unique(line))) %>% 
  ungroup() %>% 
  filter(lines_measured > 79) %>%
  filter(Trait != "paraquat.resistance.2021.f" & Trait != "starvation.resistance.high.carb.diet.m") %>% # these traits call a gcta error, we remove for now
  pull(Trait)  # find traits with 80+ replicates
 

traits_for_analysis <- traits_for_analysis[!(grepl("fitness.late", traits_for_analysis))]

# replace any slashes in the trait name, e.g. "1/3-Methylhistidine.high.yeast.f", as this is not ok to use as a file name
traits_for_analysis <- str_replace_all(traits_for_analysis, "[/]", "_") 

line_mean_phenotypes <- traits %>%
  mutate(Trait = str_replace_all(Trait, "[/]", "_")) %>%  # change to match `traits_for_analysis` list
  filter(Trait %in% traits_for_analysis) %>%
   filter(Trait != "paraquat.resistance.2021.f" & Trait != "starvation.resistance.high.carb.diet.m") %>% # these traits produce a gcta error, we remove for now
  select(line, Trait, trait_value) %>% 
  #distinct(line, Trait, .keep_all = TRUE) %>% # Remove this line once Tom resolves the duplicates
  spread(Trait, trait_value) %>% 
  mutate_at(vars(-line), ~ as.numeric(scale(.x))) %>%  # scale the traits
  mutate(line = paste("line", line, sep = "")) # add "line" in front of the line number to fit with plink/gcta formatting
```


# Use `GCTA` to estimate the genetic variances and heritabilities for each trait

Here we estimate 'SNP-heritability' for each phenotype using a genomic relatedness matrix Restricted Maximum Likelihood (GREML) approach. GREML uses a set of SNPs $S$ that are known across a population of individuals (or lines, in our case) to estimate the relatedness between random individuals (lines) and compares that with phenotypic variation across that set of individuals. This allows us to estimate the variation in phenotype that is explained by the known set of SNPs in our population. We can then divide this variance by the total variation found for the phenotype, giving the SNP-heritability. Note that SNP-heritability should be smaller than other estimates of heritability such as broad- or narrow-sense heritability, as it is limited to additive effects from a subset of the genome, while the latter two estimates encompass a broader range of genetic effects. 

We use the `GCTA` software to run this analysis. 

A few helpful exerts from the literature about GREML are provided below:

> "Data from a GWAS that are collected to detect statistical associations between SNPs and complex traits are usually analyzed by testing each SNP individually for an association with the trait. To account for the large number of significance tests carried out, a very stringent P value is used. This reduces the occurrence of false positives, but it may cause many real associations to be missed, especially if individual SNPs have a small effect on the trait." 

`r tufte::quote_footer('--- Yang et al 2010')`

> "In contrast to single-SNP association analysis, the basic concept behind the `GCTA` method is to fit the effects of all the SNPs as random effects using a mixed linear model (MLM)."

`r tufte::quote_footer('--- Yang et al 2011')`

## Make a genomic relatedness matrix for all the lines

This uses the `GCTA` command `--make-grm` to make a genomic relatedness matrix (GRM) for the DGRP SNP data. This matrix is needed to run the GREML models below.

```{r}
run_command(glue("{gcta} --bfile dgrp2_QC_all_lines",
                 " --make-grm",
                 " --out gcta_GRM"), path = "/data/gwas_data/derived/")
```

## Run univariate GREML models to find heritabilites

The below uses the `GCTA` command --reml to estimate the genetic variance (V(G)), environmental variance (V(e)), their sum (Vp), and the so-called “SNP heritability” (V(G)/Vp).

```{r}

do_univariate_greml <- function(trait1){

  # First make 'focal_data', a 2-column data frame with the line and the focal phenotype value
  focal_data <- line_mean_phenotypes %>%
    select(line, !! trait1)
  names(focal_data)[2] <- c("focal_phenotype")
  
  # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  pheno_data <- focal_data %>%
    # filter(!(line %in% lines_to_prune)) %>% 
    mutate(line_copy = line) %>%
    select(line, line_copy, focal_phenotype) %>%
    as.matrix()
  
  pheno_data %>%
    write.table(row.names = FALSE, col.names = FALSE,
                file = "data/gwas_data/derived/phenotype.txt", quote = FALSE)
  
  console_output <- suppressWarnings(capture.output(
    run_command(glue("{gcta}  --reml --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype.txt  --out delete_me"), 
                path = "/data/gwas_data/derived/")))
  
  errors <- console_output[str_detect(console_output, "[Ee]rror")]
  if(length(errors) == 0) errors <- NA
  if(length(errors) > 1) errors <- paste(errors, sep = "; ")

  rbind(read.delim("data/gwas_data/derived/delete_me.hsq", sep = "\t") , 
        data.frame(Source = "Errors", Variance = errors, SE = NA)) %>% 
    mutate(Trait = trait1) %>% select(Trait, everything())
}

# Now run the model for every trait and create a large tibble using map_dfr

if(!file.exists("data/gwas_data/variance_components.csv")){
  variance_components <-
    map_dfr(traits_for_analysis, do_univariate_greml) %>% 
    filter(Source %in% c("V(G)", "V(e)", "Vp", "V(G)/Vp")) %>% 
    mutate(Variance = as.numeric(Variance)) %>% 
    rename(Parameter = Source) %>% 
    as_tibble()
  write_csv(variance_components, file = "data/gwas_data/variance_components.csv")
} else variance_components <- read_csv("data/gwas_data/variance_components.csv")

```

Currently two phenotypes call errors when we run the univariate GREML model: `paraquat.resistance.2021.f` and `starvation.resistance.high.carb.diet.m`. The error is _Error: the information matrix is not invertible._

For now we simply don't run models for these traits, so there are no variance estimates for these traits.

$~$

## Wrangle data and save


```{r}
trait_info <- traits %>%
  mutate(Trait = str_replace_all(Trait, "[/]", "_"),
         Sex = fct_recode(Sex, Pooled = "Both")) %>% 
  select(-trait_value) %>% 
  distinct(Trait, Sex, `Trait guild`, `Trait description`, Reference)

variance_components <- left_join(variance_components, trait_info)

# wrangle so that each variance component is its own column

variance_components_wide <- 
  variance_components %>% 
  pivot_wider(names_from = Parameter,
              values_from = c(Variance, SE)) %>% 
  mutate(across(where(is.numeric), round, 2)) %>% 
  rename(`V(G)` = `Variance_V(G)`,
         `V(e)` = `Variance_V(e)`,
         `Vp` = `Variance_Vp`,
         `V(G)/Vp` = `Variance_V(G)/Vp`)
  
# Below we remove the suffix that denotes the sex the trait was measured in and replace the level 'both' with 'pooled'

SNP_heritability <-
  variance_components_wide %>% 
  mutate(Sex = recode(Sex, Both = "Pooled"),
         Trait = case_when(
           str_ends(Trait, ".f") | str_ends(Trait, ".m")  ~ gsub('.{2}$', '', Trait))) %>% 
  select(Trait, Sex, everything()) %>%  # reorder the columns for the table %>% 
  arrange(Trait)


write_csv(SNP_heritability, file = "data/SNP_heritability_cleaned.csv")
```

$~$

# Table 1

Table 1. Phenotypic variance for the traits in our dataset ($V_p$), SNP genetic variance ($V_{SNP}$), residual variance ($V_e$) and heritability ($h^2_{SNP}$).

```{r}
# Create a function to build HTML searchable tables

my_data_table <- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c("Scroller",  "Buttons"),
    options = list(
      autoWidth = TRUE,
      columnDefs = list(list(width = '300px', targets = c(4))),
      dom = 'Bfrtip',
      deferRender=TRUE,
      scrollX=TRUE, scrollY=1000,
      scrollCollapse=TRUE,
      buttons =
        list('pageLength', 'colvis', 'csv', list(
          extend = 'pdf',
          pageSize = 'A4',
          orientation = 'landscape',
          filename = 'Literature_table')),
      pageLength = 1000
    )
  )
}

my_data_table(SNP_heritability)
```

$~$

# Use GCTA to estimate genetic correlations between traits

$~$

Here we use the `gcta64` command `--reml-bivar` to estimate the genetic covariance `(C(G)_tr12)`, i.e. the co-variance explained by the known SNPs in the DGRP, and the genetic correlation (`rG`), in addition to the univariate parameters for trait 1 and trait 2 mentioned above. Note that for traits where the genetic variance is estimated to be low, the model cannot reliably estimate genetic covariance or correlations.

**I haven't run this yet because it will take AGES**
